{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "05428197",
   "metadata": {},
   "source": [
    "# Intro\n",
    "We want to ask if LCL8664 is more similar to other human and macaque LCLs than it is to other macaque tissues. To answer this we will download RNA-seq data from previously published datasets and then do a principle component analysis.  "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "12e6340b",
   "metadata": {},
   "source": [
    "## Download published data"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9865ba17",
   "metadata": {},
   "source": [
    "There are two datasets to use for now, both use data from the Gilad lab and provide human/rhesus RNA-seq data for LCLs, liver, heart, lung, and kidney. \n",
    "\n",
    "The LCL data is available at GEO number GSE24111 and the other tissues are availible at GSE112356. \n",
    "\n",
    "The LCL data has 3 truly biological replicates. But I think the data is SE reads. \n",
    "\n",
    "Human Rep 1 - SRR066745  \n",
    "Human Rep 2 - SRR066746  \n",
    "Human Rep 3 - SRR066747  \n",
    "\n",
    "Rhesus Rep 1 - SRR066751  \n",
    "Rhesus Rep 2 - SRR066752  \n",
    "Rhesus Rep 3 - SRR066753  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "f842d69e",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-03-15T14:30:24.343784Z",
     "start_time": "2022-03-15T14:09:01.332429Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Read 11546305 spots for SRR066745\n",
      "Written 11546305 spots for SRR066745\n",
      "Read 13225749 spots for SRR066746\n",
      "Written 13225749 spots for SRR066746\n",
      "Read 13693251 spots for SRR066747\n",
      "Written 13693251 spots for SRR066747\n",
      "Read 13600890 spots for SRR066751\n",
      "Written 13600890 spots for SRR066751\n",
      "Read 14354061 spots for SRR066752\n",
      "Written 14354061 spots for SRR066752\n",
      "Read 14664858 spots for SRR066753\n",
      "Written 14664858 spots for SRR066753\n"
     ]
    }
   ],
   "source": [
    "# bash kernel \n",
    "\n",
    "cd /data/hodges_lab/public_data/human-rhesus_RNA-seq/cain_2011_LCLs\n",
    "\n",
    "#don't need to split because they are SE and we are not providing a filter here. \n",
    "for ID in {066745..066747} {066751..066753}\n",
    "do\n",
    "fastq-dump --readids --dumpbase --clip SRR${ID}\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "06e432c8",
   "metadata": {},
   "source": [
    "The other tissue data has 4 replicates for each tissue type. It is also SE 50bp reads.   \n",
    "  \n",
    "Human Heart Rep 1 - SRR6900781    #this was later reclassified as liver  \n",
    "Human Kidney Rep 1 - SRR6900782    \n",
    "Human Liver Rep 1 - SRR6900783  \n",
    "Human Lung Rep 1 - SRR6900784  \n",
    "  \n",
    "Human Heart Rep 2 - SRR6900785    \n",
    "Human Kidney Rep 2 - SRR6900786    \n",
    "Human Liver Rep 2 - SRR6900787  \n",
    "Human Lung Rep 2 - SRR6900788  \n",
    "  \n",
    "Human Heart Rep 3 - SRR6900789    \n",
    "Human Kidney Rep 3 - SRR6900790   \n",
    "Human Liver Rep 3 - SRR6900791  \n",
    "Human Lung Rep 3 - SRR6900792  \n",
    "  \n",
    "Human Heart Rep 4 - SRR6900793    \n",
    "Human Kidney Rep 4 - SRR6900794   \n",
    "Human Liver Rep 4 - SRR6900795  \n",
    "Human Lung Rep 4 - SRR6900796  \n",
    "  \n",
    "------------------------------  \n",
    "  \n",
    "Macaque Heart Rep 1 - SRR6900797    \n",
    "Macaque Kidney Rep 1 - SRR6900798    \n",
    "Macaque Liver Rep 1 - SRR6900799  \n",
    "Macaque Lung Rep 1 - SRR6900800  \n",
    "  \n",
    "Macaque Heart Rep 2 - SRR6900801    \n",
    "Macaque Kidney Rep 2 - SRR6900802    \n",
    "Macaque Liver Rep 2 - SRR6900803  \n",
    "Macaque Lung Rep 2 - SRR6900804  \n",
    "  \n",
    "Macaque Heart Rep 3 - SRR6900805    \n",
    "Macaque Kidney Rep 3 - SRR6900806   \n",
    "Macaque Liver Rep 3 - SRR6900807  \n",
    "Macaque Lung Rep 3 - SRR6900808  \n",
    "  \n",
    "Macaque Heart Rep 4 - SRR6900809    \n",
    "Macaque Kidney Rep 4 - SRR6900810   \n",
    "Macaque Liver Rep 4 - SRR6900811  \n",
    "Macaque Lung Rep 4 - SRR6900812  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "949692ab",
   "metadata": {
    "ExecuteTime": {
     "start_time": "2022-03-15T14:32:54.831Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Read 49454353 spots for SRR6900781\n",
      "Written 49454353 spots for SRR6900781\n",
      "Read 51561113 spots for SRR6900782\n",
      "Written 51561113 spots for SRR6900782\n",
      "Read 52075498 spots for SRR6900783\n",
      "Written 52075498 spots for SRR6900783\n",
      "Read 37288850 spots for SRR6900784\n",
      "Written 37288850 spots for SRR6900784\n",
      "Read 54401264 spots for SRR6900785\n",
      "Written 54401264 spots for SRR6900785\n",
      "Read 39897793 spots for SRR6900786\n",
      "Written 39897793 spots for SRR6900786\n",
      "Read 45973935 spots for SRR6900787\n",
      "Written 45973935 spots for SRR6900787\n",
      "Read 55919583 spots for SRR6900788\n",
      "Written 55919583 spots for SRR6900788\n",
      "Read 44831283 spots for SRR6900789\n",
      "Written 44831283 spots for SRR6900789\n",
      "Read 53229453 spots for SRR6900790\n",
      "Written 53229453 spots for SRR6900790\n",
      "Read 56612635 spots for SRR6900791\n",
      "Written 56612635 spots for SRR6900791\n",
      "Read 54805139 spots for SRR6900792\n",
      "Written 54805139 spots for SRR6900792\n",
      "Read 53976289 spots for SRR6900793\n",
      "Written 53976289 spots for SRR6900793\n",
      "Read 62323442 spots for SRR6900794\n",
      "Written 62323442 spots for SRR6900794\n",
      "Read 57509787 spots for SRR6900795\n",
      "Written 57509787 spots for SRR6900795\n",
      "Read 66975147 spots for SRR6900796\n",
      "Written 66975147 spots for SRR6900796\n",
      "Read 36521820 spots for SRR6900797\n",
      "Written 36521820 spots for SRR6900797\n",
      "Read 34378784 spots for SRR6900798\n",
      "Written 34378784 spots for SRR6900798\n"
     ]
    }
   ],
   "source": [
    "cd /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020\n",
    "\n",
    "#I'm using code from here: https://edwards.sdsu.edu/research/fastq-dump/\n",
    "for ID in {6900781..6900812}\n",
    "do\n",
    "fastq-dump --readids --dumpbase --clip SRR${ID} \n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "9a9dfd93",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-03-16T04:42:05.421442Z",
     "start_time": "2022-03-16T01:55:47.957221Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Read 44327075 spots for SRR6900801\n",
      "Written 44327075 spots for SRR6900801\n",
      "Read 42329034 spots for SRR6900802\n",
      "Written 42329034 spots for SRR6900802\n",
      "Read 53260438 spots for SRR6900803\n",
      "Written 53260438 spots for SRR6900803\n",
      "Read 53365024 spots for SRR6900804\n",
      "Written 53365024 spots for SRR6900804\n",
      "Read 67491954 spots for SRR6900805\n",
      "Written 67491954 spots for SRR6900805\n",
      "Read 46319360 spots for SRR6900806\n",
      "Written 46319360 spots for SRR6900806\n",
      "Read 56022681 spots for SRR6900807\n",
      "Written 56022681 spots for SRR6900807\n",
      "Read 62067186 spots for SRR6900808\n",
      "Written 62067186 spots for SRR6900808\n",
      "Read 48437093 spots for SRR6900809\n",
      "Written 48437093 spots for SRR6900809\n",
      "Read 54857579 spots for SRR6900810\n",
      "Written 54857579 spots for SRR6900810\n",
      "Read 73133582 spots for SRR6900811\n",
      "Written 73133582 spots for SRR6900811\n",
      "Read 61868144 spots for SRR6900812\n",
      "Written 61868144 spots for SRR6900812\n"
     ]
    }
   ],
   "source": [
    "#the process above was terminated at file 801. Restart from 801. \n",
    "cd /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020\n",
    "\n",
    "#I'm using code from here: https://edwards.sdsu.edu/research/fastq-dump/\n",
    "for ID in {6900801..6900812}\n",
    "do\n",
    "fastq-dump --readids --dumpbase --clip SRR${ID} \n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "28c46796",
   "metadata": {},
   "source": [
    "## Trimming"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "d87495b6",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-03-16T13:52:39.166664Z",
     "start_time": "2022-03-16T13:46:10.355008Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Using an excessive number of cores has a diminishing return! It is recommended not to exceed 8 cores per trimming process (you asked for 8 cores). Please consider re-specifying\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "Cutadapt version: 1.18\n",
      "Could not detect version of Python used by Cutadapt from the first line of Cutadapt (but found this: >>>#!/bin/sh<<<)\n",
      "Letting the (modified) Cutadapt deal with the Python version instead\n",
      "pigz 2.6\n",
      "Parallel gzip (pigz) detected. Proceeding with multicore (de)compression using 8 cores\n",
      "\n",
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Output will be written into the directory: /gpfs52/data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/trimmed_reads/\n",
      "Using user-specified basename (>>Macaque_Heart_R2<<) instead of deriving the filename from the input file(s)\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900801.fastq <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Illumina\t1877\tAGATCGGAAGAGC\t1000000\t0.19\n",
      "smallRNA\t0\tTGGAATTCTCGG\t1000000\t0.00\n",
      "Nextera\t0\tCTGTCTCTTATA\t1000000\t0.00\n",
      "Using Illumina adapter for trimming (count: 1877). Second best hit was smallRNA (count: 0)\n",
      "\n",
      "Writing report to '/gpfs52/data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/trimmed_reads/SRR6900801.fastq_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900801.fastq\n",
      "Trimming mode: single-end\n",
      "Trim Galore version: 0.6.6\n",
      "Cutadapt version: 1.18\n",
      "Python version: could not detect\n",
      "Number of cores used for trimming: 8\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length before a sequence gets removed: 20 bp\n",
      "Running FastQC on the data once trimming has completed\n",
      "Running FastQC with the following extra arguments: '--outdir /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/processing_results/fastQC'\n",
      "\n",
      "Cutadapt seems to be reasonably up-to-date. Setting -j 8\n",
      "Writing final adapter and quality trimmed output to SRR6900801_trimmed.fq\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff '-q 20') and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900801.fastq <<< \n",
      "10000000 sequences processed\n",
      "20000000 sequences processed\n",
      "30000000 sequences processed\n",
      "40000000 sequences processed\n",
      "This is cutadapt 1.18 with Python 3.6.15\n",
      "Command line parameters: -j 8 -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900801.fastq\n",
      "Processing reads on 8 cores in single-end mode ...\n",
      "Finished in 176.21 s (4 us/read; 15.09 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              44,327,075\n",
      "Reads with adapters:                 3,442,825 (7.8%)\n",
      "Reads written (passing filters):    44,327,075 (100.0%)\n",
      "\n",
      "Total basepairs processed: 2,252,575,254 bp\n",
      "Quality-trimmed:             315,955,457 bp (14.0%)\n",
      "Total written (filtered):  1,920,515,472 bp (85.3%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 3442825 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 27.1%\n",
      "  C: 26.1%\n",
      "  G: 18.6%\n",
      "  T: 21.3%\n",
      "  none/other: 6.9%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t2319842\t11081768.8\t0\t2319842\n",
      "2\t659740\t2770442.2\t0\t659740\n",
      "3\t175775\t692610.5\t0\t175775\n",
      "4\t39287\t173152.6\t0\t39287\n",
      "5\t7207\t43288.2\t0\t7207\n",
      "6\t981\t10822.0\t0\t981\n",
      "7\t286\t2705.5\t0\t286\n",
      "8\t181\t676.4\t0\t181\n",
      "9\t308\t169.1\t0\t192 116\n",
      "10\t578\t42.3\t1\t87 491\n",
      "11\t587\t10.6\t1\t139 448\n",
      "12\t535\t2.6\t1\t91 444\n",
      "13\t358\t0.7\t1\t115 243\n",
      "14\t412\t0.7\t1\t364 48\n",
      "15\t345\t0.7\t1\t306 39\n",
      "16\t417\t0.7\t1\t369 48\n",
      "17\t328\t0.7\t1\t289 39\n",
      "18\t505\t0.7\t1\t436 69\n",
      "19\t303\t0.7\t1\t275 28\n",
      "20\t318\t0.7\t1\t271 47\n",
      "21\t304\t0.7\t1\t270 34\n",
      "22\t314\t0.7\t1\t274 40\n",
      "23\t333\t0.7\t1\t297 36\n",
      "24\t357\t0.7\t1\t324 33\n",
      "25\t365\t0.7\t1\t324 41\n",
      "26\t414\t0.7\t1\t364 50\n",
      "27\t368\t0.7\t1\t333 35\n",
      "28\t500\t0.7\t1\t420 80\n",
      "29\t468\t0.7\t1\t418 50\n",
      "30\t791\t0.7\t1\t696 95\n",
      "31\t693\t0.7\t1\t633 60\n",
      "32\t509\t0.7\t1\t464 45\n",
      "33\t501\t0.7\t1\t430 71\n",
      "34\t613\t0.7\t1\t551 62\n",
      "35\t656\t0.7\t1\t568 88\n",
      "36\t652\t0.7\t1\t579 73\n",
      "37\t575\t0.7\t1\t512 63\n",
      "38\t586\t0.7\t1\t507 79\n",
      "39\t710\t0.7\t1\t555 155\n",
      "40\t869\t0.7\t1\t641 228\n",
      "41\t684\t0.7\t1\t519 165\n",
      "42\t751\t0.7\t1\t591 160\n",
      "43\t576\t0.7\t1\t411 165\n",
      "44\t713\t0.7\t1\t508 205\n",
      "45\t731\t0.7\t1\t446 285\n",
      "46\t1317\t0.7\t1\t652 665\n",
      "47\t3398\t0.7\t1\t2077 1321\n",
      "48\t3921\t0.7\t1\t2357 1564\n",
      "49\t17060\t0.7\t1\t9916 7144\n",
      "50\t1549\t0.7\t1\t1123 426\n",
      "51\t193254\t0.7\t1\t133913 59341\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900801.fastq\n",
      "=============================================\n",
      "44327075 sequences processed in total\n",
      "Sequences removed because they became shorter than the length cutoff of 20 bp:\t6160155 (13.9%)\n",
      "\n",
      "Now renaming the output file SRR6900801_trimmed.fq\n",
      "\n",
      "ORIGINAL FILE 1: >>SRR6900801_trimmed.fq<<\tRENAMING TO:>>Macaque_Heart_R2_trimmed.fq<<\n",
      "\n",
      "  >>> Now running FastQC on the data <<<\n",
      "\n",
      "Started analysis of Macaque_Heart_R2_trimmed.fq\n",
      "Approx 5% complete for Macaque_Heart_R2_trimmed.fq\n",
      "Approx 10% complete for Macaque_Heart_R2_trimmed.fq\n",
      "Approx 15% complete for Macaque_Heart_R2_trimmed.fq\n",
      "Approx 20% complete for Macaque_Heart_R2_trimmed.fq\n",
      "Approx 25% complete for Macaque_Heart_R2_trimmed.fq\n",
      "Approx 30% complete for Macaque_Heart_R2_trimmed.fq\n",
      "Approx 35% complete for Macaque_Heart_R2_trimmed.fq\n",
      "Approx 40% complete for Macaque_Heart_R2_trimmed.fq\n",
      "Approx 45% complete for Macaque_Heart_R2_trimmed.fq\n",
      "Approx 50% complete for Macaque_Heart_R2_trimmed.fq\n",
      "Approx 55% complete for Macaque_Heart_R2_trimmed.fq\n",
      "Approx 60% complete for Macaque_Heart_R2_trimmed.fq\n",
      "Approx 65% complete for Macaque_Heart_R2_trimmed.fq\n",
      "Approx 70% complete for Macaque_Heart_R2_trimmed.fq\n",
      "Approx 75% complete for Macaque_Heart_R2_trimmed.fq\n",
      "Approx 80% complete for Macaque_Heart_R2_trimmed.fq\n",
      "Approx 85% complete for Macaque_Heart_R2_trimmed.fq\n",
      "Approx 90% complete for Macaque_Heart_R2_trimmed.fq\n",
      "Approx 95% complete for Macaque_Heart_R2_trimmed.fq\n",
      "Analysis complete for Macaque_Heart_R2_trimmed.fq\n"
     ]
    }
   ],
   "source": [
    "#tester\n",
    "\n",
    "DIR='/data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020'\n",
    "\n",
    "trim_galore --fastqc --fastqc_args \"--outdir ${DIR}/processing_results/fastQC\" \\\n",
    "    --dont_gzip --basename Macaque_Heart_R2 --output_dir ${DIR}/fastq/trimmed_reads \\\n",
    "    ${DIR}/fastq/SRR6900801.fastq -j 8"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "bfc2b451",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-03-16T13:55:12.875392Z",
     "start_time": "2022-03-16T13:55:12.706755Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Human_LCL_R1 SRR066745\n",
      "Human_LCL_R2 SRR066746\n",
      "Human_LCL_R3 SRR066747\n",
      "Rhesus_LCL_R1 SRR066751\n",
      "Rhesus_LCL_R2 SRR066752\n",
      "Rhesus_LCL_R3 SRR066753\n"
     ]
    }
   ],
   "source": [
    "#show file assignments \n",
    "cat /data/hodges_lab/public_data/human-rhesus_RNA-seq/LCL_names.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "95404f2b",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-03-17T13:48:54.644217Z",
     "start_time": "2022-03-17T13:48:54.522406Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Human_Heart_R1 SRR6900781\n",
      "Human_Kidney_R1 SRR6900782\n",
      "Human_Liver_R1 SRR6900783\n",
      "Human_Lung_R1 SRR6900784\n",
      "Human_Heart_R2 SRR6900785\n",
      "Human_Kidney_R2 SRR6900786\n",
      "Human_Liver_R2 SRR6900787\n",
      "Human_Lung_R2 SRR6900788\n",
      "Human_Heart_R3 SRR6900789\n",
      "Human_Kidney_R3 SRR6900790\n",
      "Human_Liver_R3 SRR6900791\n",
      "Human_Lung_R3 SRR6900792\n",
      "Human_Heart_R4 SRR6900793\n",
      "Human_Kidney_R4 SRR6900794\n",
      "Human_Liver_R4 SRR6900795\n",
      "Human_Lung_R4 SRR6900796\n",
      "Macaque_Heart_R1 SRR6900797\n",
      "Macaque_Kidney_R1 SRR6900798\n",
      "Macaque_Liver_R1 SRR6900799\n",
      "Macaque_Lung_R1 SRR6900800\n",
      "Macaque_Heart_R2 SRR6900801\n",
      "Macaque_Kidney_R2 SRR6900802\n",
      "Macaque_Liver_R2 SRR6900803\n",
      "Macaque_Lung_R2 SRR6900804\n",
      "Macaque_Heart_R3 SRR6900805\n",
      "Macaque_Kidney_R3 SRR6900806\n",
      "Macaque_Liver_R3 SRR6900807\n",
      "Macaque_Lung_R3 SRR6900808\n",
      "Macaque_Heart_R4 SRR6900809\n",
      "Macaque_Kidney_R4 SRR6900810\n",
      "Macaque_Liver_R4 SRR6900811\n",
      "Macaque_Lung_R4 SRR6900812\n"
     ]
    }
   ],
   "source": [
    "cat /data/hodges_lab/public_data/human-rhesus_RNA-seq/tissues_names.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "5c13ee9d",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-03-16T14:19:51.082713Z",
     "start_time": "2022-03-16T14:06:15.701067Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Using an excessive number of cores has a diminishing return! It is recommended not to exceed 8 cores per trimming process (you asked for 8 cores). Please consider re-specifying\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "Cutadapt version: 1.18\n",
      "Could not detect version of Python used by Cutadapt from the first line of Cutadapt (but found this: >>>#!/bin/sh<<<)\n",
      "Letting the (modified) Cutadapt deal with the Python version instead\n",
      "pigz 2.6\n",
      "Parallel gzip (pigz) detected. Proceeding with multicore (de)compression using 8 cores\n",
      "\n",
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Output will be written into the directory: /gpfs52/data/hodges_lab/public_data/human-rhesus_RNA-seq/cain_2011_LCLs/fastq/trimmed_reads/\n",
      "Using user-specified basename (>>Human_LCL_R1<<) instead of deriving the filename from the input file(s)\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> /data/hodges_lab/public_data/human-rhesus_RNA-seq/cain_2011_LCLs/fastq/SRR066745.fastq <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Illumina\t299\tAGATCGGAAGAGC\t1000000\t0.03\n",
      "smallRNA\t6\tTGGAATTCTCGG\t1000000\t0.00\n",
      "Nextera\t1\tCTGTCTCTTATA\t1000000\t0.00\n",
      "Using Illumina adapter for trimming (count: 299). Second best hit was smallRNA (count: 6)\n",
      "\n",
      "Writing report to '/gpfs52/data/hodges_lab/public_data/human-rhesus_RNA-seq/cain_2011_LCLs/fastq/trimmed_reads/SRR066745.fastq_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: /data/hodges_lab/public_data/human-rhesus_RNA-seq/cain_2011_LCLs/fastq/SRR066745.fastq\n",
      "Trimming mode: single-end\n",
      "Trim Galore version: 0.6.6\n",
      "Cutadapt version: 1.18\n",
      "Python version: could not detect\n",
      "Number of cores used for trimming: 8\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length before a sequence gets removed: 20 bp\n",
      "Running FastQC on the data once trimming has completed\n",
      "Running FastQC with the following extra arguments: '--outdir /data/hodges_lab/public_data/human-rhesus_RNA-seq/cain_2011_LCLs/processing_results/fastQC'\n",
      "\n",
      "Cutadapt seems to be reasonably up-to-date. Setting -j 8\n",
      "Writing final adapter and quality trimmed output to SRR066745_trimmed.fq\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff '-q 20') and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file /data/hodges_lab/public_data/human-rhesus_RNA-seq/cain_2011_LCLs/fastq/SRR066745.fastq <<< \n",
      "10000000 sequences processed\n",
      "This is cutadapt 1.18 with Python 3.6.15\n",
      "Command line parameters: -j 8 -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC /data/hodges_lab/public_data/human-rhesus_RNA-seq/cain_2011_LCLs/fastq/SRR066745.fastq\n",
      "Processing reads on 8 cores in single-end mode ...\n",
      "Finished in 45.95 s (4 us/read; 15.08 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              11,546,305\n",
      "Reads with adapters:                 3,485,153 (30.2%)\n",
      "Reads written (passing filters):    11,546,305 (100.0%)\n",
      "\n",
      "Total basepairs processed:   415,666,980 bp\n",
      "Quality-trimmed:               3,475,969 bp (0.8%)\n",
      "Total written (filtered):    406,416,985 bp (97.8%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 3485153 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 26.7%\n",
      "  C: 35.8%\n",
      "  G: 23.7%\n",
      "  T: 13.2%\n",
      "  none/other: 0.6%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t2298540\t2886576.2\t0\t2298540\n",
      "2\t875533\t721644.1\t0\t875533\n",
      "3\t218720\t180411.0\t0\t218720\n",
      "4\t46210\t45102.8\t0\t46210\n",
      "5\t13444\t11275.7\t0\t13444\n",
      "6\t3562\t2818.9\t0\t3562\n",
      "7\t1807\t704.7\t0\t1807\n",
      "8\t1223\t176.2\t0\t1223\n",
      "9\t1093\t44.0\t0\t875 218\n",
      "10\t930\t11.0\t1\t598 332\n",
      "11\t385\t2.8\t1\t192 193\n",
      "12\t234\t0.7\t1\t137 97\n",
      "13\t144\t0.2\t1\t133 11\n",
      "14\t302\t0.2\t1\t281 21\n",
      "15\t214\t0.2\t1\t191 23\n",
      "16\t101\t0.2\t1\t84 17\n",
      "17\t65\t0.2\t1\t50 15\n",
      "18\t24\t0.2\t1\t12 12\n",
      "19\t29\t0.2\t1\t11 18\n",
      "20\t26\t0.2\t1\t9 17\n",
      "21\t47\t0.2\t1\t12 35\n",
      "22\t64\t0.2\t1\t33 31\n",
      "23\t38\t0.2\t1\t20 18\n",
      "24\t31\t0.2\t1\t11 20\n",
      "25\t36\t0.2\t1\t9 27\n",
      "26\t86\t0.2\t1\t23 63\n",
      "27\t2860\t0.2\t1\t2545 315\n",
      "28\t134\t0.2\t1\t24 110\n",
      "29\t221\t0.2\t1\t11 210\n",
      "30\t922\t0.2\t1\t5 917\n",
      "31\t4337\t0.2\t1\t1 4336\n",
      "32\t4296\t0.2\t1\t7 4289\n",
      "33\t416\t0.2\t1\t7 409\n",
      "34\t281\t0.2\t1\t7 274\n",
      "35\t1438\t0.2\t1\t4 1434\n",
      "36\t7360\t0.2\t1\t4 7356\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: /data/hodges_lab/public_data/human-rhesus_RNA-seq/cain_2011_LCLs/fastq/SRR066745.fastq\n",
      "=============================================\n",
      "11546305 sequences processed in total\n",
      "Sequences removed because they became shorter than the length cutoff of 20 bp:\t67046 (0.6%)\n",
      "\n",
      "Now renaming the output file SRR066745_trimmed.fq\n",
      "\n",
      "ORIGINAL FILE 1: >>SRR066745_trimmed.fq<<\tRENAMING TO:>>Human_LCL_R1_trimmed.fq<<\n",
      "\n",
      "  >>> Now running FastQC on the data <<<\n",
      "\n",
      "Started analysis of Human_LCL_R1_trimmed.fq\n",
      "Approx 5% complete for Human_LCL_R1_trimmed.fq\n",
      "Approx 10% complete for Human_LCL_R1_trimmed.fq\n",
      "Approx 15% complete for Human_LCL_R1_trimmed.fq\n",
      "Approx 20% complete for Human_LCL_R1_trimmed.fq\n",
      "Approx 25% complete for Human_LCL_R1_trimmed.fq\n",
      "Approx 30% complete for Human_LCL_R1_trimmed.fq\n",
      "Approx 35% complete for Human_LCL_R1_trimmed.fq\n",
      "Approx 40% complete for Human_LCL_R1_trimmed.fq\n",
      "Approx 45% complete for Human_LCL_R1_trimmed.fq\n",
      "Approx 50% complete for Human_LCL_R1_trimmed.fq\n",
      "Approx 55% complete for Human_LCL_R1_trimmed.fq\n",
      "Approx 60% complete for Human_LCL_R1_trimmed.fq\n",
      "Approx 65% complete for Human_LCL_R1_trimmed.fq\n",
      "Approx 70% complete for Human_LCL_R1_trimmed.fq\n",
      "Approx 75% complete for Human_LCL_R1_trimmed.fq\n",
      "Approx 80% complete for Human_LCL_R1_trimmed.fq\n",
      "Approx 85% complete for Human_LCL_R1_trimmed.fq\n",
      "Approx 90% complete for Human_LCL_R1_trimmed.fq\n",
      "Approx 95% complete for Human_LCL_R1_trimmed.fq\n",
      "Analysis complete for Human_LCL_R1_trimmed.fq\n",
      "Using an excessive number of cores has a diminishing return! It is recommended not to exceed 8 cores per trimming process (you asked for 8 cores). Please consider re-specifying\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "Cutadapt version: 1.18\n",
      "Could not detect version of Python used by Cutadapt from the first line of Cutadapt (but found this: >>>#!/bin/sh<<<)\n",
      "Letting the (modified) Cutadapt deal with the Python version instead\n",
      "pigz 2.6\n",
      "Parallel gzip (pigz) detected. Proceeding with multicore (de)compression using 8 cores\n",
      "\n",
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Output will be written into the directory: /gpfs52/data/hodges_lab/public_data/human-rhesus_RNA-seq/cain_2011_LCLs/fastq/trimmed_reads/\n",
      "Using user-specified basename (>>Human_LCL_R2<<) instead of deriving the filename from the input file(s)\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> /data/hodges_lab/public_data/human-rhesus_RNA-seq/cain_2011_LCLs/fastq/SRR066746.fastq <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Illumina\t528\tAGATCGGAAGAGC\t1000000\t0.05\n",
      "smallRNA\t8\tTGGAATTCTCGG\t1000000\t0.00\n",
      "Nextera\t0\tCTGTCTCTTATA\t1000000\t0.00\n",
      "Using Illumina adapter for trimming (count: 528). Second best hit was smallRNA (count: 8)\n",
      "\n",
      "Writing report to '/gpfs52/data/hodges_lab/public_data/human-rhesus_RNA-seq/cain_2011_LCLs/fastq/trimmed_reads/SRR066746.fastq_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: /data/hodges_lab/public_data/human-rhesus_RNA-seq/cain_2011_LCLs/fastq/SRR066746.fastq\n",
      "Trimming mode: single-end\n",
      "Trim Galore version: 0.6.6\n",
      "Cutadapt version: 1.18\n",
      "Python version: could not detect\n",
      "Number of cores used for trimming: 8\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length before a sequence gets removed: 20 bp\n",
      "Running FastQC on the data once trimming has completed\n",
      "Running FastQC with the following extra arguments: '--outdir /data/hodges_lab/public_data/human-rhesus_RNA-seq/cain_2011_LCLs/processing_results/fastQC'\n",
      "\n",
      "Cutadapt seems to be reasonably up-to-date. Setting -j 8\n",
      "Writing final adapter and quality trimmed output to SRR066746_trimmed.fq\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff '-q 20') and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file /data/hodges_lab/public_data/human-rhesus_RNA-seq/cain_2011_LCLs/fastq/SRR066746.fastq <<< \n",
      "10000000 sequences processed\n",
      "This is cutadapt 1.18 with Python 3.6.15\n",
      "Command line parameters: -j 8 -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC /data/hodges_lab/public_data/human-rhesus_RNA-seq/cain_2011_LCLs/fastq/SRR066746.fastq\n",
      "Processing reads on 8 cores in single-end mode ...\n",
      "Finished in 53.78 s (4 us/read; 14.76 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              13,225,749\n",
      "Reads with adapters:                 4,117,906 (31.1%)\n",
      "Reads written (passing filters):    13,225,749 (100.0%)\n",
      "\n",
      "Total basepairs processed:   476,126,964 bp\n",
      "Quality-trimmed:               4,142,505 bp (0.9%)\n",
      "Total written (filtered):    465,210,860 bp (97.7%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 4117906 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 26.7%\n",
      "  C: 35.2%\n",
      "  G: 24.2%\n",
      "  T: 13.6%\n",
      "  none/other: 0.4%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t2678593\t3306437.2\t0\t2678593\n",
      "2\t1031614\t826609.3\t0\t1031614\n",
      "3\t259953\t206652.3\t0\t259953\n",
      "4\t65029\t51663.1\t0\t65029\n",
      "5\t24880\t12915.8\t0\t24880\n",
      "6\t12073\t3228.9\t0\t12073\n",
      "7\t8615\t807.2\t0\t8615\n",
      "8\t6324\t201.8\t0\t6324\n",
      "9\t4397\t50.5\t0\t4136 261\n",
      "10\t3575\t12.6\t1\t3140 435\n",
      "11\t1376\t3.2\t1\t1130 246\n",
      "12\t965\t0.8\t1\t813 152\n",
      "13\t878\t0.2\t1\t832 46\n",
      "14\t1985\t0.2\t1\t1902 83\n",
      "15\t1197\t0.2\t1\t1140 57\n",
      "16\t420\t0.2\t1\t397 23\n",
      "17\t137\t0.2\t1\t118 19\n",
      "18\t94\t0.2\t1\t70 24\n",
      "19\t48\t0.2\t1\t28 20\n",
      "20\t34\t0.2\t1\t20 14\n",
      "21\t43\t0.2\t1\t12 31\n",
      "22\t43\t0.2\t1\t21 22\n",
      "23\t37\t0.2\t1\t18 19\n",
      "24\t27\t0.2\t1\t7 20\n",
      "25\t31\t0.2\t1\t7 24\n",
      "26\t88\t0.2\t1\t18 70\n",
      "27\t1727\t0.2\t1\t1523 204\n",
      "28\t126\t0.2\t1\t23 103\n",
      "29\t197\t0.2\t1\t6 191\n",
      "30\t621\t0.2\t1\t9 612\n",
      "31\t2757\t0.2\t1\t7 2750\n",
      "32\t3062\t0.2\t1\t6 3056\n",
      "33\t246\t0.2\t1\t1 245\n",
      "34\t223\t0.2\t1\t6 217\n",
      "35\t1201\t0.2\t1\t2 1199\n",
      "36\t5290\t0.2\t1\t6 5284\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: /data/hodges_lab/public_data/human-rhesus_RNA-seq/cain_2011_LCLs/fastq/SRR066746.fastq\n",
      "=============================================\n",
      "13225749 sequences processed in total\n",
      "Sequences removed because they became shorter than the length cutoff of 20 bp:\t74756 (0.6%)\n",
      "\n",
      "Now renaming the output file SRR066746_trimmed.fq\n",
      "\n",
      "ORIGINAL FILE 1: >>SRR066746_trimmed.fq<<\tRENAMING TO:>>Human_LCL_R2_trimmed.fq<<\n",
      "\n",
      "  >>> Now running FastQC on the data <<<\n",
      "\n",
      "Started analysis of Human_LCL_R2_trimmed.fq\n",
      "Approx 5% complete for Human_LCL_R2_trimmed.fq\n",
      "Approx 10% complete for Human_LCL_R2_trimmed.fq\n",
      "Approx 15% complete for Human_LCL_R2_trimmed.fq\n",
      "Approx 20% complete for Human_LCL_R2_trimmed.fq\n",
      "Approx 25% complete for Human_LCL_R2_trimmed.fq\n",
      "Approx 30% complete for Human_LCL_R2_trimmed.fq\n",
      "Approx 35% complete for Human_LCL_R2_trimmed.fq\n",
      "Approx 40% complete for Human_LCL_R2_trimmed.fq\n",
      "Approx 45% complete for Human_LCL_R2_trimmed.fq\n",
      "Approx 50% complete for Human_LCL_R2_trimmed.fq\n",
      "Approx 55% complete for Human_LCL_R2_trimmed.fq\n",
      "Approx 60% complete for Human_LCL_R2_trimmed.fq\n",
      "Approx 65% complete for Human_LCL_R2_trimmed.fq\n",
      "Approx 70% complete for Human_LCL_R2_trimmed.fq\n",
      "Approx 75% complete for Human_LCL_R2_trimmed.fq\n",
      "Approx 80% complete for Human_LCL_R2_trimmed.fq\n",
      "Approx 85% complete for Human_LCL_R2_trimmed.fq\n",
      "Approx 90% complete for Human_LCL_R2_trimmed.fq\n",
      "Approx 95% complete for Human_LCL_R2_trimmed.fq\n",
      "Analysis complete for Human_LCL_R2_trimmed.fq\n",
      "Using an excessive number of cores has a diminishing return! It is recommended not to exceed 8 cores per trimming process (you asked for 8 cores). Please consider re-specifying\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "Cutadapt version: 1.18\n",
      "Could not detect version of Python used by Cutadapt from the first line of Cutadapt (but found this: >>>#!/bin/sh<<<)\n",
      "Letting the (modified) Cutadapt deal with the Python version instead\n",
      "pigz 2.6\n",
      "Parallel gzip (pigz) detected. Proceeding with multicore (de)compression using 8 cores\n",
      "\n",
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Output will be written into the directory: /gpfs52/data/hodges_lab/public_data/human-rhesus_RNA-seq/cain_2011_LCLs/fastq/trimmed_reads/\n",
      "Using user-specified basename (>>Human_LCL_R3<<) instead of deriving the filename from the input file(s)\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> /data/hodges_lab/public_data/human-rhesus_RNA-seq/cain_2011_LCLs/fastq/SRR066747.fastq <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Illumina\t641\tAGATCGGAAGAGC\t1000000\t0.06\n",
      "smallRNA\t2\tTGGAATTCTCGG\t1000000\t0.00\n",
      "Nextera\t0\tCTGTCTCTTATA\t1000000\t0.00\n",
      "Using Illumina adapter for trimming (count: 641). Second best hit was smallRNA (count: 2)\n",
      "\n",
      "Writing report to '/gpfs52/data/hodges_lab/public_data/human-rhesus_RNA-seq/cain_2011_LCLs/fastq/trimmed_reads/SRR066747.fastq_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: /data/hodges_lab/public_data/human-rhesus_RNA-seq/cain_2011_LCLs/fastq/SRR066747.fastq\n",
      "Trimming mode: single-end\n",
      "Trim Galore version: 0.6.6\n",
      "Cutadapt version: 1.18\n",
      "Python version: could not detect\n",
      "Number of cores used for trimming: 8\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length before a sequence gets removed: 20 bp\n",
      "Running FastQC on the data once trimming has completed\n",
      "Running FastQC with the following extra arguments: '--outdir /data/hodges_lab/public_data/human-rhesus_RNA-seq/cain_2011_LCLs/processing_results/fastQC'\n",
      "\n",
      "Cutadapt seems to be reasonably up-to-date. Setting -j 8\n",
      "Writing final adapter and quality trimmed output to SRR066747_trimmed.fq\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff '-q 20') and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file /data/hodges_lab/public_data/human-rhesus_RNA-seq/cain_2011_LCLs/fastq/SRR066747.fastq <<< \n",
      "10000000 sequences processed\n",
      "This is cutadapt 1.18 with Python 3.6.15\n",
      "Command line parameters: -j 8 -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC /data/hodges_lab/public_data/human-rhesus_RNA-seq/cain_2011_LCLs/fastq/SRR066747.fastq\n",
      "Processing reads on 8 cores in single-end mode ...\n",
      "Finished in 87.32 s (6 us/read; 9.41 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              13,693,251\n",
      "Reads with adapters:                 4,100,625 (29.9%)\n",
      "Reads written (passing filters):    13,693,251 (100.0%)\n",
      "\n",
      "Total basepairs processed:   492,957,036 bp\n",
      "Quality-trimmed:               4,128,535 bp (0.8%)\n",
      "Total written (filtered):    482,408,280 bp (97.9%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 4100625 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 27.0%\n",
      "  C: 35.1%\n",
      "  G: 24.1%\n",
      "  T: 13.7%\n",
      "  none/other: 0.2%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t2692751\t3423312.8\t0\t2692751\n",
      "2\t1045614\t855828.2\t0\t1045614\n",
      "3\t248286\t213957.0\t0\t248286\n",
      "4\t56285\t53489.3\t0\t56285\n",
      "5\t20569\t13372.3\t0\t20569\n",
      "6\t7124\t3343.1\t0\t7124\n",
      "7\t5024\t835.8\t0\t5024\n",
      "8\t3586\t208.9\t0\t3586\n",
      "9\t2517\t52.2\t0\t2322 195\n",
      "10\t2335\t13.1\t1\t2036 299\n",
      "11\t1144\t3.3\t1\t922 222\n",
      "12\t742\t0.8\t1\t642 100\n",
      "13\t620\t0.2\t1\t587 33\n",
      "14\t1366\t0.2\t1\t1301 65\n",
      "15\t941\t0.2\t1\t901 40\n",
      "16\t289\t0.2\t1\t258 31\n",
      "17\t128\t0.2\t1\t99 29\n",
      "18\t64\t0.2\t1\t50 14\n",
      "19\t50\t0.2\t1\t29 21\n",
      "20\t22\t0.2\t1\t11 11\n",
      "21\t29\t0.2\t1\t12 17\n",
      "22\t34\t0.2\t1\t16 18\n",
      "23\t55\t0.2\t1\t33 22\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "24\t24\t0.2\t1\t9 15\n",
      "25\t25\t0.2\t1\t9 16\n",
      "26\t70\t0.2\t1\t30 40\n",
      "27\t4959\t0.2\t1\t4475 484\n",
      "28\t119\t0.2\t1\t64 55\n",
      "29\t107\t0.2\t1\t16 91\n",
      "30\t317\t0.2\t1\t7 310\n",
      "31\t1156\t0.2\t1\t7 1149\n",
      "32\t1266\t0.2\t1\t10 1256\n",
      "33\t203\t0.2\t1\t4 199\n",
      "34\t135\t0.2\t1\t9 126\n",
      "35\t535\t0.2\t1\t10 525\n",
      "36\t2134\t0.2\t1\t2 2132\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: /data/hodges_lab/public_data/human-rhesus_RNA-seq/cain_2011_LCLs/fastq/SRR066747.fastq\n",
      "=============================================\n",
      "13693251 sequences processed in total\n",
      "Sequences removed because they became shorter than the length cutoff of 20 bp:\t63942 (0.5%)\n",
      "\n",
      "Now renaming the output file SRR066747_trimmed.fq\n",
      "\n",
      "ORIGINAL FILE 1: >>SRR066747_trimmed.fq<<\tRENAMING TO:>>Human_LCL_R3_trimmed.fq<<\n",
      "\n",
      "  >>> Now running FastQC on the data <<<\n",
      "\n",
      "Started analysis of Human_LCL_R3_trimmed.fq\n",
      "Approx 5% complete for Human_LCL_R3_trimmed.fq\n",
      "Approx 10% complete for Human_LCL_R3_trimmed.fq\n",
      "Approx 15% complete for Human_LCL_R3_trimmed.fq\n",
      "Approx 20% complete for Human_LCL_R3_trimmed.fq\n",
      "Approx 25% complete for Human_LCL_R3_trimmed.fq\n",
      "Approx 30% complete for Human_LCL_R3_trimmed.fq\n",
      "Approx 35% complete for Human_LCL_R3_trimmed.fq\n",
      "Approx 40% complete for Human_LCL_R3_trimmed.fq\n",
      "Approx 45% complete for Human_LCL_R3_trimmed.fq\n",
      "Approx 50% complete for Human_LCL_R3_trimmed.fq\n",
      "Approx 55% complete for Human_LCL_R3_trimmed.fq\n",
      "Approx 60% complete for Human_LCL_R3_trimmed.fq\n",
      "Approx 65% complete for Human_LCL_R3_trimmed.fq\n",
      "Approx 70% complete for Human_LCL_R3_trimmed.fq\n",
      "Approx 75% complete for Human_LCL_R3_trimmed.fq\n",
      "Approx 80% complete for Human_LCL_R3_trimmed.fq\n",
      "Approx 85% complete for Human_LCL_R3_trimmed.fq\n",
      "Approx 90% complete for Human_LCL_R3_trimmed.fq\n",
      "Approx 95% complete for Human_LCL_R3_trimmed.fq\n",
      "Analysis complete for Human_LCL_R3_trimmed.fq\n",
      "Using an excessive number of cores has a diminishing return! It is recommended not to exceed 8 cores per trimming process (you asked for 8 cores). Please consider re-specifying\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "Cutadapt version: 1.18\n",
      "Could not detect version of Python used by Cutadapt from the first line of Cutadapt (but found this: >>>#!/bin/sh<<<)\n",
      "Letting the (modified) Cutadapt deal with the Python version instead\n",
      "pigz 2.6\n",
      "Parallel gzip (pigz) detected. Proceeding with multicore (de)compression using 8 cores\n",
      "\n",
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Output will be written into the directory: /gpfs52/data/hodges_lab/public_data/human-rhesus_RNA-seq/cain_2011_LCLs/fastq/trimmed_reads/\n",
      "Using user-specified basename (>>Rhesus_LCL_R1<<) instead of deriving the filename from the input file(s)\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> /data/hodges_lab/public_data/human-rhesus_RNA-seq/cain_2011_LCLs/fastq/SRR066751.fastq <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Illumina\t167\tAGATCGGAAGAGC\t1000000\t0.02\n",
      "smallRNA\t1\tTGGAATTCTCGG\t1000000\t0.00\n",
      "Nextera\t1\tCTGTCTCTTATA\t1000000\t0.00\n",
      "Using Illumina adapter for trimming (count: 167). Second best hit was smallRNA (count: 1)\n",
      "\n",
      "Writing report to '/gpfs52/data/hodges_lab/public_data/human-rhesus_RNA-seq/cain_2011_LCLs/fastq/trimmed_reads/SRR066751.fastq_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: /data/hodges_lab/public_data/human-rhesus_RNA-seq/cain_2011_LCLs/fastq/SRR066751.fastq\n",
      "Trimming mode: single-end\n",
      "Trim Galore version: 0.6.6\n",
      "Cutadapt version: 1.18\n",
      "Python version: could not detect\n",
      "Number of cores used for trimming: 8\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length before a sequence gets removed: 20 bp\n",
      "Running FastQC on the data once trimming has completed\n",
      "Running FastQC with the following extra arguments: '--outdir /data/hodges_lab/public_data/human-rhesus_RNA-seq/cain_2011_LCLs/processing_results/fastQC'\n",
      "\n",
      "Cutadapt seems to be reasonably up-to-date. Setting -j 8\n",
      "Writing final adapter and quality trimmed output to SRR066751_trimmed.fq\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff '-q 20') and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file /data/hodges_lab/public_data/human-rhesus_RNA-seq/cain_2011_LCLs/fastq/SRR066751.fastq <<< \n",
      "10000000 sequences processed\n",
      "This is cutadapt 1.18 with Python 3.6.15\n",
      "Command line parameters: -j 8 -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC /data/hodges_lab/public_data/human-rhesus_RNA-seq/cain_2011_LCLs/fastq/SRR066751.fastq\n",
      "Processing reads on 8 cores in single-end mode ...\n",
      "Finished in 54.88 s (4 us/read; 14.87 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              13,600,890\n",
      "Reads with adapters:                 4,204,039 (30.9%)\n",
      "Reads written (passing filters):    13,600,890 (100.0%)\n",
      "\n",
      "Total basepairs processed:   489,632,040 bp\n",
      "Quality-trimmed:               5,417,279 bp (1.1%)\n",
      "Total written (filtered):    478,026,160 bp (97.6%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 4204039 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 28.0%\n",
      "  C: 34.2%\n",
      "  G: 23.5%\n",
      "  T: 14.2%\n",
      "  none/other: 0.1%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t2838492\t3400222.5\t0\t2838492\n",
      "2\t1002268\t850055.6\t0\t1002268\n",
      "3\t262903\t212513.9\t0\t262903\n",
      "4\t59730\t53128.5\t0\t59730\n",
      "5\t20195\t13282.1\t0\t20195\n",
      "6\t5628\t3320.5\t0\t5628\n",
      "7\t3333\t830.1\t0\t3333\n",
      "8\t2530\t207.5\t0\t2530\n",
      "9\t2166\t51.9\t0\t1865 301\n",
      "10\t1472\t13.0\t1\t1111 361\n",
      "11\t752\t3.2\t1\t517 235\n",
      "12\t382\t0.8\t1\t294 88\n",
      "13\t452\t0.2\t1\t414 38\n",
      "14\t726\t0.2\t1\t679 47\n",
      "15\t605\t0.2\t1\t558 47\n",
      "16\t135\t0.2\t1\t122 13\n",
      "17\t58\t0.2\t1\t49 9\n",
      "18\t34\t0.2\t1\t30 4\n",
      "19\t35\t0.2\t1\t20 15\n",
      "20\t14\t0.2\t1\t9 5\n",
      "21\t27\t0.2\t1\t10 17\n",
      "22\t33\t0.2\t1\t22 11\n",
      "23\t30\t0.2\t1\t23 7\n",
      "24\t13\t0.2\t1\t8 5\n",
      "25\t18\t0.2\t1\t11 7\n",
      "26\t43\t0.2\t1\t29 14\n",
      "27\t1022\t0.2\t1\t919 103\n",
      "28\t43\t0.2\t1\t16 27\n",
      "29\t20\t0.2\t1\t10 10\n",
      "30\t42\t0.2\t1\t2 40\n",
      "31\t169\t0.2\t1\t10 159\n",
      "32\t181\t0.2\t1\t12 169\n",
      "33\t68\t0.2\t1\t9 59\n",
      "34\t40\t0.2\t1\t10 30\n",
      "35\t74\t0.2\t1\t4 70\n",
      "36\t306\t0.2\t1\t2 304\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: /data/hodges_lab/public_data/human-rhesus_RNA-seq/cain_2011_LCLs/fastq/SRR066751.fastq\n",
      "=============================================\n",
      "13600890 sequences processed in total\n",
      "Sequences removed because they became shorter than the length cutoff of 20 bp:\t75902 (0.6%)\n",
      "\n",
      "Now renaming the output file SRR066751_trimmed.fq\n",
      "\n",
      "ORIGINAL FILE 1: >>SRR066751_trimmed.fq<<\tRENAMING TO:>>Rhesus_LCL_R1_trimmed.fq<<\n",
      "\n",
      "  >>> Now running FastQC on the data <<<\n",
      "\n",
      "Started analysis of Rhesus_LCL_R1_trimmed.fq\n",
      "Approx 5% complete for Rhesus_LCL_R1_trimmed.fq\n",
      "Approx 10% complete for Rhesus_LCL_R1_trimmed.fq\n",
      "Approx 15% complete for Rhesus_LCL_R1_trimmed.fq\n",
      "Approx 20% complete for Rhesus_LCL_R1_trimmed.fq\n",
      "Approx 25% complete for Rhesus_LCL_R1_trimmed.fq\n",
      "Approx 30% complete for Rhesus_LCL_R1_trimmed.fq\n",
      "Approx 35% complete for Rhesus_LCL_R1_trimmed.fq\n",
      "Approx 40% complete for Rhesus_LCL_R1_trimmed.fq\n",
      "Approx 45% complete for Rhesus_LCL_R1_trimmed.fq\n",
      "Approx 50% complete for Rhesus_LCL_R1_trimmed.fq\n",
      "Approx 55% complete for Rhesus_LCL_R1_trimmed.fq\n",
      "Approx 60% complete for Rhesus_LCL_R1_trimmed.fq\n",
      "Approx 65% complete for Rhesus_LCL_R1_trimmed.fq\n",
      "Approx 70% complete for Rhesus_LCL_R1_trimmed.fq\n",
      "Approx 75% complete for Rhesus_LCL_R1_trimmed.fq\n",
      "Approx 80% complete for Rhesus_LCL_R1_trimmed.fq\n",
      "Approx 85% complete for Rhesus_LCL_R1_trimmed.fq\n",
      "Approx 90% complete for Rhesus_LCL_R1_trimmed.fq\n",
      "Approx 95% complete for Rhesus_LCL_R1_trimmed.fq\n",
      "Analysis complete for Rhesus_LCL_R1_trimmed.fq\n",
      "Using an excessive number of cores has a diminishing return! It is recommended not to exceed 8 cores per trimming process (you asked for 8 cores). Please consider re-specifying\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "Cutadapt version: 1.18\n",
      "Could not detect version of Python used by Cutadapt from the first line of Cutadapt (but found this: >>>#!/bin/sh<<<)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Letting the (modified) Cutadapt deal with the Python version instead\n",
      "pigz 2.6\n",
      "Parallel gzip (pigz) detected. Proceeding with multicore (de)compression using 8 cores\n",
      "\n",
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Output will be written into the directory: /gpfs52/data/hodges_lab/public_data/human-rhesus_RNA-seq/cain_2011_LCLs/fastq/trimmed_reads/\n",
      "Using user-specified basename (>>Rhesus_LCL_R2<<) instead of deriving the filename from the input file(s)\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> /data/hodges_lab/public_data/human-rhesus_RNA-seq/cain_2011_LCLs/fastq/SRR066752.fastq <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Illumina\t94\tAGATCGGAAGAGC\t1000000\t0.01\n",
      "smallRNA\t2\tTGGAATTCTCGG\t1000000\t0.00\n",
      "Nextera\t0\tCTGTCTCTTATA\t1000000\t0.00\n",
      "Using Illumina adapter for trimming (count: 94). Second best hit was smallRNA (count: 2)\n",
      "\n",
      "Writing report to '/gpfs52/data/hodges_lab/public_data/human-rhesus_RNA-seq/cain_2011_LCLs/fastq/trimmed_reads/SRR066752.fastq_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: /data/hodges_lab/public_data/human-rhesus_RNA-seq/cain_2011_LCLs/fastq/SRR066752.fastq\n",
      "Trimming mode: single-end\n",
      "Trim Galore version: 0.6.6\n",
      "Cutadapt version: 1.18\n",
      "Python version: could not detect\n",
      "Number of cores used for trimming: 8\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length before a sequence gets removed: 20 bp\n",
      "Running FastQC on the data once trimming has completed\n",
      "Running FastQC with the following extra arguments: '--outdir /data/hodges_lab/public_data/human-rhesus_RNA-seq/cain_2011_LCLs/processing_results/fastQC'\n",
      "\n",
      "Cutadapt seems to be reasonably up-to-date. Setting -j 8\n",
      "Writing final adapter and quality trimmed output to SRR066752_trimmed.fq\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff '-q 20') and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file /data/hodges_lab/public_data/human-rhesus_RNA-seq/cain_2011_LCLs/fastq/SRR066752.fastq <<< \n",
      "10000000 sequences processed\n",
      "This is cutadapt 1.18 with Python 3.6.15\n",
      "Command line parameters: -j 8 -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC /data/hodges_lab/public_data/human-rhesus_RNA-seq/cain_2011_LCLs/fastq/SRR066752.fastq\n",
      "Processing reads on 8 cores in single-end mode ...\n",
      "Finished in 55.66 s (4 us/read; 15.47 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              14,354,061\n",
      "Reads with adapters:                 4,362,315 (30.4%)\n",
      "Reads written (passing filters):    14,354,061 (100.0%)\n",
      "\n",
      "Total basepairs processed:   516,746,196 bp\n",
      "Quality-trimmed:               6,626,703 bp (1.3%)\n",
      "Total written (filtered):    503,748,152 bp (97.5%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 4362315 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 28.3%\n",
      "  C: 34.4%\n",
      "  G: 23.4%\n",
      "  T: 13.8%\n",
      "  none/other: 0.1%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t2929278\t3588515.2\t0\t2929278\n",
      "2\t1072643\t897128.8\t0\t1072643\n",
      "3\t272440\t224282.2\t0\t272440\n",
      "4\t58372\t56070.6\t0\t58372\n",
      "5\t17337\t14017.6\t0\t17337\n",
      "6\t3351\t3504.4\t0\t3351\n",
      "7\t1569\t876.1\t0\t1569\n",
      "8\t1020\t219.0\t0\t1020\n",
      "9\t1131\t54.8\t0\t765 366\n",
      "10\t827\t13.7\t1\t459 368\n",
      "11\t443\t3.4\t1\t238 205\n",
      "12\t241\t0.9\t1\t149 92\n",
      "13\t223\t0.2\t1\t191 32\n",
      "14\t329\t0.2\t1\t294 35\n",
      "15\t280\t0.2\t1\t253 27\n",
      "16\t87\t0.2\t1\t76 11\n",
      "17\t26\t0.2\t1\t21 5\n",
      "18\t39\t0.2\t1\t28 11\n",
      "19\t30\t0.2\t1\t17 13\n",
      "20\t36\t0.2\t1\t20 16\n",
      "21\t41\t0.2\t1\t13 28\n",
      "22\t27\t0.2\t1\t18 9\n",
      "23\t18\t0.2\t1\t14 4\n",
      "24\t12\t0.2\t1\t4 8\n",
      "25\t31\t0.2\t1\t12 19\n",
      "26\t66\t0.2\t1\t20 46\n",
      "27\t1065\t0.2\t1\t960 105\n",
      "28\t66\t0.2\t1\t21 45\n",
      "29\t30\t0.2\t1\t5 25\n",
      "30\t63\t0.2\t1\t7 56\n",
      "31\t219\t0.2\t1\t5 214\n",
      "32\t308\t0.2\t1\t12 296\n",
      "33\t64\t0.2\t1\t10 54\n",
      "34\t42\t0.2\t1\t7 35\n",
      "35\t102\t0.2\t1\t3 99\n",
      "36\t459\t0.2\t1\t2 457\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: /data/hodges_lab/public_data/human-rhesus_RNA-seq/cain_2011_LCLs/fastq/SRR066752.fastq\n",
      "=============================================\n",
      "14354061 sequences processed in total\n",
      "Sequences removed because they became shorter than the length cutoff of 20 bp:\t97312 (0.7%)\n",
      "\n",
      "Now renaming the output file SRR066752_trimmed.fq\n",
      "\n",
      "ORIGINAL FILE 1: >>SRR066752_trimmed.fq<<\tRENAMING TO:>>Rhesus_LCL_R2_trimmed.fq<<\n",
      "\n",
      "  >>> Now running FastQC on the data <<<\n",
      "\n",
      "Started analysis of Rhesus_LCL_R2_trimmed.fq\n",
      "Approx 5% complete for Rhesus_LCL_R2_trimmed.fq\n",
      "Approx 10% complete for Rhesus_LCL_R2_trimmed.fq\n",
      "Approx 15% complete for Rhesus_LCL_R2_trimmed.fq\n",
      "Approx 20% complete for Rhesus_LCL_R2_trimmed.fq\n",
      "Approx 25% complete for Rhesus_LCL_R2_trimmed.fq\n",
      "Approx 30% complete for Rhesus_LCL_R2_trimmed.fq\n",
      "Approx 35% complete for Rhesus_LCL_R2_trimmed.fq\n",
      "Approx 40% complete for Rhesus_LCL_R2_trimmed.fq\n",
      "Approx 45% complete for Rhesus_LCL_R2_trimmed.fq\n",
      "Approx 50% complete for Rhesus_LCL_R2_trimmed.fq\n",
      "Approx 55% complete for Rhesus_LCL_R2_trimmed.fq\n",
      "Approx 60% complete for Rhesus_LCL_R2_trimmed.fq\n",
      "Approx 65% complete for Rhesus_LCL_R2_trimmed.fq\n",
      "Approx 70% complete for Rhesus_LCL_R2_trimmed.fq\n",
      "Approx 75% complete for Rhesus_LCL_R2_trimmed.fq\n",
      "Approx 80% complete for Rhesus_LCL_R2_trimmed.fq\n",
      "Approx 85% complete for Rhesus_LCL_R2_trimmed.fq\n",
      "Approx 90% complete for Rhesus_LCL_R2_trimmed.fq\n",
      "Approx 95% complete for Rhesus_LCL_R2_trimmed.fq\n",
      "Analysis complete for Rhesus_LCL_R2_trimmed.fq\n",
      "Using an excessive number of cores has a diminishing return! It is recommended not to exceed 8 cores per trimming process (you asked for 8 cores). Please consider re-specifying\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "Cutadapt version: 1.18\n",
      "Could not detect version of Python used by Cutadapt from the first line of Cutadapt (but found this: >>>#!/bin/sh<<<)\n",
      "Letting the (modified) Cutadapt deal with the Python version instead\n",
      "pigz 2.6\n",
      "Parallel gzip (pigz) detected. Proceeding with multicore (de)compression using 8 cores\n",
      "\n",
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Output will be written into the directory: /gpfs52/data/hodges_lab/public_data/human-rhesus_RNA-seq/cain_2011_LCLs/fastq/trimmed_reads/\n",
      "Using user-specified basename (>>Rhesus_LCL_R3<<) instead of deriving the filename from the input file(s)\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> /data/hodges_lab/public_data/human-rhesus_RNA-seq/cain_2011_LCLs/fastq/SRR066753.fastq <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Illumina\t219\tAGATCGGAAGAGC\t1000000\t0.02\n",
      "smallRNA\t6\tTGGAATTCTCGG\t1000000\t0.00\n",
      "Nextera\t1\tCTGTCTCTTATA\t1000000\t0.00\n",
      "Using Illumina adapter for trimming (count: 219). Second best hit was smallRNA (count: 6)\n",
      "\n",
      "Writing report to '/gpfs52/data/hodges_lab/public_data/human-rhesus_RNA-seq/cain_2011_LCLs/fastq/trimmed_reads/SRR066753.fastq_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: /data/hodges_lab/public_data/human-rhesus_RNA-seq/cain_2011_LCLs/fastq/SRR066753.fastq\n",
      "Trimming mode: single-end\n",
      "Trim Galore version: 0.6.6\n",
      "Cutadapt version: 1.18\n",
      "Python version: could not detect\n",
      "Number of cores used for trimming: 8\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length before a sequence gets removed: 20 bp\n",
      "Running FastQC on the data once trimming has completed\n",
      "Running FastQC with the following extra arguments: '--outdir /data/hodges_lab/public_data/human-rhesus_RNA-seq/cain_2011_LCLs/processing_results/fastQC'\n",
      "\n",
      "Cutadapt seems to be reasonably up-to-date. Setting -j 8\n",
      "Writing final adapter and quality trimmed output to SRR066753_trimmed.fq\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff '-q 20') and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file /data/hodges_lab/public_data/human-rhesus_RNA-seq/cain_2011_LCLs/fastq/SRR066753.fastq <<< \n",
      "10000000 sequences processed\n",
      "This is cutadapt 1.18 with Python 3.6.15\n",
      "Command line parameters: -j 8 -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC /data/hodges_lab/public_data/human-rhesus_RNA-seq/cain_2011_LCLs/fastq/SRR066753.fastq\n",
      "Processing reads on 8 cores in single-end mode ...\n",
      "Finished in 58.05 s (4 us/read; 15.16 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              14,664,858\n",
      "Reads with adapters:                 4,478,345 (30.5%)\n",
      "Reads written (passing filters):    14,664,858 (100.0%)\n",
      "\n",
      "Total basepairs processed:   527,934,888 bp\n",
      "Quality-trimmed:               8,098,539 bp (1.5%)\n",
      "Total written (filtered):    513,089,097 bp (97.2%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 4478345 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 28.6%\n",
      "  C: 34.4%\n",
      "  G: 23.7%\n",
      "  T: 13.2%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t2958841\t3666214.5\t0\t2958841\n",
      "2\t1103125\t916553.6\t0\t1103125\n",
      "3\t289066\t229138.4\t0\t289066\n",
      "4\t68902\t57284.6\t0\t68902\n",
      "5\t25697\t14321.2\t0\t25697\n",
      "6\t10214\t3580.3\t0\t10214\n",
      "7\t6744\t895.1\t0\t6744\n",
      "8\t4300\t223.8\t0\t4300\n",
      "9\t3202\t55.9\t0\t2864 338\n",
      "10\t2122\t14.0\t1\t1697 425\n",
      "11\t878\t3.5\t1\t624 254\n",
      "12\t531\t0.9\t1\t416 115\n",
      "13\t539\t0.2\t1\t494 45\n",
      "14\t797\t0.2\t1\t759 38\n",
      "15\t537\t0.2\t1\t495 42\n",
      "16\t150\t0.2\t1\t131 19\n",
      "17\t54\t0.2\t1\t44 10\n",
      "18\t38\t0.2\t1\t27 11\n",
      "19\t39\t0.2\t1\t19 20\n",
      "20\t12\t0.2\t1\t5 7\n",
      "21\t31\t0.2\t1\t8 23\n",
      "22\t36\t0.2\t1\t17 19\n",
      "23\t26\t0.2\t1\t13 13\n",
      "24\t19\t0.2\t1\t6 13\n",
      "25\t22\t0.2\t1\t10 12\n",
      "26\t61\t0.2\t1\t31 30\n",
      "27\t1950\t0.2\t1\t1764 186\n",
      "28\t44\t0.2\t1\t24 20\n",
      "29\t25\t0.2\t1\t8 17\n",
      "30\t21\t0.2\t1\t3 18\n",
      "31\t53\t0.2\t1\t6 47\n",
      "32\t69\t0.2\t1\t5 64\n",
      "33\t46\t0.2\t1\t5 41\n",
      "34\t27\t0.2\t1\t10 17\n",
      "35\t22\t0.2\t1\t1 21\n",
      "36\t105\t0.2\t1\t3 102\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: /data/hodges_lab/public_data/human-rhesus_RNA-seq/cain_2011_LCLs/fastq/SRR066753.fastq\n",
      "=============================================\n",
      "14664858 sequences processed in total\n",
      "Sequences removed because they became shorter than the length cutoff of 20 bp:\t119190 (0.8%)\n",
      "\n",
      "Now renaming the output file SRR066753_trimmed.fq\n",
      "\n",
      "ORIGINAL FILE 1: >>SRR066753_trimmed.fq<<\tRENAMING TO:>>Rhesus_LCL_R3_trimmed.fq<<\n",
      "\n",
      "  >>> Now running FastQC on the data <<<\n",
      "\n",
      "Started analysis of Rhesus_LCL_R3_trimmed.fq\n",
      "Approx 5% complete for Rhesus_LCL_R3_trimmed.fq\n",
      "Approx 10% complete for Rhesus_LCL_R3_trimmed.fq\n",
      "Approx 15% complete for Rhesus_LCL_R3_trimmed.fq\n",
      "Approx 20% complete for Rhesus_LCL_R3_trimmed.fq\n",
      "Approx 25% complete for Rhesus_LCL_R3_trimmed.fq\n",
      "Approx 30% complete for Rhesus_LCL_R3_trimmed.fq\n",
      "Approx 35% complete for Rhesus_LCL_R3_trimmed.fq\n",
      "Approx 40% complete for Rhesus_LCL_R3_trimmed.fq\n",
      "Approx 45% complete for Rhesus_LCL_R3_trimmed.fq\n",
      "Approx 50% complete for Rhesus_LCL_R3_trimmed.fq\n",
      "Approx 55% complete for Rhesus_LCL_R3_trimmed.fq\n",
      "Approx 60% complete for Rhesus_LCL_R3_trimmed.fq\n",
      "Approx 65% complete for Rhesus_LCL_R3_trimmed.fq\n",
      "Approx 70% complete for Rhesus_LCL_R3_trimmed.fq\n",
      "Approx 75% complete for Rhesus_LCL_R3_trimmed.fq\n",
      "Approx 80% complete for Rhesus_LCL_R3_trimmed.fq\n",
      "Approx 85% complete for Rhesus_LCL_R3_trimmed.fq\n",
      "Approx 90% complete for Rhesus_LCL_R3_trimmed.fq\n",
      "Approx 95% complete for Rhesus_LCL_R3_trimmed.fq\n",
      "Analysis complete for Rhesus_LCL_R3_trimmed.fq\n"
     ]
    }
   ],
   "source": [
    "# use read combined with while loop to read two variables at once.\n",
    "DIR='/data/hodges_lab/public_data/human-rhesus_RNA-seq/cain_2011_LCLs'\n",
    "\n",
    "while read -r a b ; do    \n",
    "    trim_galore -j 8 --fastqc --fastqc_args \"--outdir ${DIR}/processing_results/fastQC\" --dont_gzip --basename \"$a\" \\\n",
    "    --output_dir ${DIR}/fastq/trimmed_reads ${DIR}/fastq/\"$b\".fastq \n",
    "done < /data/hodges_lab/public_data/human-rhesus_RNA-seq/LCL_names.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "f3e8e16c",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-03-16T18:43:05.622560Z",
     "start_time": "2022-03-16T14:21:57.853960Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Using an excessive number of cores has a diminishing return! It is recommended not to exceed 8 cores per trimming process (you asked for 8 cores). Please consider re-specifying\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "Cutadapt version: 1.18\n",
      "Could not detect version of Python used by Cutadapt from the first line of Cutadapt (but found this: >>>#!/bin/sh<<<)\n",
      "Letting the (modified) Cutadapt deal with the Python version instead\n",
      "pigz 2.6\n",
      "Parallel gzip (pigz) detected. Proceeding with multicore (de)compression using 8 cores\n",
      "\n",
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Output will be written into the directory: /gpfs52/data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/trimmed_reads/\n",
      "Using user-specified basename (>>Human_Heart_R1<<) instead of deriving the filename from the input file(s)\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900781.fastq <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Illumina\t2971\tAGATCGGAAGAGC\t1000000\t0.30\n",
      "smallRNA\t2\tTGGAATTCTCGG\t1000000\t0.00\n",
      "Nextera\t0\tCTGTCTCTTATA\t1000000\t0.00\n",
      "Using Illumina adapter for trimming (count: 2971). Second best hit was smallRNA (count: 2)\n",
      "\n",
      "Writing report to '/gpfs52/data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/trimmed_reads/SRR6900781.fastq_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900781.fastq\n",
      "Trimming mode: single-end\n",
      "Trim Galore version: 0.6.6\n",
      "Cutadapt version: 1.18\n",
      "Python version: could not detect\n",
      "Number of cores used for trimming: 8\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length before a sequence gets removed: 20 bp\n",
      "Running FastQC on the data once trimming has completed\n",
      "Running FastQC with the following extra arguments: '--outdir /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/processing_results/fastQC'\n",
      "\n",
      "Cutadapt seems to be reasonably up-to-date. Setting -j 8\n",
      "Writing final adapter and quality trimmed output to SRR6900781_trimmed.fq\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff '-q 20') and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900781.fastq <<< \n",
      "10000000 sequences processed\n",
      "20000000 sequences processed\n",
      "30000000 sequences processed\n",
      "40000000 sequences processed\n",
      "This is cutadapt 1.18 with Python 3.6.15\n",
      "Command line parameters: -j 8 -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900781.fastq\n",
      "Processing reads on 8 cores in single-end mode ...\n",
      "Finished in 202.43 s (4 us/read; 14.66 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              49,454,353\n",
      "Reads with adapters:                 6,266,055 (12.7%)\n",
      "Reads written (passing filters):    49,454,353 (100.0%)\n",
      "\n",
      "Total basepairs processed: 2,507,324,274 bp\n",
      "Quality-trimmed:             315,764,324 bp (12.6%)\n",
      "Total written (filtered):  2,139,928,483 bp (85.3%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 6266055 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 27.3%\n",
      "  C: 27.0%\n",
      "  G: 17.0%\n",
      "  T: 14.5%\n",
      "  none/other: 14.2%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t3830057\t12363588.2\t0\t3830057\n",
      "2\t1148014\t3090897.1\t0\t1148014\n",
      "3\t317133\t772724.3\t0\t317133\n",
      "4\t66189\t193181.1\t0\t66189\n",
      "5\t14607\t48295.3\t0\t14607\n",
      "6\t1680\t12073.8\t0\t1680\n",
      "7\t534\t3018.5\t0\t534\n",
      "8\t376\t754.6\t0\t376\n",
      "9\t491\t188.7\t0\t221 270\n",
      "10\t1017\t47.2\t1\t196 821\n",
      "11\t825\t11.8\t1\t250 575\n",
      "12\t616\t2.9\t1\t192 424\n",
      "13\t570\t0.7\t1\t214 356\n",
      "14\t575\t0.7\t1\t383 192\n",
      "15\t538\t0.7\t1\t405 133\n",
      "16\t710\t0.7\t1\t403 307\n",
      "17\t506\t0.7\t1\t419 87\n",
      "18\t508\t0.7\t1\t382 126\n",
      "19\t469\t0.7\t1\t385 84\n",
      "20\t527\t0.7\t1\t415 112\n",
      "21\t452\t0.7\t1\t355 97\n",
      "22\t505\t0.7\t1\t399 106\n",
      "23\t503\t0.7\t1\t422 81\n",
      "24\t501\t0.7\t1\t413 88\n",
      "25\t588\t0.7\t1\t425 163\n",
      "26\t539\t0.7\t1\t439 100\n",
      "27\t650\t0.7\t1\t496 154\n",
      "28\t664\t0.7\t1\t522 142\n",
      "29\t728\t0.7\t1\t550 178\n",
      "30\t682\t0.7\t1\t548 134\n",
      "31\t621\t0.7\t1\t484 137\n",
      "32\t697\t0.7\t1\t562 135\n",
      "33\t784\t0.7\t1\t582 202\n",
      "34\t1030\t0.7\t1\t773 257\n",
      "35\t890\t0.7\t1\t651 239\n",
      "36\t1160\t0.7\t1\t781 379\n",
      "37\t1175\t0.7\t1\t782 393\n",
      "38\t1202\t0.7\t1\t648 554\n",
      "39\t1662\t0.7\t1\t1066 596\n",
      "40\t1321\t0.7\t1\t679 642\n",
      "41\t1584\t0.7\t1\t872 712\n",
      "42\t1844\t0.7\t1\t918 926\n",
      "43\t2038\t0.7\t1\t1050 988\n",
      "44\t4083\t0.7\t1\t1190 2893\n",
      "45\t9116\t0.7\t1\t5020 4096\n",
      "46\t9356\t0.7\t1\t5721 3635\n",
      "47\t10689\t0.7\t1\t6006 4683\n",
      "48\t16821\t0.7\t1\t7650 9171\n",
      "49\t105846\t0.7\t1\t53818 52028\n",
      "50\t4567\t0.7\t1\t2930 1637\n",
      "51\t697815\t0.7\t1\t411478 286337\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900781.fastq\n",
      "=============================================\n",
      "49454353 sequences processed in total\n",
      "Sequences removed because they became shorter than the length cutoff of 20 bp:\t6566513 (13.3%)\n",
      "\n",
      "Now renaming the output file SRR6900781_trimmed.fq\n",
      "\n",
      "ORIGINAL FILE 1: >>SRR6900781_trimmed.fq<<\tRENAMING TO:>>Human_Heart_R1_trimmed.fq<<\n",
      "\n",
      "  >>> Now running FastQC on the data <<<\n",
      "\n",
      "Started analysis of Human_Heart_R1_trimmed.fq\n",
      "Approx 5% complete for Human_Heart_R1_trimmed.fq\n",
      "Approx 10% complete for Human_Heart_R1_trimmed.fq\n",
      "Approx 15% complete for Human_Heart_R1_trimmed.fq\n",
      "Approx 20% complete for Human_Heart_R1_trimmed.fq\n",
      "Approx 25% complete for Human_Heart_R1_trimmed.fq\n",
      "Approx 30% complete for Human_Heart_R1_trimmed.fq\n",
      "Approx 35% complete for Human_Heart_R1_trimmed.fq\n",
      "Approx 40% complete for Human_Heart_R1_trimmed.fq\n",
      "Approx 45% complete for Human_Heart_R1_trimmed.fq\n",
      "Approx 50% complete for Human_Heart_R1_trimmed.fq\n",
      "Approx 55% complete for Human_Heart_R1_trimmed.fq\n",
      "Approx 60% complete for Human_Heart_R1_trimmed.fq\n",
      "Approx 65% complete for Human_Heart_R1_trimmed.fq\n",
      "Approx 70% complete for Human_Heart_R1_trimmed.fq\n",
      "Approx 75% complete for Human_Heart_R1_trimmed.fq\n",
      "Approx 80% complete for Human_Heart_R1_trimmed.fq\n",
      "Approx 85% complete for Human_Heart_R1_trimmed.fq\n",
      "Approx 90% complete for Human_Heart_R1_trimmed.fq\n",
      "Approx 95% complete for Human_Heart_R1_trimmed.fq\n",
      "Analysis complete for Human_Heart_R1_trimmed.fq\n",
      "Using an excessive number of cores has a diminishing return! It is recommended not to exceed 8 cores per trimming process (you asked for 8 cores). Please consider re-specifying\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "Cutadapt version: 1.18\n",
      "Could not detect version of Python used by Cutadapt from the first line of Cutadapt (but found this: >>>#!/bin/sh<<<)\n",
      "Letting the (modified) Cutadapt deal with the Python version instead\n",
      "pigz 2.6\n",
      "Parallel gzip (pigz) detected. Proceeding with multicore (de)compression using 8 cores\n",
      "\n",
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Output will be written into the directory: /gpfs52/data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/trimmed_reads/\n",
      "Using user-specified basename (>>Human_Kidney_R1<<) instead of deriving the filename from the input file(s)\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900782.fastq <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Illumina\t40\tAGATCGGAAGAGC\t1000000\t0.00\n",
      "Nextera\t3\tCTGTCTCTTATA\t1000000\t0.00\n",
      "smallRNA\t0\tTGGAATTCTCGG\t1000000\t0.00\n",
      "Using Illumina adapter for trimming (count: 40). Second best hit was Nextera (count: 3)\n",
      "\n",
      "Writing report to '/gpfs52/data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/trimmed_reads/SRR6900782.fastq_trimming_report.txt'\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900782.fastq\n",
      "Trimming mode: single-end\n",
      "Trim Galore version: 0.6.6\n",
      "Cutadapt version: 1.18\n",
      "Python version: could not detect\n",
      "Number of cores used for trimming: 8\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length before a sequence gets removed: 20 bp\n",
      "Running FastQC on the data once trimming has completed\n",
      "Running FastQC with the following extra arguments: '--outdir /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/processing_results/fastQC'\n",
      "\n",
      "Cutadapt seems to be reasonably up-to-date. Setting -j 8\n",
      "Writing final adapter and quality trimmed output to SRR6900782_trimmed.fq\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff '-q 20') and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900782.fastq <<< \n",
      "10000000 sequences processed\n",
      "20000000 sequences processed\n",
      "30000000 sequences processed\n",
      "40000000 sequences processed\n",
      "50000000 sequences processed\n",
      "This is cutadapt 1.18 with Python 3.6.15\n",
      "Command line parameters: -j 8 -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900782.fastq\n",
      "Processing reads on 8 cores in single-end mode ...\n",
      "Finished in 198.68 s (4 us/read; 15.57 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              51,561,113\n",
      "Reads with adapters:                 4,731,291 (9.2%)\n",
      "Reads written (passing filters):    51,561,113 (100.0%)\n",
      "\n",
      "Total basepairs processed: 2,618,533,390 bp\n",
      "Quality-trimmed:             643,223,966 bp (24.6%)\n",
      "Total written (filtered):  1,968,156,173 bp (75.2%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 4731291 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 29.8%\n",
      "  C: 30.3%\n",
      "  G: 20.8%\n",
      "  T: 18.3%\n",
      "  none/other: 0.8%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t3399605\t12890278.2\t0\t3399605\n",
      "2\t977648\t3222569.6\t0\t977648\n",
      "3\t272034\t805642.4\t0\t272034\n",
      "4\t52559\t201410.6\t0\t52559\n",
      "5\t11501\t50352.6\t0\t11501\n",
      "6\t1422\t12588.2\t0\t1422\n",
      "7\t237\t3147.0\t0\t237\n",
      "8\t81\t786.8\t0\t81\n",
      "9\t226\t196.7\t0\t36 190\n",
      "10\t367\t49.2\t1\t47 320\n",
      "11\t232\t12.3\t1\t51 181\n",
      "12\t150\t3.1\t1\t31 119\n",
      "13\t140\t0.8\t1\t37 103\n",
      "14\t115\t0.8\t1\t79 36\n",
      "15\t124\t0.8\t1\t88 36\n",
      "16\t133\t0.8\t1\t98 35\n",
      "17\t138\t0.8\t1\t108 30\n",
      "18\t117\t0.8\t1\t81 36\n",
      "19\t142\t0.8\t1\t101 41\n",
      "20\t136\t0.8\t1\t99 37\n",
      "21\t127\t0.8\t1\t89 38\n",
      "22\t107\t0.8\t1\t61 46\n",
      "23\t125\t0.8\t1\t92 33\n",
      "24\t121\t0.8\t1\t102 19\n",
      "25\t105\t0.8\t1\t71 34\n",
      "26\t114\t0.8\t1\t86 28\n",
      "27\t91\t0.8\t1\t62 29\n",
      "28\t85\t0.8\t1\t52 33\n",
      "29\t83\t0.8\t1\t54 29\n",
      "30\t77\t0.8\t1\t45 32\n",
      "31\t106\t0.8\t1\t63 43\n",
      "32\t80\t0.8\t1\t59 21\n",
      "33\t67\t0.8\t1\t41 26\n",
      "34\t75\t0.8\t1\t45 30\n",
      "35\t78\t0.8\t1\t51 27\n",
      "36\t73\t0.8\t1\t40 33\n",
      "37\t56\t0.8\t1\t38 18\n",
      "38\t66\t0.8\t1\t40 26\n",
      "39\t75\t0.8\t1\t35 40\n",
      "40\t78\t0.8\t1\t29 49\n",
      "41\t51\t0.8\t1\t23 28\n",
      "42\t82\t0.8\t1\t32 50\n",
      "43\t50\t0.8\t1\t12 38\n",
      "44\t92\t0.8\t1\t22 70\n",
      "45\t115\t0.8\t1\t44 71\n",
      "46\t164\t0.8\t1\t49 115\n",
      "47\t199\t0.8\t1\t58 141\n",
      "48\t339\t0.8\t1\t92 247\n",
      "49\t1284\t0.8\t1\t311 973\n",
      "50\t171\t0.8\t1\t45 126\n",
      "51\t9848\t0.8\t1\t2711 7137\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900782.fastq\n",
      "=============================================\n",
      "51561113 sequences processed in total\n",
      "Sequences removed because they became shorter than the length cutoff of 20 bp:\t12227846 (23.7%)\n",
      "\n",
      "Now renaming the output file SRR6900782_trimmed.fq\n",
      "\n",
      "ORIGINAL FILE 1: >>SRR6900782_trimmed.fq<<\tRENAMING TO:>>Human_Kidney_R1_trimmed.fq<<\n",
      "\n",
      "  >>> Now running FastQC on the data <<<\n",
      "\n",
      "Started analysis of Human_Kidney_R1_trimmed.fq\n",
      "Approx 5% complete for Human_Kidney_R1_trimmed.fq\n",
      "Approx 10% complete for Human_Kidney_R1_trimmed.fq\n",
      "Approx 15% complete for Human_Kidney_R1_trimmed.fq\n",
      "Approx 20% complete for Human_Kidney_R1_trimmed.fq\n",
      "Approx 25% complete for Human_Kidney_R1_trimmed.fq\n",
      "Approx 30% complete for Human_Kidney_R1_trimmed.fq\n",
      "Approx 35% complete for Human_Kidney_R1_trimmed.fq\n",
      "Approx 40% complete for Human_Kidney_R1_trimmed.fq\n",
      "Approx 45% complete for Human_Kidney_R1_trimmed.fq\n",
      "Approx 50% complete for Human_Kidney_R1_trimmed.fq\n",
      "Approx 55% complete for Human_Kidney_R1_trimmed.fq\n",
      "Approx 60% complete for Human_Kidney_R1_trimmed.fq\n",
      "Approx 65% complete for Human_Kidney_R1_trimmed.fq\n",
      "Approx 70% complete for Human_Kidney_R1_trimmed.fq\n",
      "Approx 75% complete for Human_Kidney_R1_trimmed.fq\n",
      "Approx 80% complete for Human_Kidney_R1_trimmed.fq\n",
      "Approx 85% complete for Human_Kidney_R1_trimmed.fq\n",
      "Approx 90% complete for Human_Kidney_R1_trimmed.fq\n",
      "Approx 95% complete for Human_Kidney_R1_trimmed.fq\n",
      "Analysis complete for Human_Kidney_R1_trimmed.fq\n",
      "Using an excessive number of cores has a diminishing return! It is recommended not to exceed 8 cores per trimming process (you asked for 8 cores). Please consider re-specifying\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "Cutadapt version: 1.18\n",
      "Could not detect version of Python used by Cutadapt from the first line of Cutadapt (but found this: >>>#!/bin/sh<<<)\n",
      "Letting the (modified) Cutadapt deal with the Python version instead\n",
      "pigz 2.6\n",
      "Parallel gzip (pigz) detected. Proceeding with multicore (de)compression using 8 cores\n",
      "\n",
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Output will be written into the directory: /gpfs52/data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/trimmed_reads/\n",
      "Using user-specified basename (>>Human_Liver_R1<<) instead of deriving the filename from the input file(s)\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900783.fastq <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Illumina\t2151\tAGATCGGAAGAGC\t1000000\t0.22\n",
      "smallRNA\t1\tTGGAATTCTCGG\t1000000\t0.00\n",
      "Nextera\t1\tCTGTCTCTTATA\t1000000\t0.00\n",
      "Using Illumina adapter for trimming (count: 2151). Second best hit was smallRNA (count: 1)\n",
      "\n",
      "Writing report to '/gpfs52/data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/trimmed_reads/SRR6900783.fastq_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900783.fastq\n",
      "Trimming mode: single-end\n",
      "Trim Galore version: 0.6.6\n",
      "Cutadapt version: 1.18\n",
      "Python version: could not detect\n",
      "Number of cores used for trimming: 8\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length before a sequence gets removed: 20 bp\n",
      "Running FastQC on the data once trimming has completed\n",
      "Running FastQC with the following extra arguments: '--outdir /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/processing_results/fastQC'\n",
      "\n",
      "Cutadapt seems to be reasonably up-to-date. Setting -j 8\n",
      "Writing final adapter and quality trimmed output to SRR6900783_trimmed.fq\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff '-q 20') and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900783.fastq <<< \n",
      "10000000 sequences processed\n",
      "20000000 sequences processed\n",
      "30000000 sequences processed\n",
      "40000000 sequences processed\n",
      "50000000 sequences processed\n",
      "This is cutadapt 1.18 with Python 3.6.15\n",
      "Command line parameters: -j 8 -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900783.fastq\n",
      "Processing reads on 8 cores in single-end mode ...\n",
      "Finished in 202.31 s (4 us/read; 15.44 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              52,075,498\n",
      "Reads with adapters:                 5,145,396 (9.9%)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Reads written (passing filters):    52,075,498 (100.0%)\n",
      "\n",
      "Total basepairs processed: 2,645,706,041 bp\n",
      "Quality-trimmed:             667,056,043 bp (25.2%)\n",
      "Total written (filtered):  1,936,012,685 bp (73.2%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 5145396 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 25.5%\n",
      "  C: 26.8%\n",
      "  G: 18.7%\n",
      "  T: 14.3%\n",
      "  none/other: 14.7%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t3149300\t13018874.5\t0\t3149300\n",
      "2\t923737\t3254718.6\t0\t923737\n",
      "3\t271929\t813679.7\t0\t271929\n",
      "4\t52800\t203419.9\t0\t52800\n",
      "5\t11607\t50855.0\t0\t11607\n",
      "6\t1435\t12713.7\t0\t1435\n",
      "7\t508\t3178.4\t0\t508\n",
      "8\t503\t794.6\t0\t503\n",
      "9\t585\t198.7\t0\t259 326\n",
      "10\t1040\t49.7\t1\t184 856\n",
      "11\t1009\t12.4\t1\t361 648\n",
      "12\t716\t3.1\t1\t226 490\n",
      "13\t816\t0.8\t1\t359 457\n",
      "14\t633\t0.8\t1\t411 222\n",
      "15\t542\t0.8\t1\t362 180\n",
      "16\t620\t0.8\t1\t293 327\n",
      "17\t444\t0.8\t1\t336 108\n",
      "18\t493\t0.8\t1\t341 152\n",
      "19\t513\t0.8\t1\t381 132\n",
      "20\t462\t0.8\t1\t331 131\n",
      "21\t426\t0.8\t1\t325 101\n",
      "22\t504\t0.8\t1\t319 185\n",
      "23\t488\t0.8\t1\t377 111\n",
      "24\t465\t0.8\t1\t360 105\n",
      "25\t494\t0.8\t1\t360 134\n",
      "26\t452\t0.8\t1\t354 98\n",
      "27\t475\t0.8\t1\t333 142\n",
      "28\t435\t0.8\t1\t338 97\n",
      "29\t495\t0.8\t1\t380 115\n",
      "30\t491\t0.8\t1\t345 146\n",
      "31\t586\t0.8\t1\t446 140\n",
      "32\t588\t0.8\t1\t401 187\n",
      "33\t663\t0.8\t1\t489 174\n",
      "34\t624\t0.8\t1\t425 199\n",
      "35\t664\t0.8\t1\t468 196\n",
      "36\t799\t0.8\t1\t502 297\n",
      "37\t807\t0.8\t1\t483 324\n",
      "38\t759\t0.8\t1\t410 349\n",
      "39\t979\t0.8\t1\t525 454\n",
      "40\t1538\t0.8\t1\t817 721\n",
      "41\t1024\t0.8\t1\t583 441\n",
      "42\t1113\t0.8\t1\t544 569\n",
      "43\t1223\t0.8\t1\t613 610\n",
      "44\t2557\t0.8\t1\t749 1808\n",
      "45\t4812\t0.8\t1\t2377 2435\n",
      "46\t5782\t0.8\t1\t2589 3193\n",
      "47\t10071\t0.8\t1\t4726 5345\n",
      "48\t14721\t0.8\t1\t6141 8580\n",
      "49\t71294\t0.8\t1\t31226 40068\n",
      "50\t8866\t0.8\t1\t5544 3322\n",
      "51\t591509\t0.8\t1\t302068 289441\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900783.fastq\n",
      "=============================================\n",
      "52075498 sequences processed in total\n",
      "Sequences removed because they became shorter than the length cutoff of 20 bp:\t13454384 (25.8%)\n",
      "\n",
      "Now renaming the output file SRR6900783_trimmed.fq\n",
      "\n",
      "ORIGINAL FILE 1: >>SRR6900783_trimmed.fq<<\tRENAMING TO:>>Human_Liver_R1_trimmed.fq<<\n",
      "\n",
      "  >>> Now running FastQC on the data <<<\n",
      "\n",
      "Started analysis of Human_Liver_R1_trimmed.fq\n",
      "Approx 5% complete for Human_Liver_R1_trimmed.fq\n",
      "Approx 10% complete for Human_Liver_R1_trimmed.fq\n",
      "Approx 15% complete for Human_Liver_R1_trimmed.fq\n",
      "Approx 20% complete for Human_Liver_R1_trimmed.fq\n",
      "Approx 25% complete for Human_Liver_R1_trimmed.fq\n",
      "Approx 30% complete for Human_Liver_R1_trimmed.fq\n",
      "Approx 35% complete for Human_Liver_R1_trimmed.fq\n",
      "Approx 40% complete for Human_Liver_R1_trimmed.fq\n",
      "Approx 45% complete for Human_Liver_R1_trimmed.fq\n",
      "Approx 50% complete for Human_Liver_R1_trimmed.fq\n",
      "Approx 55% complete for Human_Liver_R1_trimmed.fq\n",
      "Approx 60% complete for Human_Liver_R1_trimmed.fq\n",
      "Approx 65% complete for Human_Liver_R1_trimmed.fq\n",
      "Approx 70% complete for Human_Liver_R1_trimmed.fq\n",
      "Approx 75% complete for Human_Liver_R1_trimmed.fq\n",
      "Approx 80% complete for Human_Liver_R1_trimmed.fq\n",
      "Approx 85% complete for Human_Liver_R1_trimmed.fq\n",
      "Approx 90% complete for Human_Liver_R1_trimmed.fq\n",
      "Approx 95% complete for Human_Liver_R1_trimmed.fq\n",
      "Analysis complete for Human_Liver_R1_trimmed.fq\n",
      "Using an excessive number of cores has a diminishing return! It is recommended not to exceed 8 cores per trimming process (you asked for 8 cores). Please consider re-specifying\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "Cutadapt version: 1.18\n",
      "Could not detect version of Python used by Cutadapt from the first line of Cutadapt (but found this: >>>#!/bin/sh<<<)\n",
      "Letting the (modified) Cutadapt deal with the Python version instead\n",
      "pigz 2.6\n",
      "Parallel gzip (pigz) detected. Proceeding with multicore (de)compression using 8 cores\n",
      "\n",
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Output will be written into the directory: /gpfs52/data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/trimmed_reads/\n",
      "Using user-specified basename (>>Human_Lung_R1<<) instead of deriving the filename from the input file(s)\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900784.fastq <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Illumina\t96\tAGATCGGAAGAGC\t1000000\t0.01\n",
      "smallRNA\t7\tTGGAATTCTCGG\t1000000\t0.00\n",
      "Nextera\t4\tCTGTCTCTTATA\t1000000\t0.00\n",
      "Using Illumina adapter for trimming (count: 96). Second best hit was smallRNA (count: 7)\n",
      "\n",
      "Writing report to '/gpfs52/data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/trimmed_reads/SRR6900784.fastq_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900784.fastq\n",
      "Trimming mode: single-end\n",
      "Trim Galore version: 0.6.6\n",
      "Cutadapt version: 1.18\n",
      "Python version: could not detect\n",
      "Number of cores used for trimming: 8\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length before a sequence gets removed: 20 bp\n",
      "Running FastQC on the data once trimming has completed\n",
      "Running FastQC with the following extra arguments: '--outdir /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/processing_results/fastQC'\n",
      "\n",
      "Cutadapt seems to be reasonably up-to-date. Setting -j 8\n",
      "Writing final adapter and quality trimmed output to SRR6900784_trimmed.fq\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff '-q 20') and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900784.fastq <<< \n",
      "10000000 sequences processed\n",
      "20000000 sequences processed\n",
      "30000000 sequences processed\n",
      "This is cutadapt 1.18 with Python 3.6.15\n",
      "Command line parameters: -j 8 -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900784.fastq\n",
      "Processing reads on 8 cores in single-end mode ...\n",
      "Finished in 141.59 s (4 us/read; 15.80 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              37,288,850\n",
      "Reads with adapters:                 2,843,337 (7.6%)\n",
      "Reads written (passing filters):    37,288,850 (100.0%)\n",
      "\n",
      "Total basepairs processed: 1,895,758,314 bp\n",
      "Quality-trimmed:             496,033,509 bp (26.2%)\n",
      "Total written (filtered):  1,394,765,678 bp (73.6%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 2843337 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 30.0%\n",
      "  C: 30.0%\n",
      "  G: 21.2%\n",
      "  T: 17.4%\n",
      "  none/other: 1.4%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t2030370\t9322212.5\t0\t2030370\n",
      "2\t575762\t2330553.1\t0\t575762\n",
      "3\t172891\t582638.3\t0\t172891\n",
      "4\t32949\t145659.6\t0\t32949\n",
      "5\t7015\t36414.9\t0\t7015\n",
      "6\t850\t9103.7\t0\t850\n",
      "7\t161\t2275.9\t0\t161\n",
      "8\t63\t569.0\t0\t63\n",
      "9\t153\t142.2\t0\t28 125\n",
      "10\t344\t35.6\t1\t43 301\n",
      "11\t174\t8.9\t1\t20 154\n",
      "12\t125\t2.2\t1\t17 108\n",
      "13\t90\t0.6\t1\t28 62\n",
      "14\t117\t0.6\t1\t77 40\n",
      "15\t101\t0.6\t1\t70 31\n",
      "16\t97\t0.6\t1\t67 30\n",
      "17\t81\t0.6\t1\t60 21\n",
      "18\t92\t0.6\t1\t65 27\n",
      "19\t98\t0.6\t1\t68 30\n",
      "20\t99\t0.6\t1\t73 26\n",
      "21\t93\t0.6\t1\t71 22\n",
      "22\t84\t0.6\t1\t61 23\n",
      "23\t86\t0.6\t1\t57 29\n",
      "24\t85\t0.6\t1\t56 29\n",
      "25\t91\t0.6\t1\t62 29\n",
      "26\t80\t0.6\t1\t58 22\n",
      "27\t67\t0.6\t1\t42 25\n",
      "28\t76\t0.6\t1\t44 32\n",
      "29\t88\t0.6\t1\t64 24\n",
      "30\t60\t0.6\t1\t37 23\n",
      "31\t195\t0.6\t1\t172 23\n",
      "32\t68\t0.6\t1\t39 29\n",
      "33\t76\t0.6\t1\t50 26\n",
      "34\t70\t0.6\t1\t39 31\n",
      "35\t58\t0.6\t1\t36 22\n",
      "36\t59\t0.6\t1\t27 32\n",
      "37\t57\t0.6\t1\t32 25\n",
      "38\t53\t0.6\t1\t26 27\n",
      "39\t80\t0.6\t1\t31 49\n",
      "40\t88\t0.6\t1\t26 62\n",
      "41\t73\t0.6\t1\t37 36\n",
      "42\t75\t0.6\t1\t22 53\n",
      "43\t47\t0.6\t1\t14 33\n",
      "44\t57\t0.6\t1\t18 39\n",
      "45\t66\t0.6\t1\t19 47\n",
      "46\t138\t0.6\t1\t40 98\n",
      "47\t208\t0.6\t1\t54 154\n",
      "48\t288\t0.6\t1\t78 210\n",
      "49\t2003\t0.6\t1\t527 1476\n",
      "50\t185\t0.6\t1\t95 90\n",
      "51\t16951\t0.6\t1\t4881 12070\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900784.fastq\n",
      "=============================================\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "37288850 sequences processed in total\n",
      "Sequences removed because they became shorter than the length cutoff of 20 bp:\t9548427 (25.6%)\n",
      "\n",
      "Now renaming the output file SRR6900784_trimmed.fq\n",
      "\n",
      "ORIGINAL FILE 1: >>SRR6900784_trimmed.fq<<\tRENAMING TO:>>Human_Lung_R1_trimmed.fq<<\n",
      "\n",
      "  >>> Now running FastQC on the data <<<\n",
      "\n",
      "Started analysis of Human_Lung_R1_trimmed.fq\n",
      "Approx 5% complete for Human_Lung_R1_trimmed.fq\n",
      "Approx 10% complete for Human_Lung_R1_trimmed.fq\n",
      "Approx 15% complete for Human_Lung_R1_trimmed.fq\n",
      "Approx 20% complete for Human_Lung_R1_trimmed.fq\n",
      "Approx 25% complete for Human_Lung_R1_trimmed.fq\n",
      "Approx 30% complete for Human_Lung_R1_trimmed.fq\n",
      "Approx 35% complete for Human_Lung_R1_trimmed.fq\n",
      "Approx 40% complete for Human_Lung_R1_trimmed.fq\n",
      "Approx 45% complete for Human_Lung_R1_trimmed.fq\n",
      "Approx 50% complete for Human_Lung_R1_trimmed.fq\n",
      "Approx 55% complete for Human_Lung_R1_trimmed.fq\n",
      "Approx 60% complete for Human_Lung_R1_trimmed.fq\n",
      "Approx 65% complete for Human_Lung_R1_trimmed.fq\n",
      "Approx 70% complete for Human_Lung_R1_trimmed.fq\n",
      "Approx 75% complete for Human_Lung_R1_trimmed.fq\n",
      "Approx 80% complete for Human_Lung_R1_trimmed.fq\n",
      "Approx 85% complete for Human_Lung_R1_trimmed.fq\n",
      "Approx 90% complete for Human_Lung_R1_trimmed.fq\n",
      "Approx 95% complete for Human_Lung_R1_trimmed.fq\n",
      "Analysis complete for Human_Lung_R1_trimmed.fq\n",
      "Using an excessive number of cores has a diminishing return! It is recommended not to exceed 8 cores per trimming process (you asked for 8 cores). Please consider re-specifying\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "Cutadapt version: 1.18\n",
      "Could not detect version of Python used by Cutadapt from the first line of Cutadapt (but found this: >>>#!/bin/sh<<<)\n",
      "Letting the (modified) Cutadapt deal with the Python version instead\n",
      "pigz 2.6\n",
      "Parallel gzip (pigz) detected. Proceeding with multicore (de)compression using 8 cores\n",
      "\n",
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Output will be written into the directory: /gpfs52/data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/trimmed_reads/\n",
      "Using user-specified basename (>>Human_Heart_R2<<) instead of deriving the filename from the input file(s)\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900785.fastq <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Illumina\t2841\tAGATCGGAAGAGC\t1000000\t0.28\n",
      "smallRNA\t1\tTGGAATTCTCGG\t1000000\t0.00\n",
      "Nextera\t0\tCTGTCTCTTATA\t1000000\t0.00\n",
      "Using Illumina adapter for trimming (count: 2841). Second best hit was smallRNA (count: 1)\n",
      "\n",
      "Writing report to '/gpfs52/data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/trimmed_reads/SRR6900785.fastq_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900785.fastq\n",
      "Trimming mode: single-end\n",
      "Trim Galore version: 0.6.6\n",
      "Cutadapt version: 1.18\n",
      "Python version: could not detect\n",
      "Number of cores used for trimming: 8\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length before a sequence gets removed: 20 bp\n",
      "Running FastQC on the data once trimming has completed\n",
      "Running FastQC with the following extra arguments: '--outdir /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/processing_results/fastQC'\n",
      "\n",
      "Cutadapt seems to be reasonably up-to-date. Setting -j 8\n",
      "Writing final adapter and quality trimmed output to SRR6900785_trimmed.fq\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff '-q 20') and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900785.fastq <<< \n",
      "10000000 sequences processed\n",
      "20000000 sequences processed\n",
      "30000000 sequences processed\n",
      "40000000 sequences processed\n",
      "50000000 sequences processed\n",
      "This is cutadapt 1.18 with Python 3.6.15\n",
      "Command line parameters: -j 8 -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900785.fastq\n",
      "Processing reads on 8 cores in single-end mode ...\n",
      "Finished in 211.03 s (4 us/read; 15.47 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              54,401,264\n",
      "Reads with adapters:                 7,093,393 (13.0%)\n",
      "Reads written (passing filters):    54,401,264 (100.0%)\n",
      "\n",
      "Total basepairs processed: 2,758,834,048 bp\n",
      "Quality-trimmed:             619,548,175 bp (22.5%)\n",
      "Total written (filtered):  2,086,445,118 bp (75.6%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 7093393 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 25.3%\n",
      "  C: 23.0%\n",
      "  G: 17.6%\n",
      "  T: 21.2%\n",
      "  none/other: 13.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t4550597\t13600316.0\t0\t4550597\n",
      "2\t1270560\t3400079.0\t0\t1270560\n",
      "3\t303840\t850019.8\t0\t303840\n",
      "4\t59349\t212504.9\t0\t59349\n",
      "5\t10650\t53126.2\t0\t10650\n",
      "6\t1297\t13281.6\t0\t1297\n",
      "7\t435\t3320.4\t0\t435\n",
      "8\t465\t830.1\t0\t465\n",
      "9\t601\t207.5\t0\t279 322\n",
      "10\t929\t51.9\t1\t223 706\n",
      "11\t824\t13.0\t1\t327 497\n",
      "12\t720\t3.2\t1\t227 493\n",
      "13\t688\t0.8\t1\t370 318\n",
      "14\t515\t0.8\t1\t391 124\n",
      "15\t540\t0.8\t1\t451 89\n",
      "16\t427\t0.8\t1\t351 76\n",
      "17\t392\t0.8\t1\t314 78\n",
      "18\t449\t0.8\t1\t332 117\n",
      "19\t482\t0.8\t1\t374 108\n",
      "20\t435\t0.8\t1\t324 111\n",
      "21\t446\t0.8\t1\t344 102\n",
      "22\t525\t0.8\t1\t383 142\n",
      "23\t501\t0.8\t1\t397 104\n",
      "24\t487\t0.8\t1\t394 93\n",
      "25\t392\t0.8\t1\t282 110\n",
      "26\t680\t0.8\t1\t542 138\n",
      "27\t493\t0.8\t1\t347 146\n",
      "28\t587\t0.8\t1\t408 179\n",
      "29\t498\t0.8\t1\t351 147\n",
      "30\t671\t0.8\t1\t485 186\n",
      "31\t561\t0.8\t1\t404 157\n",
      "32\t587\t0.8\t1\t403 184\n",
      "33\t566\t0.8\t1\t382 184\n",
      "34\t793\t0.8\t1\t490 303\n",
      "35\t1117\t0.8\t1\t622 495\n",
      "36\t1145\t0.8\t1\t615 530\n",
      "37\t1334\t0.8\t1\t784 550\n",
      "38\t1107\t0.8\t1\t561 546\n",
      "39\t1426\t0.8\t1\t700 726\n",
      "40\t2023\t0.8\t1\t1069 954\n",
      "41\t1478\t0.8\t1\t808 670\n",
      "42\t1824\t0.8\t1\t932 892\n",
      "43\t2335\t0.8\t1\t1071 1264\n",
      "44\t5866\t0.8\t1\t1643 4223\n",
      "45\t12347\t0.8\t1\t6226 6121\n",
      "46\t11955\t0.8\t1\t5791 6164\n",
      "47\t18069\t0.8\t1\t9710 8359\n",
      "48\t24440\t0.8\t1\t10744 13696\n",
      "49\t121839\t0.8\t1\t56655 65184\n",
      "50\t11922\t0.8\t1\t6555 5367\n",
      "51\t661184\t0.8\t1\t353667 307517\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900785.fastq\n",
      "=============================================\n",
      "54401264 sequences processed in total\n",
      "Sequences removed because they became shorter than the length cutoff of 20 bp:\t12492537 (23.0%)\n",
      "\n",
      "Now renaming the output file SRR6900785_trimmed.fq\n",
      "\n",
      "ORIGINAL FILE 1: >>SRR6900785_trimmed.fq<<\tRENAMING TO:>>Human_Heart_R2_trimmed.fq<<\n",
      "\n",
      "  >>> Now running FastQC on the data <<<\n",
      "\n",
      "Started analysis of Human_Heart_R2_trimmed.fq\n",
      "Approx 5% complete for Human_Heart_R2_trimmed.fq\n",
      "Approx 10% complete for Human_Heart_R2_trimmed.fq\n",
      "Approx 15% complete for Human_Heart_R2_trimmed.fq\n",
      "Approx 20% complete for Human_Heart_R2_trimmed.fq\n",
      "Approx 25% complete for Human_Heart_R2_trimmed.fq\n",
      "Approx 30% complete for Human_Heart_R2_trimmed.fq\n",
      "Approx 35% complete for Human_Heart_R2_trimmed.fq\n",
      "Approx 40% complete for Human_Heart_R2_trimmed.fq\n",
      "Approx 45% complete for Human_Heart_R2_trimmed.fq\n",
      "Approx 50% complete for Human_Heart_R2_trimmed.fq\n",
      "Approx 55% complete for Human_Heart_R2_trimmed.fq\n",
      "Approx 60% complete for Human_Heart_R2_trimmed.fq\n",
      "Approx 65% complete for Human_Heart_R2_trimmed.fq\n",
      "Approx 70% complete for Human_Heart_R2_trimmed.fq\n",
      "Approx 75% complete for Human_Heart_R2_trimmed.fq\n",
      "Approx 80% complete for Human_Heart_R2_trimmed.fq\n",
      "Approx 85% complete for Human_Heart_R2_trimmed.fq\n",
      "Approx 90% complete for Human_Heart_R2_trimmed.fq\n",
      "Approx 95% complete for Human_Heart_R2_trimmed.fq\n",
      "Analysis complete for Human_Heart_R2_trimmed.fq\n",
      "Using an excessive number of cores has a diminishing return! It is recommended not to exceed 8 cores per trimming process (you asked for 8 cores). Please consider re-specifying\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Cutadapt version: 1.18\n",
      "Could not detect version of Python used by Cutadapt from the first line of Cutadapt (but found this: >>>#!/bin/sh<<<)\n",
      "Letting the (modified) Cutadapt deal with the Python version instead\n",
      "pigz 2.6\n",
      "Parallel gzip (pigz) detected. Proceeding with multicore (de)compression using 8 cores\n",
      "\n",
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Output will be written into the directory: /gpfs52/data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/trimmed_reads/\n",
      "Using user-specified basename (>>Human_Kidney_R2<<) instead of deriving the filename from the input file(s)\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900786.fastq <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Illumina\t5141\tAGATCGGAAGAGC\t1000000\t0.51\n",
      "smallRNA\t2\tTGGAATTCTCGG\t1000000\t0.00\n",
      "Nextera\t2\tCTGTCTCTTATA\t1000000\t0.00\n",
      "Using Illumina adapter for trimming (count: 5141). Second best hit was smallRNA (count: 2)\n",
      "\n",
      "Writing report to '/gpfs52/data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/trimmed_reads/SRR6900786.fastq_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900786.fastq\n",
      "Trimming mode: single-end\n",
      "Trim Galore version: 0.6.6\n",
      "Cutadapt version: 1.18\n",
      "Python version: could not detect\n",
      "Number of cores used for trimming: 8\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length before a sequence gets removed: 20 bp\n",
      "Running FastQC on the data once trimming has completed\n",
      "Running FastQC with the following extra arguments: '--outdir /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/processing_results/fastQC'\n",
      "\n",
      "Cutadapt seems to be reasonably up-to-date. Setting -j 8\n",
      "Writing final adapter and quality trimmed output to SRR6900786_trimmed.fq\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff '-q 20') and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900786.fastq <<< \n",
      "10000000 sequences processed\n",
      "20000000 sequences processed\n",
      "30000000 sequences processed\n",
      "This is cutadapt 1.18 with Python 3.6.15\n",
      "Command line parameters: -j 8 -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900786.fastq\n",
      "Processing reads on 8 cores in single-end mode ...\n",
      "Finished in 152.74 s (4 us/read; 15.67 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              39,897,793\n",
      "Reads with adapters:                 4,134,136 (10.4%)\n",
      "Reads written (passing filters):    39,897,793 (100.0%)\n",
      "\n",
      "Total basepairs processed: 2,028,835,502 bp\n",
      "Quality-trimmed:             529,913,392 bp (26.1%)\n",
      "Total written (filtered):  1,434,474,715 bp (70.7%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 4134136 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 19.6%\n",
      "  C: 18.1%\n",
      "  G: 15.1%\n",
      "  T: 17.0%\n",
      "  none/other: 30.1%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t2168451\t9974448.2\t0\t2168451\n",
      "2\t557187\t2493612.1\t0\t557187\n",
      "3\t163871\t623403.0\t0\t163871\n",
      "4\t29007\t155850.8\t0\t29007\n",
      "5\t5834\t38962.7\t0\t5834\n",
      "6\t918\t9740.7\t0\t918\n",
      "7\t525\t2435.2\t0\t525\n",
      "8\t722\t608.8\t0\t722\n",
      "9\t869\t152.2\t0\t398 471\n",
      "10\t1078\t38.0\t1\t290 788\n",
      "11\t1245\t9.5\t1\t579 666\n",
      "12\t1010\t2.4\t1\t336 674\n",
      "13\t1088\t0.6\t1\t493 595\n",
      "14\t620\t0.6\t1\t389 231\n",
      "15\t560\t0.6\t1\t380 180\n",
      "16\t460\t0.6\t1\t300 160\n",
      "17\t380\t0.6\t1\t273 107\n",
      "18\t555\t0.6\t1\t228 327\n",
      "19\t607\t0.6\t1\t412 195\n",
      "20\t417\t0.6\t1\t275 142\n",
      "21\t360\t0.6\t1\t199 161\n",
      "22\t423\t0.6\t1\t274 149\n",
      "23\t312\t0.6\t1\t242 70\n",
      "24\t315\t0.6\t1\t218 97\n",
      "25\t348\t0.6\t1\t211 137\n",
      "26\t495\t0.6\t1\t354 141\n",
      "27\t394\t0.6\t1\t249 145\n",
      "28\t411\t0.6\t1\t238 173\n",
      "29\t430\t0.6\t1\t292 138\n",
      "30\t591\t0.6\t1\t381 210\n",
      "31\t547\t0.6\t1\t326 221\n",
      "32\t1426\t0.6\t1\t966 460\n",
      "33\t740\t0.6\t1\t538 202\n",
      "34\t688\t0.6\t1\t426 262\n",
      "35\t662\t0.6\t1\t356 306\n",
      "36\t737\t0.6\t1\t413 324\n",
      "37\t899\t0.6\t1\t454 445\n",
      "38\t915\t0.6\t1\t412 503\n",
      "39\t1947\t0.6\t1\t643 1304\n",
      "40\t3364\t0.6\t1\t1804 1560\n",
      "41\t2070\t0.6\t1\t1235 835\n",
      "42\t1779\t0.6\t1\t996 783\n",
      "43\t1854\t0.6\t1\t912 942\n",
      "44\t2600\t0.6\t1\t1084 1516\n",
      "45\t4178\t0.6\t1\t1827 2351\n",
      "46\t6893\t0.6\t1\t2598 4295\n",
      "47\t13340\t0.6\t1\t6840 6500\n",
      "48\t18045\t0.6\t1\t7866 10179\n",
      "49\t110338\t0.6\t1\t47333 63005\n",
      "50\t12351\t0.6\t1\t7958 4393\n",
      "51\t1009280\t0.6\t1\t504540 504740\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900786.fastq\n",
      "=============================================\n",
      "39897793 sequences processed in total\n",
      "Sequences removed because they became shorter than the length cutoff of 20 bp:\t11388462 (28.5%)\n",
      "\n",
      "Now renaming the output file SRR6900786_trimmed.fq\n",
      "\n",
      "ORIGINAL FILE 1: >>SRR6900786_trimmed.fq<<\tRENAMING TO:>>Human_Kidney_R2_trimmed.fq<<\n",
      "\n",
      "  >>> Now running FastQC on the data <<<\n",
      "\n",
      "Started analysis of Human_Kidney_R2_trimmed.fq\n",
      "Approx 5% complete for Human_Kidney_R2_trimmed.fq\n",
      "Approx 10% complete for Human_Kidney_R2_trimmed.fq\n",
      "Approx 15% complete for Human_Kidney_R2_trimmed.fq\n",
      "Approx 20% complete for Human_Kidney_R2_trimmed.fq\n",
      "Approx 25% complete for Human_Kidney_R2_trimmed.fq\n",
      "Approx 30% complete for Human_Kidney_R2_trimmed.fq\n",
      "Approx 35% complete for Human_Kidney_R2_trimmed.fq\n",
      "Approx 40% complete for Human_Kidney_R2_trimmed.fq\n",
      "Approx 45% complete for Human_Kidney_R2_trimmed.fq\n",
      "Approx 50% complete for Human_Kidney_R2_trimmed.fq\n",
      "Approx 55% complete for Human_Kidney_R2_trimmed.fq\n",
      "Approx 60% complete for Human_Kidney_R2_trimmed.fq\n",
      "Approx 65% complete for Human_Kidney_R2_trimmed.fq\n",
      "Approx 70% complete for Human_Kidney_R2_trimmed.fq\n",
      "Approx 75% complete for Human_Kidney_R2_trimmed.fq\n",
      "Approx 80% complete for Human_Kidney_R2_trimmed.fq\n",
      "Approx 85% complete for Human_Kidney_R2_trimmed.fq\n",
      "Approx 90% complete for Human_Kidney_R2_trimmed.fq\n",
      "Approx 95% complete for Human_Kidney_R2_trimmed.fq\n",
      "Analysis complete for Human_Kidney_R2_trimmed.fq\n",
      "Using an excessive number of cores has a diminishing return! It is recommended not to exceed 8 cores per trimming process (you asked for 8 cores). Please consider re-specifying\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "Cutadapt version: 1.18\n",
      "Could not detect version of Python used by Cutadapt from the first line of Cutadapt (but found this: >>>#!/bin/sh<<<)\n",
      "Letting the (modified) Cutadapt deal with the Python version instead\n",
      "pigz 2.6\n",
      "Parallel gzip (pigz) detected. Proceeding with multicore (de)compression using 8 cores\n",
      "\n",
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Output will be written into the directory: /gpfs52/data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/trimmed_reads/\n",
      "Using user-specified basename (>>Human_Liver_R2<<) instead of deriving the filename from the input file(s)\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900787.fastq <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Illumina\t6244\tAGATCGGAAGAGC\t1000000\t0.62\n",
      "smallRNA\t1\tTGGAATTCTCGG\t1000000\t0.00\n",
      "Nextera\t0\tCTGTCTCTTATA\t1000000\t0.00\n",
      "Using Illumina adapter for trimming (count: 6244). Second best hit was smallRNA (count: 1)\n",
      "\n",
      "Writing report to '/gpfs52/data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/trimmed_reads/SRR6900787.fastq_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900787.fastq\n",
      "Trimming mode: single-end\n",
      "Trim Galore version: 0.6.6\n",
      "Cutadapt version: 1.18\n",
      "Python version: could not detect\n",
      "Number of cores used for trimming: 8\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length before a sequence gets removed: 20 bp\n",
      "Running FastQC on the data once trimming has completed\n",
      "Running FastQC with the following extra arguments: '--outdir /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/processing_results/fastQC'\n",
      "\n",
      "Cutadapt seems to be reasonably up-to-date. Setting -j 8\n",
      "Writing final adapter and quality trimmed output to SRR6900787_trimmed.fq\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff '-q 20') and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900787.fastq <<< \n",
      "10000000 sequences processed\n",
      "20000000 sequences processed\n",
      "30000000 sequences processed\n",
      "40000000 sequences processed\n",
      "This is cutadapt 1.18 with Python 3.6.15\n",
      "Command line parameters: -j 8 -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900787.fastq\n",
      "Processing reads on 8 cores in single-end mode ...\n",
      "Finished in 181.63 s (4 us/read; 15.19 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              45,973,935\n",
      "Reads with adapters:                 4,693,757 (10.2%)\n",
      "Reads written (passing filters):    45,973,935 (100.0%)\n",
      "\n",
      "Total basepairs processed: 2,338,045,372 bp\n",
      "Quality-trimmed:             617,434,564 bp (26.4%)\n",
      "Total written (filtered):  1,638,845,413 bp (70.1%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 4693757 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 19.5%\n",
      "  C: 19.7%\n",
      "  G: 15.3%\n",
      "  T: 12.0%\n",
      "  none/other: 33.5%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t2256996\t11493483.8\t0\t2256996\n",
      "2\t639363\t2873370.9\t0\t639363\n",
      "3\t202340\t718342.7\t0\t202340\n",
      "4\t35759\t179585.7\t0\t35759\n",
      "5\t8645\t44896.4\t0\t8645\n",
      "6\t1185\t11224.1\t0\t1185\n",
      "7\t835\t2806.0\t0\t835\n",
      "8\t582\t701.5\t0\t582\n",
      "9\t756\t175.4\t0\t393 363\n",
      "10\t1426\t43.8\t1\t260 1166\n",
      "11\t1179\t11.0\t1\t489 690\n",
      "12\t943\t2.7\t1\t255 688\n",
      "13\t991\t0.7\t1\t457 534\n",
      "14\t636\t0.7\t1\t375 261\n",
      "15\t685\t0.7\t1\t452 233\n",
      "16\t631\t0.7\t1\t324 307\n",
      "17\t593\t0.7\t1\t424 169\n",
      "18\t694\t0.7\t1\t312 382\n",
      "19\t682\t0.7\t1\t453 229\n",
      "20\t523\t0.7\t1\t340 183\n",
      "21\t500\t0.7\t1\t323 177\n",
      "22\t664\t0.7\t1\t351 313\n",
      "23\t726\t0.7\t1\t502 224\n",
      "24\t603\t0.7\t1\t422 181\n",
      "25\t1289\t0.7\t1\t805 484\n",
      "26\t508\t0.7\t1\t342 166\n",
      "27\t680\t0.7\t1\t419 261\n",
      "28\t1142\t0.7\t1\t825 317\n",
      "29\t5054\t0.7\t1\t4208 846\n",
      "30\t1065\t0.7\t1\t646 419\n",
      "31\t3253\t0.7\t1\t2554 699\n",
      "32\t1269\t0.7\t1\t859 410\n",
      "33\t837\t0.7\t1\t587 250\n",
      "34\t748\t0.7\t1\t465 283\n",
      "35\t763\t0.7\t1\t424 339\n",
      "36\t781\t0.7\t1\t459 322\n",
      "37\t1051\t0.7\t1\t481 570\n",
      "38\t1231\t0.7\t1\t518 713\n",
      "39\t1810\t0.7\t1\t758 1052\n",
      "40\t3075\t0.7\t1\t1442 1633\n",
      "41\t1901\t0.7\t1\t956 945\n",
      "42\t2146\t0.7\t1\t903 1243\n",
      "43\t2501\t0.7\t1\t1148 1353\n",
      "44\t5983\t0.7\t1\t1418 4565\n",
      "45\t9958\t0.7\t1\t4571 5387\n",
      "46\t11442\t0.7\t1\t4478 6964\n",
      "47\t19549\t0.7\t1\t8629 10920\n",
      "48\t26257\t0.7\t1\t10077 16180\n",
      "49\t126880\t0.7\t1\t48995 77885\n",
      "50\t20426\t0.7\t1\t9976 10450\n",
      "51\t1284221\t0.7\t1\t584807 699414\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900787.fastq\n",
      "=============================================\n",
      "45973935 sequences processed in total\n",
      "Sequences removed because they became shorter than the length cutoff of 20 bp:\t13396508 (29.1%)\n",
      "\n",
      "Now renaming the output file SRR6900787_trimmed.fq\n",
      "\n",
      "ORIGINAL FILE 1: >>SRR6900787_trimmed.fq<<\tRENAMING TO:>>Human_Liver_R2_trimmed.fq<<\n",
      "\n",
      "  >>> Now running FastQC on the data <<<\n",
      "\n",
      "Started analysis of Human_Liver_R2_trimmed.fq\n",
      "Approx 5% complete for Human_Liver_R2_trimmed.fq\n",
      "Approx 10% complete for Human_Liver_R2_trimmed.fq\n",
      "Approx 15% complete for Human_Liver_R2_trimmed.fq\n",
      "Approx 20% complete for Human_Liver_R2_trimmed.fq\n",
      "Approx 25% complete for Human_Liver_R2_trimmed.fq\n",
      "Approx 30% complete for Human_Liver_R2_trimmed.fq\n",
      "Approx 35% complete for Human_Liver_R2_trimmed.fq\n",
      "Approx 40% complete for Human_Liver_R2_trimmed.fq\n",
      "Approx 45% complete for Human_Liver_R2_trimmed.fq\n",
      "Approx 50% complete for Human_Liver_R2_trimmed.fq\n",
      "Approx 55% complete for Human_Liver_R2_trimmed.fq\n",
      "Approx 60% complete for Human_Liver_R2_trimmed.fq\n",
      "Approx 65% complete for Human_Liver_R2_trimmed.fq\n",
      "Approx 70% complete for Human_Liver_R2_trimmed.fq\n",
      "Approx 75% complete for Human_Liver_R2_trimmed.fq\n",
      "Approx 80% complete for Human_Liver_R2_trimmed.fq\n",
      "Approx 85% complete for Human_Liver_R2_trimmed.fq\n",
      "Approx 90% complete for Human_Liver_R2_trimmed.fq\n",
      "Approx 95% complete for Human_Liver_R2_trimmed.fq\n",
      "Analysis complete for Human_Liver_R2_trimmed.fq\n",
      "Using an excessive number of cores has a diminishing return! It is recommended not to exceed 8 cores per trimming process (you asked for 8 cores). Please consider re-specifying\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "Cutadapt version: 1.18\n",
      "Could not detect version of Python used by Cutadapt from the first line of Cutadapt (but found this: >>>#!/bin/sh<<<)\n",
      "Letting the (modified) Cutadapt deal with the Python version instead\n",
      "pigz 2.6\n",
      "Parallel gzip (pigz) detected. Proceeding with multicore (de)compression using 8 cores\n",
      "\n",
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Output will be written into the directory: /gpfs52/data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/trimmed_reads/\n",
      "Using user-specified basename (>>Human_Lung_R2<<) instead of deriving the filename from the input file(s)\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900788.fastq <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Illumina\t4515\tAGATCGGAAGAGC\t1000000\t0.45\n",
      "Nextera\t4\tCTGTCTCTTATA\t1000000\t0.00\n",
      "smallRNA\t1\tTGGAATTCTCGG\t1000000\t0.00\n",
      "Using Illumina adapter for trimming (count: 4515). Second best hit was Nextera (count: 4)\n",
      "\n",
      "Writing report to '/gpfs52/data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/trimmed_reads/SRR6900788.fastq_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900788.fastq\n",
      "Trimming mode: single-end\n",
      "Trim Galore version: 0.6.6\n",
      "Cutadapt version: 1.18\n",
      "Python version: could not detect\n",
      "Number of cores used for trimming: 8\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length before a sequence gets removed: 20 bp\n",
      "Running FastQC on the data once trimming has completed\n",
      "Running FastQC with the following extra arguments: '--outdir /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/processing_results/fastQC'\n",
      "\n",
      "Cutadapt seems to be reasonably up-to-date. Setting -j 8\n",
      "Writing final adapter and quality trimmed output to SRR6900788_trimmed.fq\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff '-q 20') and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900788.fastq <<< \n",
      "10000000 sequences processed\n",
      "20000000 sequences processed\n",
      "30000000 sequences processed\n",
      "40000000 sequences processed\n",
      "50000000 sequences processed\n",
      "This is cutadapt 1.18 with Python 3.6.15\n",
      "Command line parameters: -j 8 -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900788.fastq\n",
      "Processing reads on 8 cores in single-end mode ...\n",
      "Finished in 219.65 s (4 us/read; 15.28 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              55,919,583\n",
      "Reads with adapters:                 6,669,919 (11.9%)\n",
      "Reads written (passing filters):    55,919,583 (100.0%)\n",
      "\n",
      "Total basepairs processed: 2,840,334,189 bp\n",
      "Quality-trimmed:             698,958,066 bp (24.6%)\n",
      "Total written (filtered):  2,042,886,312 bp (71.9%)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 6669919 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 21.4%\n",
      "  C: 23.0%\n",
      "  G: 16.0%\n",
      "  T: 11.7%\n",
      "  none/other: 27.9%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t3414380\t13979895.8\t0\t3414380\n",
      "2\t1029903\t3494973.9\t0\t1029903\n",
      "3\t306483\t873743.5\t0\t306483\n",
      "4\t58452\t218435.9\t0\t58452\n",
      "5\t14773\t54609.0\t0\t14773\n",
      "6\t2145\t13652.2\t0\t2145\n",
      "7\t936\t3413.1\t0\t936\n",
      "8\t953\t853.3\t0\t953\n",
      "9\t1280\t213.3\t0\t478 802\n",
      "10\t1992\t53.3\t1\t477 1515\n",
      "11\t1907\t13.3\t1\t698 1209\n",
      "12\t1570\t3.3\t1\t478 1092\n",
      "13\t1471\t0.8\t1\t663 808\n",
      "14\t1040\t0.8\t1\t818 222\n",
      "15\t1128\t0.8\t1\t884 244\n",
      "16\t981\t0.8\t1\t769 212\n",
      "17\t944\t0.8\t1\t764 180\n",
      "18\t1049\t0.8\t1\t739 310\n",
      "19\t1140\t0.8\t1\t861 279\n",
      "20\t1006\t0.8\t1\t791 215\n",
      "21\t907\t0.8\t1\t717 190\n",
      "22\t1134\t0.8\t1\t786 348\n",
      "23\t1213\t0.8\t1\t929 284\n",
      "24\t1146\t0.8\t1\t927 219\n",
      "25\t1170\t0.8\t1\t891 279\n",
      "26\t1134\t0.8\t1\t920 214\n",
      "27\t1155\t0.8\t1\t904 251\n",
      "28\t1231\t0.8\t1\t913 318\n",
      "29\t1353\t0.8\t1\t1068 285\n",
      "30\t1546\t0.8\t1\t1138 408\n",
      "31\t1737\t0.8\t1\t1346 391\n",
      "32\t1486\t0.8\t1\t1172 314\n",
      "33\t1439\t0.8\t1\t1066 373\n",
      "34\t1679\t0.8\t1\t1230 449\n",
      "35\t1867\t0.8\t1\t1295 572\n",
      "36\t1759\t0.8\t1\t1255 504\n",
      "37\t2301\t0.8\t1\t1448 853\n",
      "38\t3065\t0.8\t1\t1399 1666\n",
      "39\t4172\t0.8\t1\t2205 1967\n",
      "40\t5553\t0.8\t1\t2684 2869\n",
      "41\t4092\t0.8\t1\t2051 2041\n",
      "42\t4224\t0.8\t1\t2308 1916\n",
      "43\t3953\t0.8\t1\t1887 2066\n",
      "44\t6927\t0.8\t1\t2031 4896\n",
      "45\t12738\t0.8\t1\t5252 7486\n",
      "46\t16937\t0.8\t1\t7618 9319\n",
      "47\t23264\t0.8\t1\t9659 13605\n",
      "48\t40316\t0.8\t1\t15918 24398\n",
      "49\t191405\t0.8\t1\t73632 117773\n",
      "50\t22600\t0.8\t1\t11456 11144\n",
      "51\t1462883\t0.8\t1\t642617 820266\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900788.fastq\n",
      "=============================================\n",
      "55919583 sequences processed in total\n",
      "Sequences removed because they became shorter than the length cutoff of 20 bp:\t15095798 (27.0%)\n",
      "\n",
      "Now renaming the output file SRR6900788_trimmed.fq\n",
      "\n",
      "ORIGINAL FILE 1: >>SRR6900788_trimmed.fq<<\tRENAMING TO:>>Human_Lung_R2_trimmed.fq<<\n",
      "\n",
      "  >>> Now running FastQC on the data <<<\n",
      "\n",
      "Started analysis of Human_Lung_R2_trimmed.fq\n",
      "Approx 5% complete for Human_Lung_R2_trimmed.fq\n",
      "Approx 10% complete for Human_Lung_R2_trimmed.fq\n",
      "Approx 15% complete for Human_Lung_R2_trimmed.fq\n",
      "Approx 20% complete for Human_Lung_R2_trimmed.fq\n",
      "Approx 25% complete for Human_Lung_R2_trimmed.fq\n",
      "Approx 30% complete for Human_Lung_R2_trimmed.fq\n",
      "Approx 35% complete for Human_Lung_R2_trimmed.fq\n",
      "Approx 40% complete for Human_Lung_R2_trimmed.fq\n",
      "Approx 45% complete for Human_Lung_R2_trimmed.fq\n",
      "Approx 50% complete for Human_Lung_R2_trimmed.fq\n",
      "Approx 55% complete for Human_Lung_R2_trimmed.fq\n",
      "Approx 60% complete for Human_Lung_R2_trimmed.fq\n",
      "Approx 65% complete for Human_Lung_R2_trimmed.fq\n",
      "Approx 70% complete for Human_Lung_R2_trimmed.fq\n",
      "Approx 75% complete for Human_Lung_R2_trimmed.fq\n",
      "Approx 80% complete for Human_Lung_R2_trimmed.fq\n",
      "Approx 85% complete for Human_Lung_R2_trimmed.fq\n",
      "Approx 90% complete for Human_Lung_R2_trimmed.fq\n",
      "Approx 95% complete for Human_Lung_R2_trimmed.fq\n",
      "Analysis complete for Human_Lung_R2_trimmed.fq\n",
      "Using an excessive number of cores has a diminishing return! It is recommended not to exceed 8 cores per trimming process (you asked for 8 cores). Please consider re-specifying\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "Cutadapt version: 1.18\n",
      "Could not detect version of Python used by Cutadapt from the first line of Cutadapt (but found this: >>>#!/bin/sh<<<)\n",
      "Letting the (modified) Cutadapt deal with the Python version instead\n",
      "pigz 2.6\n",
      "Parallel gzip (pigz) detected. Proceeding with multicore (de)compression using 8 cores\n",
      "\n",
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Output will be written into the directory: /gpfs52/data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/trimmed_reads/\n",
      "Using user-specified basename (>>Human_Heart_R3<<) instead of deriving the filename from the input file(s)\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900789.fastq <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Illumina\t532\tAGATCGGAAGAGC\t1000000\t0.05\n",
      "Nextera\t1\tCTGTCTCTTATA\t1000000\t0.00\n",
      "smallRNA\t0\tTGGAATTCTCGG\t1000000\t0.00\n",
      "Using Illumina adapter for trimming (count: 532). Second best hit was Nextera (count: 1)\n",
      "\n",
      "Writing report to '/gpfs52/data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/trimmed_reads/SRR6900789.fastq_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900789.fastq\n",
      "Trimming mode: single-end\n",
      "Trim Galore version: 0.6.6\n",
      "Cutadapt version: 1.18\n",
      "Python version: could not detect\n",
      "Number of cores used for trimming: 8\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length before a sequence gets removed: 20 bp\n",
      "Running FastQC on the data once trimming has completed\n",
      "Running FastQC with the following extra arguments: '--outdir /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/processing_results/fastQC'\n",
      "\n",
      "Cutadapt seems to be reasonably up-to-date. Setting -j 8\n",
      "Writing final adapter and quality trimmed output to SRR6900789_trimmed.fq\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff '-q 20') and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900789.fastq <<< \n",
      "10000000 sequences processed\n",
      "20000000 sequences processed\n",
      "30000000 sequences processed\n",
      "40000000 sequences processed\n",
      "This is cutadapt 1.18 with Python 3.6.15\n",
      "Command line parameters: -j 8 -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900789.fastq\n",
      "Processing reads on 8 cores in single-end mode ...\n",
      "Finished in 183.29 s (4 us/read; 14.68 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              44,831,283\n",
      "Reads with adapters:                 7,428,545 (16.6%)\n",
      "Reads written (passing filters):    44,831,283 (100.0%)\n",
      "\n",
      "Total basepairs processed: 2,265,206,838 bp\n",
      "Quality-trimmed:             397,358,835 bp (17.5%)\n",
      "Total written (filtered):  1,854,957,797 bp (81.9%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 7428545 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 27.6%\n",
      "  C: 25.7%\n",
      "  G: 19.6%\n",
      "  T: 26.0%\n",
      "  none/other: 1.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t5350100\t11207820.8\t0\t5350100\n",
      "2\t1612904\t2801955.2\t0\t1612904\n",
      "3\t319811\t700488.8\t0\t319811\n",
      "4\t73489\t175122.2\t0\t73489\n",
      "5\t9197\t43780.5\t0\t9197\n",
      "6\t1022\t10945.1\t0\t1022\n",
      "7\t192\t2736.3\t0\t192\n",
      "8\t63\t684.1\t0\t63\n",
      "9\t264\t171.0\t0\t50 214\n",
      "10\t250\t42.8\t1\t29 221\n",
      "11\t204\t10.7\t1\t51 153\n",
      "12\t94\t2.7\t1\t26 68\n",
      "13\t82\t0.7\t1\t41 41\n",
      "14\t55\t0.7\t1\t35 20\n",
      "15\t64\t0.7\t1\t49 15\n",
      "16\t54\t0.7\t1\t39 15\n",
      "17\t50\t0.7\t1\t34 16\n",
      "18\t55\t0.7\t1\t36 19\n",
      "19\t72\t0.7\t1\t48 24\n",
      "20\t59\t0.7\t1\t39 20\n",
      "21\t57\t0.7\t1\t33 24\n",
      "22\t50\t0.7\t1\t33 17\n",
      "23\t61\t0.7\t1\t36 25\n",
      "24\t49\t0.7\t1\t22 27\n",
      "25\t47\t0.7\t1\t26 21\n",
      "26\t78\t0.7\t1\t36 42\n",
      "27\t64\t0.7\t1\t30 34\n",
      "28\t72\t0.7\t1\t29 43\n",
      "29\t81\t0.7\t1\t30 51\n",
      "30\t87\t0.7\t1\t31 56\n",
      "31\t60\t0.7\t1\t22 38\n",
      "32\t89\t0.7\t1\t32 57\n",
      "33\t76\t0.7\t1\t32 44\n",
      "34\t73\t0.7\t1\t38 35\n",
      "35\t60\t0.7\t1\t28 32\n",
      "36\t81\t0.7\t1\t45 36\n",
      "37\t77\t0.7\t1\t34 43\n",
      "38\t82\t0.7\t1\t37 45\n",
      "39\t112\t0.7\t1\t45 67\n",
      "40\t182\t0.7\t1\t71 111\n",
      "41\t122\t0.7\t1\t45 77\n",
      "42\t192\t0.7\t1\t80 112\n",
      "43\t213\t0.7\t1\t97 116\n",
      "44\t419\t0.7\t1\t133 286\n",
      "45\t858\t0.7\t1\t413 445\n",
      "46\t836\t0.7\t1\t435 401\n",
      "47\t1177\t0.7\t1\t580 597\n",
      "48\t2329\t0.7\t1\t975 1354\n",
      "49\t16878\t0.7\t1\t8000 8878\n",
      "50\t1323\t0.7\t1\t499 824\n",
      "51\t34579\t0.7\t1\t18480 16099\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900789.fastq\n",
      "=============================================\n",
      "44831283 sequences processed in total\n",
      "Sequences removed because they became shorter than the length cutoff of 20 bp:\t7136904 (15.9%)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Now renaming the output file SRR6900789_trimmed.fq\n",
      "\n",
      "ORIGINAL FILE 1: >>SRR6900789_trimmed.fq<<\tRENAMING TO:>>Human_Heart_R3_trimmed.fq<<\n",
      "\n",
      "  >>> Now running FastQC on the data <<<\n",
      "\n",
      "Started analysis of Human_Heart_R3_trimmed.fq\n",
      "Approx 5% complete for Human_Heart_R3_trimmed.fq\n",
      "Approx 10% complete for Human_Heart_R3_trimmed.fq\n",
      "Approx 15% complete for Human_Heart_R3_trimmed.fq\n",
      "Approx 20% complete for Human_Heart_R3_trimmed.fq\n",
      "Approx 25% complete for Human_Heart_R3_trimmed.fq\n",
      "Approx 30% complete for Human_Heart_R3_trimmed.fq\n",
      "Approx 35% complete for Human_Heart_R3_trimmed.fq\n",
      "Approx 40% complete for Human_Heart_R3_trimmed.fq\n",
      "Approx 45% complete for Human_Heart_R3_trimmed.fq\n",
      "Approx 50% complete for Human_Heart_R3_trimmed.fq\n",
      "Approx 55% complete for Human_Heart_R3_trimmed.fq\n",
      "Approx 60% complete for Human_Heart_R3_trimmed.fq\n",
      "Approx 65% complete for Human_Heart_R3_trimmed.fq\n",
      "Approx 70% complete for Human_Heart_R3_trimmed.fq\n",
      "Approx 75% complete for Human_Heart_R3_trimmed.fq\n",
      "Approx 80% complete for Human_Heart_R3_trimmed.fq\n",
      "Approx 85% complete for Human_Heart_R3_trimmed.fq\n",
      "Approx 90% complete for Human_Heart_R3_trimmed.fq\n",
      "Approx 95% complete for Human_Heart_R3_trimmed.fq\n",
      "Analysis complete for Human_Heart_R3_trimmed.fq\n",
      "Using an excessive number of cores has a diminishing return! It is recommended not to exceed 8 cores per trimming process (you asked for 8 cores). Please consider re-specifying\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "Cutadapt version: 1.18\n",
      "Could not detect version of Python used by Cutadapt from the first line of Cutadapt (but found this: >>>#!/bin/sh<<<)\n",
      "Letting the (modified) Cutadapt deal with the Python version instead\n",
      "pigz 2.6\n",
      "Parallel gzip (pigz) detected. Proceeding with multicore (de)compression using 8 cores\n",
      "\n",
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Output will be written into the directory: /gpfs52/data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/trimmed_reads/\n",
      "Using user-specified basename (>>Human_Kidney_R3<<) instead of deriving the filename from the input file(s)\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900790.fastq <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Illumina\t39\tAGATCGGAAGAGC\t1000000\t0.00\n",
      "Nextera\t3\tCTGTCTCTTATA\t1000000\t0.00\n",
      "smallRNA\t0\tTGGAATTCTCGG\t1000000\t0.00\n",
      "Using Illumina adapter for trimming (count: 39). Second best hit was Nextera (count: 3)\n",
      "\n",
      "Writing report to '/gpfs52/data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/trimmed_reads/SRR6900790.fastq_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900790.fastq\n",
      "Trimming mode: single-end\n",
      "Trim Galore version: 0.6.6\n",
      "Cutadapt version: 1.18\n",
      "Python version: could not detect\n",
      "Number of cores used for trimming: 8\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length before a sequence gets removed: 20 bp\n",
      "Running FastQC on the data once trimming has completed\n",
      "Running FastQC with the following extra arguments: '--outdir /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/processing_results/fastQC'\n",
      "\n",
      "Cutadapt seems to be reasonably up-to-date. Setting -j 8\n",
      "Writing final adapter and quality trimmed output to SRR6900790_trimmed.fq\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff '-q 20') and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900790.fastq <<< \n",
      "10000000 sequences processed\n",
      "20000000 sequences processed\n",
      "30000000 sequences processed\n",
      "40000000 sequences processed\n",
      "50000000 sequences processed\n",
      "This is cutadapt 1.18 with Python 3.6.15\n",
      "Command line parameters: -j 8 -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900790.fastq\n",
      "Processing reads on 8 cores in single-end mode ...\n",
      "Finished in 210.97 s (4 us/read; 15.14 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              53,229,453\n",
      "Reads with adapters:                 8,361,169 (15.7%)\n",
      "Reads written (passing filters):    53,229,453 (100.0%)\n",
      "\n",
      "Total basepairs processed: 2,692,482,242 bp\n",
      "Quality-trimmed:             521,706,567 bp (19.4%)\n",
      "Total written (filtered):  2,159,082,360 bp (80.2%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 8361169 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 32.2%\n",
      "  C: 29.1%\n",
      "  G: 18.5%\n",
      "  T: 19.8%\n",
      "  none/other: 0.3%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t6060450\t13307363.2\t0\t6060450\n",
      "2\t1723675\t3326840.8\t0\t1723675\n",
      "3\t451310\t831710.2\t0\t451310\n",
      "4\t95149\t207927.6\t0\t95149\n",
      "5\t19055\t51981.9\t0\t19055\n",
      "6\t2294\t12995.5\t0\t2294\n",
      "7\t336\t3248.9\t0\t336\n",
      "8\t75\t812.2\t0\t75\n",
      "9\t348\t203.1\t0\t45 303\n",
      "10\t497\t50.8\t1\t41 456\n",
      "11\t305\t12.7\t1\t39 266\n",
      "12\t149\t3.2\t1\t40 109\n",
      "13\t95\t0.8\t1\t29 66\n",
      "14\t75\t0.8\t1\t53 22\n",
      "15\t85\t0.8\t1\t52 33\n",
      "16\t92\t0.8\t1\t63 29\n",
      "17\t85\t0.8\t1\t47 38\n",
      "18\t98\t0.8\t1\t70 28\n",
      "19\t71\t0.8\t1\t47 24\n",
      "20\t89\t0.8\t1\t55 34\n",
      "21\t80\t0.8\t1\t58 22\n",
      "22\t80\t0.8\t1\t52 28\n",
      "23\t90\t0.8\t1\t56 34\n",
      "24\t56\t0.8\t1\t32 24\n",
      "25\t75\t0.8\t1\t44 31\n",
      "26\t64\t0.8\t1\t44 20\n",
      "27\t57\t0.8\t1\t32 25\n",
      "28\t51\t0.8\t1\t32 19\n",
      "29\t57\t0.8\t1\t30 27\n",
      "30\t64\t0.8\t1\t33 31\n",
      "31\t60\t0.8\t1\t31 29\n",
      "32\t55\t0.8\t1\t20 35\n",
      "33\t53\t0.8\t1\t26 27\n",
      "34\t44\t0.8\t1\t25 19\n",
      "35\t47\t0.8\t1\t22 25\n",
      "36\t60\t0.8\t1\t23 37\n",
      "37\t51\t0.8\t1\t16 35\n",
      "38\t80\t0.8\t1\t26 54\n",
      "39\t60\t0.8\t1\t22 38\n",
      "40\t77\t0.8\t1\t25 52\n",
      "41\t60\t0.8\t1\t25 35\n",
      "42\t74\t0.8\t1\t13 61\n",
      "43\t91\t0.8\t1\t29 62\n",
      "44\t133\t0.8\t1\t19 114\n",
      "45\t171\t0.8\t1\t70 101\n",
      "46\t136\t0.8\t1\t38 98\n",
      "47\t177\t0.8\t1\t53 124\n",
      "48\t249\t0.8\t1\t73 176\n",
      "49\t824\t0.8\t1\t257 567\n",
      "50\t158\t0.8\t1\t51 107\n",
      "51\t3502\t0.8\t1\t1303 2199\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900790.fastq\n",
      "=============================================\n",
      "53229453 sequences processed in total\n",
      "Sequences removed because they became shorter than the length cutoff of 20 bp:\t9515347 (17.9%)\n",
      "\n",
      "Now renaming the output file SRR6900790_trimmed.fq\n",
      "\n",
      "ORIGINAL FILE 1: >>SRR6900790_trimmed.fq<<\tRENAMING TO:>>Human_Kidney_R3_trimmed.fq<<\n",
      "\n",
      "  >>> Now running FastQC on the data <<<\n",
      "\n",
      "Started analysis of Human_Kidney_R3_trimmed.fq\n",
      "Approx 5% complete for Human_Kidney_R3_trimmed.fq\n",
      "Approx 10% complete for Human_Kidney_R3_trimmed.fq\n",
      "Approx 15% complete for Human_Kidney_R3_trimmed.fq\n",
      "Approx 20% complete for Human_Kidney_R3_trimmed.fq\n",
      "Approx 25% complete for Human_Kidney_R3_trimmed.fq\n",
      "Approx 30% complete for Human_Kidney_R3_trimmed.fq\n",
      "Approx 35% complete for Human_Kidney_R3_trimmed.fq\n",
      "Approx 40% complete for Human_Kidney_R3_trimmed.fq\n",
      "Approx 45% complete for Human_Kidney_R3_trimmed.fq\n",
      "Approx 50% complete for Human_Kidney_R3_trimmed.fq\n",
      "Approx 55% complete for Human_Kidney_R3_trimmed.fq\n",
      "Approx 60% complete for Human_Kidney_R3_trimmed.fq\n",
      "Approx 65% complete for Human_Kidney_R3_trimmed.fq\n",
      "Approx 70% complete for Human_Kidney_R3_trimmed.fq\n",
      "Approx 75% complete for Human_Kidney_R3_trimmed.fq\n",
      "Approx 80% complete for Human_Kidney_R3_trimmed.fq\n",
      "Approx 85% complete for Human_Kidney_R3_trimmed.fq\n",
      "Approx 90% complete for Human_Kidney_R3_trimmed.fq\n",
      "Approx 95% complete for Human_Kidney_R3_trimmed.fq\n",
      "Analysis complete for Human_Kidney_R3_trimmed.fq\n",
      "Using an excessive number of cores has a diminishing return! It is recommended not to exceed 8 cores per trimming process (you asked for 8 cores). Please consider re-specifying\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "Cutadapt version: 1.18\n",
      "Could not detect version of Python used by Cutadapt from the first line of Cutadapt (but found this: >>>#!/bin/sh<<<)\n",
      "Letting the (modified) Cutadapt deal with the Python version instead\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "pigz 2.6\n",
      "Parallel gzip (pigz) detected. Proceeding with multicore (de)compression using 8 cores\n",
      "\n",
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Output will be written into the directory: /gpfs52/data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/trimmed_reads/\n",
      "Using user-specified basename (>>Human_Liver_R3<<) instead of deriving the filename from the input file(s)\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900791.fastq <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Illumina\t6182\tAGATCGGAAGAGC\t1000000\t0.62\n",
      "smallRNA\t1\tTGGAATTCTCGG\t1000000\t0.00\n",
      "Nextera\t1\tCTGTCTCTTATA\t1000000\t0.00\n",
      "Using Illumina adapter for trimming (count: 6182). Second best hit was smallRNA (count: 1)\n",
      "\n",
      "Writing report to '/gpfs52/data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/trimmed_reads/SRR6900791.fastq_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900791.fastq\n",
      "Trimming mode: single-end\n",
      "Trim Galore version: 0.6.6\n",
      "Cutadapt version: 1.18\n",
      "Python version: could not detect\n",
      "Number of cores used for trimming: 8\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length before a sequence gets removed: 20 bp\n",
      "Running FastQC on the data once trimming has completed\n",
      "Running FastQC with the following extra arguments: '--outdir /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/processing_results/fastQC'\n",
      "\n",
      "Cutadapt seems to be reasonably up-to-date. Setting -j 8\n",
      "Writing final adapter and quality trimmed output to SRR6900791_trimmed.fq\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff '-q 20') and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900791.fastq <<< \n",
      "10000000 sequences processed\n",
      "20000000 sequences processed\n",
      "30000000 sequences processed\n",
      "40000000 sequences processed\n",
      "50000000 sequences processed\n",
      "This is cutadapt 1.18 with Python 3.6.15\n",
      "Command line parameters: -j 8 -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900791.fastq\n",
      "Processing reads on 8 cores in single-end mode ...\n",
      "Finished in 229.67 s (4 us/read; 14.79 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              56,612,635\n",
      "Reads with adapters:                 9,002,430 (15.9%)\n",
      "Reads written (passing filters):    56,612,635 (100.0%)\n",
      "\n",
      "Total basepairs processed: 2,865,194,224 bp\n",
      "Quality-trimmed:             570,310,065 bp (19.9%)\n",
      "Total written (filtered):  2,242,151,587 bp (78.3%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 9002430 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 27.3%\n",
      "  C: 24.5%\n",
      "  G: 17.5%\n",
      "  T: 21.1%\n",
      "  none/other: 9.7%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t5962275\t14153158.8\t0\t5962275\n",
      "2\t1696066\t3538289.7\t0\t1696066\n",
      "3\t405086\t884572.4\t0\t405086\n",
      "4\t81594\t221143.1\t0\t81594\n",
      "5\t13100\t55285.8\t0\t13100\n",
      "6\t1589\t13821.4\t0\t1589\n",
      "7\t481\t3455.4\t0\t481\n",
      "8\t565\t863.8\t0\t565\n",
      "9\t575\t216.0\t0\t289 286\n",
      "10\t797\t54.0\t1\t195 602\n",
      "11\t865\t13.5\t1\t448 417\n",
      "12\t641\t3.4\t1\t273 368\n",
      "13\t590\t0.8\t1\t345 245\n",
      "14\t443\t0.8\t1\t292 151\n",
      "15\t454\t0.8\t1\t331 123\n",
      "16\t362\t0.8\t1\t244 118\n",
      "17\t337\t0.8\t1\t255 82\n",
      "18\t305\t0.8\t1\t169 136\n",
      "19\t401\t0.8\t1\t300 101\n",
      "20\t400\t0.8\t1\t282 118\n",
      "21\t352\t0.8\t1\t234 118\n",
      "22\t425\t0.8\t1\t259 166\n",
      "23\t383\t0.8\t1\t270 113\n",
      "24\t420\t0.8\t1\t308 112\n",
      "25\t320\t0.8\t1\t210 110\n",
      "26\t376\t0.8\t1\t275 101\n",
      "27\t369\t0.8\t1\t221 148\n",
      "28\t452\t0.8\t1\t287 165\n",
      "29\t420\t0.8\t1\t270 150\n",
      "30\t451\t0.8\t1\t287 164\n",
      "31\t474\t0.8\t1\t295 179\n",
      "32\t507\t0.8\t1\t305 202\n",
      "33\t502\t0.8\t1\t350 152\n",
      "34\t554\t0.8\t1\t303 251\n",
      "35\t728\t0.8\t1\t409 319\n",
      "36\t822\t0.8\t1\t428 394\n",
      "37\t1125\t0.8\t1\t625 500\n",
      "38\t825\t0.8\t1\t410 415\n",
      "39\t1210\t0.8\t1\t637 573\n",
      "40\t1873\t0.8\t1\t1059 814\n",
      "41\t1374\t0.8\t1\t806 568\n",
      "42\t1658\t0.8\t1\t946 712\n",
      "43\t2115\t0.8\t1\t1155 960\n",
      "44\t4380\t0.8\t1\t1462 2918\n",
      "45\t9710\t0.8\t1\t5568 4142\n",
      "46\t10276\t0.8\t1\t5997 4279\n",
      "47\t15117\t0.8\t1\t8870 6247\n",
      "48\t26768\t0.8\t1\t13334 13434\n",
      "49\t186491\t0.8\t1\t104380 82111\n",
      "50\t17347\t0.8\t1\t8656 8691\n",
      "51\t547680\t0.8\t1\t342666 205014\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900791.fastq\n",
      "=============================================\n",
      "56612635 sequences processed in total\n",
      "Sequences removed because they became shorter than the length cutoff of 20 bp:\t11277573 (19.9%)\n",
      "\n",
      "Now renaming the output file SRR6900791_trimmed.fq\n",
      "\n",
      "ORIGINAL FILE 1: >>SRR6900791_trimmed.fq<<\tRENAMING TO:>>Human_Liver_R3_trimmed.fq<<\n",
      "\n",
      "  >>> Now running FastQC on the data <<<\n",
      "\n",
      "Started analysis of Human_Liver_R3_trimmed.fq\n",
      "Approx 5% complete for Human_Liver_R3_trimmed.fq\n",
      "Approx 10% complete for Human_Liver_R3_trimmed.fq\n",
      "Approx 15% complete for Human_Liver_R3_trimmed.fq\n",
      "Approx 20% complete for Human_Liver_R3_trimmed.fq\n",
      "Approx 25% complete for Human_Liver_R3_trimmed.fq\n",
      "Approx 30% complete for Human_Liver_R3_trimmed.fq\n",
      "Approx 35% complete for Human_Liver_R3_trimmed.fq\n",
      "Approx 40% complete for Human_Liver_R3_trimmed.fq\n",
      "Approx 45% complete for Human_Liver_R3_trimmed.fq\n",
      "Approx 50% complete for Human_Liver_R3_trimmed.fq\n",
      "Approx 55% complete for Human_Liver_R3_trimmed.fq\n",
      "Approx 60% complete for Human_Liver_R3_trimmed.fq\n",
      "Approx 65% complete for Human_Liver_R3_trimmed.fq\n",
      "Approx 70% complete for Human_Liver_R3_trimmed.fq\n",
      "Approx 75% complete for Human_Liver_R3_trimmed.fq\n",
      "Approx 80% complete for Human_Liver_R3_trimmed.fq\n",
      "Approx 85% complete for Human_Liver_R3_trimmed.fq\n",
      "Approx 90% complete for Human_Liver_R3_trimmed.fq\n",
      "Approx 95% complete for Human_Liver_R3_trimmed.fq\n",
      "Analysis complete for Human_Liver_R3_trimmed.fq\n",
      "Using an excessive number of cores has a diminishing return! It is recommended not to exceed 8 cores per trimming process (you asked for 8 cores). Please consider re-specifying\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "Cutadapt version: 1.18\n",
      "Could not detect version of Python used by Cutadapt from the first line of Cutadapt (but found this: >>>#!/bin/sh<<<)\n",
      "Letting the (modified) Cutadapt deal with the Python version instead\n",
      "pigz 2.6\n",
      "Parallel gzip (pigz) detected. Proceeding with multicore (de)compression using 8 cores\n",
      "\n",
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Output will be written into the directory: /gpfs52/data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/trimmed_reads/\n",
      "Using user-specified basename (>>Human_Lung_R3<<) instead of deriving the filename from the input file(s)\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900792.fastq <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Illumina\t1458\tAGATCGGAAGAGC\t1000000\t0.15\n",
      "smallRNA\t2\tTGGAATTCTCGG\t1000000\t0.00\n",
      "Nextera\t1\tCTGTCTCTTATA\t1000000\t0.00\n",
      "Using Illumina adapter for trimming (count: 1458). Second best hit was smallRNA (count: 2)\n",
      "\n",
      "Writing report to '/gpfs52/data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/trimmed_reads/SRR6900792.fastq_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900792.fastq\n",
      "Trimming mode: single-end\n",
      "Trim Galore version: 0.6.6\n",
      "Cutadapt version: 1.18\n",
      "Python version: could not detect\n",
      "Number of cores used for trimming: 8\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length before a sequence gets removed: 20 bp\n",
      "Running FastQC on the data once trimming has completed\n",
      "Running FastQC with the following extra arguments: '--outdir /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/processing_results/fastQC'\n",
      "\n",
      "Cutadapt seems to be reasonably up-to-date. Setting -j 8\n",
      "Writing final adapter and quality trimmed output to SRR6900792_trimmed.fq\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff '-q 20') and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900792.fastq <<< \n",
      "10000000 sequences processed\n",
      "20000000 sequences processed\n",
      "30000000 sequences processed\n",
      "40000000 sequences processed\n",
      "50000000 sequences processed\n",
      "This is cutadapt 1.18 with Python 3.6.15\n",
      "Command line parameters: -j 8 -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900792.fastq\n",
      "Processing reads on 8 cores in single-end mode ...\n",
      "Finished in 217.46 s (4 us/read; 15.12 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              54,805,139\n",
      "Reads with adapters:                 8,452,933 (15.4%)\n",
      "Reads written (passing filters):    54,805,139 (100.0%)\n",
      "\n",
      "Total basepairs processed: 2,772,910,681 bp\n",
      "Quality-trimmed:             544,855,518 bp (19.6%)\n",
      "Total written (filtered):  2,203,701,420 bp (79.5%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 8452933 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 30.5%\n",
      "  C: 30.5%\n",
      "  G: 19.3%\n",
      "  T: 16.5%\n",
      "  none/other: 3.3%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t5805653\t13701284.8\t0\t5805653\n",
      "2\t1771173\t3425321.2\t0\t1771173\n",
      "3\t483705\t856330.3\t0\t483705\n",
      "4\t98781\t214082.6\t0\t98781\n",
      "5\t22841\t53520.6\t0\t22841\n",
      "6\t2886\t13380.2\t0\t2886\n",
      "7\t549\t3345.0\t0\t549\n",
      "8\t267\t836.3\t0\t267\n",
      "9\t656\t209.1\t0\t178 478\n",
      "10\t966\t52.3\t1\t145 821\n",
      "11\t600\t13.1\t1\t141 459\n",
      "12\t359\t3.3\t1\t113 246\n",
      "13\t275\t0.8\t1\t123 152\n",
      "14\t283\t0.8\t1\t203 80\n",
      "15\t280\t0.8\t1\t226 54\n",
      "16\t275\t0.8\t1\t214 61\n",
      "17\t260\t0.8\t1\t202 58\n",
      "18\t242\t0.8\t1\t173 69\n",
      "19\t252\t0.8\t1\t199 53\n",
      "20\t234\t0.8\t1\t160 74\n",
      "21\t239\t0.8\t1\t175 64\n",
      "22\t268\t0.8\t1\t180 88\n",
      "23\t273\t0.8\t1\t202 71\n",
      "24\t273\t0.8\t1\t204 69\n",
      "25\t289\t0.8\t1\t196 93\n",
      "26\t261\t0.8\t1\t169 92\n",
      "27\t357\t0.8\t1\t200 157\n",
      "28\t405\t0.8\t1\t246 159\n",
      "29\t494\t0.8\t1\t337 157\n",
      "30\t372\t0.8\t1\t253 119\n",
      "31\t456\t0.8\t1\t323 133\n",
      "32\t350\t0.8\t1\t244 106\n",
      "33\t500\t0.8\t1\t225 275\n",
      "34\t653\t0.8\t1\t294 359\n",
      "35\t625\t0.8\t1\t350 275\n",
      "36\t596\t0.8\t1\t297 299\n",
      "37\t583\t0.8\t1\t310 273\n",
      "38\t585\t0.8\t1\t240 345\n",
      "39\t735\t0.8\t1\t324 411\n",
      "40\t989\t0.8\t1\t402 587\n",
      "41\t833\t0.8\t1\t398 435\n",
      "42\t783\t0.8\t1\t335 448\n",
      "43\t827\t0.8\t1\t333 494\n",
      "44\t1893\t0.8\t1\t433 1460\n",
      "45\t3445\t0.8\t1\t1236 2209\n",
      "46\t4022\t0.8\t1\t1893 2129\n",
      "47\t5209\t0.8\t1\t1834 3375\n",
      "48\t9133\t0.8\t1\t2903 6230\n",
      "49\t60349\t0.8\t1\t20409 39940\n",
      "50\t5251\t0.8\t1\t1803 3448\n",
      "51\t161348\t0.8\t1\t62105 99243\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900792.fastq\n",
      "=============================================\n",
      "54805139 sequences processed in total\n",
      "Sequences removed because they became shorter than the length cutoff of 20 bp:\t10205809 (18.6%)\n",
      "\n",
      "Now renaming the output file SRR6900792_trimmed.fq\n",
      "\n",
      "ORIGINAL FILE 1: >>SRR6900792_trimmed.fq<<\tRENAMING TO:>>Human_Lung_R3_trimmed.fq<<\n",
      "\n",
      "  >>> Now running FastQC on the data <<<\n",
      "\n",
      "Started analysis of Human_Lung_R3_trimmed.fq\n",
      "Approx 5% complete for Human_Lung_R3_trimmed.fq\n",
      "Approx 10% complete for Human_Lung_R3_trimmed.fq\n",
      "Approx 15% complete for Human_Lung_R3_trimmed.fq\n",
      "Approx 20% complete for Human_Lung_R3_trimmed.fq\n",
      "Approx 25% complete for Human_Lung_R3_trimmed.fq\n",
      "Approx 30% complete for Human_Lung_R3_trimmed.fq\n",
      "Approx 35% complete for Human_Lung_R3_trimmed.fq\n",
      "Approx 40% complete for Human_Lung_R3_trimmed.fq\n",
      "Approx 45% complete for Human_Lung_R3_trimmed.fq\n",
      "Approx 50% complete for Human_Lung_R3_trimmed.fq\n",
      "Approx 55% complete for Human_Lung_R3_trimmed.fq\n",
      "Approx 60% complete for Human_Lung_R3_trimmed.fq\n",
      "Approx 65% complete for Human_Lung_R3_trimmed.fq\n",
      "Approx 70% complete for Human_Lung_R3_trimmed.fq\n",
      "Approx 75% complete for Human_Lung_R3_trimmed.fq\n",
      "Approx 80% complete for Human_Lung_R3_trimmed.fq\n",
      "Approx 85% complete for Human_Lung_R3_trimmed.fq\n",
      "Approx 90% complete for Human_Lung_R3_trimmed.fq\n",
      "Approx 95% complete for Human_Lung_R3_trimmed.fq\n",
      "Analysis complete for Human_Lung_R3_trimmed.fq\n",
      "Using an excessive number of cores has a diminishing return! It is recommended not to exceed 8 cores per trimming process (you asked for 8 cores). Please consider re-specifying\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "Cutadapt version: 1.18\n",
      "Could not detect version of Python used by Cutadapt from the first line of Cutadapt (but found this: >>>#!/bin/sh<<<)\n",
      "Letting the (modified) Cutadapt deal with the Python version instead\n",
      "pigz 2.6\n",
      "Parallel gzip (pigz) detected. Proceeding with multicore (de)compression using 8 cores\n",
      "\n",
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Output will be written into the directory: /gpfs52/data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/trimmed_reads/\n",
      "Using user-specified basename (>>Human_Heart_R4<<) instead of deriving the filename from the input file(s)\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900793.fastq <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Illumina\t4420\tAGATCGGAAGAGC\t1000000\t0.44\n",
      "smallRNA\t1\tTGGAATTCTCGG\t1000000\t0.00\n",
      "Nextera\t1\tCTGTCTCTTATA\t1000000\t0.00\n",
      "Using Illumina adapter for trimming (count: 4420). Second best hit was smallRNA (count: 1)\n",
      "\n",
      "Writing report to '/gpfs52/data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/trimmed_reads/SRR6900793.fastq_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900793.fastq\n",
      "Trimming mode: single-end\n",
      "Trim Galore version: 0.6.6\n",
      "Cutadapt version: 1.18\n",
      "Python version: could not detect\n",
      "Number of cores used for trimming: 8\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length before a sequence gets removed: 20 bp\n",
      "Running FastQC on the data once trimming has completed\n",
      "Running FastQC with the following extra arguments: '--outdir /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/processing_results/fastQC'\n",
      "\n",
      "Cutadapt seems to be reasonably up-to-date. Setting -j 8\n",
      "Writing final adapter and quality trimmed output to SRR6900793_trimmed.fq\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff '-q 20') and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900793.fastq <<< \n",
      "10000000 sequences processed\n",
      "20000000 sequences processed\n",
      "30000000 sequences processed\n",
      "40000000 sequences processed\n",
      "50000000 sequences processed\n",
      "This is cutadapt 1.18 with Python 3.6.15\n",
      "Command line parameters: -j 8 -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900793.fastq\n",
      "Processing reads on 8 cores in single-end mode ...\n",
      "Finished in 215.94 s (4 us/read; 15.00 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              53,976,289\n",
      "Reads with adapters:                 7,342,017 (13.6%)\n",
      "Reads written (passing filters):    53,976,289 (100.0%)\n",
      "\n",
      "Total basepairs processed: 2,735,231,403 bp\n",
      "Quality-trimmed:             589,740,114 bp (21.6%)\n",
      "Total written (filtered):  2,103,315,344 bp (76.9%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 7342017 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  A: 27.0%\n",
      "  C: 26.5%\n",
      "  G: 18.6%\n",
      "  T: 18.6%\n",
      "  none/other: 9.4%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t4796998\t13494072.2\t0\t4796998\n",
      "2\t1431032\t3373518.1\t0\t1431032\n",
      "3\t357868\t843379.5\t0\t357868\n",
      "4\t71500\t210844.9\t0\t71500\n",
      "5\t14633\t52711.2\t0\t14633\n",
      "6\t1886\t13177.8\t0\t1886\n",
      "7\t491\t3294.5\t0\t491\n",
      "8\t355\t823.6\t0\t355\n",
      "9\t720\t205.9\t0\t228 492\n",
      "10\t891\t51.5\t1\t256 635\n",
      "11\t797\t12.9\t1\t348 449\n",
      "12\t610\t3.2\t1\t221 389\n",
      "13\t533\t0.8\t1\t309 224\n",
      "14\t372\t0.8\t1\t217 155\n",
      "15\t432\t0.8\t1\t314 118\n",
      "16\t334\t0.8\t1\t238 96\n",
      "17\t308\t0.8\t1\t208 100\n",
      "18\t324\t0.8\t1\t192 132\n",
      "19\t371\t0.8\t1\t255 116\n",
      "20\t387\t0.8\t1\t259 128\n",
      "21\t350\t0.8\t1\t231 119\n",
      "22\t332\t0.8\t1\t221 111\n",
      "23\t364\t0.8\t1\t270 94\n",
      "24\t328\t0.8\t1\t238 90\n",
      "25\t348\t0.8\t1\t236 112\n",
      "26\t426\t0.8\t1\t264 162\n",
      "27\t396\t0.8\t1\t206 190\n",
      "28\t506\t0.8\t1\t274 232\n",
      "29\t497\t0.8\t1\t272 225\n",
      "30\t481\t0.8\t1\t289 192\n",
      "31\t477\t0.8\t1\t260 217\n",
      "32\t488\t0.8\t1\t264 224\n",
      "33\t489\t0.8\t1\t266 223\n",
      "34\t605\t0.8\t1\t338 267\n",
      "35\t735\t0.8\t1\t361 374\n",
      "36\t712\t0.8\t1\t398 314\n",
      "37\t719\t0.8\t1\t361 358\n",
      "38\t904\t0.8\t1\t444 460\n",
      "39\t1755\t0.8\t1\t575 1180\n",
      "40\t3072\t0.8\t1\t1568 1504\n",
      "41\t2344\t0.8\t1\t1216 1128\n",
      "42\t2178\t0.8\t1\t1117 1061\n",
      "43\t2330\t0.8\t1\t1058 1272\n",
      "44\t3000\t0.8\t1\t1391 1609\n",
      "45\t3614\t0.8\t1\t1683 1931\n",
      "46\t8419\t0.8\t1\t2143 6276\n",
      "47\t17182\t0.8\t1\t8996 8186\n",
      "48\t22061\t0.8\t1\t10512 11549\n",
      "49\t100203\t0.8\t1\t47555 52648\n",
      "50\t14861\t0.8\t1\t7543 7318\n",
      "51\t470999\t0.8\t1\t248540 222459\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900793.fastq\n",
      "=============================================\n",
      "53976289 sequences processed in total\n",
      "Sequences removed because they became shorter than the length cutoff of 20 bp:\t11618630 (21.5%)\n",
      "\n",
      "Now renaming the output file SRR6900793_trimmed.fq\n",
      "\n",
      "ORIGINAL FILE 1: >>SRR6900793_trimmed.fq<<\tRENAMING TO:>>Human_Heart_R4_trimmed.fq<<\n",
      "\n",
      "  >>> Now running FastQC on the data <<<\n",
      "\n",
      "Started analysis of Human_Heart_R4_trimmed.fq\n",
      "Approx 5% complete for Human_Heart_R4_trimmed.fq\n",
      "Approx 10% complete for Human_Heart_R4_trimmed.fq\n",
      "Approx 15% complete for Human_Heart_R4_trimmed.fq\n",
      "Approx 20% complete for Human_Heart_R4_trimmed.fq\n",
      "Approx 25% complete for Human_Heart_R4_trimmed.fq\n",
      "Approx 30% complete for Human_Heart_R4_trimmed.fq\n",
      "Approx 35% complete for Human_Heart_R4_trimmed.fq\n",
      "Approx 40% complete for Human_Heart_R4_trimmed.fq\n",
      "Approx 45% complete for Human_Heart_R4_trimmed.fq\n",
      "Approx 50% complete for Human_Heart_R4_trimmed.fq\n",
      "Approx 55% complete for Human_Heart_R4_trimmed.fq\n",
      "Approx 60% complete for Human_Heart_R4_trimmed.fq\n",
      "Approx 65% complete for Human_Heart_R4_trimmed.fq\n",
      "Approx 70% complete for Human_Heart_R4_trimmed.fq\n",
      "Approx 75% complete for Human_Heart_R4_trimmed.fq\n",
      "Approx 80% complete for Human_Heart_R4_trimmed.fq\n",
      "Approx 85% complete for Human_Heart_R4_trimmed.fq\n",
      "Approx 90% complete for Human_Heart_R4_trimmed.fq\n",
      "Approx 95% complete for Human_Heart_R4_trimmed.fq\n",
      "Analysis complete for Human_Heart_R4_trimmed.fq\n",
      "Using an excessive number of cores has a diminishing return! It is recommended not to exceed 8 cores per trimming process (you asked for 8 cores). Please consider re-specifying\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "Cutadapt version: 1.18\n",
      "Could not detect version of Python used by Cutadapt from the first line of Cutadapt (but found this: >>>#!/bin/sh<<<)\n",
      "Letting the (modified) Cutadapt deal with the Python version instead\n",
      "pigz 2.6\n",
      "Parallel gzip (pigz) detected. Proceeding with multicore (de)compression using 8 cores\n",
      "\n",
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Output will be written into the directory: /gpfs52/data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/trimmed_reads/\n",
      "Using user-specified basename (>>Human_Kidney_R4<<) instead of deriving the filename from the input file(s)\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900794.fastq <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Illumina\t9329\tAGATCGGAAGAGC\t1000000\t0.93\n",
      "Nextera\t3\tCTGTCTCTTATA\t1000000\t0.00\n",
      "smallRNA\t0\tTGGAATTCTCGG\t1000000\t0.00\n",
      "Using Illumina adapter for trimming (count: 9329). Second best hit was Nextera (count: 3)\n",
      "\n",
      "Writing report to '/gpfs52/data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/trimmed_reads/SRR6900794.fastq_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900794.fastq\n",
      "Trimming mode: single-end\n",
      "Trim Galore version: 0.6.6\n",
      "Cutadapt version: 1.18\n",
      "Python version: could not detect\n",
      "Number of cores used for trimming: 8\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length before a sequence gets removed: 20 bp\n",
      "Running FastQC on the data once trimming has completed\n",
      "Running FastQC with the following extra arguments: '--outdir /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/processing_results/fastQC'\n",
      "\n",
      "Cutadapt seems to be reasonably up-to-date. Setting -j 8\n",
      "Writing final adapter and quality trimmed output to SRR6900794_trimmed.fq\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff '-q 20') and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900794.fastq <<< \n",
      "10000000 sequences processed\n",
      "20000000 sequences processed\n",
      "30000000 sequences processed\n",
      "40000000 sequences processed\n",
      "50000000 sequences processed\n",
      "60000000 sequences processed\n",
      "This is cutadapt 1.18 with Python 3.6.15\n",
      "Command line parameters: -j 8 -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900794.fastq\n",
      "Processing reads on 8 cores in single-end mode ...\n",
      "Finished in 247.28 s (4 us/read; 15.12 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              62,323,442\n",
      "Reads with adapters:                 9,136,649 (14.7%)\n",
      "Reads written (passing filters):    62,323,442 (100.0%)\n",
      "\n",
      "Total basepairs processed: 3,158,477,985 bp\n",
      "Quality-trimmed:             686,709,216 bp (21.7%)\n",
      "Total written (filtered):  2,386,222,498 bp (75.5%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 9136649 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 25.2%\n",
      "  C: 24.4%\n",
      "  G: 16.6%\n",
      "  T: 16.9%\n",
      "  none/other: 16.9%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t5506876\t15580860.5\t0\t5506876\n",
      "2\t1596411\t3895215.1\t0\t1596411\n",
      "3\t416727\t973803.8\t0\t416727\n",
      "4\t83138\t243450.9\t0\t83138\n",
      "5\t17127\t60862.7\t0\t17127\n",
      "6\t2612\t15215.7\t0\t2612\n",
      "7\t1009\t3803.9\t0\t1009\n",
      "8\t791\t951.0\t0\t791\n",
      "9\t1108\t237.7\t0\t599 509\n",
      "10\t1412\t59.4\t1\t408 1004\n",
      "11\t1461\t14.9\t1\t550 911\n",
      "12\t1096\t3.7\t1\t372 724\n",
      "13\t1116\t0.9\t1\t573 543\n",
      "14\t735\t0.9\t1\t486 249\n",
      "15\t688\t0.9\t1\t465 223\n",
      "16\t718\t0.9\t1\t529 189\n",
      "17\t780\t0.9\t1\t565 215\n",
      "18\t923\t0.9\t1\t570 353\n",
      "19\t1108\t0.9\t1\t862 246\n",
      "20\t573\t0.9\t1\t412 161\n",
      "21\t525\t0.9\t1\t385 140\n",
      "22\t545\t0.9\t1\t327 218\n",
      "23\t649\t0.9\t1\t418 231\n",
      "24\t670\t0.9\t1\t465 205\n",
      "25\t555\t0.9\t1\t360 195\n",
      "26\t556\t0.9\t1\t406 150\n",
      "27\t598\t0.9\t1\t333 265\n",
      "28\t741\t0.9\t1\t425 316\n",
      "29\t604\t0.9\t1\t378 226\n",
      "30\t604\t0.9\t1\t399 205\n",
      "31\t700\t0.9\t1\t473 227\n",
      "32\t823\t0.9\t1\t554 269\n",
      "33\t706\t0.9\t1\t433 273\n",
      "34\t927\t0.9\t1\t511 416\n",
      "35\t973\t0.9\t1\t562 411\n",
      "36\t980\t0.9\t1\t566 414\n",
      "37\t921\t0.9\t1\t506 415\n",
      "38\t1122\t0.9\t1\t523 599\n",
      "39\t1493\t0.9\t1\t807 686\n",
      "40\t2352\t0.9\t1\t1084 1268\n",
      "41\t1570\t0.9\t1\t707 863\n",
      "42\t2252\t0.9\t1\t979 1273\n",
      "43\t2674\t0.9\t1\t1354 1320\n",
      "44\t3298\t0.9\t1\t1337 1961\n",
      "45\t4985\t0.9\t1\t2060 2925\n",
      "46\t25583\t0.9\t1\t2789 22794\n",
      "47\t55655\t0.9\t1\t25131 30524\n",
      "48\t67426\t0.9\t1\t35149 32277\n",
      "49\t259454\t0.9\t1\t110936 148518\n",
      "50\t30174\t0.9\t1\t13001 17173\n",
      "51\t1030125\t0.9\t1\t495330 534795\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900794.fastq\n",
      "=============================================\n",
      "62323442 sequences processed in total\n",
      "Sequences removed because they became shorter than the length cutoff of 20 bp:\t14288184 (22.9%)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Now renaming the output file SRR6900794_trimmed.fq\n",
      "\n",
      "ORIGINAL FILE 1: >>SRR6900794_trimmed.fq<<\tRENAMING TO:>>Human_Kidney_R4_trimmed.fq<<\n",
      "\n",
      "  >>> Now running FastQC on the data <<<\n",
      "\n",
      "Started analysis of Human_Kidney_R4_trimmed.fq\n",
      "Approx 5% complete for Human_Kidney_R4_trimmed.fq\n",
      "Approx 10% complete for Human_Kidney_R4_trimmed.fq\n",
      "Approx 15% complete for Human_Kidney_R4_trimmed.fq\n",
      "Approx 20% complete for Human_Kidney_R4_trimmed.fq\n",
      "Approx 25% complete for Human_Kidney_R4_trimmed.fq\n",
      "Approx 30% complete for Human_Kidney_R4_trimmed.fq\n",
      "Approx 35% complete for Human_Kidney_R4_trimmed.fq\n",
      "Approx 40% complete for Human_Kidney_R4_trimmed.fq\n",
      "Approx 45% complete for Human_Kidney_R4_trimmed.fq\n",
      "Approx 50% complete for Human_Kidney_R4_trimmed.fq\n",
      "Approx 55% complete for Human_Kidney_R4_trimmed.fq\n",
      "Approx 60% complete for Human_Kidney_R4_trimmed.fq\n",
      "Approx 65% complete for Human_Kidney_R4_trimmed.fq\n",
      "Approx 70% complete for Human_Kidney_R4_trimmed.fq\n",
      "Approx 75% complete for Human_Kidney_R4_trimmed.fq\n",
      "Approx 80% complete for Human_Kidney_R4_trimmed.fq\n",
      "Approx 85% complete for Human_Kidney_R4_trimmed.fq\n",
      "Approx 90% complete for Human_Kidney_R4_trimmed.fq\n",
      "Approx 95% complete for Human_Kidney_R4_trimmed.fq\n",
      "Analysis complete for Human_Kidney_R4_trimmed.fq\n",
      "Using an excessive number of cores has a diminishing return! It is recommended not to exceed 8 cores per trimming process (you asked for 8 cores). Please consider re-specifying\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "Cutadapt version: 1.18\n",
      "Could not detect version of Python used by Cutadapt from the first line of Cutadapt (but found this: >>>#!/bin/sh<<<)\n",
      "Letting the (modified) Cutadapt deal with the Python version instead\n",
      "pigz 2.6\n",
      "Parallel gzip (pigz) detected. Proceeding with multicore (de)compression using 8 cores\n",
      "\n",
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Output will be written into the directory: /gpfs52/data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/trimmed_reads/\n",
      "Using user-specified basename (>>Human_Liver_R4<<) instead of deriving the filename from the input file(s)\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900795.fastq <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Illumina\t7280\tAGATCGGAAGAGC\t1000000\t0.73\n",
      "smallRNA\t0\tTGGAATTCTCGG\t1000000\t0.00\n",
      "Nextera\t0\tCTGTCTCTTATA\t1000000\t0.00\n",
      "Using Illumina adapter for trimming (count: 7280). Second best hit was smallRNA (count: 0)\n",
      "\n",
      "Writing report to '/gpfs52/data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/trimmed_reads/SRR6900795.fastq_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900795.fastq\n",
      "Trimming mode: single-end\n",
      "Trim Galore version: 0.6.6\n",
      "Cutadapt version: 1.18\n",
      "Python version: could not detect\n",
      "Number of cores used for trimming: 8\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length before a sequence gets removed: 20 bp\n",
      "Running FastQC on the data once trimming has completed\n",
      "Running FastQC with the following extra arguments: '--outdir /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/processing_results/fastQC'\n",
      "\n",
      "Cutadapt seems to be reasonably up-to-date. Setting -j 8\n",
      "Writing final adapter and quality trimmed output to SRR6900795_trimmed.fq\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff '-q 20') and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900795.fastq <<< \n",
      "10000000 sequences processed\n",
      "20000000 sequences processed\n",
      "30000000 sequences processed\n",
      "40000000 sequences processed\n",
      "50000000 sequences processed\n",
      "This is cutadapt 1.18 with Python 3.6.15\n",
      "Command line parameters: -j 8 -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900795.fastq\n",
      "Processing reads on 8 cores in single-end mode ...\n",
      "Finished in 645.50 s (11 us/read; 5.35 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              57,509,787\n",
      "Reads with adapters:                10,330,098 (18.0%)\n",
      "Reads written (passing filters):    57,509,787 (100.0%)\n",
      "\n",
      "Total basepairs processed: 2,909,255,194 bp\n",
      "Quality-trimmed:             557,098,772 bp (19.1%)\n",
      "Total written (filtered):  2,267,205,533 bp (77.9%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 10330098 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 27.3%\n",
      "  C: 24.1%\n",
      "  G: 16.3%\n",
      "  T: 17.6%\n",
      "  none/other: 14.7%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t6414503\t14377446.8\t0\t6414503\n",
      "2\t1827533\t3594361.7\t0\t1827533\n",
      "3\t470823\t898590.4\t0\t470823\n",
      "4\t94040\t224647.6\t0\t94040\n",
      "5\t17754\t56161.9\t0\t17754\n",
      "6\t2442\t14040.5\t0\t2442\n",
      "7\t611\t3510.1\t0\t611\n",
      "8\t396\t877.5\t0\t396\n",
      "9\t680\t219.4\t0\t213 467\n",
      "10\t1157\t54.8\t1\t244 913\n",
      "11\t621\t13.7\t1\t157 464\n",
      "12\t516\t3.4\t1\t172 344\n",
      "13\t382\t0.9\t1\t214 168\n",
      "14\t319\t0.9\t1\t185 134\n",
      "15\t401\t0.9\t1\t300 101\n",
      "16\t262\t0.9\t1\t174 88\n",
      "17\t322\t0.9\t1\t210 112\n",
      "18\t329\t0.9\t1\t197 132\n",
      "19\t310\t0.9\t1\t229 81\n",
      "20\t312\t0.9\t1\t188 124\n",
      "21\t363\t0.9\t1\t218 145\n",
      "22\t337\t0.9\t1\t199 138\n",
      "23\t405\t0.9\t1\t210 195\n",
      "24\t559\t0.9\t1\t315 244\n",
      "25\t605\t0.9\t1\t305 300\n",
      "26\t586\t0.9\t1\t320 266\n",
      "27\t780\t0.9\t1\t345 435\n",
      "28\t1180\t0.9\t1\t703 477\n",
      "29\t5891\t0.9\t1\t4972 919\n",
      "30\t1301\t0.9\t1\t895 406\n",
      "31\t4832\t0.9\t1\t4309 523\n",
      "32\t684\t0.9\t1\t357 327\n",
      "33\t849\t0.9\t1\t234 615\n",
      "34\t1612\t0.9\t1\t460 1152\n",
      "35\t5967\t0.9\t1\t828 5139\n",
      "36\t8206\t0.9\t1\t2754 5452\n",
      "37\t5753\t0.9\t1\t2832 2921\n",
      "38\t5728\t0.9\t1\t1337 4391\n",
      "39\t10015\t0.9\t1\t2380 7635\n",
      "40\t47003\t0.9\t1\t4530 42473\n",
      "41\t56212\t0.9\t1\t22367 33845\n",
      "42\t28071\t0.9\t1\t16244 11827\n",
      "43\t11332\t0.9\t1\t5419 5913\n",
      "44\t13722\t0.9\t1\t3310 10412\n",
      "45\t14153\t0.9\t1\t6052 8101\n",
      "46\t32854\t0.9\t1\t4427 28427\n",
      "47\t46380\t0.9\t1\t19973 26407\n",
      "48\t49878\t0.9\t1\t19671 30207\n",
      "49\t173930\t0.9\t1\t53416 120514\n",
      "50\t30656\t0.9\t1\t11311 19345\n",
      "51\t936541\t0.9\t1\t339007 597534\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900795.fastq\n",
      "=============================================\n",
      "57509787 sequences processed in total\n",
      "Sequences removed because they became shorter than the length cutoff of 20 bp:\t11560453 (20.1%)\n",
      "\n",
      "Now renaming the output file SRR6900795_trimmed.fq\n",
      "\n",
      "ORIGINAL FILE 1: >>SRR6900795_trimmed.fq<<\tRENAMING TO:>>Human_Liver_R4_trimmed.fq<<\n",
      "\n",
      "  >>> Now running FastQC on the data <<<\n",
      "\n",
      "Started analysis of Human_Liver_R4_trimmed.fq\n",
      "Approx 5% complete for Human_Liver_R4_trimmed.fq\n",
      "Approx 10% complete for Human_Liver_R4_trimmed.fq\n",
      "Approx 15% complete for Human_Liver_R4_trimmed.fq\n",
      "Approx 20% complete for Human_Liver_R4_trimmed.fq\n",
      "Approx 25% complete for Human_Liver_R4_trimmed.fq\n",
      "Approx 30% complete for Human_Liver_R4_trimmed.fq\n",
      "Approx 35% complete for Human_Liver_R4_trimmed.fq\n",
      "Approx 40% complete for Human_Liver_R4_trimmed.fq\n",
      "Approx 45% complete for Human_Liver_R4_trimmed.fq\n",
      "Approx 50% complete for Human_Liver_R4_trimmed.fq\n",
      "Approx 55% complete for Human_Liver_R4_trimmed.fq\n",
      "Approx 60% complete for Human_Liver_R4_trimmed.fq\n",
      "Approx 65% complete for Human_Liver_R4_trimmed.fq\n",
      "Approx 70% complete for Human_Liver_R4_trimmed.fq\n",
      "Approx 75% complete for Human_Liver_R4_trimmed.fq\n",
      "Approx 80% complete for Human_Liver_R4_trimmed.fq\n",
      "Approx 85% complete for Human_Liver_R4_trimmed.fq\n",
      "Approx 90% complete for Human_Liver_R4_trimmed.fq\n",
      "Approx 95% complete for Human_Liver_R4_trimmed.fq\n",
      "Analysis complete for Human_Liver_R4_trimmed.fq\n",
      "Using an excessive number of cores has a diminishing return! It is recommended not to exceed 8 cores per trimming process (you asked for 8 cores). Please consider re-specifying\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "Cutadapt version: 1.18\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Could not detect version of Python used by Cutadapt from the first line of Cutadapt (but found this: >>>#!/bin/sh<<<)\n",
      "Letting the (modified) Cutadapt deal with the Python version instead\n",
      "pigz 2.6\n",
      "Parallel gzip (pigz) detected. Proceeding with multicore (de)compression using 8 cores\n",
      "\n",
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Output will be written into the directory: /gpfs52/data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/trimmed_reads/\n",
      "Using user-specified basename (>>Human_Lung_R4<<) instead of deriving the filename from the input file(s)\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900796.fastq <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Illumina\t8364\tAGATCGGAAGAGC\t1000000\t0.84\n",
      "smallRNA\t3\tTGGAATTCTCGG\t1000000\t0.00\n",
      "Nextera\t1\tCTGTCTCTTATA\t1000000\t0.00\n",
      "Using Illumina adapter for trimming (count: 8364). Second best hit was smallRNA (count: 3)\n",
      "\n",
      "Writing report to '/gpfs52/data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/trimmed_reads/SRR6900796.fastq_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900796.fastq\n",
      "Trimming mode: single-end\n",
      "Trim Galore version: 0.6.6\n",
      "Cutadapt version: 1.18\n",
      "Python version: could not detect\n",
      "Number of cores used for trimming: 8\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length before a sequence gets removed: 20 bp\n",
      "Running FastQC on the data once trimming has completed\n",
      "Running FastQC with the following extra arguments: '--outdir /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/processing_results/fastQC'\n",
      "\n",
      "Cutadapt seems to be reasonably up-to-date. Setting -j 8\n",
      "Writing final adapter and quality trimmed output to SRR6900796_trimmed.fq\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff '-q 20') and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900796.fastq <<< \n",
      "10000000 sequences processed\n",
      "20000000 sequences processed\n",
      "30000000 sequences processed\n",
      "40000000 sequences processed\n",
      "50000000 sequences processed\n",
      "60000000 sequences processed\n",
      "This is cutadapt 1.18 with Python 3.6.15\n",
      "Command line parameters: -j 8 -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900796.fastq\n",
      "Processing reads on 8 cores in single-end mode ...\n",
      "Finished in 352.82 s (5 us/read; 11.39 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              66,975,147\n",
      "Reads with adapters:                11,500,843 (17.2%)\n",
      "Reads written (passing filters):    66,975,147 (100.0%)\n",
      "\n",
      "Total basepairs processed: 3,387,970,812 bp\n",
      "Quality-trimmed:             654,223,411 bp (19.3%)\n",
      "Total written (filtered):  2,648,600,974 bp (78.2%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 11500843 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 27.3%\n",
      "  C: 27.5%\n",
      "  G: 17.3%\n",
      "  T: 15.2%\n",
      "  none/other: 12.7%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t7148104\t16743786.8\t0\t7148104\n",
      "2\t2160338\t4185946.7\t0\t2160338\n",
      "3\t596108\t1046486.7\t0\t596108\n",
      "4\t120591\t261621.7\t0\t120591\n",
      "5\t27726\t65405.4\t0\t27726\n",
      "6\t4032\t16351.4\t0\t4032\n",
      "7\t1191\t4087.8\t0\t1191\n",
      "8\t967\t1022.0\t0\t967\n",
      "9\t1440\t255.5\t0\t615 825\n",
      "10\t1978\t63.9\t1\t554 1424\n",
      "11\t1851\t16.0\t1\t750 1101\n",
      "12\t1398\t4.0\t1\t574 824\n",
      "13\t1246\t1.0\t1\t622 624\n",
      "14\t905\t1.0\t1\t676 229\n",
      "15\t970\t1.0\t1\t767 203\n",
      "16\t792\t1.0\t1\t614 178\n",
      "17\t973\t1.0\t1\t790 183\n",
      "18\t973\t1.0\t1\t723 250\n",
      "19\t941\t1.0\t1\t735 206\n",
      "20\t912\t1.0\t1\t699 213\n",
      "21\t996\t1.0\t1\t751 245\n",
      "22\t1074\t1.0\t1\t815 259\n",
      "23\t1060\t1.0\t1\t855 205\n",
      "24\t887\t1.0\t1\t718 169\n",
      "25\t1028\t1.0\t1\t796 232\n",
      "26\t956\t1.0\t1\t738 218\n",
      "27\t1111\t1.0\t1\t810 301\n",
      "28\t1309\t1.0\t1\t960 349\n",
      "29\t1313\t1.0\t1\t1023 290\n",
      "30\t1224\t1.0\t1\t929 295\n",
      "31\t1254\t1.0\t1\t967 287\n",
      "32\t1243\t1.0\t1\t954 289\n",
      "33\t1357\t1.0\t1\t1031 326\n",
      "34\t1454\t1.0\t1\t1026 428\n",
      "35\t1459\t1.0\t1\t1078 381\n",
      "36\t1598\t1.0\t1\t1083 515\n",
      "37\t1870\t1.0\t1\t1220 650\n",
      "38\t1864\t1.0\t1\t1224 640\n",
      "39\t1766\t1.0\t1\t1160 606\n",
      "40\t2892\t1.0\t1\t1529 1363\n",
      "41\t2199\t1.0\t1\t1168 1031\n",
      "42\t3263\t1.0\t1\t1450 1813\n",
      "43\t3777\t1.0\t1\t2064 1713\n",
      "44\t3927\t1.0\t1\t1647 2280\n",
      "45\t5042\t1.0\t1\t2484 2558\n",
      "46\t13671\t1.0\t1\t3016 10655\n",
      "47\t38077\t1.0\t1\t17123 20954\n",
      "48\t61402\t1.0\t1\t33843 27559\n",
      "49\t332820\t1.0\t1\t150037 182783\n",
      "50\t30796\t1.0\t1\t14383 16413\n",
      "51\t904718\t1.0\t1\t466385 438333\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900796.fastq\n",
      "=============================================\n",
      "66975147 sequences processed in total\n",
      "Sequences removed because they became shorter than the length cutoff of 20 bp:\t13299991 (19.9%)\n",
      "\n",
      "Now renaming the output file SRR6900796_trimmed.fq\n",
      "\n",
      "ORIGINAL FILE 1: >>SRR6900796_trimmed.fq<<\tRENAMING TO:>>Human_Lung_R4_trimmed.fq<<\n",
      "\n",
      "  >>> Now running FastQC on the data <<<\n",
      "\n",
      "Started analysis of Human_Lung_R4_trimmed.fq\n",
      "Approx 5% complete for Human_Lung_R4_trimmed.fq\n",
      "Approx 10% complete for Human_Lung_R4_trimmed.fq\n",
      "Approx 15% complete for Human_Lung_R4_trimmed.fq\n",
      "Approx 20% complete for Human_Lung_R4_trimmed.fq\n",
      "Approx 25% complete for Human_Lung_R4_trimmed.fq\n",
      "Approx 30% complete for Human_Lung_R4_trimmed.fq\n",
      "Approx 35% complete for Human_Lung_R4_trimmed.fq\n",
      "Approx 40% complete for Human_Lung_R4_trimmed.fq\n",
      "Approx 45% complete for Human_Lung_R4_trimmed.fq\n",
      "Approx 50% complete for Human_Lung_R4_trimmed.fq\n",
      "Approx 55% complete for Human_Lung_R4_trimmed.fq\n",
      "Approx 60% complete for Human_Lung_R4_trimmed.fq\n",
      "Approx 65% complete for Human_Lung_R4_trimmed.fq\n",
      "Approx 70% complete for Human_Lung_R4_trimmed.fq\n",
      "Approx 75% complete for Human_Lung_R4_trimmed.fq\n",
      "Approx 80% complete for Human_Lung_R4_trimmed.fq\n",
      "Approx 85% complete for Human_Lung_R4_trimmed.fq\n",
      "Approx 90% complete for Human_Lung_R4_trimmed.fq\n",
      "Approx 95% complete for Human_Lung_R4_trimmed.fq\n",
      "Analysis complete for Human_Lung_R4_trimmed.fq\n",
      "Using an excessive number of cores has a diminishing return! It is recommended not to exceed 8 cores per trimming process (you asked for 8 cores). Please consider re-specifying\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "Cutadapt version: 1.18\n",
      "Could not detect version of Python used by Cutadapt from the first line of Cutadapt (but found this: >>>#!/bin/sh<<<)\n",
      "Letting the (modified) Cutadapt deal with the Python version instead\n",
      "pigz 2.6\n",
      "Parallel gzip (pigz) detected. Proceeding with multicore (de)compression using 8 cores\n",
      "\n",
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Output will be written into the directory: /gpfs52/data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/trimmed_reads/\n",
      "Using user-specified basename (>>Macaque_Heart_R1<<) instead of deriving the filename from the input file(s)\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900797.fastq <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Illumina\t521\tAGATCGGAAGAGC\t1000000\t0.05\n",
      "smallRNA\t2\tTGGAATTCTCGG\t1000000\t0.00\n",
      "Nextera\t0\tCTGTCTCTTATA\t1000000\t0.00\n",
      "Using Illumina adapter for trimming (count: 521). Second best hit was smallRNA (count: 2)\n",
      "\n",
      "Writing report to '/gpfs52/data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/trimmed_reads/SRR6900797.fastq_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900797.fastq\n",
      "Trimming mode: single-end\n",
      "Trim Galore version: 0.6.6\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Cutadapt version: 1.18\n",
      "Python version: could not detect\n",
      "Number of cores used for trimming: 8\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length before a sequence gets removed: 20 bp\n",
      "Running FastQC on the data once trimming has completed\n",
      "Running FastQC with the following extra arguments: '--outdir /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/processing_results/fastQC'\n",
      "\n",
      "Cutadapt seems to be reasonably up-to-date. Setting -j 8\n",
      "Writing final adapter and quality trimmed output to SRR6900797_trimmed.fq\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff '-q 20') and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900797.fastq <<< \n",
      "10000000 sequences processed\n",
      "20000000 sequences processed\n",
      "30000000 sequences processed\n",
      "This is cutadapt 1.18 with Python 3.6.15\n",
      "Command line parameters: -j 8 -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900797.fastq\n",
      "Processing reads on 8 cores in single-end mode ...\n",
      "Finished in 154.62 s (4 us/read; 14.17 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              36,521,820\n",
      "Reads with adapters:                 2,967,697 (8.1%)\n",
      "Reads written (passing filters):    36,521,820 (100.0%)\n",
      "\n",
      "Total basepairs processed: 1,855,132,078 bp\n",
      "Quality-trimmed:             254,793,487 bp (13.7%)\n",
      "Total written (filtered):  1,593,702,000 bp (85.9%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 2967697 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 28.7%\n",
      "  C: 27.7%\n",
      "  G: 19.5%\n",
      "  T: 22.2%\n",
      "  none/other: 1.9%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t2105819\t9130455.0\t0\t2105819\n",
      "2\t600805\t2282613.8\t0\t600805\n",
      "3\t159124\t570653.4\t0\t159124\n",
      "4\t36220\t142663.4\t0\t36220\n",
      "5\t6541\t35665.8\t0\t6541\n",
      "6\t849\t8916.5\t0\t849\n",
      "7\t196\t2229.1\t0\t196\n",
      "8\t121\t557.3\t0\t121\n",
      "9\t286\t139.3\t0\t159 127\n",
      "10\t507\t34.8\t1\t55 452\n",
      "11\t403\t8.7\t1\t64 339\n",
      "12\t418\t2.2\t1\t41 377\n",
      "13\t246\t0.5\t1\t47 199\n",
      "14\t284\t0.5\t1\t252 32\n",
      "15\t230\t0.5\t1\t206 24\n",
      "16\t304\t0.5\t1\t267 37\n",
      "17\t249\t0.5\t1\t227 22\n",
      "18\t302\t0.5\t1\t262 40\n",
      "19\t259\t0.5\t1\t229 30\n",
      "20\t232\t0.5\t1\t209 23\n",
      "21\t235\t0.5\t1\t215 20\n",
      "22\t251\t0.5\t1\t222 29\n",
      "23\t298\t0.5\t1\t260 38\n",
      "24\t321\t0.5\t1\t301 20\n",
      "25\t263\t0.5\t1\t247 16\n",
      "26\t271\t0.5\t1\t241 30\n",
      "27\t275\t0.5\t1\t252 23\n",
      "28\t327\t0.5\t1\t284 43\n",
      "29\t293\t0.5\t1\t260 33\n",
      "30\t369\t0.5\t1\t313 56\n",
      "31\t374\t0.5\t1\t343 31\n",
      "32\t305\t0.5\t1\t277 28\n",
      "33\t317\t0.5\t1\t285 32\n",
      "34\t361\t0.5\t1\t331 30\n",
      "35\t402\t0.5\t1\t361 41\n",
      "36\t399\t0.5\t1\t363 36\n",
      "37\t377\t0.5\t1\t341 36\n",
      "38\t368\t0.5\t1\t334 34\n",
      "39\t354\t0.5\t1\t330 24\n",
      "40\t293\t0.5\t1\t258 35\n",
      "41\t264\t0.5\t1\t234 30\n",
      "42\t370\t0.5\t1\t307 63\n",
      "43\t189\t0.5\t1\t151 38\n",
      "44\t229\t0.5\t1\t155 74\n",
      "45\t270\t0.5\t1\t156 114\n",
      "46\t379\t0.5\t1\t253 126\n",
      "47\t469\t0.5\t1\t252 217\n",
      "48\t970\t0.5\t1\t540 430\n",
      "49\t4241\t0.5\t1\t2089 2152\n",
      "50\t237\t0.5\t1\t117 120\n",
      "51\t40231\t0.5\t1\t22381 17850\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900797.fastq\n",
      "=============================================\n",
      "36521820 sequences processed in total\n",
      "Sequences removed because they became shorter than the length cutoff of 20 bp:\t4803113 (13.2%)\n",
      "\n",
      "Now renaming the output file SRR6900797_trimmed.fq\n",
      "\n",
      "ORIGINAL FILE 1: >>SRR6900797_trimmed.fq<<\tRENAMING TO:>>Macaque_Heart_R1_trimmed.fq<<\n",
      "\n",
      "  >>> Now running FastQC on the data <<<\n",
      "\n",
      "Started analysis of Macaque_Heart_R1_trimmed.fq\n",
      "Approx 5% complete for Macaque_Heart_R1_trimmed.fq\n",
      "Approx 10% complete for Macaque_Heart_R1_trimmed.fq\n",
      "Approx 15% complete for Macaque_Heart_R1_trimmed.fq\n",
      "Approx 20% complete for Macaque_Heart_R1_trimmed.fq\n",
      "Approx 25% complete for Macaque_Heart_R1_trimmed.fq\n",
      "Approx 30% complete for Macaque_Heart_R1_trimmed.fq\n",
      "Approx 35% complete for Macaque_Heart_R1_trimmed.fq\n",
      "Approx 40% complete for Macaque_Heart_R1_trimmed.fq\n",
      "Approx 45% complete for Macaque_Heart_R1_trimmed.fq\n",
      "Approx 50% complete for Macaque_Heart_R1_trimmed.fq\n",
      "Approx 55% complete for Macaque_Heart_R1_trimmed.fq\n",
      "Approx 60% complete for Macaque_Heart_R1_trimmed.fq\n",
      "Approx 65% complete for Macaque_Heart_R1_trimmed.fq\n",
      "Approx 70% complete for Macaque_Heart_R1_trimmed.fq\n",
      "Approx 75% complete for Macaque_Heart_R1_trimmed.fq\n",
      "Approx 80% complete for Macaque_Heart_R1_trimmed.fq\n",
      "Approx 85% complete for Macaque_Heart_R1_trimmed.fq\n",
      "Approx 90% complete for Macaque_Heart_R1_trimmed.fq\n",
      "Approx 95% complete for Macaque_Heart_R1_trimmed.fq\n",
      "Analysis complete for Macaque_Heart_R1_trimmed.fq\n",
      "Using an excessive number of cores has a diminishing return! It is recommended not to exceed 8 cores per trimming process (you asked for 8 cores). Please consider re-specifying\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "Cutadapt version: 1.18\n",
      "Could not detect version of Python used by Cutadapt from the first line of Cutadapt (but found this: >>>#!/bin/sh<<<)\n",
      "Letting the (modified) Cutadapt deal with the Python version instead\n",
      "pigz 2.6\n",
      "Parallel gzip (pigz) detected. Proceeding with multicore (de)compression using 8 cores\n",
      "\n",
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Output will be written into the directory: /gpfs52/data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/trimmed_reads/\n",
      "Using user-specified basename (>>Macaque_Kidney_R1<<) instead of deriving the filename from the input file(s)\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900798.fastq <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Illumina\t681\tAGATCGGAAGAGC\t1000000\t0.07\n",
      "smallRNA\t1\tTGGAATTCTCGG\t1000000\t0.00\n",
      "Nextera\t1\tCTGTCTCTTATA\t1000000\t0.00\n",
      "Using Illumina adapter for trimming (count: 681). Second best hit was smallRNA (count: 1)\n",
      "\n",
      "Writing report to '/gpfs52/data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/trimmed_reads/SRR6900798.fastq_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900798.fastq\n",
      "Trimming mode: single-end\n",
      "Trim Galore version: 0.6.6\n",
      "Cutadapt version: 1.18\n",
      "Python version: could not detect\n",
      "Number of cores used for trimming: 8\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length before a sequence gets removed: 20 bp\n",
      "Running FastQC on the data once trimming has completed\n",
      "Running FastQC with the following extra arguments: '--outdir /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/processing_results/fastQC'\n",
      "\n",
      "Cutadapt seems to be reasonably up-to-date. Setting -j 8\n",
      "Writing final adapter and quality trimmed output to SRR6900798_trimmed.fq\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff '-q 20') and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900798.fastq <<< \n",
      "10000000 sequences processed\n",
      "20000000 sequences processed\n",
      "30000000 sequences processed\n",
      "This is cutadapt 1.18 with Python 3.6.15\n",
      "Command line parameters: -j 8 -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900798.fastq\n",
      "Processing reads on 8 cores in single-end mode ...\n",
      "Finished in 137.87 s (4 us/read; 14.96 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              34,378,784\n",
      "Reads with adapters:                 2,257,392 (6.6%)\n",
      "Reads written (passing filters):    34,378,784 (100.0%)\n",
      "\n",
      "Total basepairs processed: 1,747,959,071 bp\n",
      "Quality-trimmed:             252,212,957 bp (14.4%)\n",
      "Total written (filtered):  1,490,532,172 bp (85.3%)\n",
      "\n",
      "=== Adapter 1 ===\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 2257392 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 30.4%\n",
      "  C: 28.6%\n",
      "  G: 19.0%\n",
      "  T: 19.7%\n",
      "  none/other: 2.2%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t1607355\t8594696.0\t0\t1607355\n",
      "2\t443638\t2148674.0\t0\t443638\n",
      "3\t126578\t537168.5\t0\t126578\n",
      "4\t26620\t134292.1\t0\t26620\n",
      "5\t5170\t33573.0\t0\t5170\n",
      "6\t586\t8393.3\t0\t586\n",
      "7\t132\t2098.3\t0\t132\n",
      "8\t80\t524.6\t0\t80\n",
      "9\t157\t131.1\t0\t72 85\n",
      "10\t347\t32.8\t1\t25 322\n",
      "11\t294\t8.2\t1\t42 252\n",
      "12\t240\t2.0\t1\t26 214\n",
      "13\t191\t0.5\t1\t28 163\n",
      "14\t213\t0.5\t1\t184 29\n",
      "15\t184\t0.5\t1\t162 22\n",
      "16\t195\t0.5\t1\t176 19\n",
      "17\t176\t0.5\t1\t152 24\n",
      "18\t194\t0.5\t1\t167 27\n",
      "19\t206\t0.5\t1\t183 23\n",
      "20\t196\t0.5\t1\t159 37\n",
      "21\t164\t0.5\t1\t145 19\n",
      "22\t172\t0.5\t1\t154 18\n",
      "23\t178\t0.5\t1\t161 17\n",
      "24\t196\t0.5\t1\t173 23\n",
      "25\t184\t0.5\t1\t166 18\n",
      "26\t223\t0.5\t1\t199 24\n",
      "27\t187\t0.5\t1\t164 23\n",
      "28\t219\t0.5\t1\t191 28\n",
      "29\t221\t0.5\t1\t197 24\n",
      "30\t243\t0.5\t1\t212 31\n",
      "31\t207\t0.5\t1\t184 23\n",
      "32\t230\t0.5\t1\t206 24\n",
      "33\t268\t0.5\t1\t234 34\n",
      "34\t229\t0.5\t1\t196 33\n",
      "35\t247\t0.5\t1\t217 30\n",
      "36\t256\t0.5\t1\t219 37\n",
      "37\t288\t0.5\t1\t251 37\n",
      "38\t314\t0.5\t1\t268 46\n",
      "39\t306\t0.5\t1\t245 61\n",
      "40\t245\t0.5\t1\t189 56\n",
      "41\t227\t0.5\t1\t182 45\n",
      "42\t323\t0.5\t1\t251 72\n",
      "43\t206\t0.5\t1\t140 66\n",
      "44\t240\t0.5\t1\t109 131\n",
      "45\t425\t0.5\t1\t254 171\n",
      "46\t409\t0.5\t1\t225 184\n",
      "47\t504\t0.5\t1\t290 214\n",
      "48\t781\t0.5\t1\t456 325\n",
      "49\t1883\t0.5\t1\t910 973\n",
      "50\t375\t0.5\t1\t199 176\n",
      "51\t34690\t0.5\t1\t19473 15217\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900798.fastq\n",
      "=============================================\n",
      "34378784 sequences processed in total\n",
      "Sequences removed because they became shorter than the length cutoff of 20 bp:\t4819208 (14.0%)\n",
      "\n",
      "Now renaming the output file SRR6900798_trimmed.fq\n",
      "\n",
      "ORIGINAL FILE 1: >>SRR6900798_trimmed.fq<<\tRENAMING TO:>>Macaque_Kidney_R1_trimmed.fq<<\n",
      "\n",
      "  >>> Now running FastQC on the data <<<\n",
      "\n",
      "Started analysis of Macaque_Kidney_R1_trimmed.fq\n",
      "Approx 5% complete for Macaque_Kidney_R1_trimmed.fq\n",
      "Approx 10% complete for Macaque_Kidney_R1_trimmed.fq\n",
      "Approx 15% complete for Macaque_Kidney_R1_trimmed.fq\n",
      "Approx 20% complete for Macaque_Kidney_R1_trimmed.fq\n",
      "Approx 25% complete for Macaque_Kidney_R1_trimmed.fq\n",
      "Approx 30% complete for Macaque_Kidney_R1_trimmed.fq\n",
      "Approx 35% complete for Macaque_Kidney_R1_trimmed.fq\n",
      "Approx 40% complete for Macaque_Kidney_R1_trimmed.fq\n",
      "Approx 45% complete for Macaque_Kidney_R1_trimmed.fq\n",
      "Approx 50% complete for Macaque_Kidney_R1_trimmed.fq\n",
      "Approx 55% complete for Macaque_Kidney_R1_trimmed.fq\n",
      "Approx 60% complete for Macaque_Kidney_R1_trimmed.fq\n",
      "Approx 65% complete for Macaque_Kidney_R1_trimmed.fq\n",
      "Approx 70% complete for Macaque_Kidney_R1_trimmed.fq\n",
      "Approx 75% complete for Macaque_Kidney_R1_trimmed.fq\n",
      "Approx 80% complete for Macaque_Kidney_R1_trimmed.fq\n",
      "Approx 85% complete for Macaque_Kidney_R1_trimmed.fq\n",
      "Approx 90% complete for Macaque_Kidney_R1_trimmed.fq\n",
      "Approx 95% complete for Macaque_Kidney_R1_trimmed.fq\n",
      "Analysis complete for Macaque_Kidney_R1_trimmed.fq\n",
      "Using an excessive number of cores has a diminishing return! It is recommended not to exceed 8 cores per trimming process (you asked for 8 cores). Please consider re-specifying\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "Cutadapt version: 1.18\n",
      "Could not detect version of Python used by Cutadapt from the first line of Cutadapt (but found this: >>>#!/bin/sh<<<)\n",
      "Letting the (modified) Cutadapt deal with the Python version instead\n",
      "pigz 2.6\n",
      "Parallel gzip (pigz) detected. Proceeding with multicore (de)compression using 8 cores\n",
      "\n",
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Output will be written into the directory: /gpfs52/data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/trimmed_reads/\n",
      "Using user-specified basename (>>Macaque_Liver_R1<<) instead of deriving the filename from the input file(s)\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900799.fastq <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Illumina\t2314\tAGATCGGAAGAGC\t1000000\t0.23\n",
      "smallRNA\t0\tTGGAATTCTCGG\t1000000\t0.00\n",
      "Nextera\t0\tCTGTCTCTTATA\t1000000\t0.00\n",
      "Using Illumina adapter for trimming (count: 2314). Second best hit was smallRNA (count: 0)\n",
      "\n",
      "Writing report to '/gpfs52/data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/trimmed_reads/SRR6900799.fastq_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900799.fastq\n",
      "Trimming mode: single-end\n",
      "Trim Galore version: 0.6.6\n",
      "Cutadapt version: 1.18\n",
      "Python version: could not detect\n",
      "Number of cores used for trimming: 8\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length before a sequence gets removed: 20 bp\n",
      "Running FastQC on the data once trimming has completed\n",
      "Running FastQC with the following extra arguments: '--outdir /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/processing_results/fastQC'\n",
      "\n",
      "Cutadapt seems to be reasonably up-to-date. Setting -j 8\n",
      "Writing final adapter and quality trimmed output to SRR6900799_trimmed.fq\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff '-q 20') and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900799.fastq <<< \n",
      "10000000 sequences processed\n",
      "20000000 sequences processed\n",
      "30000000 sequences processed\n",
      "This is cutadapt 1.18 with Python 3.6.15\n",
      "Command line parameters: -j 8 -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900799.fastq\n",
      "Processing reads on 8 cores in single-end mode ...\n",
      "Finished in 149.28 s (4 us/read; 14.94 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              37,159,831\n",
      "Reads with adapters:                 2,347,555 (6.3%)\n",
      "Reads written (passing filters):    37,159,831 (100.0%)\n",
      "\n",
      "Total basepairs processed: 1,889,890,709 bp\n",
      "Quality-trimmed:             276,577,058 bp (14.6%)\n",
      "Total written (filtered):  1,602,097,784 bp (84.8%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 2347555 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 29.1%\n",
      "  C: 29.1%\n",
      "  G: 19.3%\n",
      "  T: 15.6%\n",
      "  none/other: 6.9%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t1563803\t9289957.8\t0\t1563803\n",
      "2\t450296\t2322489.4\t0\t450296\n",
      "3\t128269\t580622.4\t0\t128269\n",
      "4\t25969\t145155.6\t0\t25969\n",
      "5\t5601\t36288.9\t0\t5601\n",
      "6\t533\t9072.2\t0\t533\n",
      "7\t172\t2268.1\t0\t172\n",
      "8\t150\t567.0\t0\t150\n",
      "9\t210\t141.8\t0\t120 90\n",
      "10\t744\t35.4\t1\t105 639\n",
      "11\t600\t8.9\t1\t119 481\n",
      "12\t557\t2.2\t1\t93 464\n",
      "13\t536\t0.6\t1\t133 403\n",
      "14\t567\t0.6\t1\t516 51\n",
      "15\t507\t0.6\t1\t471 36\n",
      "16\t505\t0.6\t1\t456 49\n",
      "17\t526\t0.6\t1\t486 40\n",
      "18\t598\t0.6\t1\t548 50\n",
      "19\t549\t0.6\t1\t508 41\n",
      "20\t533\t0.6\t1\t484 49\n",
      "21\t528\t0.6\t1\t484 44\n",
      "22\t518\t0.6\t1\t467 51\n",
      "23\t653\t0.6\t1\t603 50\n",
      "24\t612\t0.6\t1\t571 41\n",
      "25\t643\t0.6\t1\t594 49\n",
      "26\t583\t0.6\t1\t546 37\n",
      "27\t1130\t0.6\t1\t1083 47\n",
      "28\t705\t0.6\t1\t664 41\n",
      "29\t681\t0.6\t1\t629 52\n",
      "30\t781\t0.6\t1\t732 49\n",
      "31\t759\t0.6\t1\t716 43\n",
      "32\t731\t0.6\t1\t682 49\n",
      "33\t837\t0.6\t1\t781 56\n",
      "34\t729\t0.6\t1\t685 44\n",
      "35\t883\t0.6\t1\t825 58\n",
      "36\t833\t0.6\t1\t759 74\n",
      "37\t793\t0.6\t1\t720 73\n",
      "38\t703\t0.6\t1\t641 62\n",
      "39\t808\t0.6\t1\t747 61\n",
      "40\t595\t0.6\t1\t513 82\n",
      "41\t444\t0.6\t1\t376 68\n",
      "42\t557\t0.6\t1\t461 96\n",
      "43\t344\t0.6\t1\t259 85\n",
      "44\t417\t0.6\t1\t223 194\n",
      "45\t621\t0.6\t1\t358 263\n",
      "46\t819\t0.6\t1\t478 341\n",
      "47\t1179\t0.6\t1\t703 476\n",
      "48\t2259\t0.6\t1\t1274 985\n",
      "49\t10566\t0.6\t1\t5411 5155\n",
      "50\t766\t0.6\t1\t476 290\n",
      "51\t133853\t0.6\t1\t76992 56861\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900799.fastq\n",
      "=============================================\n",
      "37159831 sequences processed in total\n",
      "Sequences removed because they became shorter than the length cutoff of 20 bp:\t5419462 (14.6%)\n",
      "\n",
      "Now renaming the output file SRR6900799_trimmed.fq\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "ORIGINAL FILE 1: >>SRR6900799_trimmed.fq<<\tRENAMING TO:>>Macaque_Liver_R1_trimmed.fq<<\n",
      "\n",
      "  >>> Now running FastQC on the data <<<\n",
      "\n",
      "Started analysis of Macaque_Liver_R1_trimmed.fq\n",
      "Approx 5% complete for Macaque_Liver_R1_trimmed.fq\n",
      "Approx 10% complete for Macaque_Liver_R1_trimmed.fq\n",
      "Approx 15% complete for Macaque_Liver_R1_trimmed.fq\n",
      "Approx 20% complete for Macaque_Liver_R1_trimmed.fq\n",
      "Approx 25% complete for Macaque_Liver_R1_trimmed.fq\n",
      "Approx 30% complete for Macaque_Liver_R1_trimmed.fq\n",
      "Approx 35% complete for Macaque_Liver_R1_trimmed.fq\n",
      "Approx 40% complete for Macaque_Liver_R1_trimmed.fq\n",
      "Approx 45% complete for Macaque_Liver_R1_trimmed.fq\n",
      "Approx 50% complete for Macaque_Liver_R1_trimmed.fq\n",
      "Approx 55% complete for Macaque_Liver_R1_trimmed.fq\n",
      "Approx 60% complete for Macaque_Liver_R1_trimmed.fq\n",
      "Approx 65% complete for Macaque_Liver_R1_trimmed.fq\n",
      "Approx 70% complete for Macaque_Liver_R1_trimmed.fq\n",
      "Approx 75% complete for Macaque_Liver_R1_trimmed.fq\n",
      "Approx 80% complete for Macaque_Liver_R1_trimmed.fq\n",
      "Approx 85% complete for Macaque_Liver_R1_trimmed.fq\n",
      "Approx 90% complete for Macaque_Liver_R1_trimmed.fq\n",
      "Approx 95% complete for Macaque_Liver_R1_trimmed.fq\n",
      "Analysis complete for Macaque_Liver_R1_trimmed.fq\n",
      "Using an excessive number of cores has a diminishing return! It is recommended not to exceed 8 cores per trimming process (you asked for 8 cores). Please consider re-specifying\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "Cutadapt version: 1.18\n",
      "Could not detect version of Python used by Cutadapt from the first line of Cutadapt (but found this: >>>#!/bin/sh<<<)\n",
      "Letting the (modified) Cutadapt deal with the Python version instead\n",
      "pigz 2.6\n",
      "Parallel gzip (pigz) detected. Proceeding with multicore (de)compression using 8 cores\n",
      "\n",
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Output will be written into the directory: /gpfs52/data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/trimmed_reads/\n",
      "Using user-specified basename (>>Macaque_Lung_R1<<) instead of deriving the filename from the input file(s)\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900800.fastq <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Illumina\t1347\tAGATCGGAAGAGC\t1000000\t0.13\n",
      "smallRNA\t3\tTGGAATTCTCGG\t1000000\t0.00\n",
      "Nextera\t1\tCTGTCTCTTATA\t1000000\t0.00\n",
      "Using Illumina adapter for trimming (count: 1347). Second best hit was smallRNA (count: 3)\n",
      "\n",
      "Writing report to '/gpfs52/data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/trimmed_reads/SRR6900800.fastq_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900800.fastq\n",
      "Trimming mode: single-end\n",
      "Trim Galore version: 0.6.6\n",
      "Cutadapt version: 1.18\n",
      "Python version: could not detect\n",
      "Number of cores used for trimming: 8\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length before a sequence gets removed: 20 bp\n",
      "Running FastQC on the data once trimming has completed\n",
      "Running FastQC with the following extra arguments: '--outdir /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/processing_results/fastQC'\n",
      "\n",
      "Cutadapt seems to be reasonably up-to-date. Setting -j 8\n",
      "Writing final adapter and quality trimmed output to SRR6900800_trimmed.fq\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff '-q 20') and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900800.fastq <<< \n",
      "10000000 sequences processed\n",
      "20000000 sequences processed\n",
      "30000000 sequences processed\n",
      "40000000 sequences processed\n",
      "This is cutadapt 1.18 with Python 3.6.15\n",
      "Command line parameters: -j 8 -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900800.fastq\n",
      "Processing reads on 8 cores in single-end mode ...\n",
      "Finished in 180.98 s (4 us/read; 14.47 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              43,638,825\n",
      "Reads with adapters:                 2,718,443 (6.2%)\n",
      "Reads written (passing filters):    43,638,825 (100.0%)\n",
      "\n",
      "Total basepairs processed: 2,219,360,909 bp\n",
      "Quality-trimmed:             323,708,243 bp (14.6%)\n",
      "Total written (filtered):  1,885,411,764 bp (85.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 2718443 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 29.2%\n",
      "  C: 29.5%\n",
      "  G: 19.5%\n",
      "  T: 16.6%\n",
      "  none/other: 5.1%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t1848347\t10909706.2\t0\t1848347\n",
      "2\t540573\t2727426.6\t0\t540573\n",
      "3\t153915\t681856.6\t0\t153915\n",
      "4\t30182\t170464.2\t0\t30182\n",
      "5\t6444\t42616.0\t0\t6444\n",
      "6\t793\t10654.0\t0\t793\n",
      "7\t191\t2663.5\t0\t191\n",
      "8\t97\t665.9\t0\t97\n",
      "9\t236\t166.5\t0\t72 164\n",
      "10\t482\t41.6\t1\t59 423\n",
      "11\t372\t10.4\t1\t69 303\n",
      "12\t307\t2.6\t1\t66 241\n",
      "13\t326\t0.7\t1\t61 265\n",
      "14\t320\t0.7\t1\t262 58\n",
      "15\t304\t0.7\t1\t261 43\n",
      "16\t284\t0.7\t1\t251 33\n",
      "17\t289\t0.7\t1\t248 41\n",
      "18\t295\t0.7\t1\t257 38\n",
      "19\t266\t0.7\t1\t232 34\n",
      "20\t262\t0.7\t1\t214 48\n",
      "21\t275\t0.7\t1\t241 34\n",
      "22\t290\t0.7\t1\t244 46\n",
      "23\t305\t0.7\t1\t265 40\n",
      "24\t278\t0.7\t1\t250 28\n",
      "25\t310\t0.7\t1\t273 37\n",
      "26\t291\t0.7\t1\t260 31\n",
      "27\t312\t0.7\t1\t261 51\n",
      "28\t335\t0.7\t1\t270 65\n",
      "29\t454\t0.7\t1\t389 65\n",
      "30\t374\t0.7\t1\t310 64\n",
      "31\t680\t0.7\t1\t620 60\n",
      "32\t333\t0.7\t1\t293 40\n",
      "33\t387\t0.7\t1\t331 56\n",
      "34\t493\t0.7\t1\t392 101\n",
      "35\t409\t0.7\t1\t348 61\n",
      "36\t452\t0.7\t1\t390 62\n",
      "37\t400\t0.7\t1\t333 67\n",
      "38\t393\t0.7\t1\t318 75\n",
      "39\t440\t0.7\t1\t340 100\n",
      "40\t363\t0.7\t1\t260 103\n",
      "41\t370\t0.7\t1\t270 100\n",
      "42\t431\t0.7\t1\t306 125\n",
      "43\t300\t0.7\t1\t193 107\n",
      "44\t425\t0.7\t1\t203 222\n",
      "45\t744\t0.7\t1\t398 346\n",
      "46\t1015\t0.7\t1\t614 401\n",
      "47\t1031\t0.7\t1\t534 497\n",
      "48\t2033\t0.7\t1\t1093 940\n",
      "49\t11077\t0.7\t1\t5608 5469\n",
      "50\t746\t0.7\t1\t451 295\n",
      "51\t108412\t0.7\t1\t61680 46732\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900800.fastq\n",
      "=============================================\n",
      "43638825 sequences processed in total\n",
      "Sequences removed because they became shorter than the length cutoff of 20 bp:\t6278108 (14.4%)\n",
      "\n",
      "Now renaming the output file SRR6900800_trimmed.fq\n",
      "\n",
      "ORIGINAL FILE 1: >>SRR6900800_trimmed.fq<<\tRENAMING TO:>>Macaque_Lung_R1_trimmed.fq<<\n",
      "\n",
      "  >>> Now running FastQC on the data <<<\n",
      "\n",
      "Started analysis of Macaque_Lung_R1_trimmed.fq\n",
      "Approx 5% complete for Macaque_Lung_R1_trimmed.fq\n",
      "Approx 10% complete for Macaque_Lung_R1_trimmed.fq\n",
      "Approx 15% complete for Macaque_Lung_R1_trimmed.fq\n",
      "Approx 20% complete for Macaque_Lung_R1_trimmed.fq\n",
      "Approx 25% complete for Macaque_Lung_R1_trimmed.fq\n",
      "Approx 30% complete for Macaque_Lung_R1_trimmed.fq\n",
      "Approx 35% complete for Macaque_Lung_R1_trimmed.fq\n",
      "Approx 40% complete for Macaque_Lung_R1_trimmed.fq\n",
      "Approx 45% complete for Macaque_Lung_R1_trimmed.fq\n",
      "Approx 50% complete for Macaque_Lung_R1_trimmed.fq\n",
      "Approx 55% complete for Macaque_Lung_R1_trimmed.fq\n",
      "Approx 60% complete for Macaque_Lung_R1_trimmed.fq\n",
      "Approx 65% complete for Macaque_Lung_R1_trimmed.fq\n",
      "Approx 70% complete for Macaque_Lung_R1_trimmed.fq\n",
      "Approx 75% complete for Macaque_Lung_R1_trimmed.fq\n",
      "Approx 80% complete for Macaque_Lung_R1_trimmed.fq\n",
      "Approx 85% complete for Macaque_Lung_R1_trimmed.fq\n",
      "Approx 90% complete for Macaque_Lung_R1_trimmed.fq\n",
      "Approx 95% complete for Macaque_Lung_R1_trimmed.fq\n",
      "Analysis complete for Macaque_Lung_R1_trimmed.fq\n",
      "Using an excessive number of cores has a diminishing return! It is recommended not to exceed 8 cores per trimming process (you asked for 8 cores). Please consider re-specifying\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "Cutadapt version: 1.18\n",
      "Could not detect version of Python used by Cutadapt from the first line of Cutadapt (but found this: >>>#!/bin/sh<<<)\n",
      "Letting the (modified) Cutadapt deal with the Python version instead\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "pigz 2.6\n",
      "Parallel gzip (pigz) detected. Proceeding with multicore (de)compression using 8 cores\n",
      "\n",
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Output will be written into the directory: /gpfs52/data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/trimmed_reads/\n",
      "Using user-specified basename (>>Macaque_Heart_R2<<) instead of deriving the filename from the input file(s)\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900801.fastq <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Illumina\t1877\tAGATCGGAAGAGC\t1000000\t0.19\n",
      "smallRNA\t0\tTGGAATTCTCGG\t1000000\t0.00\n",
      "Nextera\t0\tCTGTCTCTTATA\t1000000\t0.00\n",
      "Using Illumina adapter for trimming (count: 1877). Second best hit was smallRNA (count: 0)\n",
      "\n",
      "Writing report to '/gpfs52/data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/trimmed_reads/SRR6900801.fastq_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900801.fastq\n",
      "Trimming mode: single-end\n",
      "Trim Galore version: 0.6.6\n",
      "Cutadapt version: 1.18\n",
      "Python version: could not detect\n",
      "Number of cores used for trimming: 8\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length before a sequence gets removed: 20 bp\n",
      "Running FastQC on the data once trimming has completed\n",
      "Running FastQC with the following extra arguments: '--outdir /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/processing_results/fastQC'\n",
      "\n",
      "Cutadapt seems to be reasonably up-to-date. Setting -j 8\n",
      "Writing final adapter and quality trimmed output to SRR6900801_trimmed.fq\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff '-q 20') and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900801.fastq <<< \n",
      "10000000 sequences processed\n",
      "20000000 sequences processed\n",
      "30000000 sequences processed\n",
      "40000000 sequences processed\n",
      "This is cutadapt 1.18 with Python 3.6.15\n",
      "Command line parameters: -j 8 -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900801.fastq\n",
      "Processing reads on 8 cores in single-end mode ...\n",
      "Finished in 179.87 s (4 us/read; 14.79 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              44,327,075\n",
      "Reads with adapters:                 3,442,825 (7.8%)\n",
      "Reads written (passing filters):    44,327,075 (100.0%)\n",
      "\n",
      "Total basepairs processed: 2,252,575,254 bp\n",
      "Quality-trimmed:             315,955,457 bp (14.0%)\n",
      "Total written (filtered):  1,920,515,472 bp (85.3%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 3442825 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 27.1%\n",
      "  C: 26.1%\n",
      "  G: 18.6%\n",
      "  T: 21.3%\n",
      "  none/other: 6.9%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t2319842\t11081768.8\t0\t2319842\n",
      "2\t659740\t2770442.2\t0\t659740\n",
      "3\t175775\t692610.5\t0\t175775\n",
      "4\t39287\t173152.6\t0\t39287\n",
      "5\t7207\t43288.2\t0\t7207\n",
      "6\t981\t10822.0\t0\t981\n",
      "7\t286\t2705.5\t0\t286\n",
      "8\t181\t676.4\t0\t181\n",
      "9\t308\t169.1\t0\t192 116\n",
      "10\t578\t42.3\t1\t87 491\n",
      "11\t587\t10.6\t1\t139 448\n",
      "12\t535\t2.6\t1\t91 444\n",
      "13\t358\t0.7\t1\t115 243\n",
      "14\t412\t0.7\t1\t364 48\n",
      "15\t345\t0.7\t1\t306 39\n",
      "16\t417\t0.7\t1\t369 48\n",
      "17\t328\t0.7\t1\t289 39\n",
      "18\t505\t0.7\t1\t436 69\n",
      "19\t303\t0.7\t1\t275 28\n",
      "20\t318\t0.7\t1\t271 47\n",
      "21\t304\t0.7\t1\t270 34\n",
      "22\t314\t0.7\t1\t274 40\n",
      "23\t333\t0.7\t1\t297 36\n",
      "24\t357\t0.7\t1\t324 33\n",
      "25\t365\t0.7\t1\t324 41\n",
      "26\t414\t0.7\t1\t364 50\n",
      "27\t368\t0.7\t1\t333 35\n",
      "28\t500\t0.7\t1\t420 80\n",
      "29\t468\t0.7\t1\t418 50\n",
      "30\t791\t0.7\t1\t696 95\n",
      "31\t693\t0.7\t1\t633 60\n",
      "32\t509\t0.7\t1\t464 45\n",
      "33\t501\t0.7\t1\t430 71\n",
      "34\t613\t0.7\t1\t551 62\n",
      "35\t656\t0.7\t1\t568 88\n",
      "36\t652\t0.7\t1\t579 73\n",
      "37\t575\t0.7\t1\t512 63\n",
      "38\t586\t0.7\t1\t507 79\n",
      "39\t710\t0.7\t1\t555 155\n",
      "40\t869\t0.7\t1\t641 228\n",
      "41\t684\t0.7\t1\t519 165\n",
      "42\t751\t0.7\t1\t591 160\n",
      "43\t576\t0.7\t1\t411 165\n",
      "44\t713\t0.7\t1\t508 205\n",
      "45\t731\t0.7\t1\t446 285\n",
      "46\t1317\t0.7\t1\t652 665\n",
      "47\t3398\t0.7\t1\t2077 1321\n",
      "48\t3921\t0.7\t1\t2357 1564\n",
      "49\t17060\t0.7\t1\t9916 7144\n",
      "50\t1549\t0.7\t1\t1123 426\n",
      "51\t193254\t0.7\t1\t133913 59341\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900801.fastq\n",
      "=============================================\n",
      "44327075 sequences processed in total\n",
      "Sequences removed because they became shorter than the length cutoff of 20 bp:\t6160155 (13.9%)\n",
      "\n",
      "Now renaming the output file SRR6900801_trimmed.fq\n",
      "\n",
      "ORIGINAL FILE 1: >>SRR6900801_trimmed.fq<<\tRENAMING TO:>>Macaque_Heart_R2_trimmed.fq<<\n",
      "\n",
      "  >>> Now running FastQC on the data <<<\n",
      "\n",
      "Started analysis of Macaque_Heart_R2_trimmed.fq\n",
      "Approx 5% complete for Macaque_Heart_R2_trimmed.fq\n",
      "Approx 10% complete for Macaque_Heart_R2_trimmed.fq\n",
      "Approx 15% complete for Macaque_Heart_R2_trimmed.fq\n",
      "Approx 20% complete for Macaque_Heart_R2_trimmed.fq\n",
      "Approx 25% complete for Macaque_Heart_R2_trimmed.fq\n",
      "Approx 30% complete for Macaque_Heart_R2_trimmed.fq\n",
      "Approx 35% complete for Macaque_Heart_R2_trimmed.fq\n",
      "Approx 40% complete for Macaque_Heart_R2_trimmed.fq\n",
      "Approx 45% complete for Macaque_Heart_R2_trimmed.fq\n",
      "Approx 50% complete for Macaque_Heart_R2_trimmed.fq\n",
      "Approx 55% complete for Macaque_Heart_R2_trimmed.fq\n",
      "Approx 60% complete for Macaque_Heart_R2_trimmed.fq\n",
      "Approx 65% complete for Macaque_Heart_R2_trimmed.fq\n",
      "Approx 70% complete for Macaque_Heart_R2_trimmed.fq\n",
      "Approx 75% complete for Macaque_Heart_R2_trimmed.fq\n",
      "Approx 80% complete for Macaque_Heart_R2_trimmed.fq\n",
      "Approx 85% complete for Macaque_Heart_R2_trimmed.fq\n",
      "Approx 90% complete for Macaque_Heart_R2_trimmed.fq\n",
      "Approx 95% complete for Macaque_Heart_R2_trimmed.fq\n",
      "Analysis complete for Macaque_Heart_R2_trimmed.fq\n",
      "Using an excessive number of cores has a diminishing return! It is recommended not to exceed 8 cores per trimming process (you asked for 8 cores). Please consider re-specifying\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "Cutadapt version: 1.18\n",
      "Could not detect version of Python used by Cutadapt from the first line of Cutadapt (but found this: >>>#!/bin/sh<<<)\n",
      "Letting the (modified) Cutadapt deal with the Python version instead\n",
      "pigz 2.6\n",
      "Parallel gzip (pigz) detected. Proceeding with multicore (de)compression using 8 cores\n",
      "\n",
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Output will be written into the directory: /gpfs52/data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/trimmed_reads/\n",
      "Using user-specified basename (>>Macaque_Kidney_R2<<) instead of deriving the filename from the input file(s)\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900802.fastq <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Illumina\t4036\tAGATCGGAAGAGC\t1000000\t0.40\n",
      "smallRNA\t1\tTGGAATTCTCGG\t1000000\t0.00\n",
      "Nextera\t0\tCTGTCTCTTATA\t1000000\t0.00\n",
      "Using Illumina adapter for trimming (count: 4036). Second best hit was smallRNA (count: 1)\n",
      "\n",
      "Writing report to '/gpfs52/data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/trimmed_reads/SRR6900802.fastq_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900802.fastq\n",
      "Trimming mode: single-end\n",
      "Trim Galore version: 0.6.6\n",
      "Cutadapt version: 1.18\n",
      "Python version: could not detect\n",
      "Number of cores used for trimming: 8\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length before a sequence gets removed: 20 bp\n",
      "Running FastQC on the data once trimming has completed\n",
      "Running FastQC with the following extra arguments: '--outdir /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/processing_results/fastQC'\n",
      "\n",
      "Cutadapt seems to be reasonably up-to-date. Setting -j 8\n",
      "Writing final adapter and quality trimmed output to SRR6900802_trimmed.fq\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff '-q 20') and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900802.fastq <<< \n",
      "10000000 sequences processed\n",
      "20000000 sequences processed\n",
      "30000000 sequences processed\n",
      "40000000 sequences processed\n",
      "This is cutadapt 1.18 with Python 3.6.15\n",
      "Command line parameters: -j 8 -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900802.fastq\n",
      "Processing reads on 8 cores in single-end mode ...\n",
      "Finished in 175.52 s (4 us/read; 14.47 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              42,329,034\n",
      "Reads with adapters:                 4,189,680 (9.9%)\n",
      "Reads written (passing filters):    42,329,034 (100.0%)\n",
      "\n",
      "Total basepairs processed: 2,149,410,712 bp\n",
      "Quality-trimmed:             293,662,936 bp (13.7%)\n",
      "Total written (filtered):  1,821,277,376 bp (84.7%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 4189680 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 26.7%\n",
      "  C: 25.4%\n",
      "  G: 16.6%\n",
      "  T: 17.2%\n",
      "  none/other: 14.1%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t2599124\t10582258.5\t0\t2599124\n",
      "2\t730607\t2645564.6\t0\t730607\n",
      "3\t206571\t661391.2\t0\t206571\n",
      "4\t44491\t165347.8\t0\t44491\n",
      "5\t9156\t41336.9\t0\t9156\n",
      "6\t1264\t10334.2\t0\t1264\n",
      "7\t494\t2583.6\t0\t494\n",
      "8\t356\t645.9\t0\t356\n",
      "9\t602\t161.5\t0\t416 186\n",
      "10\t1045\t40.4\t1\t205 840\n",
      "11\t1028\t10.1\t1\t227 801\n",
      "12\t793\t2.5\t1\t214 579\n",
      "13\t740\t0.6\t1\t255 485\n",
      "14\t625\t0.6\t1\t543 82\n",
      "15\t589\t0.6\t1\t515 74\n",
      "16\t657\t0.6\t1\t588 69\n",
      "17\t733\t0.6\t1\t634 99\n",
      "18\t1026\t0.6\t1\t851 175\n",
      "19\t1227\t0.6\t1\t1090 137\n",
      "20\t614\t0.6\t1\t541 73\n",
      "21\t590\t0.6\t1\t539 51\n",
      "22\t557\t0.6\t1\t486 71\n",
      "23\t624\t0.6\t1\t539 85\n",
      "24\t603\t0.6\t1\t531 72\n",
      "25\t596\t0.6\t1\t527 69\n",
      "26\t545\t0.6\t1\t476 69\n",
      "27\t624\t0.6\t1\t547 77\n",
      "28\t649\t0.6\t1\t585 64\n",
      "29\t709\t0.6\t1\t642 67\n",
      "30\t760\t0.6\t1\t671 89\n",
      "31\t709\t0.6\t1\t613 96\n",
      "32\t1025\t0.6\t1\t867 158\n",
      "33\t790\t0.6\t1\t708 82\n",
      "34\t826\t0.6\t1\t697 129\n",
      "35\t913\t0.6\t1\t779 134\n",
      "36\t866\t0.6\t1\t751 115\n",
      "37\t796\t0.6\t1\t681 115\n",
      "38\t778\t0.6\t1\t646 132\n",
      "39\t935\t0.6\t1\t757 178\n",
      "40\t774\t0.6\t1\t575 199\n",
      "41\t732\t0.6\t1\t571 161\n",
      "42\t827\t0.6\t1\t623 204\n",
      "43\t778\t0.6\t1\t540 238\n",
      "44\t774\t0.6\t1\t444 330\n",
      "45\t1190\t0.6\t1\t720 470\n",
      "46\t3836\t0.6\t1\t1020 2816\n",
      "47\t12610\t0.6\t1\t7134 5476\n",
      "48\t15807\t0.6\t1\t9855 5952\n",
      "49\t65712\t0.6\t1\t35261 30451\n",
      "50\t2349\t0.6\t1\t1606 743\n",
      "51\t468654\t0.6\t1\t300315 168339\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900802.fastq\n",
      "=============================================\n",
      "42329034 sequences processed in total\n",
      "Sequences removed because they became shorter than the length cutoff of 20 bp:\t6030653 (14.2%)\n",
      "\n",
      "Now renaming the output file SRR6900802_trimmed.fq\n",
      "\n",
      "ORIGINAL FILE 1: >>SRR6900802_trimmed.fq<<\tRENAMING TO:>>Macaque_Kidney_R2_trimmed.fq<<\n",
      "\n",
      "  >>> Now running FastQC on the data <<<\n",
      "\n",
      "Started analysis of Macaque_Kidney_R2_trimmed.fq\n",
      "Approx 5% complete for Macaque_Kidney_R2_trimmed.fq\n",
      "Approx 10% complete for Macaque_Kidney_R2_trimmed.fq\n",
      "Approx 15% complete for Macaque_Kidney_R2_trimmed.fq\n",
      "Approx 20% complete for Macaque_Kidney_R2_trimmed.fq\n",
      "Approx 25% complete for Macaque_Kidney_R2_trimmed.fq\n",
      "Approx 30% complete for Macaque_Kidney_R2_trimmed.fq\n",
      "Approx 35% complete for Macaque_Kidney_R2_trimmed.fq\n",
      "Approx 40% complete for Macaque_Kidney_R2_trimmed.fq\n",
      "Approx 45% complete for Macaque_Kidney_R2_trimmed.fq\n",
      "Approx 50% complete for Macaque_Kidney_R2_trimmed.fq\n",
      "Approx 55% complete for Macaque_Kidney_R2_trimmed.fq\n",
      "Approx 60% complete for Macaque_Kidney_R2_trimmed.fq\n",
      "Approx 65% complete for Macaque_Kidney_R2_trimmed.fq\n",
      "Approx 70% complete for Macaque_Kidney_R2_trimmed.fq\n",
      "Approx 75% complete for Macaque_Kidney_R2_trimmed.fq\n",
      "Approx 80% complete for Macaque_Kidney_R2_trimmed.fq\n",
      "Approx 85% complete for Macaque_Kidney_R2_trimmed.fq\n",
      "Approx 90% complete for Macaque_Kidney_R2_trimmed.fq\n",
      "Approx 95% complete for Macaque_Kidney_R2_trimmed.fq\n",
      "Analysis complete for Macaque_Kidney_R2_trimmed.fq\n",
      "Using an excessive number of cores has a diminishing return! It is recommended not to exceed 8 cores per trimming process (you asked for 8 cores). Please consider re-specifying\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "Cutadapt version: 1.18\n",
      "Could not detect version of Python used by Cutadapt from the first line of Cutadapt (but found this: >>>#!/bin/sh<<<)\n",
      "Letting the (modified) Cutadapt deal with the Python version instead\n",
      "pigz 2.6\n",
      "Parallel gzip (pigz) detected. Proceeding with multicore (de)compression using 8 cores\n",
      "\n",
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Output will be written into the directory: /gpfs52/data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/trimmed_reads/\n",
      "Using user-specified basename (>>Macaque_Liver_R2<<) instead of deriving the filename from the input file(s)\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900803.fastq <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Illumina\t704\tAGATCGGAAGAGC\t1000000\t0.07\n",
      "smallRNA\t1\tTGGAATTCTCGG\t1000000\t0.00\n",
      "Nextera\t0\tCTGTCTCTTATA\t1000000\t0.00\n",
      "Using Illumina adapter for trimming (count: 704). Second best hit was smallRNA (count: 1)\n",
      "\n",
      "Writing report to '/gpfs52/data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/trimmed_reads/SRR6900803.fastq_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900803.fastq\n",
      "Trimming mode: single-end\n",
      "Trim Galore version: 0.6.6\n",
      "Cutadapt version: 1.18\n",
      "Python version: could not detect\n",
      "Number of cores used for trimming: 8\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length before a sequence gets removed: 20 bp\n",
      "Running FastQC on the data once trimming has completed\n",
      "Running FastQC with the following extra arguments: '--outdir /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/processing_results/fastQC'\n",
      "\n",
      "Cutadapt seems to be reasonably up-to-date. Setting -j 8\n",
      "Writing final adapter and quality trimmed output to SRR6900803_trimmed.fq\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff '-q 20') and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900803.fastq <<< \n",
      "10000000 sequences processed\n",
      "20000000 sequences processed\n",
      "30000000 sequences processed\n",
      "40000000 sequences processed\n",
      "50000000 sequences processed\n",
      "This is cutadapt 1.18 with Python 3.6.15\n",
      "Command line parameters: -j 8 -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900803.fastq\n",
      "Processing reads on 8 cores in single-end mode ...\n",
      "Finished in 224.56 s (4 us/read; 14.23 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              53,260,438\n",
      "Reads with adapters:                 3,631,541 (6.8%)\n",
      "Reads written (passing filters):    53,260,438 (100.0%)\n",
      "\n",
      "Total basepairs processed: 2,707,551,067 bp\n",
      "Quality-trimmed:             388,011,456 bp (14.3%)\n",
      "Total written (filtered):  2,308,857,125 bp (85.3%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 3631541 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 30.5%\n",
      "  C: 30.4%\n",
      "  G: 19.7%\n",
      "  T: 16.2%\n",
      "  none/other: 3.3%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t2517944\t13315109.5\t0\t2517944\n",
      "2\t731238\t3328777.4\t0\t731238\n",
      "3\t203089\t832194.3\t0\t203089\n",
      "4\t41164\t208048.6\t0\t41164\n",
      "5\t8889\t52012.1\t0\t8889\n",
      "6\t903\t13003.0\t0\t903\n",
      "7\t208\t3250.8\t0\t208\n",
      "8\t120\t812.7\t0\t120\n",
      "9\t257\t203.2\t0\t124 133\n",
      "10\t747\t50.8\t1\t85 662\n",
      "11\t655\t12.7\t1\t92 563\n",
      "12\t608\t3.2\t1\t111 497\n",
      "13\t486\t0.8\t1\t96 390\n",
      "14\t550\t0.8\t1\t514 36\n",
      "15\t572\t0.8\t1\t536 36\n",
      "16\t511\t0.8\t1\t476 35\n",
      "17\t533\t0.8\t1\t498 35\n",
      "18\t591\t0.8\t1\t544 47\n",
      "19\t607\t0.8\t1\t567 40\n",
      "20\t478\t0.8\t1\t435 43\n",
      "21\t500\t0.8\t1\t460 40\n",
      "22\t550\t0.8\t1\t509 41\n",
      "23\t584\t0.8\t1\t546 38\n",
      "24\t627\t0.8\t1\t591 36\n",
      "25\t688\t0.8\t1\t646 42\n",
      "26\t623\t0.8\t1\t583 40\n",
      "27\t1290\t0.8\t1\t1233 57\n",
      "28\t808\t0.8\t1\t767 41\n",
      "29\t811\t0.8\t1\t753 58\n",
      "30\t919\t0.8\t1\t860 59\n",
      "31\t1311\t0.8\t1\t1233 78\n",
      "32\t820\t0.8\t1\t784 36\n",
      "33\t899\t0.8\t1\t843 56\n",
      "34\t908\t0.8\t1\t856 52\n",
      "35\t1043\t0.8\t1\t966 77\n",
      "36\t1171\t0.8\t1\t1034 137\n",
      "37\t1036\t0.8\t1\t941 95\n",
      "38\t942\t0.8\t1\t854 88\n",
      "39\t1063\t0.8\t1\t911 152\n",
      "40\t1237\t0.8\t1\t728 509\n",
      "41\t1885\t0.8\t1\t1315 570\n",
      "42\t1427\t0.8\t1\t1153 274\n",
      "43\t579\t0.8\t1\t427 152\n",
      "44\t471\t0.8\t1\t301 170\n",
      "45\t569\t0.8\t1\t394 175\n",
      "46\t750\t0.8\t1\t441 309\n",
      "47\t1306\t0.8\t1\t833 473\n",
      "48\t2056\t0.8\t1\t1415 641\n",
      "49\t3358\t0.8\t1\t2035 1323\n",
      "50\t870\t0.8\t1\t647 223\n",
      "51\t90290\t0.8\t1\t60872 29418\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900803.fastq\n",
      "=============================================\n",
      "53260438 sequences processed in total\n",
      "Sequences removed because they became shorter than the length cutoff of 20 bp:\t7439356 (14.0%)\n",
      "\n",
      "Now renaming the output file SRR6900803_trimmed.fq\n",
      "\n",
      "ORIGINAL FILE 1: >>SRR6900803_trimmed.fq<<\tRENAMING TO:>>Macaque_Liver_R2_trimmed.fq<<\n",
      "\n",
      "  >>> Now running FastQC on the data <<<\n",
      "\n",
      "Started analysis of Macaque_Liver_R2_trimmed.fq\n",
      "Approx 5% complete for Macaque_Liver_R2_trimmed.fq\n",
      "Approx 10% complete for Macaque_Liver_R2_trimmed.fq\n",
      "Approx 15% complete for Macaque_Liver_R2_trimmed.fq\n",
      "Approx 20% complete for Macaque_Liver_R2_trimmed.fq\n",
      "Approx 25% complete for Macaque_Liver_R2_trimmed.fq\n",
      "Approx 30% complete for Macaque_Liver_R2_trimmed.fq\n",
      "Approx 35% complete for Macaque_Liver_R2_trimmed.fq\n",
      "Approx 40% complete for Macaque_Liver_R2_trimmed.fq\n",
      "Approx 45% complete for Macaque_Liver_R2_trimmed.fq\n",
      "Approx 50% complete for Macaque_Liver_R2_trimmed.fq\n",
      "Approx 55% complete for Macaque_Liver_R2_trimmed.fq\n",
      "Approx 60% complete for Macaque_Liver_R2_trimmed.fq\n",
      "Approx 65% complete for Macaque_Liver_R2_trimmed.fq\n",
      "Approx 70% complete for Macaque_Liver_R2_trimmed.fq\n",
      "Approx 75% complete for Macaque_Liver_R2_trimmed.fq\n",
      "Approx 80% complete for Macaque_Liver_R2_trimmed.fq\n",
      "Approx 85% complete for Macaque_Liver_R2_trimmed.fq\n",
      "Approx 90% complete for Macaque_Liver_R2_trimmed.fq\n",
      "Approx 95% complete for Macaque_Liver_R2_trimmed.fq\n",
      "Analysis complete for Macaque_Liver_R2_trimmed.fq\n",
      "Using an excessive number of cores has a diminishing return! It is recommended not to exceed 8 cores per trimming process (you asked for 8 cores). Please consider re-specifying\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "Cutadapt version: 1.18\n",
      "Could not detect version of Python used by Cutadapt from the first line of Cutadapt (but found this: >>>#!/bin/sh<<<)\n",
      "Letting the (modified) Cutadapt deal with the Python version instead\n",
      "pigz 2.6\n",
      "Parallel gzip (pigz) detected. Proceeding with multicore (de)compression using 8 cores\n",
      "\n",
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Output will be written into the directory: /gpfs52/data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/trimmed_reads/\n",
      "Using user-specified basename (>>Macaque_Lung_R2<<) instead of deriving the filename from the input file(s)\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900804.fastq <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Illumina\t1745\tAGATCGGAAGAGC\t1000000\t0.17\n",
      "Nextera\t5\tCTGTCTCTTATA\t1000000\t0.00\n",
      "smallRNA\t1\tTGGAATTCTCGG\t1000000\t0.00\n",
      "Using Illumina adapter for trimming (count: 1745). Second best hit was Nextera (count: 5)\n",
      "\n",
      "Writing report to '/gpfs52/data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/trimmed_reads/SRR6900804.fastq_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900804.fastq\n",
      "Trimming mode: single-end\n",
      "Trim Galore version: 0.6.6\n",
      "Cutadapt version: 1.18\n",
      "Python version: could not detect\n",
      "Number of cores used for trimming: 8\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length before a sequence gets removed: 20 bp\n",
      "Running FastQC on the data once trimming has completed\n",
      "Running FastQC with the following extra arguments: '--outdir /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/processing_results/fastQC'\n",
      "\n",
      "Cutadapt seems to be reasonably up-to-date. Setting -j 8\n",
      "Writing final adapter and quality trimmed output to SRR6900804_trimmed.fq\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff '-q 20') and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900804.fastq <<< \n",
      "10000000 sequences processed\n",
      "20000000 sequences processed\n",
      "30000000 sequences processed\n",
      "40000000 sequences processed\n",
      "50000000 sequences processed\n",
      "This is cutadapt 1.18 with Python 3.6.15\n",
      "Command line parameters: -j 8 -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900804.fastq\n",
      "Processing reads on 8 cores in single-end mode ...\n",
      "Finished in 224.65 s (4 us/read; 14.25 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              53,365,024\n",
      "Reads with adapters:                 4,938,493 (9.3%)\n",
      "Reads written (passing filters):    53,365,024 (100.0%)\n",
      "\n",
      "Total basepairs processed: 2,709,765,258 bp\n",
      "Quality-trimmed:             368,274,262 bp (13.6%)\n",
      "Total written (filtered):  2,315,175,607 bp (85.4%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 4938493 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 28.9%\n",
      "  C: 28.9%\n",
      "  G: 18.3%\n",
      "  T: 16.2%\n",
      "  none/other: 7.7%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t3229424\t13341256.0\t0\t3229424\n",
      "2\t951333\t3335314.0\t0\t951333\n",
      "3\t266416\t833828.5\t0\t266416\n",
      "4\t54299\t208457.1\t0\t54299\n",
      "5\t11963\t52114.3\t0\t11963\n",
      "6\t1586\t13028.6\t0\t1586\n",
      "7\t478\t3257.1\t0\t478\n",
      "8\t360\t814.3\t0\t360\n",
      "9\t491\t203.6\t0\t294 197\n",
      "10\t1460\t50.9\t1\t260 1200\n",
      "11\t1239\t12.7\t1\t286 953\n",
      "12\t1189\t3.2\t1\t247 942\n",
      "13\t1158\t0.8\t1\t268 890\n",
      "14\t1233\t0.8\t1\t1129 104\n",
      "15\t1159\t0.8\t1\t1046 113\n",
      "16\t1256\t0.8\t1\t1160 96\n",
      "17\t1227\t0.8\t1\t1125 102\n",
      "18\t1165\t0.8\t1\t1056 109\n",
      "19\t1235\t0.8\t1\t1123 112\n",
      "20\t1185\t0.8\t1\t1088 97\n",
      "21\t1305\t0.8\t1\t1200 105\n",
      "22\t1248\t0.8\t1\t1149 99\n",
      "23\t1453\t0.8\t1\t1334 119\n",
      "24\t1557\t0.8\t1\t1454 103\n",
      "25\t1494\t0.8\t1\t1388 106\n",
      "26\t1617\t0.8\t1\t1516 101\n",
      "27\t1589\t0.8\t1\t1477 112\n",
      "28\t1788\t0.8\t1\t1654 134\n",
      "29\t1871\t0.8\t1\t1734 137\n",
      "30\t2212\t0.8\t1\t2082 130\n",
      "31\t2391\t0.8\t1\t2262 129\n",
      "32\t2251\t0.8\t1\t2124 127\n",
      "33\t2385\t0.8\t1\t2257 128\n",
      "34\t2593\t0.8\t1\t2429 164\n",
      "35\t2587\t0.8\t1\t2431 156\n",
      "36\t2574\t0.8\t1\t2393 181\n",
      "37\t2600\t0.8\t1\t2413 187\n",
      "38\t2175\t0.8\t1\t2013 162\n",
      "39\t2265\t0.8\t1\t2099 166\n",
      "40\t1732\t0.8\t1\t1511 221\n",
      "41\t1497\t0.8\t1\t1317 180\n",
      "42\t1904\t0.8\t1\t1626 278\n",
      "43\t1136\t0.8\t1\t871 265\n",
      "44\t1071\t0.8\t1\t817 254\n",
      "45\t936\t0.8\t1\t636 300\n",
      "46\t2019\t0.8\t1\t1150 869\n",
      "47\t4682\t0.8\t1\t2610 2072\n",
      "48\t8394\t0.8\t1\t5627 2767\n",
      "49\t35896\t0.8\t1\t20155 15741\n",
      "50\t2075\t0.8\t1\t1471 604\n",
      "51\t309340\t0.8\t1\t206236 103104\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900804.fastq\n",
      "=============================================\n",
      "53365024 sequences processed in total\n",
      "Sequences removed because they became shorter than the length cutoff of 20 bp:\t7212626 (13.5%)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Now renaming the output file SRR6900804_trimmed.fq\n",
      "\n",
      "ORIGINAL FILE 1: >>SRR6900804_trimmed.fq<<\tRENAMING TO:>>Macaque_Lung_R2_trimmed.fq<<\n",
      "\n",
      "  >>> Now running FastQC on the data <<<\n",
      "\n",
      "Started analysis of Macaque_Lung_R2_trimmed.fq\n",
      "Approx 5% complete for Macaque_Lung_R2_trimmed.fq\n",
      "Approx 10% complete for Macaque_Lung_R2_trimmed.fq\n",
      "Approx 15% complete for Macaque_Lung_R2_trimmed.fq\n",
      "Approx 20% complete for Macaque_Lung_R2_trimmed.fq\n",
      "Approx 25% complete for Macaque_Lung_R2_trimmed.fq\n",
      "Approx 30% complete for Macaque_Lung_R2_trimmed.fq\n",
      "Approx 35% complete for Macaque_Lung_R2_trimmed.fq\n",
      "Approx 40% complete for Macaque_Lung_R2_trimmed.fq\n",
      "Approx 45% complete for Macaque_Lung_R2_trimmed.fq\n",
      "Approx 50% complete for Macaque_Lung_R2_trimmed.fq\n",
      "Approx 55% complete for Macaque_Lung_R2_trimmed.fq\n",
      "Approx 60% complete for Macaque_Lung_R2_trimmed.fq\n",
      "Approx 65% complete for Macaque_Lung_R2_trimmed.fq\n",
      "Approx 70% complete for Macaque_Lung_R2_trimmed.fq\n",
      "Approx 75% complete for Macaque_Lung_R2_trimmed.fq\n",
      "Approx 80% complete for Macaque_Lung_R2_trimmed.fq\n",
      "Approx 85% complete for Macaque_Lung_R2_trimmed.fq\n",
      "Approx 90% complete for Macaque_Lung_R2_trimmed.fq\n",
      "Approx 95% complete for Macaque_Lung_R2_trimmed.fq\n",
      "Analysis complete for Macaque_Lung_R2_trimmed.fq\n",
      "Using an excessive number of cores has a diminishing return! It is recommended not to exceed 8 cores per trimming process (you asked for 8 cores). Please consider re-specifying\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "Cutadapt version: 1.18\n",
      "Could not detect version of Python used by Cutadapt from the first line of Cutadapt (but found this: >>>#!/bin/sh<<<)\n",
      "Letting the (modified) Cutadapt deal with the Python version instead\n",
      "pigz 2.6\n",
      "Parallel gzip (pigz) detected. Proceeding with multicore (de)compression using 8 cores\n",
      "\n",
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Output will be written into the directory: /gpfs52/data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/trimmed_reads/\n",
      "Using user-specified basename (>>Macaque_Heart_R3<<) instead of deriving the filename from the input file(s)\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900805.fastq <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Illumina\t1405\tAGATCGGAAGAGC\t1000000\t0.14\n",
      "smallRNA\t1\tTGGAATTCTCGG\t1000000\t0.00\n",
      "Nextera\t1\tCTGTCTCTTATA\t1000000\t0.00\n",
      "Using Illumina adapter for trimming (count: 1405). Second best hit was smallRNA (count: 1)\n",
      "\n",
      "Writing report to '/gpfs52/data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/trimmed_reads/SRR6900805.fastq_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900805.fastq\n",
      "Trimming mode: single-end\n",
      "Trim Galore version: 0.6.6\n",
      "Cutadapt version: 1.18\n",
      "Python version: could not detect\n",
      "Number of cores used for trimming: 8\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length before a sequence gets removed: 20 bp\n",
      "Running FastQC on the data once trimming has completed\n",
      "Running FastQC with the following extra arguments: '--outdir /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/processing_results/fastQC'\n",
      "\n",
      "Cutadapt seems to be reasonably up-to-date. Setting -j 8\n",
      "Writing final adapter and quality trimmed output to SRR6900805_trimmed.fq\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff '-q 20') and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900805.fastq <<< \n",
      "10000000 sequences processed\n",
      "20000000 sequences processed\n",
      "30000000 sequences processed\n",
      "40000000 sequences processed\n",
      "50000000 sequences processed\n",
      "60000000 sequences processed\n",
      "This is cutadapt 1.18 with Python 3.6.15\n",
      "Command line parameters: -j 8 -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900805.fastq\n",
      "Processing reads on 8 cores in single-end mode ...\n",
      "Finished in 258.27 s (4 us/read; 15.68 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              67,491,954\n",
      "Reads with adapters:                13,861,647 (20.5%)\n",
      "Reads written (passing filters):    67,491,954 (100.0%)\n",
      "\n",
      "Total basepairs processed: 3,413,108,374 bp\n",
      "Quality-trimmed:              38,049,630 bp (1.1%)\n",
      "Total written (filtered):  3,350,662,618 bp (98.2%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 13861647 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 28.4%\n",
      "  C: 28.2%\n",
      "  G: 19.6%\n",
      "  T: 23.0%\n",
      "  none/other: 0.7%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t9793387\t16872988.5\t0\t9793387\n",
      "2\t3006027\t4218247.1\t0\t3006027\n",
      "3\t741585\t1054561.8\t0\t741585\n",
      "4\t159826\t263640.4\t0\t159826\n",
      "5\t35095\t65910.1\t0\t35095\n",
      "6\t4308\t16477.5\t0\t4308\n",
      "7\t968\t4119.4\t0\t968\n",
      "8\t389\t1029.8\t0\t389\n",
      "9\t913\t257.5\t0\t486 427\n",
      "10\t1152\t64.4\t1\t297 855\n",
      "11\t906\t16.1\t1\t316 590\n",
      "12\t1508\t4.0\t1\t254 1254\n",
      "13\t465\t1.0\t1\t302 163\n",
      "14\t410\t1.0\t1\t368 42\n",
      "15\t435\t1.0\t1\t397 38\n",
      "16\t421\t1.0\t1\t384 37\n",
      "17\t630\t1.0\t1\t543 87\n",
      "18\t432\t1.0\t1\t395 37\n",
      "19\t395\t1.0\t1\t341 54\n",
      "20\t426\t1.0\t1\t382 44\n",
      "21\t407\t1.0\t1\t357 50\n",
      "22\t420\t1.0\t1\t366 54\n",
      "23\t545\t1.0\t1\t508 37\n",
      "24\t431\t1.0\t1\t391 40\n",
      "25\t411\t1.0\t1\t371 40\n",
      "26\t383\t1.0\t1\t349 34\n",
      "27\t484\t1.0\t1\t412 72\n",
      "28\t419\t1.0\t1\t368 51\n",
      "29\t631\t1.0\t1\t530 101\n",
      "30\t580\t1.0\t1\t506 74\n",
      "31\t441\t1.0\t1\t396 45\n",
      "32\t451\t1.0\t1\t404 47\n",
      "33\t497\t1.0\t1\t429 68\n",
      "34\t558\t1.0\t1\t483 75\n",
      "35\t615\t1.0\t1\t556 59\n",
      "36\t532\t1.0\t1\t460 72\n",
      "37\t663\t1.0\t1\t592 71\n",
      "38\t314\t1.0\t1\t274 40\n",
      "39\t392\t1.0\t1\t333 59\n",
      "40\t359\t1.0\t1\t321 38\n",
      "41\t444\t1.0\t1\t366 78\n",
      "42\t233\t1.0\t1\t195 38\n",
      "43\t206\t1.0\t1\t165 41\n",
      "44\t167\t1.0\t1\t122 45\n",
      "45\t267\t1.0\t1\t182 85\n",
      "46\t370\t1.0\t1\t180 190\n",
      "47\t1188\t1.0\t1\t639 549\n",
      "48\t1408\t1.0\t1\t1067 341\n",
      "49\t1537\t1.0\t1\t826 711\n",
      "50\t82983\t1.0\t1\t43554 39429\n",
      "51\t13633\t1.0\t1\t7423 6210\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900805.fastq\n",
      "=============================================\n",
      "67491954 sequences processed in total\n",
      "Sequences removed because they became shorter than the length cutoff of 20 bp:\t128021 (0.2%)\n",
      "\n",
      "Now renaming the output file SRR6900805_trimmed.fq\n",
      "\n",
      "ORIGINAL FILE 1: >>SRR6900805_trimmed.fq<<\tRENAMING TO:>>Macaque_Heart_R3_trimmed.fq<<\n",
      "\n",
      "  >>> Now running FastQC on the data <<<\n",
      "\n",
      "Started analysis of Macaque_Heart_R3_trimmed.fq\n",
      "Approx 5% complete for Macaque_Heart_R3_trimmed.fq\n",
      "Approx 10% complete for Macaque_Heart_R3_trimmed.fq\n",
      "Approx 15% complete for Macaque_Heart_R3_trimmed.fq\n",
      "Approx 20% complete for Macaque_Heart_R3_trimmed.fq\n",
      "Approx 25% complete for Macaque_Heart_R3_trimmed.fq\n",
      "Approx 30% complete for Macaque_Heart_R3_trimmed.fq\n",
      "Approx 35% complete for Macaque_Heart_R3_trimmed.fq\n",
      "Approx 40% complete for Macaque_Heart_R3_trimmed.fq\n",
      "Approx 45% complete for Macaque_Heart_R3_trimmed.fq\n",
      "Approx 50% complete for Macaque_Heart_R3_trimmed.fq\n",
      "Approx 55% complete for Macaque_Heart_R3_trimmed.fq\n",
      "Approx 60% complete for Macaque_Heart_R3_trimmed.fq\n",
      "Approx 65% complete for Macaque_Heart_R3_trimmed.fq\n",
      "Approx 70% complete for Macaque_Heart_R3_trimmed.fq\n",
      "Approx 75% complete for Macaque_Heart_R3_trimmed.fq\n",
      "Approx 80% complete for Macaque_Heart_R3_trimmed.fq\n",
      "Approx 85% complete for Macaque_Heart_R3_trimmed.fq\n",
      "Approx 90% complete for Macaque_Heart_R3_trimmed.fq\n",
      "Approx 95% complete for Macaque_Heart_R3_trimmed.fq\n",
      "Analysis complete for Macaque_Heart_R3_trimmed.fq\n",
      "Using an excessive number of cores has a diminishing return! It is recommended not to exceed 8 cores per trimming process (you asked for 8 cores). Please consider re-specifying\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "Cutadapt version: 1.18\n",
      "Could not detect version of Python used by Cutadapt from the first line of Cutadapt (but found this: >>>#!/bin/sh<<<)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Letting the (modified) Cutadapt deal with the Python version instead\n",
      "pigz 2.6\n",
      "Parallel gzip (pigz) detected. Proceeding with multicore (de)compression using 8 cores\n",
      "\n",
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Output will be written into the directory: /gpfs52/data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/trimmed_reads/\n",
      "Using user-specified basename (>>Macaque_Kidney_R3<<) instead of deriving the filename from the input file(s)\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900806.fastq <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Illumina\t513\tAGATCGGAAGAGC\t1000000\t0.05\n",
      "smallRNA\t1\tTGGAATTCTCGG\t1000000\t0.00\n",
      "Nextera\t1\tCTGTCTCTTATA\t1000000\t0.00\n",
      "Using Illumina adapter for trimming (count: 513). Second best hit was smallRNA (count: 1)\n",
      "\n",
      "Writing report to '/gpfs52/data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/trimmed_reads/SRR6900806.fastq_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900806.fastq\n",
      "Trimming mode: single-end\n",
      "Trim Galore version: 0.6.6\n",
      "Cutadapt version: 1.18\n",
      "Python version: could not detect\n",
      "Number of cores used for trimming: 8\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length before a sequence gets removed: 20 bp\n",
      "Running FastQC on the data once trimming has completed\n",
      "Running FastQC with the following extra arguments: '--outdir /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/processing_results/fastQC'\n",
      "\n",
      "Cutadapt seems to be reasonably up-to-date. Setting -j 8\n",
      "Writing final adapter and quality trimmed output to SRR6900806_trimmed.fq\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff '-q 20') and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900806.fastq <<< \n",
      "10000000 sequences processed\n",
      "20000000 sequences processed\n",
      "30000000 sequences processed\n",
      "40000000 sequences processed\n",
      "This is cutadapt 1.18 with Python 3.6.15\n",
      "Command line parameters: -j 8 -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900806.fastq\n",
      "Processing reads on 8 cores in single-end mode ...\n",
      "Finished in 232.58 s (5 us/read; 11.95 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              46,319,360\n",
      "Reads with adapters:                15,719,176 (33.9%)\n",
      "Reads written (passing filters):    46,319,360 (100.0%)\n",
      "\n",
      "Total basepairs processed: 2,315,968,000 bp\n",
      "Quality-trimmed:              79,604,650 bp (3.4%)\n",
      "Total written (filtered):  2,212,779,406 bp (95.5%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 15719176 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 32.1%\n",
      "  C: 29.8%\n",
      "  G: 18.0%\n",
      "  T: 19.9%\n",
      "  none/other: 0.2%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t11211537\t11579840.0\t0\t11211537\n",
      "2\t3350305\t2894960.0\t0\t3350305\n",
      "3\t863964\t723740.0\t0\t863964\n",
      "4\t200247\t180935.0\t0\t200247\n",
      "5\t38738\t45233.8\t0\t38738\n",
      "6\t4691\t11308.4\t0\t4691\n",
      "7\t1090\t2827.1\t0\t1090\n",
      "8\t407\t706.8\t0\t407\n",
      "9\t1296\t176.7\t0\t661 635\n",
      "10\t1220\t44.2\t1\t237 983\n",
      "11\t719\t11.0\t1\t224 495\n",
      "12\t505\t2.8\t1\t239 266\n",
      "13\t205\t0.7\t1\t173 32\n",
      "14\t254\t0.7\t1\t224 30\n",
      "15\t271\t0.7\t1\t227 44\n",
      "16\t288\t0.7\t1\t253 35\n",
      "17\t263\t0.7\t1\t228 35\n",
      "18\t320\t0.7\t1\t259 61\n",
      "19\t165\t0.7\t1\t135 30\n",
      "20\t210\t0.7\t1\t168 42\n",
      "21\t257\t0.7\t1\t197 60\n",
      "22\t229\t0.7\t1\t196 33\n",
      "23\t182\t0.7\t1\t149 33\n",
      "24\t257\t0.7\t1\t219 38\n",
      "25\t226\t0.7\t1\t183 43\n",
      "26\t214\t0.7\t1\t169 45\n",
      "27\t270\t0.7\t1\t202 68\n",
      "28\t305\t0.7\t1\t248 57\n",
      "29\t233\t0.7\t1\t188 45\n",
      "30\t244\t0.7\t1\t206 38\n",
      "31\t192\t0.7\t1\t153 39\n",
      "32\t208\t0.7\t1\t159 49\n",
      "33\t220\t0.7\t1\t186 34\n",
      "34\t229\t0.7\t1\t176 53\n",
      "35\t186\t0.7\t1\t149 37\n",
      "36\t263\t0.7\t1\t206 57\n",
      "37\t294\t0.7\t1\t213 81\n",
      "38\t187\t0.7\t1\t157 30\n",
      "39\t193\t0.7\t1\t147 46\n",
      "40\t262\t0.7\t1\t180 82\n",
      "41\t173\t0.7\t1\t100 73\n",
      "42\t148\t0.7\t1\t76 72\n",
      "43\t196\t0.7\t1\t94 102\n",
      "44\t309\t0.7\t1\t192 117\n",
      "45\t368\t0.7\t1\t147 221\n",
      "46\t1294\t0.7\t1\t348 946\n",
      "47\t2965\t0.7\t1\t1492 1473\n",
      "48\t4208\t0.7\t1\t2143 2065\n",
      "49\t26837\t0.7\t1\t12351 14486\n",
      "50\t1332\t0.7\t1\t481 851\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900806.fastq\n",
      "=============================================\n",
      "46319360 sequences processed in total\n",
      "Sequences removed because they became shorter than the length cutoff of 20 bp:\t97765 (0.2%)\n",
      "\n",
      "Now renaming the output file SRR6900806_trimmed.fq\n",
      "\n",
      "ORIGINAL FILE 1: >>SRR6900806_trimmed.fq<<\tRENAMING TO:>>Macaque_Kidney_R3_trimmed.fq<<\n",
      "\n",
      "  >>> Now running FastQC on the data <<<\n",
      "\n",
      "Started analysis of Macaque_Kidney_R3_trimmed.fq\n",
      "Approx 5% complete for Macaque_Kidney_R3_trimmed.fq\n",
      "Approx 10% complete for Macaque_Kidney_R3_trimmed.fq\n",
      "Approx 15% complete for Macaque_Kidney_R3_trimmed.fq\n",
      "Approx 20% complete for Macaque_Kidney_R3_trimmed.fq\n",
      "Approx 25% complete for Macaque_Kidney_R3_trimmed.fq\n",
      "Approx 30% complete for Macaque_Kidney_R3_trimmed.fq\n",
      "Approx 35% complete for Macaque_Kidney_R3_trimmed.fq\n",
      "Approx 40% complete for Macaque_Kidney_R3_trimmed.fq\n",
      "Approx 45% complete for Macaque_Kidney_R3_trimmed.fq\n",
      "Approx 50% complete for Macaque_Kidney_R3_trimmed.fq\n",
      "Approx 55% complete for Macaque_Kidney_R3_trimmed.fq\n",
      "Approx 60% complete for Macaque_Kidney_R3_trimmed.fq\n",
      "Approx 65% complete for Macaque_Kidney_R3_trimmed.fq\n",
      "Approx 70% complete for Macaque_Kidney_R3_trimmed.fq\n",
      "Approx 75% complete for Macaque_Kidney_R3_trimmed.fq\n",
      "Approx 80% complete for Macaque_Kidney_R3_trimmed.fq\n",
      "Approx 85% complete for Macaque_Kidney_R3_trimmed.fq\n",
      "Approx 90% complete for Macaque_Kidney_R3_trimmed.fq\n",
      "Approx 95% complete for Macaque_Kidney_R3_trimmed.fq\n",
      "Analysis complete for Macaque_Kidney_R3_trimmed.fq\n",
      "Using an excessive number of cores has a diminishing return! It is recommended not to exceed 8 cores per trimming process (you asked for 8 cores). Please consider re-specifying\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "Cutadapt version: 1.18\n",
      "Could not detect version of Python used by Cutadapt from the first line of Cutadapt (but found this: >>>#!/bin/sh<<<)\n",
      "Letting the (modified) Cutadapt deal with the Python version instead\n",
      "pigz 2.6\n",
      "Parallel gzip (pigz) detected. Proceeding with multicore (de)compression using 8 cores\n",
      "\n",
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Output will be written into the directory: /gpfs52/data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/trimmed_reads/\n",
      "Using user-specified basename (>>Macaque_Liver_R3<<) instead of deriving the filename from the input file(s)\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900807.fastq <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Illumina\t4082\tAGATCGGAAGAGC\t1000000\t0.41\n",
      "smallRNA\t1\tTGGAATTCTCGG\t1000000\t0.00\n",
      "Nextera\t1\tCTGTCTCTTATA\t1000000\t0.00\n",
      "Using Illumina adapter for trimming (count: 4082). Second best hit was smallRNA (count: 1)\n",
      "\n",
      "Writing report to '/gpfs52/data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/trimmed_reads/SRR6900807.fastq_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900807.fastq\n",
      "Trimming mode: single-end\n",
      "Trim Galore version: 0.6.6\n",
      "Cutadapt version: 1.18\n",
      "Python version: could not detect\n",
      "Number of cores used for trimming: 8\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length before a sequence gets removed: 20 bp\n",
      "Running FastQC on the data once trimming has completed\n",
      "Running FastQC with the following extra arguments: '--outdir /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/processing_results/fastQC'\n",
      "\n",
      "Cutadapt seems to be reasonably up-to-date. Setting -j 8\n",
      "Writing final adapter and quality trimmed output to SRR6900807_trimmed.fq\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff '-q 20') and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900807.fastq <<< \n",
      "10000000 sequences processed\n",
      "20000000 sequences processed\n",
      "30000000 sequences processed\n",
      "40000000 sequences processed\n",
      "50000000 sequences processed\n",
      "This is cutadapt 1.18 with Python 3.6.15\n",
      "Command line parameters: -j 8 -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900807.fastq\n",
      "Processing reads on 8 cores in single-end mode ...\n",
      "Finished in 274.70 s (5 us/read; 12.24 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              56,022,681\n",
      "Reads with adapters:                11,946,283 (21.3%)\n",
      "Reads written (passing filters):    56,022,681 (100.0%)\n",
      "\n",
      "Total basepairs processed: 2,833,951,179 bp\n",
      "Quality-trimmed:              34,048,833 bp (1.2%)\n",
      "Total written (filtered):  2,767,895,172 bp (97.7%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 11946283 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 30.6%\n",
      "  C: 31.0%\n",
      "  G: 19.4%\n",
      "  T: 16.4%\n",
      "  none/other: 2.6%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t8234267\t14005670.2\t0\t8234267\n",
      "2\t2512867\t3501417.6\t0\t2512867\n",
      "3\t696253\t875354.4\t0\t696253\n",
      "4\t142780\t218838.6\t0\t142780\n",
      "5\t29671\t54709.6\t0\t29671\n",
      "6\t2689\t13677.4\t0\t2689\n",
      "7\t749\t3419.4\t0\t749\n",
      "8\t465\t854.8\t0\t465\n",
      "9\t918\t213.7\t0\t402 516\n",
      "10\t1458\t53.4\t1\t331 1127\n",
      "11\t1195\t13.4\t1\t235 960\n",
      "12\t754\t3.3\t1\t250 504\n",
      "13\t416\t0.8\t1\t261 155\n",
      "14\t416\t0.8\t1\t383 33\n",
      "15\t395\t0.8\t1\t359 36\n",
      "16\t362\t0.8\t1\t325 37\n",
      "17\t449\t0.8\t1\t403 46\n",
      "18\t443\t0.8\t1\t412 31\n",
      "19\t391\t0.8\t1\t357 34\n",
      "20\t349\t0.8\t1\t331 18\n",
      "21\t388\t0.8\t1\t347 41\n",
      "22\t347\t0.8\t1\t316 31\n",
      "23\t368\t0.8\t1\t327 41\n",
      "24\t348\t0.8\t1\t306 42\n",
      "25\t328\t0.8\t1\t303 25\n",
      "26\t593\t0.8\t1\t542 51\n",
      "27\t394\t0.8\t1\t352 42\n",
      "28\t370\t0.8\t1\t331 39\n",
      "29\t362\t0.8\t1\t336 26\n",
      "30\t386\t0.8\t1\t349 37\n",
      "31\t365\t0.8\t1\t325 40\n",
      "32\t381\t0.8\t1\t343 38\n",
      "33\t325\t0.8\t1\t296 29\n",
      "34\t398\t0.8\t1\t360 38\n",
      "35\t443\t0.8\t1\t403 40\n",
      "36\t412\t0.8\t1\t364 48\n",
      "37\t343\t0.8\t1\t303 40\n",
      "38\t407\t0.8\t1\t364 43\n",
      "39\t287\t0.8\t1\t254 33\n",
      "40\t277\t0.8\t1\t210 67\n",
      "41\t395\t0.8\t1\t309 86\n",
      "42\t276\t0.8\t1\t220 56\n",
      "43\t201\t0.8\t1\t140 61\n",
      "44\t324\t0.8\t1\t261 63\n",
      "45\t307\t0.8\t1\t201 106\n",
      "46\t560\t0.8\t1\t269 291\n",
      "47\t1399\t0.8\t1\t896 503\n",
      "48\t2014\t0.8\t1\t1431 583\n",
      "49\t3787\t0.8\t1\t2190 1597\n",
      "50\t262991\t0.8\t1\t141844 121147\n",
      "51\t39220\t0.8\t1\t22216 17004\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900807.fastq\n",
      "=============================================\n",
      "56022681 sequences processed in total\n",
      "Sequences removed because they became shorter than the length cutoff of 20 bp:\t365944 (0.7%)\n",
      "\n",
      "Now renaming the output file SRR6900807_trimmed.fq\n",
      "\n",
      "ORIGINAL FILE 1: >>SRR6900807_trimmed.fq<<\tRENAMING TO:>>Macaque_Liver_R3_trimmed.fq<<\n",
      "\n",
      "  >>> Now running FastQC on the data <<<\n",
      "\n",
      "Started analysis of Macaque_Liver_R3_trimmed.fq\n",
      "Approx 5% complete for Macaque_Liver_R3_trimmed.fq\n",
      "Approx 10% complete for Macaque_Liver_R3_trimmed.fq\n",
      "Approx 15% complete for Macaque_Liver_R3_trimmed.fq\n",
      "Approx 20% complete for Macaque_Liver_R3_trimmed.fq\n",
      "Approx 25% complete for Macaque_Liver_R3_trimmed.fq\n",
      "Approx 30% complete for Macaque_Liver_R3_trimmed.fq\n",
      "Approx 35% complete for Macaque_Liver_R3_trimmed.fq\n",
      "Approx 40% complete for Macaque_Liver_R3_trimmed.fq\n",
      "Approx 45% complete for Macaque_Liver_R3_trimmed.fq\n",
      "Approx 50% complete for Macaque_Liver_R3_trimmed.fq\n",
      "Approx 55% complete for Macaque_Liver_R3_trimmed.fq\n",
      "Approx 60% complete for Macaque_Liver_R3_trimmed.fq\n",
      "Approx 65% complete for Macaque_Liver_R3_trimmed.fq\n",
      "Approx 70% complete for Macaque_Liver_R3_trimmed.fq\n",
      "Approx 75% complete for Macaque_Liver_R3_trimmed.fq\n",
      "Approx 80% complete for Macaque_Liver_R3_trimmed.fq\n",
      "Approx 85% complete for Macaque_Liver_R3_trimmed.fq\n",
      "Approx 90% complete for Macaque_Liver_R3_trimmed.fq\n",
      "Approx 95% complete for Macaque_Liver_R3_trimmed.fq\n",
      "Analysis complete for Macaque_Liver_R3_trimmed.fq\n",
      "Using an excessive number of cores has a diminishing return! It is recommended not to exceed 8 cores per trimming process (you asked for 8 cores). Please consider re-specifying\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "Cutadapt version: 1.18\n",
      "Could not detect version of Python used by Cutadapt from the first line of Cutadapt (but found this: >>>#!/bin/sh<<<)\n",
      "Letting the (modified) Cutadapt deal with the Python version instead\n",
      "pigz 2.6\n",
      "Parallel gzip (pigz) detected. Proceeding with multicore (de)compression using 8 cores\n",
      "\n",
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Output will be written into the directory: /gpfs52/data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/trimmed_reads/\n",
      "Using user-specified basename (>>Macaque_Lung_R3<<) instead of deriving the filename from the input file(s)\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900808.fastq <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Illumina\t5409\tAGATCGGAAGAGC\t1000000\t0.54\n",
      "smallRNA\t2\tTGGAATTCTCGG\t1000000\t0.00\n",
      "Nextera\t2\tCTGTCTCTTATA\t1000000\t0.00\n",
      "Using Illumina adapter for trimming (count: 5409). Second best hit was smallRNA (count: 2)\n",
      "\n",
      "Writing report to '/gpfs52/data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/trimmed_reads/SRR6900808.fastq_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900808.fastq\n",
      "Trimming mode: single-end\n",
      "Trim Galore version: 0.6.6\n",
      "Cutadapt version: 1.18\n",
      "Python version: could not detect\n",
      "Number of cores used for trimming: 8\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length before a sequence gets removed: 20 bp\n",
      "Running FastQC on the data once trimming has completed\n",
      "Running FastQC with the following extra arguments: '--outdir /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/processing_results/fastQC'\n",
      "\n",
      "Cutadapt seems to be reasonably up-to-date. Setting -j 8\n",
      "Writing final adapter and quality trimmed output to SRR6900808_trimmed.fq\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff '-q 20') and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900808.fastq <<< \n",
      "10000000 sequences processed\n",
      "20000000 sequences processed\n",
      "30000000 sequences processed\n",
      "40000000 sequences processed\n",
      "50000000 sequences processed\n",
      "60000000 sequences processed\n",
      "This is cutadapt 1.18 with Python 3.6.15\n",
      "Command line parameters: -j 8 -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900808.fastq\n",
      "Processing reads on 8 cores in single-end mode ...\n",
      "Finished in 299.37 s (5 us/read; 12.44 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              62,067,186\n",
      "Reads with adapters:                13,095,060 (21.1%)\n",
      "Reads written (passing filters):    62,067,186 (100.0%)\n",
      "\n",
      "Total basepairs processed: 3,140,209,860 bp\n",
      "Quality-trimmed:              39,015,240 bp (1.2%)\n",
      "Total written (filtered):  3,062,849,041 bp (97.5%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 13095060 times.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 29.4%\n",
      "  C: 32.0%\n",
      "  G: 19.7%\n",
      "  T: 15.9%\n",
      "  none/other: 3.1%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t8864107\t15516796.5\t0\t8864107\n",
      "2\t2849076\t3879199.1\t0\t2849076\n",
      "3\t757534\t969799.8\t0\t757534\n",
      "4\t157472\t242449.9\t0\t157472\n",
      "5\t34338\t60612.5\t0\t34338\n",
      "6\t4084\t15153.1\t0\t4084\n",
      "7\t1137\t3788.3\t0\t1137\n",
      "8\t511\t947.1\t0\t511\n",
      "9\t1118\t236.8\t0\t503 615\n",
      "10\t1620\t59.2\t1\t479 1141\n",
      "11\t999\t14.8\t1\t416 583\n",
      "12\t843\t3.7\t1\t463 380\n",
      "13\t688\t0.9\t1\t422 266\n",
      "14\t696\t0.9\t1\t630 66\n",
      "15\t689\t0.9\t1\t620 69\n",
      "16\t695\t0.9\t1\t626 69\n",
      "17\t704\t0.9\t1\t635 69\n",
      "18\t663\t0.9\t1\t608 55\n",
      "19\t624\t0.9\t1\t539 85\n",
      "20\t631\t0.9\t1\t580 51\n",
      "21\t647\t0.9\t1\t554 93\n",
      "22\t679\t0.9\t1\t601 78\n",
      "23\t694\t0.9\t1\t636 58\n",
      "24\t724\t0.9\t1\t654 70\n",
      "25\t745\t0.9\t1\t662 83\n",
      "26\t730\t0.9\t1\t645 85\n",
      "27\t809\t0.9\t1\t719 90\n",
      "28\t800\t0.9\t1\t720 80\n",
      "29\t788\t0.9\t1\t702 86\n",
      "30\t868\t0.9\t1\t785 83\n",
      "31\t802\t0.9\t1\t738 64\n",
      "32\t861\t0.9\t1\t784 77\n",
      "33\t862\t0.9\t1\t776 86\n",
      "34\t912\t0.9\t1\t831 81\n",
      "35\t923\t0.9\t1\t820 103\n",
      "36\t936\t0.9\t1\t852 84\n",
      "37\t762\t0.9\t1\t669 93\n",
      "38\t794\t0.9\t1\t683 111\n",
      "39\t673\t0.9\t1\t572 101\n",
      "40\t582\t0.9\t1\t507 75\n",
      "41\t639\t0.9\t1\t489 150\n",
      "42\t498\t0.9\t1\t333 165\n",
      "43\t472\t0.9\t1\t307 165\n",
      "44\t480\t0.9\t1\t307 173\n",
      "45\t742\t0.9\t1\t446 296\n",
      "46\t1429\t0.9\t1\t679 750\n",
      "47\t4753\t0.9\t1\t2554 2199\n",
      "48\t6294\t0.9\t1\t5012 1282\n",
      "49\t6076\t0.9\t1\t3393 2683\n",
      "50\t325253\t0.9\t1\t186347 138906\n",
      "51\t54604\t0.9\t1\t32439 22165\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900808.fastq\n",
      "=============================================\n",
      "62067186 sequences processed in total\n",
      "Sequences removed because they became shorter than the length cutoff of 20 bp:\t450795 (0.7%)\n",
      "\n",
      "Now renaming the output file SRR6900808_trimmed.fq\n",
      "\n",
      "ORIGINAL FILE 1: >>SRR6900808_trimmed.fq<<\tRENAMING TO:>>Macaque_Lung_R3_trimmed.fq<<\n",
      "\n",
      "  >>> Now running FastQC on the data <<<\n",
      "\n",
      "Started analysis of Macaque_Lung_R3_trimmed.fq\n",
      "Approx 5% complete for Macaque_Lung_R3_trimmed.fq\n",
      "Approx 10% complete for Macaque_Lung_R3_trimmed.fq\n",
      "Approx 15% complete for Macaque_Lung_R3_trimmed.fq\n",
      "Approx 20% complete for Macaque_Lung_R3_trimmed.fq\n",
      "Approx 25% complete for Macaque_Lung_R3_trimmed.fq\n",
      "Approx 30% complete for Macaque_Lung_R3_trimmed.fq\n",
      "Approx 35% complete for Macaque_Lung_R3_trimmed.fq\n",
      "Approx 40% complete for Macaque_Lung_R3_trimmed.fq\n",
      "Approx 45% complete for Macaque_Lung_R3_trimmed.fq\n",
      "Approx 50% complete for Macaque_Lung_R3_trimmed.fq\n",
      "Approx 55% complete for Macaque_Lung_R3_trimmed.fq\n",
      "Approx 60% complete for Macaque_Lung_R3_trimmed.fq\n",
      "Approx 65% complete for Macaque_Lung_R3_trimmed.fq\n",
      "Approx 70% complete for Macaque_Lung_R3_trimmed.fq\n",
      "Approx 75% complete for Macaque_Lung_R3_trimmed.fq\n",
      "Approx 80% complete for Macaque_Lung_R3_trimmed.fq\n",
      "Approx 85% complete for Macaque_Lung_R3_trimmed.fq\n",
      "Approx 90% complete for Macaque_Lung_R3_trimmed.fq\n",
      "Approx 95% complete for Macaque_Lung_R3_trimmed.fq\n",
      "Analysis complete for Macaque_Lung_R3_trimmed.fq\n",
      "Using an excessive number of cores has a diminishing return! It is recommended not to exceed 8 cores per trimming process (you asked for 8 cores). Please consider re-specifying\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "Cutadapt version: 1.18\n",
      "Could not detect version of Python used by Cutadapt from the first line of Cutadapt (but found this: >>>#!/bin/sh<<<)\n",
      "Letting the (modified) Cutadapt deal with the Python version instead\n",
      "pigz 2.6\n",
      "Parallel gzip (pigz) detected. Proceeding with multicore (de)compression using 8 cores\n",
      "\n",
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Output will be written into the directory: /gpfs52/data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/trimmed_reads/\n",
      "Using user-specified basename (>>Macaque_Heart_R4<<) instead of deriving the filename from the input file(s)\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900809.fastq <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Illumina\t2197\tAGATCGGAAGAGC\t1000000\t0.22\n",
      "smallRNA\t0\tTGGAATTCTCGG\t1000000\t0.00\n",
      "Nextera\t0\tCTGTCTCTTATA\t1000000\t0.00\n",
      "Using Illumina adapter for trimming (count: 2197). Second best hit was smallRNA (count: 0)\n",
      "\n",
      "Writing report to '/gpfs52/data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/trimmed_reads/SRR6900809.fastq_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900809.fastq\n",
      "Trimming mode: single-end\n",
      "Trim Galore version: 0.6.6\n",
      "Cutadapt version: 1.18\n",
      "Python version: could not detect\n",
      "Number of cores used for trimming: 8\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length before a sequence gets removed: 20 bp\n",
      "Running FastQC on the data once trimming has completed\n",
      "Running FastQC with the following extra arguments: '--outdir /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/processing_results/fastQC'\n",
      "\n",
      "Cutadapt seems to be reasonably up-to-date. Setting -j 8\n",
      "Writing final adapter and quality trimmed output to SRR6900809_trimmed.fq\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff '-q 20') and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900809.fastq <<< \n",
      "10000000 sequences processed\n",
      "20000000 sequences processed\n",
      "30000000 sequences processed\n",
      "40000000 sequences processed\n",
      "This is cutadapt 1.18 with Python 3.6.15\n",
      "Command line parameters: -j 8 -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900809.fastq\n",
      "Processing reads on 8 cores in single-end mode ...\n",
      "Finished in 192.25 s (4 us/read; 15.12 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              48,437,093\n",
      "Reads with adapters:                 9,077,351 (18.7%)\n",
      "Reads written (passing filters):    48,437,093 (100.0%)\n",
      "\n",
      "Total basepairs processed: 2,446,762,479 bp\n",
      "Quality-trimmed:              26,075,383 bp (1.1%)\n",
      "Total written (filtered):  2,402,873,235 bp (98.2%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 9077351 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 28.6%\n",
      "  C: 29.1%\n",
      "  G: 19.5%\n",
      "  T: 21.7%\n",
      "  none/other: 1.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t6379831\t12109273.2\t0\t6379831\n",
      "2\t1951899\t3027318.3\t0\t1951899\n",
      "3\t488671\t756829.6\t0\t488671\n",
      "4\t106009\t189207.4\t0\t106009\n",
      "5\t23337\t47301.8\t0\t23337\n",
      "6\t2913\t11825.5\t0\t2913\n",
      "7\t802\t2956.4\t0\t802\n",
      "8\t354\t739.1\t0\t354\n",
      "9\t686\t184.8\t0\t401 285\n",
      "10\t1046\t46.2\t1\t323 723\n",
      "11\t892\t11.5\t1\t342 550\n",
      "12\t1159\t2.9\t1\t342 817\n",
      "13\t662\t0.7\t1\t359 303\n",
      "14\t585\t0.7\t1\t525 60\n",
      "15\t626\t0.7\t1\t572 54\n",
      "16\t549\t0.7\t1\t508 41\n",
      "17\t882\t0.7\t1\t756 126\n",
      "18\t622\t0.7\t1\t556 66\n",
      "19\t573\t0.7\t1\t503 70\n",
      "20\t574\t0.7\t1\t522 52\n",
      "21\t554\t0.7\t1\t500 54\n",
      "22\t618\t0.7\t1\t561 57\n",
      "23\t901\t0.7\t1\t833 68\n",
      "24\t631\t0.7\t1\t584 47\n",
      "25\t712\t0.7\t1\t657 55\n",
      "26\t684\t0.7\t1\t624 60\n",
      "27\t861\t0.7\t1\t750 111\n",
      "28\t741\t0.7\t1\t676 65\n",
      "29\t1143\t0.7\t1\t1008 135\n",
      "30\t2348\t0.7\t1\t2081 267\n",
      "31\t1072\t0.7\t1\t991 81\n",
      "32\t786\t0.7\t1\t724 62\n",
      "33\t960\t0.7\t1\t884 76\n",
      "34\t1063\t0.7\t1\t962 101\n",
      "35\t999\t0.7\t1\t908 91\n",
      "36\t814\t0.7\t1\t746 68\n",
      "37\t836\t0.7\t1\t732 104\n",
      "38\t799\t0.7\t1\t726 73\n",
      "39\t711\t0.7\t1\t592 119\n",
      "40\t677\t0.7\t1\t594 83\n",
      "41\t756\t0.7\t1\t619 137\n",
      "42\t399\t0.7\t1\t325 74\n",
      "43\t365\t0.7\t1\t315 50\n",
      "44\t255\t0.7\t1\t180 75\n",
      "45\t432\t0.7\t1\t316 116\n",
      "46\t497\t0.7\t1\t305 192\n",
      "47\t1267\t0.7\t1\t868 399\n",
      "48\t1579\t0.7\t1\t1296 283\n",
      "49\t1248\t0.7\t1\t800 448\n",
      "50\t79133\t0.7\t1\t48435 30698\n",
      "51\t10838\t0.7\t1\t7014 3824\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900809.fastq\n",
      "=============================================\n",
      "48437093 sequences processed in total\n",
      "Sequences removed because they became shorter than the length cutoff of 20 bp:\t142705 (0.3%)\n",
      "\n",
      "Now renaming the output file SRR6900809_trimmed.fq\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "ORIGINAL FILE 1: >>SRR6900809_trimmed.fq<<\tRENAMING TO:>>Macaque_Heart_R4_trimmed.fq<<\n",
      "\n",
      "  >>> Now running FastQC on the data <<<\n",
      "\n",
      "Started analysis of Macaque_Heart_R4_trimmed.fq\n",
      "Approx 5% complete for Macaque_Heart_R4_trimmed.fq\n",
      "Approx 10% complete for Macaque_Heart_R4_trimmed.fq\n",
      "Approx 15% complete for Macaque_Heart_R4_trimmed.fq\n",
      "Approx 20% complete for Macaque_Heart_R4_trimmed.fq\n",
      "Approx 25% complete for Macaque_Heart_R4_trimmed.fq\n",
      "Approx 30% complete for Macaque_Heart_R4_trimmed.fq\n",
      "Approx 35% complete for Macaque_Heart_R4_trimmed.fq\n",
      "Approx 40% complete for Macaque_Heart_R4_trimmed.fq\n",
      "Approx 45% complete for Macaque_Heart_R4_trimmed.fq\n",
      "Approx 50% complete for Macaque_Heart_R4_trimmed.fq\n",
      "Approx 55% complete for Macaque_Heart_R4_trimmed.fq\n",
      "Approx 60% complete for Macaque_Heart_R4_trimmed.fq\n",
      "Approx 65% complete for Macaque_Heart_R4_trimmed.fq\n",
      "Approx 70% complete for Macaque_Heart_R4_trimmed.fq\n",
      "Approx 75% complete for Macaque_Heart_R4_trimmed.fq\n",
      "Approx 80% complete for Macaque_Heart_R4_trimmed.fq\n",
      "Approx 85% complete for Macaque_Heart_R4_trimmed.fq\n",
      "Approx 90% complete for Macaque_Heart_R4_trimmed.fq\n",
      "Approx 95% complete for Macaque_Heart_R4_trimmed.fq\n",
      "Analysis complete for Macaque_Heart_R4_trimmed.fq\n",
      "Using an excessive number of cores has a diminishing return! It is recommended not to exceed 8 cores per trimming process (you asked for 8 cores). Please consider re-specifying\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "Cutadapt version: 1.18\n",
      "Could not detect version of Python used by Cutadapt from the first line of Cutadapt (but found this: >>>#!/bin/sh<<<)\n",
      "Letting the (modified) Cutadapt deal with the Python version instead\n",
      "pigz 2.6\n",
      "Parallel gzip (pigz) detected. Proceeding with multicore (de)compression using 8 cores\n",
      "\n",
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Output will be written into the directory: /gpfs52/data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/trimmed_reads/\n",
      "Using user-specified basename (>>Macaque_Kidney_R4<<) instead of deriving the filename from the input file(s)\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900810.fastq <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Illumina\t1593\tAGATCGGAAGAGC\t1000000\t0.16\n",
      "smallRNA\t0\tTGGAATTCTCGG\t1000000\t0.00\n",
      "Nextera\t0\tCTGTCTCTTATA\t1000000\t0.00\n",
      "Using Illumina adapter for trimming (count: 1593). Second best hit was smallRNA (count: 0)\n",
      "\n",
      "Writing report to '/gpfs52/data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/trimmed_reads/SRR6900810.fastq_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900810.fastq\n",
      "Trimming mode: single-end\n",
      "Trim Galore version: 0.6.6\n",
      "Cutadapt version: 1.18\n",
      "Python version: could not detect\n",
      "Number of cores used for trimming: 8\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length before a sequence gets removed: 20 bp\n",
      "Running FastQC on the data once trimming has completed\n",
      "Running FastQC with the following extra arguments: '--outdir /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/processing_results/fastQC'\n",
      "\n",
      "Cutadapt seems to be reasonably up-to-date. Setting -j 8\n",
      "Writing final adapter and quality trimmed output to SRR6900810_trimmed.fq\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff '-q 20') and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900810.fastq <<< \n",
      "10000000 sequences processed\n",
      "20000000 sequences processed\n",
      "30000000 sequences processed\n",
      "40000000 sequences processed\n",
      "50000000 sequences processed\n",
      "This is cutadapt 1.18 with Python 3.6.15\n",
      "Command line parameters: -j 8 -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900810.fastq\n",
      "Processing reads on 8 cores in single-end mode ...\n",
      "Finished in 227.94 s (4 us/read; 14.44 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              54,857,579\n",
      "Reads with adapters:                12,688,661 (23.1%)\n",
      "Reads written (passing filters):    54,857,579 (100.0%)\n",
      "\n",
      "Total basepairs processed: 2,778,861,721 bp\n",
      "Quality-trimmed:              36,572,862 bp (1.3%)\n",
      "Total written (filtered):  2,720,115,346 bp (97.9%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 12688661 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 30.2%\n",
      "  C: 31.0%\n",
      "  G: 19.3%\n",
      "  T: 18.8%\n",
      "  none/other: 0.7%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t8922515\t13714394.8\t0\t8922515\n",
      "2\t2744333\t3428598.7\t0\t2744333\n",
      "3\t725342\t857149.7\t0\t725342\n",
      "4\t155978\t214287.4\t0\t155978\n",
      "5\t32678\t53571.9\t0\t32678\n",
      "6\t3608\t13393.0\t0\t3608\n",
      "7\t848\t3348.2\t0\t848\n",
      "8\t353\t837.1\t0\t353\n",
      "9\t813\t209.3\t0\t334 479\n",
      "10\t1178\t52.3\t1\t216 962\n",
      "11\t727\t13.1\t1\t204 523\n",
      "12\t829\t3.3\t1\t257 572\n",
      "13\t299\t0.8\t1\t183 116\n",
      "14\t355\t0.8\t1\t316 39\n",
      "15\t318\t0.8\t1\t289 29\n",
      "16\t342\t0.8\t1\t300 42\n",
      "17\t375\t0.8\t1\t323 52\n",
      "18\t357\t0.8\t1\t313 44\n",
      "19\t294\t0.8\t1\t255 39\n",
      "20\t322\t0.8\t1\t286 36\n",
      "21\t320\t0.8\t1\t274 46\n",
      "22\t342\t0.8\t1\t305 37\n",
      "23\t375\t0.8\t1\t326 49\n",
      "24\t351\t0.8\t1\t301 50\n",
      "25\t337\t0.8\t1\t302 35\n",
      "26\t341\t0.8\t1\t288 53\n",
      "27\t364\t0.8\t1\t316 48\n",
      "28\t423\t0.8\t1\t371 52\n",
      "29\t392\t0.8\t1\t343 49\n",
      "30\t408\t0.8\t1\t366 42\n",
      "31\t400\t0.8\t1\t353 47\n",
      "32\t423\t0.8\t1\t382 41\n",
      "33\t399\t0.8\t1\t350 49\n",
      "34\t440\t0.8\t1\t391 49\n",
      "35\t498\t0.8\t1\t452 46\n",
      "36\t369\t0.8\t1\t329 40\n",
      "37\t330\t0.8\t1\t294 36\n",
      "38\t433\t0.8\t1\t378 55\n",
      "39\t270\t0.8\t1\t238 32\n",
      "40\t249\t0.8\t1\t217 32\n",
      "41\t338\t0.8\t1\t281 57\n",
      "42\t213\t0.8\t1\t165 48\n",
      "43\t175\t0.8\t1\t131 44\n",
      "44\t262\t0.8\t1\t227 35\n",
      "45\t254\t0.8\t1\t194 60\n",
      "46\t393\t0.8\t1\t259 134\n",
      "47\t1254\t0.8\t1\t873 381\n",
      "48\t1527\t0.8\t1\t1269 258\n",
      "49\t1430\t0.8\t1\t968 462\n",
      "50\t71673\t0.8\t1\t48158 23515\n",
      "51\t11814\t0.8\t1\t8271 3543\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900810.fastq\n",
      "=============================================\n",
      "54857579 sequences processed in total\n",
      "Sequences removed because they became shorter than the length cutoff of 20 bp:\t122057 (0.2%)\n",
      "\n",
      "Now renaming the output file SRR6900810_trimmed.fq\n",
      "\n",
      "ORIGINAL FILE 1: >>SRR6900810_trimmed.fq<<\tRENAMING TO:>>Macaque_Kidney_R4_trimmed.fq<<\n",
      "\n",
      "  >>> Now running FastQC on the data <<<\n",
      "\n",
      "Started analysis of Macaque_Kidney_R4_trimmed.fq\n",
      "Approx 5% complete for Macaque_Kidney_R4_trimmed.fq\n",
      "Approx 10% complete for Macaque_Kidney_R4_trimmed.fq\n",
      "Approx 15% complete for Macaque_Kidney_R4_trimmed.fq\n",
      "Approx 20% complete for Macaque_Kidney_R4_trimmed.fq\n",
      "Approx 25% complete for Macaque_Kidney_R4_trimmed.fq\n",
      "Approx 30% complete for Macaque_Kidney_R4_trimmed.fq\n",
      "Approx 35% complete for Macaque_Kidney_R4_trimmed.fq\n",
      "Approx 40% complete for Macaque_Kidney_R4_trimmed.fq\n",
      "Approx 45% complete for Macaque_Kidney_R4_trimmed.fq\n",
      "Approx 50% complete for Macaque_Kidney_R4_trimmed.fq\n",
      "Approx 55% complete for Macaque_Kidney_R4_trimmed.fq\n",
      "Approx 60% complete for Macaque_Kidney_R4_trimmed.fq\n",
      "Approx 65% complete for Macaque_Kidney_R4_trimmed.fq\n",
      "Approx 70% complete for Macaque_Kidney_R4_trimmed.fq\n",
      "Approx 75% complete for Macaque_Kidney_R4_trimmed.fq\n",
      "Approx 80% complete for Macaque_Kidney_R4_trimmed.fq\n",
      "Approx 85% complete for Macaque_Kidney_R4_trimmed.fq\n",
      "Approx 90% complete for Macaque_Kidney_R4_trimmed.fq\n",
      "Approx 95% complete for Macaque_Kidney_R4_trimmed.fq\n",
      "Analysis complete for Macaque_Kidney_R4_trimmed.fq\n",
      "Using an excessive number of cores has a diminishing return! It is recommended not to exceed 8 cores per trimming process (you asked for 8 cores). Please consider re-specifying\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "Cutadapt version: 1.18\n",
      "Could not detect version of Python used by Cutadapt from the first line of Cutadapt (but found this: >>>#!/bin/sh<<<)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Letting the (modified) Cutadapt deal with the Python version instead\n",
      "pigz 2.6\n",
      "Parallel gzip (pigz) detected. Proceeding with multicore (de)compression using 8 cores\n",
      "\n",
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Output will be written into the directory: /gpfs52/data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/trimmed_reads/\n",
      "Using user-specified basename (>>Macaque_Liver_R4<<) instead of deriving the filename from the input file(s)\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900811.fastq <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Illumina\t3657\tAGATCGGAAGAGC\t1000000\t0.37\n",
      "Nextera\t1\tCTGTCTCTTATA\t1000000\t0.00\n",
      "smallRNA\t0\tTGGAATTCTCGG\t1000000\t0.00\n",
      "Using Illumina adapter for trimming (count: 3657). Second best hit was Nextera (count: 1)\n",
      "\n",
      "Writing report to '/gpfs52/data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/trimmed_reads/SRR6900811.fastq_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900811.fastq\n",
      "Trimming mode: single-end\n",
      "Trim Galore version: 0.6.6\n",
      "Cutadapt version: 1.18\n",
      "Python version: could not detect\n",
      "Number of cores used for trimming: 8\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length before a sequence gets removed: 20 bp\n",
      "Running FastQC on the data once trimming has completed\n",
      "Running FastQC with the following extra arguments: '--outdir /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/processing_results/fastQC'\n",
      "\n",
      "Cutadapt seems to be reasonably up-to-date. Setting -j 8\n",
      "Writing final adapter and quality trimmed output to SRR6900811_trimmed.fq\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff '-q 20') and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900811.fastq <<< \n",
      "10000000 sequences processed\n",
      "20000000 sequences processed\n",
      "30000000 sequences processed\n",
      "40000000 sequences processed\n",
      "50000000 sequences processed\n",
      "60000000 sequences processed\n",
      "70000000 sequences processed\n",
      "This is cutadapt 1.18 with Python 3.6.15\n",
      "Command line parameters: -j 8 -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900811.fastq\n",
      "Processing reads on 8 cores in single-end mode ...\n",
      "Finished in 343.91 s (5 us/read; 12.76 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              73,133,582\n",
      "Reads with adapters:                24,255,964 (33.2%)\n",
      "Reads written (passing filters):    73,133,582 (100.0%)\n",
      "\n",
      "Total basepairs processed: 3,656,679,100 bp\n",
      "Quality-trimmed:             128,885,866 bp (3.5%)\n",
      "Total written (filtered):  3,474,345,015 bp (95.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 24255964 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 31.5%\n",
      "  C: 32.3%\n",
      "  G: 19.1%\n",
      "  T: 15.3%\n",
      "  none/other: 1.7%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t16585510\t18283395.5\t0\t16585510\n",
      "2\t5489686\t4570848.9\t0\t5489686\n",
      "3\t1356790\t1142712.2\t0\t1356790\n",
      "4\t291576\t285678.1\t0\t291576\n",
      "5\t64442\t71419.5\t0\t64442\n",
      "6\t7787\t17854.9\t0\t7787\n",
      "7\t2540\t4463.7\t0\t2540\n",
      "8\t1353\t1115.9\t0\t1353\n",
      "9\t2551\t279.0\t0\t1440 1111\n",
      "10\t3619\t69.7\t1\t1257 2362\n",
      "11\t1950\t17.4\t1\t995 955\n",
      "12\t1551\t4.4\t1\t1161 390\n",
      "13\t890\t1.1\t1\t799 91\n",
      "14\t1193\t1.1\t1\t1129 64\n",
      "15\t986\t1.1\t1\t911 75\n",
      "16\t1156\t1.1\t1\t1062 94\n",
      "17\t1386\t1.1\t1\t1289 97\n",
      "18\t1278\t1.1\t1\t1160 118\n",
      "19\t715\t1.1\t1\t643 72\n",
      "20\t1000\t1.1\t1\t888 112\n",
      "21\t1154\t1.1\t1\t1020 134\n",
      "22\t1123\t1.1\t1\t990 133\n",
      "23\t1142\t1.1\t1\t979 163\n",
      "24\t1344\t1.1\t1\t1136 208\n",
      "25\t2370\t1.1\t1\t2087 283\n",
      "26\t1399\t1.1\t1\t1106 293\n",
      "27\t1900\t1.1\t1\t1259 641\n",
      "28\t2603\t1.1\t1\t1981 622\n",
      "29\t2319\t1.1\t1\t1647 672\n",
      "30\t2327\t1.1\t1\t1946 381\n",
      "31\t1489\t1.1\t1\t1006 483\n",
      "32\t1937\t1.1\t1\t1460 477\n",
      "33\t3141\t1.1\t1\t1456 1685\n",
      "34\t4925\t1.1\t1\t3113 1812\n",
      "35\t3401\t1.1\t1\t2661 740\n",
      "36\t1910\t1.1\t1\t1138 772\n",
      "37\t4812\t1.1\t1\t1834 2978\n",
      "38\t6625\t1.1\t1\t3578 3047\n",
      "39\t5629\t1.1\t1\t3383 2246\n",
      "40\t6247\t1.1\t1\t2720 3527\n",
      "41\t6350\t1.1\t1\t3987 2363\n",
      "42\t5775\t1.1\t1\t2446 3329\n",
      "43\t7395\t1.1\t1\t3901 3494\n",
      "44\t9313\t1.1\t1\t4428 4885\n",
      "45\t14201\t1.1\t1\t6522 7679\n",
      "46\t41014\t1.1\t1\t7769 33245\n",
      "47\t64907\t1.1\t1\t32885 32022\n",
      "48\t64791\t1.1\t1\t40992 23799\n",
      "49\t162413\t1.1\t1\t76509 85904\n",
      "50\t4049\t1.1\t1\t1423 2626\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900811.fastq\n",
      "=============================================\n",
      "73133582 sequences processed in total\n",
      "Sequences removed because they became shorter than the length cutoff of 20 bp:\t502340 (0.7%)\n",
      "\n",
      "Now renaming the output file SRR6900811_trimmed.fq\n",
      "\n",
      "ORIGINAL FILE 1: >>SRR6900811_trimmed.fq<<\tRENAMING TO:>>Macaque_Liver_R4_trimmed.fq<<\n",
      "\n",
      "  >>> Now running FastQC on the data <<<\n",
      "\n",
      "Started analysis of Macaque_Liver_R4_trimmed.fq\n",
      "Approx 5% complete for Macaque_Liver_R4_trimmed.fq\n",
      "Approx 10% complete for Macaque_Liver_R4_trimmed.fq\n",
      "Approx 15% complete for Macaque_Liver_R4_trimmed.fq\n",
      "Approx 20% complete for Macaque_Liver_R4_trimmed.fq\n",
      "Approx 25% complete for Macaque_Liver_R4_trimmed.fq\n",
      "Approx 30% complete for Macaque_Liver_R4_trimmed.fq\n",
      "Approx 35% complete for Macaque_Liver_R4_trimmed.fq\n",
      "Approx 40% complete for Macaque_Liver_R4_trimmed.fq\n",
      "Approx 45% complete for Macaque_Liver_R4_trimmed.fq\n",
      "Approx 50% complete for Macaque_Liver_R4_trimmed.fq\n",
      "Approx 55% complete for Macaque_Liver_R4_trimmed.fq\n",
      "Approx 60% complete for Macaque_Liver_R4_trimmed.fq\n",
      "Approx 65% complete for Macaque_Liver_R4_trimmed.fq\n",
      "Approx 70% complete for Macaque_Liver_R4_trimmed.fq\n",
      "Approx 75% complete for Macaque_Liver_R4_trimmed.fq\n",
      "Approx 80% complete for Macaque_Liver_R4_trimmed.fq\n",
      "Approx 85% complete for Macaque_Liver_R4_trimmed.fq\n",
      "Approx 90% complete for Macaque_Liver_R4_trimmed.fq\n",
      "Approx 95% complete for Macaque_Liver_R4_trimmed.fq\n",
      "Analysis complete for Macaque_Liver_R4_trimmed.fq\n",
      "Using an excessive number of cores has a diminishing return! It is recommended not to exceed 8 cores per trimming process (you asked for 8 cores). Please consider re-specifying\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "Cutadapt version: 1.18\n",
      "Could not detect version of Python used by Cutadapt from the first line of Cutadapt (but found this: >>>#!/bin/sh<<<)\n",
      "Letting the (modified) Cutadapt deal with the Python version instead\n",
      "pigz 2.6\n",
      "Parallel gzip (pigz) detected. Proceeding with multicore (de)compression using 8 cores\n",
      "\n",
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Output will be written into the directory: /gpfs52/data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/trimmed_reads/\n",
      "Using user-specified basename (>>Macaque_Lung_R4<<) instead of deriving the filename from the input file(s)\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900812.fastq <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Illumina\t8020\tAGATCGGAAGAGC\t1000000\t0.80\n",
      "smallRNA\t1\tTGGAATTCTCGG\t1000000\t0.00\n",
      "Nextera\t0\tCTGTCTCTTATA\t1000000\t0.00\n",
      "Using Illumina adapter for trimming (count: 8020). Second best hit was smallRNA (count: 1)\n",
      "\n",
      "Writing report to '/gpfs52/data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/trimmed_reads/SRR6900812.fastq_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900812.fastq\n",
      "Trimming mode: single-end\n",
      "Trim Galore version: 0.6.6\n",
      "Cutadapt version: 1.18\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Python version: could not detect\n",
      "Number of cores used for trimming: 8\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length before a sequence gets removed: 20 bp\n",
      "Running FastQC on the data once trimming has completed\n",
      "Running FastQC with the following extra arguments: '--outdir /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/processing_results/fastQC'\n",
      "\n",
      "Cutadapt seems to be reasonably up-to-date. Setting -j 8\n",
      "Writing final adapter and quality trimmed output to SRR6900812_trimmed.fq\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff '-q 20') and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900812.fastq <<< \n",
      "10000000 sequences processed\n",
      "20000000 sequences processed\n",
      "30000000 sequences processed\n",
      "40000000 sequences processed\n",
      "50000000 sequences processed\n",
      "60000000 sequences processed\n",
      "This is cutadapt 1.18 with Python 3.6.15\n",
      "Command line parameters: -j 8 -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900812.fastq\n",
      "Processing reads on 8 cores in single-end mode ...\n",
      "Finished in 249.34 s (4 us/read; 14.89 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              61,868,144\n",
      "Reads with adapters:                13,428,988 (21.7%)\n",
      "Reads written (passing filters):    61,868,144 (100.0%)\n",
      "\n",
      "Total basepairs processed: 3,131,332,080 bp\n",
      "Quality-trimmed:              40,217,249 bp (1.3%)\n",
      "Total written (filtered):  3,046,550,409 bp (97.3%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 13428988 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 28.8%\n",
      "  C: 32.1%\n",
      "  G: 19.8%\n",
      "  T: 15.6%\n",
      "  none/other: 3.7%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t8994640\t15467036.0\t0\t8994640\n",
      "2\t2903163\t3866759.0\t0\t2903163\n",
      "3\t767092\t966689.8\t0\t767092\n",
      "4\t160687\t241672.4\t0\t160687\n",
      "5\t35404\t60418.1\t0\t35404\n",
      "6\t4675\t15104.5\t0\t4675\n",
      "7\t1651\t3776.1\t0\t1651\n",
      "8\t1116\t944.0\t0\t1116\n",
      "9\t1760\t236.0\t0\t1180 580\n",
      "10\t2328\t59.0\t1\t1046 1282\n",
      "11\t1705\t14.8\t1\t1032 673\n",
      "12\t1515\t3.7\t1\t1063 452\n",
      "13\t1294\t0.9\t1\t918 376\n",
      "14\t1455\t0.9\t1\t1342 113\n",
      "15\t1405\t0.9\t1\t1292 113\n",
      "16\t1405\t0.9\t1\t1281 124\n",
      "17\t1478\t0.9\t1\t1372 106\n",
      "18\t1531\t0.9\t1\t1414 117\n",
      "19\t1363\t0.9\t1\t1236 127\n",
      "20\t1431\t0.9\t1\t1294 137\n",
      "21\t1499\t0.9\t1\t1355 144\n",
      "22\t1588\t0.9\t1\t1454 134\n",
      "23\t1777\t0.9\t1\t1651 126\n",
      "24\t1809\t0.9\t1\t1685 124\n",
      "25\t1813\t0.9\t1\t1662 151\n",
      "26\t1704\t0.9\t1\t1565 139\n",
      "27\t1931\t0.9\t1\t1778 153\n",
      "28\t2027\t0.9\t1\t1904 123\n",
      "29\t2018\t0.9\t1\t1837 181\n",
      "30\t2247\t0.9\t1\t2097 150\n",
      "31\t1911\t0.9\t1\t1786 125\n",
      "32\t2200\t0.9\t1\t2059 141\n",
      "33\t2286\t0.9\t1\t2090 196\n",
      "34\t2336\t0.9\t1\t2141 195\n",
      "35\t2613\t0.9\t1\t2414 199\n",
      "36\t1978\t0.9\t1\t1802 176\n",
      "37\t1927\t0.9\t1\t1702 225\n",
      "38\t2141\t0.9\t1\t1903 238\n",
      "39\t1658\t0.9\t1\t1449 209\n",
      "40\t1246\t0.9\t1\t1047 199\n",
      "41\t1651\t0.9\t1\t1381 270\n",
      "42\t963\t0.9\t1\t769 194\n",
      "43\t840\t0.9\t1\t630 210\n",
      "44\t897\t0.9\t1\t625 272\n",
      "45\t1400\t0.9\t1\t987 413\n",
      "46\t1842\t0.9\t1\t1040 802\n",
      "47\t5285\t0.9\t1\t3242 2043\n",
      "48\t9033\t0.9\t1\t7473 1560\n",
      "49\t7640\t0.9\t1\t4517 3123\n",
      "50\t402533\t0.9\t1\t249164 153369\n",
      "51\t71097\t0.9\t1\t45690 25407\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/SRR6900812.fastq\n",
      "=============================================\n",
      "61868144 sequences processed in total\n",
      "Sequences removed because they became shorter than the length cutoff of 20 bp:\t570113 (0.9%)\n",
      "\n",
      "Now renaming the output file SRR6900812_trimmed.fq\n",
      "\n",
      "ORIGINAL FILE 1: >>SRR6900812_trimmed.fq<<\tRENAMING TO:>>Macaque_Lung_R4_trimmed.fq<<\n",
      "\n",
      "  >>> Now running FastQC on the data <<<\n",
      "\n",
      "Started analysis of Macaque_Lung_R4_trimmed.fq\n",
      "Approx 5% complete for Macaque_Lung_R4_trimmed.fq\n",
      "Approx 10% complete for Macaque_Lung_R4_trimmed.fq\n",
      "Approx 15% complete for Macaque_Lung_R4_trimmed.fq\n",
      "Approx 20% complete for Macaque_Lung_R4_trimmed.fq\n",
      "Approx 25% complete for Macaque_Lung_R4_trimmed.fq\n",
      "Approx 30% complete for Macaque_Lung_R4_trimmed.fq\n",
      "Approx 35% complete for Macaque_Lung_R4_trimmed.fq\n",
      "Approx 40% complete for Macaque_Lung_R4_trimmed.fq\n",
      "Approx 45% complete for Macaque_Lung_R4_trimmed.fq\n",
      "Approx 50% complete for Macaque_Lung_R4_trimmed.fq\n",
      "Approx 55% complete for Macaque_Lung_R4_trimmed.fq\n",
      "Approx 60% complete for Macaque_Lung_R4_trimmed.fq\n",
      "Approx 65% complete for Macaque_Lung_R4_trimmed.fq\n",
      "Approx 70% complete for Macaque_Lung_R4_trimmed.fq\n",
      "Approx 75% complete for Macaque_Lung_R4_trimmed.fq\n",
      "Approx 80% complete for Macaque_Lung_R4_trimmed.fq\n",
      "Approx 85% complete for Macaque_Lung_R4_trimmed.fq\n",
      "Approx 90% complete for Macaque_Lung_R4_trimmed.fq\n",
      "Approx 95% complete for Macaque_Lung_R4_trimmed.fq\n",
      "Analysis complete for Macaque_Lung_R4_trimmed.fq\n"
     ]
    }
   ],
   "source": [
    "# use read combined with while loop to read two variables at once.\n",
    "DIR='/data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020'\n",
    "\n",
    "while read -r a b ; do    \n",
    "    trim_galore -j 8 --fastqc --fastqc_args \"--outdir ${DIR}/processing_results/fastQC\" --dont_gzip --basename \"$a\" \\\n",
    "    --output_dir ${DIR}/fastq/trimmed_reads ${DIR}/fastq/\"$b\".fastq \n",
    "done < /data/hodges_lab/public_data/human-rhesus_RNA-seq/tissues_names.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "35c7b95c",
   "metadata": {},
   "source": [
    "## Mapping\n",
    "\n",
    "See GM12878_vs_LCL8664.ipynb for generating the modified GTFs."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "36117341",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-03-16T19:44:20.636335Z",
     "start_time": "2022-03-16T19:44:20.354741Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "#!/bin/bash\n",
      "#SBATCH --mail-user=tyler.j.hansen@vanderbilt.edu\n",
      "#SBATCH --mail-type=ALL\n",
      "#SBATCH --nodes=1\n",
      "#SBATCH --ntasks=1\n",
      "#SBATCH --cpus-per-task=8\n",
      "#SBATCH --time=24:00:00\n",
      "#SBATCH --mem=64G\n",
      "\n",
      "#Generate indexes for 50bp and 36bp read lengths.\n",
      "HG38_FA='/data/hodges_lab/hg38_genome/hg38.fa'\n",
      "HG38_GTF='/data/hodges_lab/hg38_genome/Homo_sapiens.GRCh38.104.modified.gtf'\n",
      "RHEMAC10_FA='/data/hodges_lab/rheMac10_genome/rheMac10.fa'\n",
      "RHEMAC10=GTF='/data/hodges_lab/rheMac10_genome/Macaca_mulatta.Mmul_10.104.modified.gtf'\n",
      "\n",
      "## 50bp ##\n",
      "#hg38\n",
      "STAR --runThreadN 8 \\\n",
      "--runMode genomeGenerate \\\n",
      "--genomeDir /data/hodges_lab/hg38_genome/hg38_GTF-ensembl.104_index_49 \\\n",
      "--genomeFastaFiles $HG38_FA \\\n",
      "--sjdbGTFfile $HG38_GTF \\\n",
      "--sjdbOverhang 49 #overhang is maximum readlength-1. \n",
      "\n",
      "#rheMac10\n",
      "STAR --runThreadN 8 \\\n",
      "--runMode genomeGenerate \\\n",
      "--genomeDir /data/hodges_lab/rheMac10_genome/rheMac10_GTF-ensembl.104_index_49 \\\n",
      "--genomeFastaFiles $RHEMAC10_FA \\\n",
      "--sjdbGTFfile $RHEMAC10_GTF \\\n",
      "--sjdbOverhang 49 #overhang is maximum readlength-1.\n",
      "\n",
      "## 36bp ##\n",
      "#hg38\n",
      "STAR --runThreadN 8 \\\n",
      "--runMode genomeGenerate \\\n",
      "--genomeDir /data/hodges_lab/hg38_genome/hg38_GTF-ensembl.104_index_35 \\\n",
      "--genomeFastaFiles $HG38_FA \\\n",
      "--sjdbGTFfile $HG38_GTF \\\n",
      "--sjdbOverhang 35 #overhang is maximum readlength-1. \n",
      "\n",
      "#rheMac10\n",
      "STAR --runThreadN 8 \\\n",
      "--runMode genomeGenerate \\\n",
      "--genomeDir /data/hodges_lab/rheMac10_genome/rheMac10_GTF-ensembl.104_index_35 \\\n",
      "--genomeFastaFiles $RHEMAC10_FA \\\n",
      "--sjdbGTFfile $RHEMAC10_GTF \\\n",
      "--sjdbOverhang 35 #overhang is maximum readlength-1.\n"
     ]
    }
   ],
   "source": [
    "#Generate indexes for 50bp and 36bp read lengths.\n",
    "\n",
    "#ran as job: \n",
    "cat generate_indexes.slrm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "ecdd0d34",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-03-17T14:29:59.409501Z",
     "start_time": "2022-03-17T13:49:07.659234Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\tSTAR --runMode alignReads --runThreadN 8 --genomeDir /data/hodges_lab/hg38_genome/hg38_GTF-ensembl.104_index_35 --readFilesIn /data/hodges_lab/public_data/human-rhesus_RNA-seq/cain_2011_LCLs/fastq/trimmed_reads/Human_LCL_R1_trimmed.fq --outFileNamePrefix /data/hodges_lab/public_data/human-rhesus_RNA-seq/cain_2011_LCLs/star-output/Human_LCL_R1_ensembl.104_ --outSAMtype BAM SortedByCoordinate\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Mar 17 08:49:08 ..... started STAR run\n",
      "Mar 17 08:49:08 ..... loading genome\n",
      "Mar 17 08:54:19 ..... started mapping\n",
      "Mar 17 08:55:41 ..... finished mapping\n",
      "Mar 17 08:55:46 ..... started sorting BAM\n",
      "Mar 17 08:56:24 ..... finished successfully\n",
      "\tSTAR --runMode alignReads --runThreadN 8 --genomeDir /data/hodges_lab/hg38_genome/hg38_GTF-ensembl.104_index_35 --readFilesIn /data/hodges_lab/public_data/human-rhesus_RNA-seq/cain_2011_LCLs/fastq/trimmed_reads/Human_LCL_R2_trimmed.fq --outFileNamePrefix /data/hodges_lab/public_data/human-rhesus_RNA-seq/cain_2011_LCLs/star-output/Human_LCL_R2_ensembl.104_ --outSAMtype BAM SortedByCoordinate\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Mar 17 08:56:24 ..... started STAR run\n",
      "Mar 17 08:56:24 ..... loading genome\n",
      "Mar 17 09:00:44 ..... started mapping\n",
      "Mar 17 09:02:12 ..... finished mapping\n",
      "Mar 17 09:02:17 ..... started sorting BAM\n",
      "Mar 17 09:03:02 ..... finished successfully\n",
      "\tSTAR --runMode alignReads --runThreadN 8 --genomeDir /data/hodges_lab/hg38_genome/hg38_GTF-ensembl.104_index_35 --readFilesIn /data/hodges_lab/public_data/human-rhesus_RNA-seq/cain_2011_LCLs/fastq/trimmed_reads/Human_LCL_R3_trimmed.fq --outFileNamePrefix /data/hodges_lab/public_data/human-rhesus_RNA-seq/cain_2011_LCLs/star-output/Human_LCL_R3_ensembl.104_ --outSAMtype BAM SortedByCoordinate\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Mar 17 09:03:02 ..... started STAR run\n",
      "Mar 17 09:03:02 ..... loading genome\n",
      "Mar 17 09:07:25 ..... started mapping\n",
      "Mar 17 09:09:02 ..... finished mapping\n",
      "Mar 17 09:09:07 ..... started sorting BAM\n",
      "Mar 17 09:09:58 ..... finished successfully\n",
      "\tSTAR --runMode alignReads --runThreadN 8 --genomeDir /data/hodges_lab/rheMac10_genome/rheMac10_GTF-ensembl.104_index_35 --readFilesIn /data/hodges_lab/public_data/human-rhesus_RNA-seq/cain_2011_LCLs/fastq/trimmed_reads/Rhesus_LCL_R1_trimmed.fq --outFileNamePrefix /data/hodges_lab/public_data/human-rhesus_RNA-seq/cain_2011_LCLs/star-output/Rhesus_LCL_R1_ensembl.104_ --outSAMtype BAM SortedByCoordinate\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Mar 17 09:09:59 ..... started STAR run\n",
      "Mar 17 09:09:59 ..... loading genome\n",
      "Mar 17 09:14:16 ..... started mapping\n",
      "Mar 17 09:15:49 ..... finished mapping\n",
      "Mar 17 09:15:53 ..... started sorting BAM\n",
      "Mar 17 09:16:42 ..... finished successfully\n",
      "\tSTAR --runMode alignReads --runThreadN 8 --genomeDir /data/hodges_lab/rheMac10_genome/rheMac10_GTF-ensembl.104_index_35 --readFilesIn /data/hodges_lab/public_data/human-rhesus_RNA-seq/cain_2011_LCLs/fastq/trimmed_reads/Rhesus_LCL_R2_trimmed.fq --outFileNamePrefix /data/hodges_lab/public_data/human-rhesus_RNA-seq/cain_2011_LCLs/star-output/Rhesus_LCL_R2_ensembl.104_ --outSAMtype BAM SortedByCoordinate\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Mar 17 09:16:42 ..... started STAR run\n",
      "Mar 17 09:16:42 ..... loading genome\n",
      "Mar 17 09:20:49 ..... started mapping\n",
      "Mar 17 09:22:30 ..... finished mapping\n",
      "Mar 17 09:22:34 ..... started sorting BAM\n",
      "Mar 17 09:23:19 ..... finished successfully\n",
      "\tSTAR --runMode alignReads --runThreadN 8 --genomeDir /data/hodges_lab/rheMac10_genome/rheMac10_GTF-ensembl.104_index_35 --readFilesIn /data/hodges_lab/public_data/human-rhesus_RNA-seq/cain_2011_LCLs/fastq/trimmed_reads/Rhesus_LCL_R3_trimmed.fq --outFileNamePrefix /data/hodges_lab/public_data/human-rhesus_RNA-seq/cain_2011_LCLs/star-output/Rhesus_LCL_R3_ensembl.104_ --outSAMtype BAM SortedByCoordinate\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Mar 17 09:23:20 ..... started STAR run\n",
      "Mar 17 09:23:20 ..... loading genome\n",
      "Mar 17 09:27:32 ..... started mapping\n",
      "Mar 17 09:29:10 ..... finished mapping\n",
      "Mar 17 09:29:14 ..... started sorting BAM\n",
      "Mar 17 09:29:59 ..... finished successfully\n"
     ]
    }
   ],
   "source": [
    "## Map to transcriptiome with STAR aligner. \n",
    "DIR='/data/hodges_lab/public_data/human-rhesus_RNA-seq/cain_2011_LCLs'\n",
    "\n",
    "#Index paths\n",
    "HG38_36='/data/hodges_lab/hg38_genome/hg38_GTF-ensembl.104_index_35'\n",
    "RHEMAC10_36='/data/hodges_lab/rheMac10_genome/rheMac10_GTF-ensembl.104_index_35'\n",
    "\n",
    "#hg38\n",
    "for EXPT in Human_LCL_R1 Human_LCL_R2 Human_LCL_R3\n",
    "do\n",
    "    STAR --runMode alignReads --runThreadN 8 --genomeDir $HG38_36 \\\n",
    "    --readFilesIn ${DIR}/fastq/trimmed_reads/${EXPT}_trimmed.fq \\\n",
    "    --outFileNamePrefix ${DIR}/star-output/${EXPT}_ensembl.104_ --outSAMtype BAM SortedByCoordinate \n",
    "done\n",
    "\n",
    "#rheMac10\n",
    "for EXPT in Rhesus_LCL_R1 Rhesus_LCL_R2 Rhesus_LCL_R3\n",
    "do\n",
    "    STAR --runMode alignReads --runThreadN 8 --genomeDir $RHEMAC10_36 \\\n",
    "    --readFilesIn ${DIR}/fastq/trimmed_reads/${EXPT}_trimmed.fq \\\n",
    "    --outFileNamePrefix ${DIR}/star-output/${EXPT}_ensembl.104_ --outSAMtype BAM SortedByCoordinate \n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e78b9ab6",
   "metadata": {
    "ExecuteTime": {
     "start_time": "2022-03-17T14:33:47.412Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\tSTAR --runMode alignReads --runThreadN 8 --genomeDir /data/hodges_lab/hg38_genome/hg38_GTF-ensembl.104_index_49 --readFilesIn /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/trimmed_reads/Human_Heart_R1_trimmed.fq --outFileNamePrefix /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/star-output/Human_Heart_R1_ensembl.104_ --outSAMtype BAM SortedByCoordinate\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Mar 17 09:33:39 ..... started STAR run\n",
      "Mar 17 09:33:39 ..... loading genome\n",
      "Mar 17 09:37:55 ..... started mapping\n",
      "Mar 17 09:42:39 ..... finished mapping\n",
      "Mar 17 09:42:44 ..... started sorting BAM\n",
      "Mar 17 09:45:10 ..... finished successfully\n",
      "\tSTAR --runMode alignReads --runThreadN 8 --genomeDir /data/hodges_lab/hg38_genome/hg38_GTF-ensembl.104_index_49 --readFilesIn /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/trimmed_reads/Human_Heart_R2_trimmed.fq --outFileNamePrefix /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/star-output/Human_Heart_R2_ensembl.104_ --outSAMtype BAM SortedByCoordinate\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Mar 17 09:45:10 ..... started STAR run\n",
      "Mar 17 09:45:10 ..... loading genome\n",
      "Mar 17 09:52:04 ..... started mapping\n",
      "Mar 17 09:57:10 ..... finished mapping\n",
      "Mar 17 09:57:15 ..... started sorting BAM\n",
      "Mar 17 09:59:38 ..... finished successfully\n",
      "\tSTAR --runMode alignReads --runThreadN 8 --genomeDir /data/hodges_lab/hg38_genome/hg38_GTF-ensembl.104_index_49 --readFilesIn /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/trimmed_reads/Human_Heart_R3_trimmed.fq --outFileNamePrefix /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/star-output/Human_Heart_R3_ensembl.104_ --outSAMtype BAM SortedByCoordinate\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Mar 17 09:59:38 ..... started STAR run\n",
      "Mar 17 09:59:38 ..... loading genome\n",
      "Mar 17 10:06:02 ..... started mapping\n",
      "Mar 17 10:09:55 ..... finished mapping\n",
      "Mar 17 10:09:59 ..... started sorting BAM\n",
      "Mar 17 10:11:53 ..... finished successfully\n",
      "\tSTAR --runMode alignReads --runThreadN 8 --genomeDir /data/hodges_lab/hg38_genome/hg38_GTF-ensembl.104_index_49 --readFilesIn /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/trimmed_reads/Human_Heart_R4_trimmed.fq --outFileNamePrefix /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/star-output/Human_Heart_R4_ensembl.104_ --outSAMtype BAM SortedByCoordinate\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Mar 17 10:11:54 ..... started STAR run\n",
      "Mar 17 10:11:54 ..... loading genome\n",
      "Mar 17 10:18:18 ..... started mapping\n",
      "Mar 17 10:22:49 ..... finished mapping\n",
      "Mar 17 10:22:54 ..... started sorting BAM\n",
      "Mar 17 10:24:57 ..... finished successfully\n",
      "\tSTAR --runMode alignReads --runThreadN 8 --genomeDir /data/hodges_lab/hg38_genome/hg38_GTF-ensembl.104_index_49 --readFilesIn /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/trimmed_reads/Human_Kidney_R1_trimmed.fq --outFileNamePrefix /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/star-output/Human_Kidney_R1_ensembl.104_ --outSAMtype BAM SortedByCoordinate\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Mar 17 10:24:58 ..... started STAR run\n",
      "Mar 17 10:24:58 ..... loading genome\n",
      "Mar 17 10:29:28 ..... started mapping\n",
      "Mar 17 10:33:58 ..... finished mapping\n",
      "Mar 17 10:34:03 ..... started sorting BAM\n",
      "Mar 17 10:36:54 ..... finished successfully\n",
      "\tSTAR --runMode alignReads --runThreadN 8 --genomeDir /data/hodges_lab/hg38_genome/hg38_GTF-ensembl.104_index_49 --readFilesIn /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/trimmed_reads/Human_Kidney_R2_trimmed.fq --outFileNamePrefix /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/star-output/Human_Kidney_R2_ensembl.104_ --outSAMtype BAM SortedByCoordinate\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Mar 17 10:36:55 ..... started STAR run\n",
      "Mar 17 10:36:55 ..... loading genome\n",
      "Mar 17 10:47:17 ..... started mapping\n",
      "Mar 17 10:50:16 ..... finished mapping\n",
      "Mar 17 10:50:21 ..... started sorting BAM\n",
      "Mar 17 10:52:08 ..... finished successfully\n",
      "\tSTAR --runMode alignReads --runThreadN 8 --genomeDir /data/hodges_lab/hg38_genome/hg38_GTF-ensembl.104_index_49 --readFilesIn /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/trimmed_reads/Human_Kidney_R3_trimmed.fq --outFileNamePrefix /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/star-output/Human_Kidney_R3_ensembl.104_ --outSAMtype BAM SortedByCoordinate\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Mar 17 10:52:08 ..... started STAR run\n",
      "Mar 17 10:52:08 ..... loading genome\n",
      "Mar 17 11:01:53 ..... started mapping\n",
      "Mar 17 11:06:40 ..... finished mapping\n",
      "Mar 17 11:06:44 ..... started sorting BAM\n",
      "Mar 17 11:08:59 ..... finished successfully\n",
      "\tSTAR --runMode alignReads --runThreadN 8 --genomeDir /data/hodges_lab/hg38_genome/hg38_GTF-ensembl.104_index_49 --readFilesIn /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/trimmed_reads/Human_Kidney_R4_trimmed.fq --outFileNamePrefix /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/star-output/Human_Kidney_R4_ensembl.104_ --outSAMtype BAM SortedByCoordinate\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Mar 17 11:09:00 ..... started STAR run\n",
      "Mar 17 11:09:00 ..... loading genome\n",
      "Mar 17 11:15:42 ..... started mapping\n",
      "Mar 17 11:21:56 ..... finished mapping\n",
      "Mar 17 11:22:01 ..... started sorting BAM\n",
      "Mar 17 11:24:33 ..... finished successfully\n",
      "\tSTAR --runMode alignReads --runThreadN 8 --genomeDir /data/hodges_lab/hg38_genome/hg38_GTF-ensembl.104_index_49 --readFilesIn /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/trimmed_reads/Human_Liver_R1_trimmed.fq --outFileNamePrefix /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/star-output/Human_Liver_R1_ensembl.104_ --outSAMtype BAM SortedByCoordinate\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Mar 17 11:24:33 ..... started STAR run\n",
      "Mar 17 11:24:33 ..... loading genome\n",
      "Mar 17 11:37:05 ..... started mapping\n",
      "Mar 17 11:42:13 ..... finished mapping\n",
      "Mar 17 11:42:17 ..... started sorting BAM\n",
      "Mar 17 11:44:20 ..... finished successfully\n",
      "\tSTAR --runMode alignReads --runThreadN 8 --genomeDir /data/hodges_lab/hg38_genome/hg38_GTF-ensembl.104_index_49 --readFilesIn /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/trimmed_reads/Human_Liver_R2_trimmed.fq --outFileNamePrefix /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/star-output/Human_Liver_R2_ensembl.104_ --outSAMtype BAM SortedByCoordinate\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Mar 17 11:44:20 ..... started STAR run\n",
      "Mar 17 11:44:20 ..... loading genome\n",
      "Mar 17 11:51:14 ..... started mapping\n",
      "Mar 17 11:55:00 ..... finished mapping\n",
      "Mar 17 11:55:05 ..... started sorting BAM\n",
      "Mar 17 11:57:03 ..... finished successfully\n",
      "\tSTAR --runMode alignReads --runThreadN 8 --genomeDir /data/hodges_lab/hg38_genome/hg38_GTF-ensembl.104_index_49 --readFilesIn /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/trimmed_reads/Human_Liver_R3_trimmed.fq --outFileNamePrefix /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/star-output/Human_Liver_R3_ensembl.104_ --outSAMtype BAM SortedByCoordinate\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Mar 17 11:57:04 ..... started STAR run\n",
      "Mar 17 11:57:04 ..... loading genome\n",
      "Mar 17 12:01:24 ..... started mapping\n",
      "Mar 17 12:06:36 ..... finished mapping\n",
      "Mar 17 12:06:40 ..... started sorting BAM\n",
      "Mar 17 12:09:09 ..... finished successfully\n",
      "\tSTAR --runMode alignReads --runThreadN 8 --genomeDir /data/hodges_lab/hg38_genome/hg38_GTF-ensembl.104_index_49 --readFilesIn /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/trimmed_reads/Human_Liver_R4_trimmed.fq --outFileNamePrefix /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/star-output/Human_Liver_R4_ensembl.104_ --outSAMtype BAM SortedByCoordinate\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Mar 17 12:09:10 ..... started STAR run\n",
      "Mar 17 12:09:10 ..... loading genome\n",
      "Mar 17 12:15:54 ..... started mapping\n",
      "Mar 17 12:21:29 ..... finished mapping\n",
      "Mar 17 12:21:34 ..... started sorting BAM\n",
      "Mar 17 12:23:58 ..... finished successfully\n",
      "\tSTAR --runMode alignReads --runThreadN 8 --genomeDir /data/hodges_lab/hg38_genome/hg38_GTF-ensembl.104_index_49 --readFilesIn /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/trimmed_reads/Human_Lung_R1_trimmed.fq --outFileNamePrefix /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/star-output/Human_Lung_R1_ensembl.104_ --outSAMtype BAM SortedByCoordinate\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Mar 17 12:23:58 ..... started STAR run\n",
      "Mar 17 12:23:58 ..... loading genome\n",
      "Mar 17 12:28:15 ..... started mapping\n",
      "Mar 17 12:31:12 ..... finished mapping\n",
      "Mar 17 12:31:17 ..... started sorting BAM\n",
      "Mar 17 12:32:55 ..... finished successfully\n",
      "\tSTAR --runMode alignReads --runThreadN 8 --genomeDir /data/hodges_lab/hg38_genome/hg38_GTF-ensembl.104_index_49 --readFilesIn /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/trimmed_reads/Human_Lung_R2_trimmed.fq --outFileNamePrefix /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/star-output/Human_Lung_R2_ensembl.104_ --outSAMtype BAM SortedByCoordinate\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Mar 17 12:32:55 ..... started STAR run\n",
      "Mar 17 12:32:55 ..... loading genome\n",
      "Mar 17 12:37:14 ..... started mapping\n",
      "Mar 17 12:43:11 ..... finished mapping\n",
      "Mar 17 12:43:16 ..... started sorting BAM\n",
      "Mar 17 12:45:39 ..... finished successfully\n",
      "\tSTAR --runMode alignReads --runThreadN 8 --genomeDir /data/hodges_lab/hg38_genome/hg38_GTF-ensembl.104_index_49 --readFilesIn /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/trimmed_reads/Human_Lung_R3_trimmed.fq --outFileNamePrefix /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/star-output/Human_Lung_R3_ensembl.104_ --outSAMtype BAM SortedByCoordinate\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Mar 17 12:45:39 ..... started STAR run\n",
      "Mar 17 12:45:39 ..... loading genome\n",
      "Mar 17 12:50:04 ..... started mapping\n",
      "Mar 17 12:55:18 ..... finished mapping\n",
      "Mar 17 12:55:24 ..... started sorting BAM\n",
      "Mar 17 13:00:14 ..... finished successfully\n",
      "\tSTAR --runMode alignReads --runThreadN 8 --genomeDir /data/hodges_lab/hg38_genome/hg38_GTF-ensembl.104_index_49 --readFilesIn /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/trimmed_reads/Human_Lung_R4_trimmed.fq --outFileNamePrefix /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/star-output/Human_Lung_R4_ensembl.104_ --outSAMtype BAM SortedByCoordinate\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Mar 17 13:00:14 ..... started STAR run\n",
      "Mar 17 13:00:14 ..... loading genome\n",
      "Mar 17 13:06:11 ..... started mapping\n",
      "Mar 17 13:17:53 ..... finished mapping\n",
      "Mar 17 13:17:58 ..... started sorting BAM\n",
      "Mar 17 13:20:39 ..... finished successfully\n",
      "\tSTAR --runMode alignReads --runThreadN 8 --genomeDir /data/hodges_lab/rheMac10_genome/rheMac10_GTF-ensembl.104_index_49 --readFilesIn /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/trimmed_reads/Macaque_Heart_R1_trimmed.fq --outFileNamePrefix /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/star-output/Macaque_Heart_R1_ensembl.104_ --outSAMtype BAM SortedByCoordinate\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Mar 17 13:20:40 ..... started STAR run\n",
      "Mar 17 13:20:40 ..... loading genome\n",
      "Mar 17 13:24:52 ..... started mapping\n",
      "Mar 17 13:30:03 ..... finished mapping\n",
      "Mar 17 13:30:08 ..... started sorting BAM\n",
      "Mar 17 13:33:15 ..... finished successfully\n",
      "\tSTAR --runMode alignReads --runThreadN 8 --genomeDir /data/hodges_lab/rheMac10_genome/rheMac10_GTF-ensembl.104_index_49 --readFilesIn /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/trimmed_reads/Macaque_Heart_R2_trimmed.fq --outFileNamePrefix /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/star-output/Macaque_Heart_R2_ensembl.104_ --outSAMtype BAM SortedByCoordinate\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Mar 17 13:33:16 ..... started STAR run\n",
      "Mar 17 13:33:16 ..... loading genome\n",
      "Mar 17 13:40:56 ..... started mapping\n",
      "Mar 17 13:47:20 ..... finished mapping\n",
      "Mar 17 13:47:24 ..... started sorting BAM\n",
      "Mar 17 13:51:20 ..... finished successfully\n",
      "\tSTAR --runMode alignReads --runThreadN 8 --genomeDir /data/hodges_lab/rheMac10_genome/rheMac10_GTF-ensembl.104_index_49 --readFilesIn /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/trimmed_reads/Macaque_Heart_R3_trimmed.fq --outFileNamePrefix /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/star-output/Macaque_Heart_R3_ensembl.104_ --outSAMtype BAM SortedByCoordinate\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Mar 17 13:51:21 ..... started STAR run\n",
      "Mar 17 13:51:21 ..... loading genome\n",
      "Mar 17 14:05:50 ..... started mapping\n",
      "Mar 17 14:16:28 ..... finished mapping\n",
      "Mar 17 14:16:33 ..... started sorting BAM\n",
      "Mar 17 14:23:56 ..... finished successfully\n",
      "\tSTAR --runMode alignReads --runThreadN 8 --genomeDir /data/hodges_lab/rheMac10_genome/rheMac10_GTF-ensembl.104_index_49 --readFilesIn /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/trimmed_reads/Macaque_Heart_R4_trimmed.fq --outFileNamePrefix /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/star-output/Macaque_Heart_R4_ensembl.104_ --outSAMtype BAM SortedByCoordinate\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Mar 17 14:23:56 ..... started STAR run\n",
      "Mar 17 14:23:56 ..... loading genome\n",
      "Mar 17 14:30:27 ..... started mapping\n",
      "Mar 17 14:37:18 ..... finished mapping\n",
      "Mar 17 14:37:23 ..... started sorting BAM\n",
      "Mar 17 14:42:19 ..... finished successfully\n",
      "\tSTAR --runMode alignReads --runThreadN 8 --genomeDir /data/hodges_lab/rheMac10_genome/rheMac10_GTF-ensembl.104_index_49 --readFilesIn /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/trimmed_reads/Macaque_Kidney_R1_trimmed.fq --outFileNamePrefix /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/star-output/Macaque_Kidney_R1_ensembl.104_ --outSAMtype BAM SortedByCoordinate\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Mar 17 14:42:20 ..... started STAR run\n",
      "Mar 17 14:42:20 ..... loading genome\n",
      "Mar 17 14:48:44 ..... started mapping\n",
      "Mar 17 14:52:13 ..... finished mapping\n",
      "Mar 17 14:52:17 ..... started sorting BAM\n",
      "Mar 17 14:54:40 ..... finished successfully\n",
      "\tSTAR --runMode alignReads --runThreadN 8 --genomeDir /data/hodges_lab/rheMac10_genome/rheMac10_GTF-ensembl.104_index_49 --readFilesIn /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/trimmed_reads/Macaque_Kidney_R2_trimmed.fq --outFileNamePrefix /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/star-output/Macaque_Kidney_R2_ensembl.104_ --outSAMtype BAM SortedByCoordinate\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Mar 17 14:54:40 ..... started STAR run\n",
      "Mar 17 14:54:40 ..... loading genome\n",
      "Mar 17 15:03:04 ..... started mapping\n",
      "Mar 17 15:09:09 ..... finished mapping\n",
      "Mar 17 15:09:14 ..... started sorting BAM\n",
      "Mar 17 15:11:38 ..... finished successfully\n",
      "\tSTAR --runMode alignReads --runThreadN 8 --genomeDir /data/hodges_lab/rheMac10_genome/rheMac10_GTF-ensembl.104_index_49 --readFilesIn /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/trimmed_reads/Macaque_Kidney_R3_trimmed.fq --outFileNamePrefix /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/star-output/Macaque_Kidney_R3_ensembl.104_ --outSAMtype BAM SortedByCoordinate\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Mar 17 15:11:38 ..... started STAR run\n",
      "Mar 17 15:11:38 ..... loading genome\n",
      "Mar 17 15:20:07 ..... started mapping\n",
      "Mar 17 15:26:29 ..... finished mapping\n",
      "Mar 17 15:26:36 ..... started sorting BAM\n",
      "Mar 17 15:31:24 ..... finished successfully\n",
      "\tSTAR --runMode alignReads --runThreadN 8 --genomeDir /data/hodges_lab/rheMac10_genome/rheMac10_GTF-ensembl.104_index_49 --readFilesIn /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/trimmed_reads/Macaque_Kidney_R4_trimmed.fq --outFileNamePrefix /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/star-output/Macaque_Kidney_R4_ensembl.104_ --outSAMtype BAM SortedByCoordinate\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Mar 17 15:31:25 ..... started STAR run\n",
      "Mar 17 15:31:25 ..... loading genome\n",
      "Mar 17 15:42:25 ..... started mapping\n",
      "Mar 17 15:50:06 ..... finished mapping\n",
      "Mar 17 15:50:10 ..... started sorting BAM\n",
      "Mar 17 15:54:06 ..... finished successfully\n",
      "\tSTAR --runMode alignReads --runThreadN 8 --genomeDir /data/hodges_lab/rheMac10_genome/rheMac10_GTF-ensembl.104_index_49 --readFilesIn /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/trimmed_reads/Macaque_Liver_R1_trimmed.fq --outFileNamePrefix /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/star-output/Macaque_Liver_R1_ensembl.104_ --outSAMtype BAM SortedByCoordinate\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Mar 17 15:54:06 ..... started STAR run\n",
      "Mar 17 15:54:06 ..... loading genome\n",
      "Mar 17 15:59:33 ..... started mapping\n",
      "Mar 17 16:02:42 ..... finished mapping\n",
      "Mar 17 16:02:46 ..... started sorting BAM\n",
      "Mar 17 16:04:23 ..... finished successfully\n",
      "\tSTAR --runMode alignReads --runThreadN 8 --genomeDir /data/hodges_lab/rheMac10_genome/rheMac10_GTF-ensembl.104_index_49 --readFilesIn /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/trimmed_reads/Macaque_Liver_R2_trimmed.fq --outFileNamePrefix /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/star-output/Macaque_Liver_R2_ensembl.104_ --outSAMtype BAM SortedByCoordinate\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Mar 17 16:04:24 ..... started STAR run\n",
      "Mar 17 16:04:24 ..... loading genome\n",
      "Mar 17 16:10:58 ..... started mapping\n",
      "Mar 17 16:15:42 ..... finished mapping\n",
      "Mar 17 16:15:47 ..... started sorting BAM\n",
      "Mar 17 16:18:00 ..... finished successfully\n",
      "\tSTAR --runMode alignReads --runThreadN 8 --genomeDir /data/hodges_lab/rheMac10_genome/rheMac10_GTF-ensembl.104_index_49 --readFilesIn /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/trimmed_reads/Macaque_Liver_R3_trimmed.fq --outFileNamePrefix /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/star-output/Macaque_Liver_R3_ensembl.104_ --outSAMtype BAM SortedByCoordinate\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Mar 17 16:18:00 ..... started STAR run\n",
      "Mar 17 16:18:00 ..... loading genome\n"
     ]
    }
   ],
   "source": [
    "## Map to transcriptiome with STAR aligner. \n",
    "DIR='/data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020'\n",
    "\n",
    "#Index paths\n",
    "HG38_50='/data/hodges_lab/hg38_genome/hg38_GTF-ensembl.104_index_49'\n",
    "RHEMAC10_50='/data/hodges_lab/rheMac10_genome/rheMac10_GTF-ensembl.104_index_49'\n",
    "\n",
    "#hg38\n",
    "for TISSUE in Heart Kidney Liver Lung \n",
    "do\n",
    "    for REP in R1 R2 R3 R4\n",
    "    do\n",
    "        EXPT=\"Human_${TISSUE}_${REP}\"\n",
    "        STAR --runMode alignReads --runThreadN 8 --genomeDir $HG38_50 \\\n",
    "        --readFilesIn ${DIR}/fastq/trimmed_reads/${EXPT}_trimmed.fq \\\n",
    "        --outFileNamePrefix ${DIR}/star-output/${EXPT}_ensembl.104_ --outSAMtype BAM SortedByCoordinate \n",
    "    done\n",
    "done\n",
    "\n",
    "#rheMac10\n",
    "for TISSUE in Heart Kidney Liver Lung \n",
    "do\n",
    "    for REP in R1 R2 R3 R4\n",
    "    do\n",
    "        EXPT=\"Macaque_${TISSUE}_${REP}\"\n",
    "        STAR --runMode alignReads --runThreadN 8 --genomeDir $RHEMAC10_50 \\\n",
    "        --readFilesIn ${DIR}/fastq/trimmed_reads/${EXPT}_trimmed.fq \\\n",
    "        --outFileNamePrefix ${DIR}/star-output/${EXPT}_ensembl.104_ --outSAMtype BAM SortedByCoordinate \n",
    "    done\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "72507d2b",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-03-18T14:23:49.916972Z",
     "start_time": "2022-03-18T13:46:27.509766Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\tSTAR --runMode alignReads --runThreadN 8 --genomeDir /data/hodges_lab/rheMac10_genome/rheMac10_GTF-ensembl.104_index_49 --readFilesIn /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/trimmed_reads/Macaque_Lung_R2_trimmed.fq --outFileNamePrefix /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/star-output/Macaque_Lung_R2_ensembl.104_ --outSAMtype BAM SortedByCoordinate\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Mar 18 08:46:28 ..... started STAR run\n",
      "Mar 18 08:46:29 ..... loading genome\n",
      "Mar 18 08:51:04 ..... started mapping\n",
      "Mar 18 08:55:36 ..... finished mapping\n",
      "Mar 18 08:55:40 ..... started sorting BAM\n",
      "Mar 18 08:57:40 ..... finished successfully\n",
      "\tSTAR --runMode alignReads --runThreadN 8 --genomeDir /data/hodges_lab/rheMac10_genome/rheMac10_GTF-ensembl.104_index_49 --readFilesIn /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/trimmed_reads/Macaque_Lung_R3_trimmed.fq --outFileNamePrefix /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/star-output/Macaque_Lung_R3_ensembl.104_ --outSAMtype BAM SortedByCoordinate\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Mar 18 08:57:41 ..... started STAR run\n",
      "Mar 18 08:57:41 ..... loading genome\n",
      "Mar 18 09:02:01 ..... started mapping\n",
      "Mar 18 09:08:03 ..... finished mapping\n",
      "Mar 18 09:08:07 ..... started sorting BAM\n",
      "Mar 18 09:10:59 ..... finished successfully\n",
      "\tSTAR --runMode alignReads --runThreadN 8 --genomeDir /data/hodges_lab/rheMac10_genome/rheMac10_GTF-ensembl.104_index_49 --readFilesIn /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/fastq/trimmed_reads/Macaque_Lung_R4_trimmed.fq --outFileNamePrefix /data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020/star-output/Macaque_Lung_R4_ensembl.104_ --outSAMtype BAM SortedByCoordinate\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Mar 18 09:11:00 ..... started STAR run\n",
      "Mar 18 09:11:00 ..... loading genome\n",
      "Mar 18 09:15:23 ..... started mapping\n",
      "Mar 18 09:21:13 ..... finished mapping\n",
      "Mar 18 09:21:17 ..... started sorting BAM\n",
      "Mar 18 09:23:49 ..... finished successfully\n"
     ]
    }
   ],
   "source": [
    "#stopped at Mac Lung R2 so I could go home (the above output is truncated). restart from there. \n",
    "## Map to transcriptiome with STAR aligner. \n",
    "DIR='/data/hodges_lab/public_data/human-rhesus_RNA-seq/blake_2020'\n",
    "\n",
    "#Index paths\n",
    "HG38_50='/data/hodges_lab/hg38_genome/hg38_GTF-ensembl.104_index_49'\n",
    "RHEMAC10_50='/data/hodges_lab/rheMac10_genome/rheMac10_GTF-ensembl.104_index_49'\n",
    "\n",
    "#rheMac10\n",
    "for TISSUE in Lung \n",
    "do\n",
    "    for REP in R2 R3 R4\n",
    "    do\n",
    "        EXPT=\"Macaque_${TISSUE}_${REP}\"\n",
    "        STAR --runMode alignReads --runThreadN 8 --genomeDir $RHEMAC10_50 \\\n",
    "        --readFilesIn ${DIR}/fastq/trimmed_reads/${EXPT}_trimmed.fq \\\n",
    "        --outFileNamePrefix ${DIR}/star-output/${EXPT}_ensembl.104_ --outSAMtype BAM SortedByCoordinate \n",
    "    done\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f2579fc1",
   "metadata": {},
   "source": [
    "## Count reads per gene with featureCounts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "25a9f0bf",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-03-18T15:37:15.702328Z",
     "start_time": "2022-03-18T14:56:45.318577Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "       \u001b[44;37m ========== \u001b[0m\u001b[36m    _____ _    _ ____  _____  ______          _____  \n",
      "       \u001b[44;37m =====      \u001b[0m\u001b[36m   / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "       \u001b[44;37m   =====    \u001b[0m\u001b[36m  | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "       \u001b[44;37m     ====   \u001b[0m\u001b[36m   \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "       \u001b[44;37m       ==== \u001b[0m\u001b[36m   ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "       \u001b[44;37m ========== \u001b[0m\u001b[36m  |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\u001b[0m\n",
      "\t  v2.0.0\n",
      "\n",
      "//==========================\u001b[36m featureCounts setting \u001b[0m===========================\\\\\n",
      "||  \u001b[0m                                                                          ||\n",
      "||             Input files : \u001b[36m2 BAM files  \u001b[0m \u001b[0m                                   ||\n",
      "||                           \u001b[32mo\u001b[36m GM12878_Rep1_RNA-seq_ensembl.104_Aligned.s\u001b[0m ... \u001b[0m||\n",
      "||                           \u001b[32mo\u001b[36m GM12878_Rep2_RNA-seq_ensembl.104_Aligned.s\u001b[0m ... \u001b[0m||\n",
      "||  \u001b[0m                                                                          ||\n",
      "||             Output file : \u001b[36mHuman_cross-tissue_RNA-seq_counts_XSAnno_PE.tsv\u001b[0m \u001b[0m ||\n",
      "||                 Summary : \u001b[36mHuman_cross-tissue_RNA-seq_counts_XSAnno_PE.\u001b[0m ...\u001b[0m \u001b[0m||\n",
      "||              Annotation : \u001b[36mEnsembl.v64.fullTransExon.hg19TorheMac2.hg38\u001b[0m ...\u001b[0m \u001b[0m||\n",
      "||      Dir for temp files : \u001b[36m/data/hodges_lab/ATAC-STARR_B-cells/data/RNA\u001b[0m ...\u001b[0m \u001b[0m||\n",
      "||  \u001b[0m                                                                          ||\n",
      "||                 Threads : \u001b[36m8\u001b[0m \u001b[0m                                               ||\n",
      "||                   Level : \u001b[36mmeta-feature level\u001b[0m \u001b[0m                              ||\n",
      "||              Paired-end : \u001b[36myes\u001b[0m \u001b[0m                                             ||\n",
      "||      Multimapping reads : \u001b[36mnot counted\u001b[0m \u001b[0m                                     ||\n",
      "|| Multi-overlapping reads : \u001b[36mcounted\u001b[0m \u001b[0m                                         ||\n",
      "||   Min overlapping bases : \u001b[36m1\u001b[0m \u001b[0m                                               ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "||          Chimeric reads : \u001b[36mcounted\u001b[0m \u001b[0m                                         ||\n",
      "||        Both ends mapped : \u001b[36mrequired\u001b[0m \u001b[0m                                        ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "\\\\============================================================================//\n",
      "\n",
      "//=================================\u001b[36m Running \u001b[0m==================================\\\\\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Load annotation file Ensembl.v64.fullTransExon.hg19TorheMac2.hg38.gtf \u001b[0m... \u001b[0m ||\n",
      "||    Features : \u001b[36m529754\u001b[0m \u001b[0m                                                      ||\n",
      "||    Meta-features : \u001b[36m27898\u001b[0m \u001b[0m                                                  ||\n",
      "||    Chromosomes/contigs : \u001b[36m25\u001b[0m \u001b[0m                                               ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Load FASTA contigs from hg38.fa... \u001b[0m                                        ||\n",
      "||    454 contigs were loaded \u001b[0m                                                ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file GM12878_Rep1_RNA-seq_ensembl.104_Aligned.sortedByCoor\u001b[0m ... \u001b[0m||\n",
      "||    Paired-end reads are included. \u001b[0m                                         ||\n",
      "||    Total alignments : \u001b[36m80947593\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m15329694 (18.9%)\u001b[0m \u001b[0m                    ||\n",
      "||    Running time : \u001b[36m3.87 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file GM12878_Rep2_RNA-seq_ensembl.104_Aligned.sortedByCoor\u001b[0m ... \u001b[0m||\n",
      "||    Paired-end reads are included. \u001b[0m                                         ||\n",
      "||    Total alignments : \u001b[36m67483119\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m13424254 (19.9%)\u001b[0m \u001b[0m                    ||\n",
      "||    Running time : \u001b[36m2.79 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Summary of counting results can be found in file \"/data/hodges_lab/ATAC-S \u001b[0m ||\n",
      "|| TARR_B-cells/data/RNA-seq/count_matricies/Human_cross-tissue_RNA-seq_coun \u001b[0m ||\n",
      "|| ts_XSAnno_PE.tsv.summary\" \u001b[0m                                                 ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "\\\\============================================================================//\n",
      "\n",
      "\n",
      "       \u001b[44;37m ========== \u001b[0m\u001b[36m    _____ _    _ ____  _____  ______          _____  \n",
      "       \u001b[44;37m =====      \u001b[0m\u001b[36m   / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "       \u001b[44;37m   =====    \u001b[0m\u001b[36m  | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "       \u001b[44;37m     ====   \u001b[0m\u001b[36m   \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "       \u001b[44;37m       ==== \u001b[0m\u001b[36m   ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "       \u001b[44;37m ========== \u001b[0m\u001b[36m  |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\u001b[0m\n",
      "\t  v2.0.0\n",
      "\n",
      "//==========================\u001b[36m featureCounts setting \u001b[0m===========================\\\\\n",
      "||  \u001b[0m                                                                          ||\n",
      "||             Input files : \u001b[36m19 BAM files  \u001b[0m \u001b[0m                                  ||\n",
      "||                           \u001b[32mo\u001b[36m Human_LCL_R1_ensembl.104_Aligned.sortedByC\u001b[0m ... \u001b[0m||\n",
      "||                           \u001b[32mo\u001b[36m Human_LCL_R2_ensembl.104_Aligned.sortedByC\u001b[0m ... \u001b[0m||\n",
      "||                           \u001b[32mo\u001b[36m Human_LCL_R3_ensembl.104_Aligned.sortedByC\u001b[0m ... \u001b[0m||\n",
      "||                           \u001b[32mo\u001b[36m Human_Heart_R1_ensembl.104_Aligned.sortedB\u001b[0m ... \u001b[0m||\n",
      "||                           \u001b[32mo\u001b[36m Human_Heart_R2_ensembl.104_Aligned.sortedB\u001b[0m ... \u001b[0m||\n",
      "||                           \u001b[32mo\u001b[36m Human_Heart_R3_ensembl.104_Aligned.sortedB\u001b[0m ... \u001b[0m||\n",
      "||                           \u001b[32mo\u001b[36m Human_Heart_R4_ensembl.104_Aligned.sortedB\u001b[0m ... \u001b[0m||\n",
      "||                           \u001b[32mo\u001b[36m Human_Kidney_R1_ensembl.104_Aligned.sorted\u001b[0m ... \u001b[0m||\n",
      "||                           \u001b[32mo\u001b[36m Human_Kidney_R2_ensembl.104_Aligned.sorted\u001b[0m ... \u001b[0m||\n",
      "||                           \u001b[32mo\u001b[36m Human_Kidney_R3_ensembl.104_Aligned.sorted\u001b[0m ... \u001b[0m||\n",
      "||                           \u001b[32mo\u001b[36m Human_Kidney_R4_ensembl.104_Aligned.sorted\u001b[0m ... \u001b[0m||\n",
      "||                           \u001b[32mo\u001b[36m Human_Liver_R1_ensembl.104_Aligned.sortedB\u001b[0m ... \u001b[0m||\n",
      "||                           \u001b[32mo\u001b[36m Human_Liver_R2_ensembl.104_Aligned.sortedB\u001b[0m ... \u001b[0m||\n",
      "||                           \u001b[32mo\u001b[36m Human_Liver_R3_ensembl.104_Aligned.sortedB\u001b[0m ... \u001b[0m||\n",
      "||                           \u001b[32mo\u001b[36m Human_Liver_R4_ensembl.104_Aligned.sortedB\u001b[0m ... \u001b[0m||\n",
      "||                           \u001b[32mo\u001b[36m Human_Lung_R1_ensembl.104_Aligned.sortedBy\u001b[0m ... \u001b[0m||\n",
      "||                           \u001b[32mo\u001b[36m Human_Lung_R2_ensembl.104_Aligned.sortedBy\u001b[0m ... \u001b[0m||\n",
      "||                           \u001b[32mo\u001b[36m Human_Lung_R3_ensembl.104_Aligned.sortedBy\u001b[0m ... \u001b[0m||\n",
      "||                           \u001b[32mo\u001b[36m Human_Lung_R4_ensembl.104_Aligned.sortedBy\u001b[0m ... \u001b[0m||\n",
      "||  \u001b[0m                                                                          ||\n",
      "||             Output file : \u001b[36mHuman_cross-tissue_RNA-seq_counts_XSAnno_SE.tsv\u001b[0m \u001b[0m ||\n",
      "||                 Summary : \u001b[36mHuman_cross-tissue_RNA-seq_counts_XSAnno_SE.\u001b[0m ...\u001b[0m \u001b[0m||\n",
      "||              Annotation : \u001b[36mEnsembl.v64.fullTransExon.hg19TorheMac2.hg38\u001b[0m ...\u001b[0m \u001b[0m||\n",
      "||      Dir for temp files : \u001b[36m/data/hodges_lab/ATAC-STARR_B-cells/data/RNA\u001b[0m ...\u001b[0m \u001b[0m||\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "||  \u001b[0m                                                                          ||\n",
      "||                 Threads : \u001b[36m8\u001b[0m \u001b[0m                                               ||\n",
      "||                   Level : \u001b[36mmeta-feature level\u001b[0m \u001b[0m                              ||\n",
      "||              Paired-end : \u001b[36mno\u001b[0m \u001b[0m                                              ||\n",
      "||      Multimapping reads : \u001b[36mnot counted\u001b[0m \u001b[0m                                     ||\n",
      "|| Multi-overlapping reads : \u001b[36mcounted\u001b[0m \u001b[0m                                         ||\n",
      "||   Min overlapping bases : \u001b[36m1\u001b[0m \u001b[0m                                               ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "\\\\============================================================================//\n",
      "\n",
      "//=================================\u001b[36m Running \u001b[0m==================================\\\\\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Load annotation file Ensembl.v64.fullTransExon.hg19TorheMac2.hg38.gtf \u001b[0m... \u001b[0m ||\n",
      "||    Features : \u001b[36m529754\u001b[0m \u001b[0m                                                      ||\n",
      "||    Meta-features : \u001b[36m27898\u001b[0m \u001b[0m                                                  ||\n",
      "||    Chromosomes/contigs : \u001b[36m25\u001b[0m \u001b[0m                                               ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Load FASTA contigs from hg38.fa... \u001b[0m                                        ||\n",
      "||    454 contigs were loaded \u001b[0m                                                ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file Human_LCL_R1_ensembl.104_Aligned.sortedByCoord.out.ba\u001b[0m ... \u001b[0m||\n",
      "||    Single-end reads are included. \u001b[0m                                         ||\n",
      "||    Total alignments : \u001b[36m18858544\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m4791984 (25.4%)\u001b[0m \u001b[0m                     ||\n",
      "||    Running time : \u001b[36m0.10 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file Human_LCL_R2_ensembl.104_Aligned.sortedByCoord.out.ba\u001b[0m ... \u001b[0m||\n",
      "||    Single-end reads are included. \u001b[0m                                         ||\n",
      "||    Total alignments : \u001b[36m22573726\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m5308136 (23.5%)\u001b[0m \u001b[0m                     ||\n",
      "||    Running time : \u001b[36m0.12 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file Human_LCL_R3_ensembl.104_Aligned.sortedByCoord.out.ba\u001b[0m ... \u001b[0m||\n",
      "||    Single-end reads are included. \u001b[0m                                         ||\n",
      "||    Total alignments : \u001b[36m24602752\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m5194340 (21.1%)\u001b[0m \u001b[0m                     ||\n",
      "||    Running time : \u001b[36m0.13 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file Human_Heart_R1_ensembl.104_Aligned.sortedByCoord.out.\u001b[0m ... \u001b[0m||\n",
      "||    Single-end reads are included. \u001b[0m                                         ||\n",
      "||    Total alignments : \u001b[36m55517344\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m19621478 (35.3%)\u001b[0m \u001b[0m                    ||\n",
      "||    Running time : \u001b[36m0.30 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file Human_Heart_R2_ensembl.104_Aligned.sortedByCoord.out.\u001b[0m ... \u001b[0m||\n",
      "||    Single-end reads are included. \u001b[0m                                         ||\n",
      "||    Total alignments : \u001b[36m56452859\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m10605710 (18.8%)\u001b[0m \u001b[0m                    ||\n",
      "||    Running time : \u001b[36m0.29 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file Human_Heart_R3_ensembl.104_Aligned.sortedByCoord.out.\u001b[0m ... \u001b[0m||\n",
      "||    Single-end reads are included. \u001b[0m                                         ||\n",
      "||    Total alignments : \u001b[36m50807520\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m6834025 (13.5%)\u001b[0m \u001b[0m                     ||\n",
      "||    Running time : \u001b[36m0.27 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file Human_Heart_R4_ensembl.104_Aligned.sortedByCoord.out.\u001b[0m ... \u001b[0m||\n",
      "||    Single-end reads are included. \u001b[0m                                         ||\n",
      "||    Total alignments : \u001b[36m53498106\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m15536688 (29.0%)\u001b[0m \u001b[0m                    ||\n",
      "||    Running time : \u001b[36m0.30 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file Human_Kidney_R1_ensembl.104_Aligned.sortedByCoord.out\u001b[0m ... \u001b[0m||\n",
      "||    Single-end reads are included. \u001b[0m                                         ||\n",
      "||    Total alignments : \u001b[36m56569588\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m16401831 (29.0%)\u001b[0m \u001b[0m                    ||\n",
      "||    Running time : \u001b[36m0.31 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file Human_Kidney_R2_ensembl.104_Aligned.sortedByCoord.out\u001b[0m ... \u001b[0m||\n",
      "||    Single-end reads are included. \u001b[0m                                         ||\n",
      "||    Total alignments : \u001b[36m37303257\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m7268816 (19.5%)\u001b[0m \u001b[0m                     ||\n",
      "||    Running time : \u001b[36m0.19 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file Human_Kidney_R3_ensembl.104_Aligned.sortedByCoord.out\u001b[0m ... \u001b[0m||\n",
      "||    Single-end reads are included. \u001b[0m                                         ||\n",
      "||    Total alignments : \u001b[36m56702185\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m17321759 (30.5%)\u001b[0m \u001b[0m                    ||\n",
      "||    Running time : \u001b[36m0.34 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file Human_Kidney_R4_ensembl.104_Aligned.sortedByCoord.out\u001b[0m ... \u001b[0m||\n",
      "||    Single-end reads are included. \u001b[0m                                         ||\n",
      "||    Total alignments : \u001b[36m62206857\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m17990606 (28.9%)\u001b[0m \u001b[0m                    ||\n",
      "||    Running time : \u001b[36m0.36 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file Human_Liver_R1_ensembl.104_Aligned.sortedByCoord.out.\u001b[0m ... \u001b[0m||\n",
      "||    Single-end reads are included. \u001b[0m                                         ||\n",
      "||    Total alignments : \u001b[36m49112987\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m18159566 (37.0%)\u001b[0m \u001b[0m                    ||\n",
      "||    Running time : \u001b[36m0.27 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "|| Process BAM file Human_Liver_R2_ensembl.104_Aligned.sortedByCoord.out.\u001b[0m ... \u001b[0m||\n",
      "||    Single-end reads are included. \u001b[0m                                         ||\n",
      "||    Total alignments : \u001b[36m49741040\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m12168826 (24.5%)\u001b[0m \u001b[0m                    ||\n",
      "||    Running time : \u001b[36m0.25 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file Human_Liver_R3_ensembl.104_Aligned.sortedByCoord.out.\u001b[0m ... \u001b[0m||\n",
      "||    Single-end reads are included. \u001b[0m                                         ||\n",
      "||    Total alignments : \u001b[36m70248304\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m11886244 (16.9%)\u001b[0m \u001b[0m                    ||\n",
      "||    Running time : \u001b[36m0.37 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file Human_Liver_R4_ensembl.104_Aligned.sortedByCoord.out.\u001b[0m ... \u001b[0m||\n",
      "||    Single-end reads are included. \u001b[0m                                         ||\n",
      "||    Total alignments : \u001b[36m61456888\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m14464330 (23.5%)\u001b[0m \u001b[0m                    ||\n",
      "||    Running time : \u001b[36m0.34 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file Human_Lung_R1_ensembl.104_Aligned.sortedByCoord.out.b\u001b[0m ... \u001b[0m||\n",
      "||    Single-end reads are included. \u001b[0m                                         ||\n",
      "||    Total alignments : \u001b[36m37704498\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m12710326 (33.7%)\u001b[0m \u001b[0m                    ||\n",
      "||    Running time : \u001b[36m0.20 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file Human_Lung_R2_ensembl.104_Aligned.sortedByCoord.out.b\u001b[0m ... \u001b[0m||\n",
      "||    Single-end reads are included. \u001b[0m                                         ||\n",
      "||    Total alignments : \u001b[36m60791400\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m19239752 (31.6%)\u001b[0m \u001b[0m                    ||\n",
      "||    Running time : \u001b[36m0.33 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file Human_Lung_R3_ensembl.104_Aligned.sortedByCoord.out.b\u001b[0m ... \u001b[0m||\n",
      "||    Single-end reads are included. \u001b[0m                                         ||\n",
      "||    Total alignments : \u001b[36m60462143\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m20592858 (34.1%)\u001b[0m \u001b[0m                    ||\n",
      "||    Running time : \u001b[36m0.37 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file Human_Lung_R4_ensembl.104_Aligned.sortedByCoord.out.b\u001b[0m ... \u001b[0m||\n",
      "||    Single-end reads are included. \u001b[0m                                         ||\n",
      "||    Total alignments : \u001b[36m71715498\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m24860272 (34.7%)\u001b[0m \u001b[0m                    ||\n",
      "||    Running time : \u001b[36m0.44 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Summary of counting results can be found in file \"/data/hodges_lab/ATAC-S \u001b[0m ||\n",
      "|| TARR_B-cells/data/RNA-seq/count_matricies/Human_cross-tissue_RNA-seq_coun \u001b[0m ||\n",
      "|| ts_XSAnno_SE.tsv.summary\" \u001b[0m                                                 ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "\\\\============================================================================//\n",
      "\n",
      "\n",
      "       \u001b[44;37m ========== \u001b[0m\u001b[36m    _____ _    _ ____  _____  ______          _____  \n",
      "       \u001b[44;37m =====      \u001b[0m\u001b[36m   / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "       \u001b[44;37m   =====    \u001b[0m\u001b[36m  | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "       \u001b[44;37m     ====   \u001b[0m\u001b[36m   \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "       \u001b[44;37m       ==== \u001b[0m\u001b[36m   ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "       \u001b[44;37m ========== \u001b[0m\u001b[36m  |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\u001b[0m\n",
      "\t  v2.0.0\n",
      "\n",
      "//==========================\u001b[36m featureCounts setting \u001b[0m===========================\\\\\n",
      "||  \u001b[0m                                                                          ||\n",
      "||             Input files : \u001b[36m2 BAM files  \u001b[0m \u001b[0m                                   ||\n",
      "||                           \u001b[32mo\u001b[36m LCL8664_Rep1_RNA-seq_ensembl.104_Aligned.s\u001b[0m ... \u001b[0m||\n",
      "||                           \u001b[32mo\u001b[36m LCL8664_Rep2_RNA-seq_ensembl.104_Aligned.s\u001b[0m ... \u001b[0m||\n",
      "||  \u001b[0m                                                                          ||\n",
      "||             Output file : \u001b[36mMacaque_cross-tissue_RNA-seq_counts_XSAnno_P\u001b[0m ...\u001b[0m \u001b[0m||\n",
      "||                 Summary : \u001b[36mMacaque_cross-tissue_RNA-seq_counts_XSAnno_P\u001b[0m ...\u001b[0m \u001b[0m||\n",
      "||              Annotation : \u001b[36mEnsembl.v64.fullTransExon.hg19TorheMac2.rheM\u001b[0m ...\u001b[0m \u001b[0m||\n",
      "||      Dir for temp files : \u001b[36m/data/hodges_lab/ATAC-STARR_B-cells/data/RNA\u001b[0m ...\u001b[0m \u001b[0m||\n",
      "||  \u001b[0m                                                                          ||\n",
      "||                 Threads : \u001b[36m8\u001b[0m \u001b[0m                                               ||\n",
      "||                   Level : \u001b[36mmeta-feature level\u001b[0m \u001b[0m                              ||\n",
      "||              Paired-end : \u001b[36myes\u001b[0m \u001b[0m                                             ||\n",
      "||      Multimapping reads : \u001b[36mnot counted\u001b[0m \u001b[0m                                     ||\n",
      "|| Multi-overlapping reads : \u001b[36mcounted\u001b[0m \u001b[0m                                         ||\n",
      "||   Min overlapping bases : \u001b[36m1\u001b[0m \u001b[0m                                               ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "||          Chimeric reads : \u001b[36mcounted\u001b[0m \u001b[0m                                         ||\n",
      "||        Both ends mapped : \u001b[36mrequired\u001b[0m \u001b[0m                                        ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "\\\\============================================================================//\n",
      "\n",
      "//=================================\u001b[36m Running \u001b[0m==================================\\\\\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Load annotation file Ensembl.v64.fullTransExon.hg19TorheMac2.rheMac10.\u001b[0m ... \u001b[0m||\n",
      "||    Features : \u001b[36m526805\u001b[0m \u001b[0m                                                      ||\n",
      "||    Meta-features : \u001b[36m27762\u001b[0m \u001b[0m                                                  ||\n",
      "||    Chromosomes/contigs : \u001b[36m31\u001b[0m \u001b[0m                                               ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Load FASTA contigs from rheMac10.fa... \u001b[0m                                    ||\n",
      "||    2938 contigs were loaded \u001b[0m                                               ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file LCL8664_Rep1_RNA-seq_ensembl.104_Aligned.sortedByCoor\u001b[0m ... \u001b[0m||\n",
      "||    Paired-end reads are included. \u001b[0m                                         ||\n",
      "||    Total alignments : \u001b[36m115968397\u001b[0m \u001b[0m                                           ||\n",
      "||    Successfully assigned alignments : \u001b[36m14508128 (12.5%)\u001b[0m \u001b[0m                    ||\n",
      "||    Running time : \u001b[36m10.51 minutes\u001b[0m \u001b[0m                                           ||\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file LCL8664_Rep2_RNA-seq_ensembl.104_Aligned.sortedByCoor\u001b[0m ... \u001b[0m||\n",
      "||    Paired-end reads are included. \u001b[0m                                         ||\n",
      "||    Total alignments : \u001b[36m83772041\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m11363843 (13.6%)\u001b[0m \u001b[0m                    ||\n",
      "||    Running time : \u001b[36m6.27 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Summary of counting results can be found in file \"/data/hodges_lab/ATAC-S \u001b[0m ||\n",
      "|| TARR_B-cells/data/RNA-seq/count_matricies/Macaque_cross-tissue_RNA-seq_co \u001b[0m ||\n",
      "|| unts_XSAnno_PE.tsv.summary\" \u001b[0m                                               ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "\\\\============================================================================//\n",
      "\n",
      "\n",
      "       \u001b[44;37m ========== \u001b[0m\u001b[36m    _____ _    _ ____  _____  ______          _____  \n",
      "       \u001b[44;37m =====      \u001b[0m\u001b[36m   / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "       \u001b[44;37m   =====    \u001b[0m\u001b[36m  | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "       \u001b[44;37m     ====   \u001b[0m\u001b[36m   \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "       \u001b[44;37m       ==== \u001b[0m\u001b[36m   ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "       \u001b[44;37m ========== \u001b[0m\u001b[36m  |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\u001b[0m\n",
      "\t  v2.0.0\n",
      "\n",
      "//==========================\u001b[36m featureCounts setting \u001b[0m===========================\\\\\n",
      "||  \u001b[0m                                                                          ||\n",
      "||             Input files : \u001b[36m19 BAM files  \u001b[0m \u001b[0m                                  ||\n",
      "||                           \u001b[32mo\u001b[36m Rhesus_LCL_R1_ensembl.104_Aligned.sortedBy\u001b[0m ... \u001b[0m||\n",
      "||                           \u001b[32mo\u001b[36m Rhesus_LCL_R2_ensembl.104_Aligned.sortedBy\u001b[0m ... \u001b[0m||\n",
      "||                           \u001b[32mo\u001b[36m Rhesus_LCL_R3_ensembl.104_Aligned.sortedBy\u001b[0m ... \u001b[0m||\n",
      "||                           \u001b[32mo\u001b[36m Macaque_Heart_R1_ensembl.104_Aligned.sorte\u001b[0m ... \u001b[0m||\n",
      "||                           \u001b[32mo\u001b[36m Macaque_Heart_R2_ensembl.104_Aligned.sorte\u001b[0m ... \u001b[0m||\n",
      "||                           \u001b[32mo\u001b[36m Macaque_Heart_R3_ensembl.104_Aligned.sorte\u001b[0m ... \u001b[0m||\n",
      "||                           \u001b[32mo\u001b[36m Macaque_Heart_R4_ensembl.104_Aligned.sorte\u001b[0m ... \u001b[0m||\n",
      "||                           \u001b[32mo\u001b[36m Macaque_Kidney_R1_ensembl.104_Aligned.sort\u001b[0m ... \u001b[0m||\n",
      "||                           \u001b[32mo\u001b[36m Macaque_Kidney_R2_ensembl.104_Aligned.sort\u001b[0m ... \u001b[0m||\n",
      "||                           \u001b[32mo\u001b[36m Macaque_Kidney_R3_ensembl.104_Aligned.sort\u001b[0m ... \u001b[0m||\n",
      "||                           \u001b[32mo\u001b[36m Macaque_Kidney_R4_ensembl.104_Aligned.sort\u001b[0m ... \u001b[0m||\n",
      "||                           \u001b[32mo\u001b[36m Macaque_Liver_R1_ensembl.104_Aligned.sorte\u001b[0m ... \u001b[0m||\n",
      "||                           \u001b[32mo\u001b[36m Macaque_Liver_R2_ensembl.104_Aligned.sorte\u001b[0m ... \u001b[0m||\n",
      "||                           \u001b[32mo\u001b[36m Macaque_Liver_R3_ensembl.104_Aligned.sorte\u001b[0m ... \u001b[0m||\n",
      "||                           \u001b[32mo\u001b[36m Macaque_Liver_R4_ensembl.104_Aligned.sorte\u001b[0m ... \u001b[0m||\n",
      "||                           \u001b[32mo\u001b[36m Macaque_Lung_R1_ensembl.104_Aligned.sorted\u001b[0m ... \u001b[0m||\n",
      "||                           \u001b[32mo\u001b[36m Macaque_Lung_R2_ensembl.104_Aligned.sorted\u001b[0m ... \u001b[0m||\n",
      "||                           \u001b[32mo\u001b[36m Macaque_Lung_R3_ensembl.104_Aligned.sorted\u001b[0m ... \u001b[0m||\n",
      "||                           \u001b[32mo\u001b[36m Macaque_Lung_R4_ensembl.104_Aligned.sorted\u001b[0m ... \u001b[0m||\n",
      "||  \u001b[0m                                                                          ||\n",
      "||             Output file : \u001b[36mMacaque_cross-tissue_RNA-seq_counts_XSAnno_S\u001b[0m ...\u001b[0m \u001b[0m||\n",
      "||                 Summary : \u001b[36mMacaque_cross-tissue_RNA-seq_counts_XSAnno_S\u001b[0m ...\u001b[0m \u001b[0m||\n",
      "||              Annotation : \u001b[36mEnsembl.v64.fullTransExon.hg19TorheMac2.rheM\u001b[0m ...\u001b[0m \u001b[0m||\n",
      "||      Dir for temp files : \u001b[36m/data/hodges_lab/ATAC-STARR_B-cells/data/RNA\u001b[0m ...\u001b[0m \u001b[0m||\n",
      "||  \u001b[0m                                                                          ||\n",
      "||                 Threads : \u001b[36m8\u001b[0m \u001b[0m                                               ||\n",
      "||                   Level : \u001b[36mmeta-feature level\u001b[0m \u001b[0m                              ||\n",
      "||              Paired-end : \u001b[36mno\u001b[0m \u001b[0m                                              ||\n",
      "||      Multimapping reads : \u001b[36mnot counted\u001b[0m \u001b[0m                                     ||\n",
      "|| Multi-overlapping reads : \u001b[36mcounted\u001b[0m \u001b[0m                                         ||\n",
      "||   Min overlapping bases : \u001b[36m1\u001b[0m \u001b[0m                                               ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "\\\\============================================================================//\n",
      "\n",
      "//=================================\u001b[36m Running \u001b[0m==================================\\\\\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Load annotation file Ensembl.v64.fullTransExon.hg19TorheMac2.rheMac10.\u001b[0m ... \u001b[0m||\n",
      "||    Features : \u001b[36m526805\u001b[0m \u001b[0m                                                      ||\n",
      "||    Meta-features : \u001b[36m27762\u001b[0m \u001b[0m                                                  ||\n",
      "||    Chromosomes/contigs : \u001b[36m31\u001b[0m \u001b[0m                                               ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Load FASTA contigs from rheMac10.fa... \u001b[0m                                    ||\n",
      "||    2938 contigs were loaded \u001b[0m                                               ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file Rhesus_LCL_R1_ensembl.104_Aligned.sortedByCoord.out.b\u001b[0m ... \u001b[0m||\n",
      "||    Single-end reads are included. \u001b[0m                                         ||\n",
      "||    Total alignments : \u001b[36m23039794\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m5698672 (24.7%)\u001b[0m \u001b[0m                     ||\n",
      "||    Running time : \u001b[36m0.12 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file Rhesus_LCL_R2_ensembl.104_Aligned.sortedByCoord.out.b\u001b[0m ... \u001b[0m||\n",
      "||    Single-end reads are included. \u001b[0m                                         ||\n",
      "||    Total alignments : \u001b[36m24134614\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m6073513 (25.2%)\u001b[0m \u001b[0m                     ||\n",
      "||    Running time : \u001b[36m0.13 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file Rhesus_LCL_R3_ensembl.104_Aligned.sortedByCoord.out.b\u001b[0m ... \u001b[0m||\n",
      "||    Single-end reads are included. \u001b[0m                                         ||\n",
      "||    Total alignments : \u001b[36m22368237\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m6458505 (28.9%)\u001b[0m \u001b[0m                     ||\n",
      "||    Running time : \u001b[36m0.14 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file Macaque_Heart_R1_ensembl.104_Aligned.sortedByCoord.ou\u001b[0m ... \u001b[0m||\n",
      "||    Single-end reads are included. \u001b[0m                                         ||\n",
      "||    Total alignments : \u001b[36m89642755\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m10656758 (11.9%)\u001b[0m \u001b[0m                    ||\n",
      "||    Running time : \u001b[36m0.38 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file Macaque_Heart_R2_ensembl.104_Aligned.sortedByCoord.ou\u001b[0m ... \u001b[0m||\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "||    Single-end reads are included. \u001b[0m                                         ||\n",
      "||    Total alignments : \u001b[36m107671006\u001b[0m \u001b[0m                                           ||\n",
      "||    Successfully assigned alignments : \u001b[36m12430042 (11.5%)\u001b[0m \u001b[0m                    ||\n",
      "||    Running time : \u001b[36m0.44 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file Macaque_Heart_R3_ensembl.104_Aligned.sortedByCoord.ou\u001b[0m ... \u001b[0m||\n",
      "||    Single-end reads are included. \u001b[0m                                         ||\n",
      "||    Total alignments : \u001b[36m212315529\u001b[0m \u001b[0m                                           ||\n",
      "||    Successfully assigned alignments : \u001b[36m19732409 (9.3%)\u001b[0m \u001b[0m                     ||\n",
      "||    Running time : \u001b[36m0.94 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file Macaque_Heart_R4_ensembl.104_Aligned.sortedByCoord.ou\u001b[0m ... \u001b[0m||\n",
      "||    Single-end reads are included. \u001b[0m                                         ||\n",
      "||    Total alignments : \u001b[36m138677611\u001b[0m \u001b[0m                                           ||\n",
      "||    Successfully assigned alignments : \u001b[36m15905939 (11.5%)\u001b[0m \u001b[0m                    ||\n",
      "||    Running time : \u001b[36m0.65 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file Macaque_Kidney_R1_ensembl.104_Aligned.sortedByCoord.o\u001b[0m ... \u001b[0m||\n",
      "||    Single-end reads are included. \u001b[0m                                         ||\n",
      "||    Total alignments : \u001b[36m53483384\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m12304720 (23.0%)\u001b[0m \u001b[0m                    ||\n",
      "||    Running time : \u001b[36m0.25 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file Macaque_Kidney_R2_ensembl.104_Aligned.sortedByCoord.o\u001b[0m ... \u001b[0m||\n",
      "||    Single-end reads are included. \u001b[0m                                         ||\n",
      "||    Total alignments : \u001b[36m67014035\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m14958307 (22.3%)\u001b[0m \u001b[0m                    ||\n",
      "||    Running time : \u001b[36m0.32 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file Macaque_Kidney_R3_ensembl.104_Aligned.sortedByCoord.o\u001b[0m ... \u001b[0m||\n",
      "||    Single-end reads are included. \u001b[0m                                         ||\n",
      "||    Total alignments : \u001b[36m92103113\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m18459874 (20.0%)\u001b[0m \u001b[0m                    ||\n",
      "||    Running time : \u001b[36m0.57 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file Macaque_Kidney_R4_ensembl.104_Aligned.sortedByCoord.o\u001b[0m ... \u001b[0m||\n",
      "||    Single-end reads are included. \u001b[0m                                         ||\n",
      "||    Total alignments : \u001b[36m98087011\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m23166024 (23.6%)\u001b[0m \u001b[0m                    ||\n",
      "||    Running time : \u001b[36m0.49 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file Macaque_Liver_R1_ensembl.104_Aligned.sortedByCoord.ou\u001b[0m ... \u001b[0m||\n",
      "||    Single-end reads are included. \u001b[0m                                         ||\n",
      "||    Total alignments : \u001b[36m43751237\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m16434031 (37.6%)\u001b[0m \u001b[0m                    ||\n",
      "||    Running time : \u001b[36m0.21 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file Macaque_Liver_R2_ensembl.104_Aligned.sortedByCoord.ou\u001b[0m ... \u001b[0m||\n",
      "||    Single-end reads are included. \u001b[0m                                         ||\n",
      "||    Total alignments : \u001b[36m63069736\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m23674855 (37.5%)\u001b[0m \u001b[0m                    ||\n",
      "||    Running time : \u001b[36m0.29 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file Macaque_Liver_R3_ensembl.104_Aligned.sortedByCoord.ou\u001b[0m ... \u001b[0m||\n",
      "||    Single-end reads are included. \u001b[0m                                         ||\n",
      "||    Total alignments : \u001b[36m81352916\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m28308132 (34.8%)\u001b[0m \u001b[0m                    ||\n",
      "||    Running time : \u001b[36m0.42 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file Macaque_Liver_R4_ensembl.104_Aligned.sortedByCoord.ou\u001b[0m ... \u001b[0m||\n",
      "||    Single-end reads are included. \u001b[0m                                         ||\n",
      "||    Total alignments : \u001b[36m99293622\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m37101281 (37.4%)\u001b[0m \u001b[0m                    ||\n",
      "||    Running time : \u001b[36m0.63 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file Macaque_Lung_R1_ensembl.104_Aligned.sortedByCoord.out\u001b[0m ... \u001b[0m||\n",
      "||    Single-end reads are included. \u001b[0m                                         ||\n",
      "||    Total alignments : \u001b[36m47258728\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m17558469 (37.2%)\u001b[0m \u001b[0m                    ||\n",
      "||    Running time : \u001b[36m0.25 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file Macaque_Lung_R2_ensembl.104_Aligned.sortedByCoord.out\u001b[0m ... \u001b[0m||\n",
      "||    Single-end reads are included. \u001b[0m                                         ||\n",
      "||    Total alignments : \u001b[36m57404822\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m21097069 (36.8%)\u001b[0m \u001b[0m                    ||\n",
      "||    Running time : \u001b[36m0.31 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file Macaque_Lung_R3_ensembl.104_Aligned.sortedByCoord.out\u001b[0m ... \u001b[0m||\n",
      "||    Single-end reads are included. \u001b[0m                                         ||\n",
      "||    Total alignments : \u001b[36m78436893\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m28640023 (36.5%)\u001b[0m \u001b[0m                    ||\n",
      "||    Running time : \u001b[36m0.45 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file Macaque_Lung_R4_ensembl.104_Aligned.sortedByCoord.out\u001b[0m ... \u001b[0m||\n",
      "||    Single-end reads are included. \u001b[0m                                         ||\n",
      "||    Total alignments : \u001b[36m76389551\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m28901729 (37.8%)\u001b[0m \u001b[0m                    ||\n",
      "||    Running time : \u001b[36m0.45 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Summary of counting results can be found in file \"/data/hodges_lab/ATAC-S \u001b[0m ||\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "|| TARR_B-cells/data/RNA-seq/count_matricies/Macaque_cross-tissue_RNA-seq_co \u001b[0m ||\n",
      "|| unts_XSAnno_SE.tsv.summary\" \u001b[0m                                               ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "\\\\============================================================================//\n",
      "\n"
     ]
    }
   ],
   "source": [
    "HG38_FA='/data/hodges_lab/hg38_genome/hg38.fa'\n",
    "HG38_GTF='/data/hodges_lab/ATAC-STARR_B-cells/data/RNA-seq/Ensembl.v64.fullTransExon.hg19TorheMac2.hg38.gtf'\n",
    "RHEMAC10_FA='/data/hodges_lab/rheMac10_genome/rheMac10.fa'\n",
    "RHEMAC10_GTF='/data/hodges_lab/ATAC-STARR_B-cells/data/RNA-seq/Ensembl.v64.fullTransExon.hg19TorheMac2.rheMac10.gtf'\n",
    "INT_DIR='/data/hodges_lab/ATAC-STARR_B-cells/data/RNA-seq'\n",
    "PUB_DIR='/data/hodges_lab/public_data/human-rhesus_RNA-seq'\n",
    "\n",
    "#Seperate into two files, one for paired end reads (ours) the other for single end reads (public data). Join afterwards. \n",
    "# For the PE file, will need to assess if fragments or individual reads are counted. \n",
    "# For both files, do unstranded mapping since I'm not sure if the public data is forward or reverse stranded. \n",
    "\n",
    "#Human\n",
    "featureCounts -s 0 -p -B -O -T 8 -F GTF -G $HG38_FA -t exon -g gene_id \\\n",
    "        -a $HG38_GTF -o ${INT_DIR}/count_matricies/Human_cross-tissue_RNA-seq_counts_XSAnno_PE.tsv \\\n",
    "        ${INT_DIR}/STAR-output/GM12878_Rep{1..2}_RNA-seq_ensembl.104_Aligned.sortedByCoord.out.bam \n",
    "\n",
    "featureCounts -s 0 -B -O -T 8 -F GTF -G $HG38_FA -t exon -g gene_id \\\n",
    "        -a $HG38_GTF -o ${INT_DIR}/count_matricies/Human_cross-tissue_RNA-seq_counts_XSAnno_SE.tsv \\\n",
    "        ${PUB_DIR}/cain_2011_LCLs/star-output/Human_LCL_R{1..3}_ensembl.104_Aligned.sortedByCoord.out.bam \\\n",
    "        ${PUB_DIR}/blake_2020/star-output/Human_Heart_R{1..4}_ensembl.104_Aligned.sortedByCoord.out.bam \\\n",
    "        ${PUB_DIR}/blake_2020/star-output/Human_Kidney_R{1..4}_ensembl.104_Aligned.sortedByCoord.out.bam \\\n",
    "        ${PUB_DIR}/blake_2020/star-output/Human_Liver_R{1..4}_ensembl.104_Aligned.sortedByCoord.out.bam \\\n",
    "        ${PUB_DIR}/blake_2020/star-output/Human_Lung_R{1..4}_ensembl.104_Aligned.sortedByCoord.out.bam \n",
    "        \n",
    "#LCL8664\n",
    "featureCounts -s 0 -p -B -O -T 8 -F GTF -G $RHEMAC10_FA -t exon -g gene_id \\\n",
    "        -a $RHEMAC10_GTF -o ${INT_DIR}/count_matricies/Macaque_cross-tissue_RNA-seq_counts_XSAnno_PE.tsv \\\n",
    "        ${INT_DIR}/STAR-output/LCL8664_Rep{1..2}_RNA-seq_ensembl.104_Aligned.sortedByCoord.out.bam\n",
    "\n",
    "featureCounts -s 0 -B -O -T 8 -F GTF -G $RHEMAC10_FA -t exon -g gene_id \\\n",
    "        -a $RHEMAC10_GTF -o ${INT_DIR}/count_matricies/Macaque_cross-tissue_RNA-seq_counts_XSAnno_SE.tsv \\\n",
    "        ${PUB_DIR}/cain_2011_LCLs/star-output/Rhesus_LCL_R{1..3}_ensembl.104_Aligned.sortedByCoord.out.bam \\\n",
    "        ${PUB_DIR}/blake_2020/star-output/Macaque_Heart_R{1..4}_ensembl.104_Aligned.sortedByCoord.out.bam \\\n",
    "        ${PUB_DIR}/blake_2020/star-output/Macaque_Kidney_R{1..4}_ensembl.104_Aligned.sortedByCoord.out.bam \\\n",
    "        ${PUB_DIR}/blake_2020/star-output/Macaque_Liver_R{1..4}_ensembl.104_Aligned.sortedByCoord.out.bam \\\n",
    "        ${PUB_DIR}/blake_2020/star-output/Macaque_Lung_R{1..4}_ensembl.104_Aligned.sortedByCoord.out.bam "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "02556359",
   "metadata": {},
   "source": [
    "## PCA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "82ff666e",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-07-27T14:56:23.987474Z",
     "start_time": "2022-07-27T14:56:21.177Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1mRows: \u001b[22m\u001b[34m27898\u001b[39m \u001b[1mColumns: \u001b[22m\u001b[34m8\u001b[39m\n",
      "\n",
      "\u001b[36m\u001b[39m \u001b[1mColumn specification\u001b[22m \u001b[36m\u001b[39m\n",
      "\u001b[1mDelimiter:\u001b[22m \"\\t\"\n",
      "\u001b[31mchr\u001b[39m (5): gene_id, Chr, Start, End, Strand\n",
      "\u001b[32mdbl\u001b[39m (3): Length, GM12878_R1, GM12878_R2\n",
      "\n",
      "\n",
      "\u001b[36m\u001b[39m Use `spec()` to retrieve the full column specification for this data.\n",
      "\u001b[36m\u001b[39m Specify the column types or set `show_col_types = FALSE` to quiet this message.\n",
      "\n",
      "\u001b[1mRows: \u001b[22m\u001b[34m27898\u001b[39m \u001b[1mColumns: \u001b[22m\u001b[34m25\u001b[39m\n",
      "\n",
      "\u001b[36m\u001b[39m \u001b[1mColumn specification\u001b[22m \u001b[36m\u001b[39m\n",
      "\u001b[1mDelimiter:\u001b[22m \"\\t\"\n",
      "\u001b[31mchr\u001b[39m  (5): gene_id, Chr, Start, End, Strand\n",
      "\u001b[32mdbl\u001b[39m (20): Length, Human_LCL_R1, Human_LCL_R2, Human_LCL_R3, Human_Heart_R1, ...\n",
      "\n",
      "\n",
      "\u001b[36m\u001b[39m Use `spec()` to retrieve the full column specification for this data.\n",
      "\u001b[36m\u001b[39m Specify the column types or set `show_col_types = FALSE` to quiet this message.\n",
      "\n",
      "\u001b[1mRows: \u001b[22m\u001b[34m27762\u001b[39m \u001b[1mColumns: \u001b[22m\u001b[34m8\u001b[39m\n",
      "\n",
      "\u001b[36m\u001b[39m \u001b[1mColumn specification\u001b[22m \u001b[36m\u001b[39m\n",
      "\u001b[1mDelimiter:\u001b[22m \"\\t\"\n",
      "\u001b[31mchr\u001b[39m (5): gene_id, Chr, Start, End, Strand\n",
      "\u001b[32mdbl\u001b[39m (3): Length, LCL8664_R1, LCL8664_R2\n",
      "\n",
      "\n",
      "\u001b[36m\u001b[39m Use `spec()` to retrieve the full column specification for this data.\n",
      "\u001b[36m\u001b[39m Specify the column types or set `show_col_types = FALSE` to quiet this message.\n",
      "\n",
      "\u001b[1mRows: \u001b[22m\u001b[34m27762\u001b[39m \u001b[1mColumns: \u001b[22m\u001b[34m25\u001b[39m\n",
      "\n",
      "\u001b[36m\u001b[39m \u001b[1mColumn specification\u001b[22m \u001b[36m\u001b[39m\n",
      "\u001b[1mDelimiter:\u001b[22m \"\\t\"\n",
      "\u001b[31mchr\u001b[39m  (5): gene_id, Chr, Start, End, Strand\n",
      "\u001b[32mdbl\u001b[39m (20): Length, Macaque_LCL_R1, Macaque_LCL_R2, Macaque_LCL_R3, Macaque_He...\n",
      "\n",
      "\n",
      "\u001b[36m\u001b[39m Use `spec()` to retrieve the full column specification for this data.\n",
      "\u001b[36m\u001b[39m Specify the column types or set `show_col_types = FALSE` to quiet this message.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# r kernel\n",
    "suppressPackageStartupMessages(library(tidyverse))\n",
    "\n",
    "dir <- \"/data/hodges_lab/ATAC-STARR_B-cells/data/RNA-seq/count_matricies/\"\n",
    "\n",
    "#read in cts\n",
    "H_PE <- read_tsv(paste0(dir,\"Human_cross-tissue_RNA-seq_counts_XSAnno_PE.tsv\"), \n",
    "                     col_names = c(\"gene_id\", \"Chr\", \"Start\", \"End\", \"Strand\", \"Length\", \"GM12878_R1\", \"GM12878_R2\"), \n",
    "                     skip = 2) %>% dplyr::select(-Chr, -Start, -End, -Strand, -Length)\n",
    "                     \n",
    "H_SE <- read_tsv(paste0(dir,\"Human_cross-tissue_RNA-seq_counts_XSAnno_SE.tsv\"), \n",
    "                     col_names = c(\"gene_id\", \"Chr\", \"Start\", \"End\", \"Strand\", \"Length\", \n",
    "                     \"Human_LCL_R1\", \"Human_LCL_R2\", \"Human_LCL_R3\",\n",
    "                     \"Human_Heart_R1\", \"Human_Heart_R2\", \"Human_Heart_R3\", \"Human_Heart_R4\",\n",
    "                     \"Human_Kidney_R1\", \"Human_Kidney_R2\", \"Human_Kidney_R3\", \"Human_Kidney_R4\",\n",
    "                     \"Human_Liver_R1\", \"Human_Liver_R2\", \"Human_Liver_R3\", \"Human_Liver_R4\",\n",
    "                     \"Human_Lung_R1\", \"Human_Lung_R2\", \"Human_Lung_R3\", \"Human_Lung_R4\"), \n",
    "                     skip = 2) %>% dplyr::select(-Chr, -Start, -End, -Strand, -Length)\n",
    "\n",
    "M_PE <- read_tsv(paste0(dir,\"Macaque_cross-tissue_RNA-seq_counts_XSAnno_PE.tsv\"), \n",
    "                     col_names = c(\"gene_id\", \"Chr\", \"Start\", \"End\", \"Strand\", \"Length\", \"LCL8664_R1\", \"LCL8664_R2\"), \n",
    "                     skip = 2) %>% dplyr::select(-Chr, -Start, -End, -Strand, -Length)\n",
    "                     \n",
    "M_SE <- read_tsv(paste0(dir,\"Macaque_cross-tissue_RNA-seq_counts_XSAnno_SE.tsv\"), \n",
    "                     col_names = c(\"gene_id\", \"Chr\", \"Start\", \"End\", \"Strand\", \"Length\", \n",
    "                     \"Macaque_LCL_R1\", \"Macaque_LCL_R2\", \"Macaque_LCL_R3\",\n",
    "                     \"Macaque_Heart_R1\", \"Macaque_Heart_R2\", \"Macaque_Heart_R3\", \"Macaque_Heart_R4\",\n",
    "                     \"Macaque_Kidney_R1\", \"Macaque_Kidney_R2\", \"Macaque_Kidney_R3\", \"Macaque_Kidney_R4\",\n",
    "                     \"Macaque_Liver_R1\", \"Macaque_Liver_R2\", \"Macaque_Liver_R3\", \"Macaque_Liver_R4\",\n",
    "                     \"Macaque_Lung_R1\", \"Macaque_Lung_R2\", \"Macaque_Lung_R3\", \"Macaque_Lung_R4\"), \n",
    "                     skip = 2) %>% dplyr::select(-Chr, -Start, -End, -Strand, -Length)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "707fa8fd",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-07-27T14:56:24.099981Z",
     "start_time": "2022-07-27T14:56:21.180Z"
    }
   },
   "outputs": [],
   "source": [
    "#join counts and assign gene_id to row_names\n",
    "cts <- inner_join(H_PE, H_SE, by = \"gene_id\") %>%\n",
    "    inner_join(M_PE, by = \"gene_id\") %>%\n",
    "    inner_join(M_SE, by = \"gene_id\") %>% \n",
    "    column_to_rownames(var = \"gene_id\") %>%\n",
    "    as.matrix()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "01cd1459",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-07-27T14:56:24.172539Z",
     "start_time": "2022-07-27T14:56:21.184Z"
    }
   },
   "outputs": [],
   "source": [
    "# Prepare dataframe of sample info\n",
    "RNames <- c(\"GM12878_R1\", \"GM12878_R2\",\n",
    "             \"Human_LCL_R1\", \"Human_LCL_R2\", \"Human_LCL_R3\",\n",
    "             \"Human_Heart_R1\", \"Human_Heart_R2\", \"Human_Heart_R3\", \"Human_Heart_R4\",\n",
    "             \"Human_Kidney_R1\", \"Human_Kidney_R2\", \"Human_Kidney_R3\", \"Human_Kidney_R4\",\n",
    "             \"Human_Liver_R1\", \"Human_Liver_R2\", \"Human_Liver_R3\", \"Human_Liver_R4\",\n",
    "             \"Human_Lung_R1\", \"Human_Lung_R2\", \"Human_Lung_R3\", \"Human_Lung_R4\", \n",
    "             \"LCL8664_R1\", \"LCL8664_R2\", \n",
    "             \"Macaque_LCL_R1\", \"Macaque_LCL_R2\", \"Macaque_LCL_R3\",\n",
    "             \"Macaque_Heart_R1\", \"Macaque_Heart_R2\", \"Macaque_Heart_R3\", \"Macaque_Heart_R4\",\n",
    "             \"Macaque_Kidney_R1\", \"Macaque_Kidney_R2\", \"Macaque_Kidney_R3\", \"Macaque_Kidney_R4\",\n",
    "             \"Macaque_Liver_R1\", \"Macaque_Liver_R2\", \"Macaque_Liver_R3\", \"Macaque_Liver_R4\",\n",
    "             \"Macaque_Lung_R1\", \"Macaque_Lung_R2\", \"Macaque_Lung_R3\", \"Macaque_Lung_R4\")\n",
    "\n",
    "species <- c(\"Human\", \"Human\",\n",
    "             \"Human\", \"Human\", \"Human\",\n",
    "             \"Human\", \"Human\", \"Human\", \"Human\",\n",
    "             \"Human\", \"Human\", \"Human\", \"Human\",\n",
    "             \"Human\", \"Human\", \"Human\", \"Human\",\n",
    "             \"Human\", \"Human\", \"Human\", \"Human\", \n",
    "             \"Macaque\", \"Macaque\", \n",
    "             \"Macaque\", \"Macaque\", \"Macaque\",\n",
    "             \"Macaque\", \"Macaque\", \"Macaque\", \"Macaque\",\n",
    "             \"Macaque\", \"Macaque\", \"Macaque\", \"Macaque\",\n",
    "             \"Macaque\", \"Macaque\", \"Macaque\", \"Macaque\",\n",
    "             \"Macaque\", \"Macaque\", \"Macaque\", \"Macaque\")\n",
    "\n",
    "tissue <- c(\"GM12878\", \"GM12878\",\n",
    "             \"LCL\", \"LCL\", \"LCL\",\n",
    "             \"Heart\", \"Heart\", \"Heart\", \"Heart\",\n",
    "             \"Kidney\", \"Kidney\", \"Kidney\", \"Kidney\",\n",
    "             \"Liver\", \"Liver\", \"Liver\", \"Liver\",\n",
    "             \"Lung\", \"Lung\", \"Lung\", \"Lung\", \n",
    "             \"LCL8664\", \"LCL8664\",\n",
    "             \"LCL\", \"LCL\", \"LCL\",\n",
    "             \"Heart\", \"Heart\", \"Heart\", \"Heart\",\n",
    "             \"Kidney\", \"Kidney\", \"Kidney\", \"Kidney\",\n",
    "             \"Liver\", \"Liver\", \"Liver\", \"Liver\",\n",
    "             \"Lung\", \"Lung\", \"Lung\", \"Lung\")\n",
    "             \n",
    "replicates <- c(\"Rep1\", \"Rep2\",\n",
    "             \"Rep1\", \"Rep2\", \"Rep3\",\n",
    "             \"Rep1\", \"Rep2\", \"Rep3\", \"Rep4\", \n",
    "             \"Rep1\", \"Rep2\", \"Rep3\", \"Rep4\",\n",
    "             \"Rep1\", \"Rep2\", \"Rep3\", \"Rep4\",\n",
    "             \"Rep1\", \"Rep2\", \"Rep3\", \"Rep4\", \n",
    "             \"Rep1\", \"Rep2\",\n",
    "             \"Rep1\", \"Rep2\", \"Rep3\",\n",
    "             \"Rep1\", \"Rep2\", \"Rep3\", \"Rep4\", \n",
    "             \"Rep1\", \"Rep2\", \"Rep3\", \"Rep4\",\n",
    "             \"Rep1\", \"Rep2\", \"Rep3\", \"Rep4\",\n",
    "             \"Rep1\", \"Rep2\", \"Rep3\", \"Rep4\")\n",
    "\n",
    "coldata <- data.frame(row.names = RNames, species, tissue, replicates)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "5f8aa897",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-07-27T14:56:24.193189Z",
     "start_time": "2022-07-27T14:56:21.186Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] TRUE\n"
     ]
    }
   ],
   "source": [
    "#check that the coldata matches data. It should print TRUE. \n",
    "print(all(rownames(coldata) == colnames(cts))) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "ad7003bd",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-07-27T14:56:34.339779Z",
     "start_time": "2022-07-27T14:56:21.189Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: S4Vectors\n",
      "\n",
      "Loading required package: stats4\n",
      "\n",
      "Loading required package: BiocGenerics\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "\n",
      "The following objects are masked from package:dplyr:\n",
      "\n",
      "    combine, intersect, setdiff, union\n",
      "\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, append, as.data.frame, basename, cbind, colnames,\n",
      "    dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,\n",
      "    grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,\n",
      "    union, unique, unsplit, which.max, which.min\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "\n",
      "The following objects are masked from package:dplyr:\n",
      "\n",
      "    first, rename\n",
      "\n",
      "\n",
      "The following object is masked from package:tidyr:\n",
      "\n",
      "    expand\n",
      "\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Loading required package: IRanges\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "\n",
      "The following objects are masked from package:dplyr:\n",
      "\n",
      "    collapse, desc, slice\n",
      "\n",
      "\n",
      "The following object is masked from package:purrr:\n",
      "\n",
      "    reduce\n",
      "\n",
      "\n",
      "Loading required package: GenomicRanges\n",
      "\n",
      "Loading required package: GenomeInfoDb\n",
      "\n",
      "Loading required package: SummarizedExperiment\n",
      "\n",
      "Loading required package: MatrixGenerics\n",
      "\n",
      "Loading required package: matrixStats\n",
      "\n",
      "\n",
      "Attaching package: matrixStats\n",
      "\n",
      "\n",
      "The following object is masked from package:dplyr:\n",
      "\n",
      "    count\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: MatrixGenerics\n",
      "\n",
      "\n",
      "The following objects are masked from package:matrixStats:\n",
      "\n",
      "    colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,\n",
      "    colCounts, colCummaxs, colCummins, colCumprods, colCumsums,\n",
      "    colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,\n",
      "    colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,\n",
      "    colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,\n",
      "    colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,\n",
      "    colWeightedMeans, colWeightedMedians, colWeightedSds,\n",
      "    colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,\n",
      "    rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,\n",
      "    rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,\n",
      "    rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,\n",
      "    rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,\n",
      "    rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,\n",
      "    rowWeightedMads, rowWeightedMeans, rowWeightedMedians,\n",
      "    rowWeightedSds, rowWeightedVars\n",
      "\n",
      "\n",
      "Loading required package: Biobase\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: Biobase\n",
      "\n",
      "\n",
      "The following object is masked from package:MatrixGenerics:\n",
      "\n",
      "    rowMedians\n",
      "\n",
      "\n",
      "The following objects are masked from package:matrixStats:\n",
      "\n",
      "    anyMissing, rowMedians\n",
      "\n",
      "\n",
      "converting counts to integer mode\n",
      "\n",
      "Warning message in DESeqDataSet(se, design = design, ignoreRank):\n",
      "some variables in design formula are characters, converting to factors\n"
     ]
    }
   ],
   "source": [
    "#normalize counts with DESeq2:\n",
    "library(\"DESeq2\")\n",
    "dds <- DESeqDataSetFromMatrix(countData = cts,\n",
    "                              colData = coldata,\n",
    "                              design = ~ species + tissue)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "03506b3b",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-07-27T14:56:38.862399Z",
     "start_time": "2022-07-27T14:56:21.193Z"
    }
   },
   "outputs": [],
   "source": [
    "#VST normalize and use variance accross all samples\n",
    "vsd <- vst(dds, blind=TRUE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "1f4a82dd",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-07-27T14:57:17.362779Z",
     "start_time": "2022-07-27T14:57:16.477Z"
    }
   },
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAeAAAAFoCAIAAAAAVb93AAAACXBIWXMAABJ0AAASdAHeZh94\nAAAgAElEQVR4nO3deVgTR/8A8MkdINxXFOQSrCICIt6i4gVq+0o9URBB20o9oWJfUTxaqXer\nVntoqQeKV8GjWtBq1WotKFpBkSoiUvFALiPkItlkf3/s++aXF0K4kmVDvp+nT5/NZHdmdpN8\nHWZnZ2g4jiMAAADUQ+/oCgAAjN3IkSO3bdvW0bWgIgjQAACyDRs2bMeOHaqXoaGhPj4+HVgf\nymJ2dAUAAMZuxYoVHV0FioIWNACAVJGRkTdu3IiPj6fRaHw+H/1vF0d6erq3tzeXy3VwcIiM\njNSS6Obmlp6ersqWx+P98ccfCCEMw1auXOnk5MTj8UaMGJGfn0/q6ekUtKABAKQ6fPhwaWnp\n1KlT4+LiGrxVXV0dERGRkpIyatSo6urqnJycphK1WLNmzaVLl44dO9a1a9e0tLSQkJCioiIL\nCwt9nY8+UStA//XXXzU1NS3fXy6XM5lMGo2mvyo1oFAoGAwGacURJdJoNDqdvL91iIE9ZF5V\npVKJ4zj5F5bMEnEcxzCsc3xdHRwcfH19dZ4tQqi8vFypVI4fP97Ozs7JyYkoRWNiU8Ri8fbt\n22/fvt27d2+E0Jo1a06dOpWVlTVjxgx9VFjfqBWgjx8/PmXKlBbujON4bW2tmZkZk0neWUgk\nEi6XS+ZvTCgUslgsDodDWokYhuE4zmKxSCuxvr5eLpfzeDzSSsRxXCqVmpiYkFaiQqEQCoUW\nFhZkfnnEYrGpqanOsz127JieAnSvXr1GjBjh5eU1ceLE8ePHT5kyhcvlakxsKofHjx9LpdIG\ntxyfPn2qj9qSgFoBmsViDRgwoIU74zheXV1taWlJZigRCoVmZmZk/sYEAgGbzdbHz6wpMpkM\nx3Ey/0kQi8UymczKyoq0EnEcF4lEZP6TgGGYQCCwsbEh84+huro6c3NznWf7888/6zxPAp1O\n//XXX69du5aZmblmzZqNGzfm5uaamJhoTGxwJRUKBUJIKBQihKqrq21sbPRUSTLBTUIAANnY\nbDYRTxuj0+kjR47csmVLfn7+o0ePcnNzm0q0t7dX9YiWlZVJpVKEkLe3N5PJzMrKIutU9AsC\nNACAbG5ubn/88cfLly8b3HPKzc3dtGlTfn7+ixcvTpw4QaPRPDw8NCYihEaOHJmamioWi8Vi\ncUJCAtGgtra2XrBgQXx8/NGjR0tKSnJychISEgoLCzvmPNsNAjQAgGwJCQnPnj1zd3f39vZW\nT7ewsLh8+fKYMWM8PT2//vrr9PR0Z2dnjYkIocTEREdHRxcXF39//7CwMNUdha+++mrx4sWJ\niYk9e/acNm1aRUWFnZ1dB5ykLtAoNRdHUlJScnJyC3eGPmg9gT5ofehMfdCt+p2C9oAWNAAA\nUBQEaAAAoCgI0KDjfT0r9bs5Rzu6FgBQjl7GQdfX1+/Zsyc3N5dOp7/33ntTp05FCJWUlOzc\nuVMkEjk5OSUkJOijawwAADoTvbSg9+/fj2HYvn37Dhw4MGzYMCJx165d4eHhKSkp7u7uaWlp\n+igXGKKtU35osAEAIOi+BS2VSi9duvTjjz8SgyuI2arKy8urqqoGDRqEEBo/fnxCQkJsbKzO\niwadTOOQvTzjww6pCQAdQvcB+tWrVxYWFhkZGffv3+fxeDExMR4eHlVVVXZ2dsToNHt7+9ra\nWrlcTkRwqVT64Yf/+dXxeDyBQNCq4oRCIcnT+mAYRlpxCCGFQiGVSmUyGWklEiMvJRIJCWUR\nXc9EiRiGbZz03ccHZ6rebXypG+zQHkqlsrVftvYgzvHt27ckT5akj3Mk+SdgzHQfoBUKRVVV\nlYeHx9y5c7Ozszds2JCSkoLjeFPfSwaDMWbMGGI7Ly+vVcNvMQxjsVhkzkkmk8lYLBaZvzGJ\nRMJkMskc661QKHAcJ2cKKmJQMDGbHbGt+gJ8PStV45BhnQzQxnFcLpez2ez2Z9VCSqVSIpFw\nOBwyvzz19fX6GM9O5lBuI6f7H6G9vT1CKCgoCCE0ePDgrVu31tbW2tvbV1ZWEmG6srLSwsJC\nFXFYLNacOXOI7cePH7d8gjHiWQMOh0Ny8DIxMSH5N8ZkMsmcd420B1W2TvmB+KkTrUtie3fk\nYaIfo6kooJNLgeO4Uqkk86piGEZMhUhmdMMwTB/nCAGaNLq/0JaWlj4+Pnfv3kUI3bt3z8LC\nwtzcnM/n29raEjNtZ2VlDRkyROflAsOi5ZYg3C0EgKCXfwkXLlx4/Pjx2NjYAwcOJCYmEu3N\nJUuWHDlyZO7cuSUlJbNmzdJHuaAzgfuBAOiln9HJyWnr1q0NEj08PHbt2qWP4oAhUo+/LZ+L\nA6I2MCrQlwSoa3nGh+oRGaIzMDbUWlEFgMYgLgOjBS1oAACgKAjQAABAURCgAQCAoqAPGgBA\nOYK3kpfltQjHnbpaWVpwO7o6HQZa0AAAChEK6/cezJ75waFP1/786bpz4fNS9x7MFgrr25ab\nRCJJSEhwdXX18PDo3bt3fHx8bW0tQojH4w0cOFC1W2xsLI1GKygoQAgtXrzYxcWFRqNVVVUR\n71ZUVLz77rt8Pt/Z2TkqKkosFhPpJ06c8PHx8fHxCQwMJJ7Ce/78ufN/WVtb+/n5NbVnC0GA\nBgBQhUKh/ObHG+cvPXTtZu3uauvuauPmYpP129/f7ruhUCjbkOGMGTOeP3+en59fUlJy584d\nd3f3ly9fEm/J5fJHjx4hhOrr6y9fvuzi4kKkT58+/ebNm2ZmZqpMlErl/Pnznz9//uTJk7dv\n327atIk4PCYm5syZMwUFBfHx8UuWLEEIOTs7P/+vsLCw6dOnN7VnC0GABgBQxe27ZddzSrp2\nsWCx/jMDGpNJd+JbXst+knv3WWtzu3fv3vXr11NSUohnoLhc7pIlS3r27Em8GxUVlZqaihA6\ne/ZsSEiIakqfoKCgLl26qOfD5/Pfe+89JpPJ4XBGjBhRVlaGEFIqlXQ6XSQSIYSEQmHXrl3V\nDxGLxadOnYqKimp2T+2gDxoAQBUPiyvMzRpO+Eej0czNuI8eVw4KdGtVbnl5eb17925q7fap\nU6eOGzcuOTk5NTV19erVWVlZzWYolUoPHDjwxRdfIIQ4HE5aWlpQUJCFhQWdTv/jjz/U90xP\nT+/fv3+3bt2a3VM7aEEDAKhCIpHTGRqmimQwaGJJq6dEJ2ZJJKSnp7u5uTk4OBw6dIhI4fF4\nAQEBGRkZpaWl/fv3bzY3DMNmzZoVEhLy3nvvIYRkMtnGjRuvXLlSVlaWlJQUERGhvvOBAwei\no6OJbe17agcBGgBAFdZWpnKZhr5mmVxpZdnqeVP9/f0fPHggFAoRQlOnTi0tLR03blx9/f/f\nb4yKivroo49mzmx+CQiFQhEZGeng4KCaZejmzZtisTggIAAhFBERcf36ddWqGs+ePbt79+7k\nyZOb3bNZEKABAFTR19epTiSVyRXqiTK5ok4o7efXrbW5+fn5DR069IMPPqipqUEIKRQKYgiH\nypgxY1avXh0TE6M9HxzH582bx+Vyv/vuO1Wii4tLSUlJcXExQujs2bNeXl6q9R8OHjw4depU\n1UzcWvZsFvRBAwCookd3+6gZgYdP3LG1MTMxYSGEJBJ5zRvx7OmBPTzt25DhiRMnkpKS/Pz8\n6HS6ra3t6NGjw8LCVO/S6fT4+PgGh8yfP/+XX34Ri8W+vr59+vS5cOFCbm7uwYMHu3TpQvQp\nBwcHHzp0yNXVdfv27RMnTkQIWVpaqi+EnZqaeuDAAdVLLXs2i6beTdPhkpKSkpOTW7gzjuPV\n1dWWlpZkrqgiFArNzMzIXFFFIBCw2WxTU1PSSiRtRRWVlk83qgUxzb/6zEpbp/zQ1ERLxHI8\nTd0+0gcMwwQCgY2NDZnLkdTV1Zmbm+s821b9TlsLx/E/cp5e+/PJ60ohQsjRnhc0pHvQIHcy\nf3TUAS1o0JlpidGAmmg0WtBgj6DBHhimxBHOYpK34igFQR806AxUq2Rp2QCGhcmkG3l0RhCg\nAQCAsiBAA4PXoI28dcoPjVPIrREAugEBGhg2jcG37MEr8msCgM5BgAZUtGndL5vW/dK2YzVG\nZ2hEA0MEozgAtajHZWJ7xbqJTe3cOOyqonPZg1fdendpdMT/H6JQKFacjm1nbQHQK2hBAwrR\n2Gpuc1O6AeibBgYHWtDAgLVqjLPGcAwDpQGVQQsaAAAoCgI0AABQFARoAACgKAjQgEI0DtjQ\nMoqjVaCvGRgcuEkIqIUIx80OsGub5RkfNrhVCFEbUBkEaEBFOg/NKkREJn+6UQDaALo4AACA\noqjVgsYwrK6urlWHiMViMmdAxzCMWOKMNAqFQiaTKRSK5nfVEaVSieN4y5dNaz+FQqFQKFr7\n0bcHjuPkl4gQEgqFZE48L5fL9XGOZH4bjRy1AjSTyWz5AhA4jtfX15uamsKKKrrVUSuq6GPt\nj6Z0yIoqMpmMx+N1ghVVGAxjn6aZNNDFAQAAFEWtFjQAACAMw27dUpaUIIToHh7MAQMQ00gj\nFbSggbGAqZEMgrKkRBgVJQwLkyQlSZKShGFhwqgo5ZMnbcuNx+OVl5erp0gkkk8++cTFxcXd\n3d3X1/e7777TuNuJEyd8fHx8fHwCAwNzcnKIRLFYPG/ePAcHhy5dumzatElLImHRokU0Gq2q\nqqptlUfQggadnnpcbrzyN6AUXCgUL1+uvHeP4e2NiDs9OK68fVv86admBw/SdHHPIDw83NTU\ntKCgwMLCorKycu/evY33kcvlMTEx9+7d6969e1pa2pIlS27duoUQWr58uUwmKysrY7FYpaWl\nxM4aExFC165dq6mpaee9HAjQwOhsnfJDQvoHHV0LoAGWlYVlZzPeeef/k2g0mpMTlp0t/+UX\n9owZ7cz//v37165dKysrI+4P29vbr1q1qvFuSqWSTqeLRCKEkFAo7Nq1K7Gxf//+f/75h4i5\nHh4eTSUihCQSyaeffvrzzz+fPHmyPRWGAA06M+jWMCyKwkKahUXjdLqlpeLBg/bnf/fu3d69\nezc7eofD4aSlpQUFBVlYWNDp9D/++AMh9OTJEzs7u02bNl29etXGxmbr1q3+/v4aExFCSUlJ\nsbGxDg4O7aww9EEDAKgCl8loGoch0mi4LgbmE6PRmyWTyTZu3HjlypWysrKkpKSIiAiEEIZh\nZWVlAQEBd+7cWbBgQVhYWFOJN2/efPDgQXR0dPsrDAEaAEAVdBcXXCxunI6LxXQXl/bn37dv\n3wcPHjT7rNnNmzfFYnFAQABCKCIi4vr16zKZrFu3bgih6dOnI4Tef//9V69eVVVVaUy8fv36\nvXv33Nzc3Nzc6uvrAwICCgsL21ZhCNAANAP6SUjDCgmhe3mhBk8/1tUxvLxYoaHtz9/X13fo\n0KEffvihQCBACFVWVq5fv77xbi4uLiUlJcXFxQihs2fPenl5sdlsBweHESNG/Prrrwihy5cv\n29nZ2draakxMSEh4+fJlaWlpaWkph8P566+/vL2921Zh6IMGnVnj6euIxBb+qYsgOpOL7ubG\nWbBANG8e3caGbmaGI4SLRHhNjWlKCuO/999aKyAggHh6s0uXLrm5uSdOnFi5cqWPjw+NRrOy\nslqyZInG3bZv3z5x4kSEkKWlZVpaGrHPnj17oqOj4+PjLS0tMzIyiCeKNSbqCq3l31QSJCUl\nJScnt3BnHMerq6stLS3hUW/d6qhHva2srPRXRIMBdi1/1FsVoNs5OA/DMIFAYGNj0wke9W7V\n77QNFA8eyH7+WfWgCvtf/2L07q2/4qgMWtDAKLQtvELzuUMwevc2MdaI3AD0QQOgWYPoDMEa\nkA8CNAAAUBQEaAA00NhehkY0IBkEaEA2Yr1BAwUxGpAJbhIC8qhCs2pDf2sPtgdEYUAREKAB\nSQyo4QzT3QGKgC4O0JEMKGoDQD4I0AAAQFHQxQH0SL2B/Ky02sXNtgMrA4DBgRY00JfG3RfP\nSqsbpFDzJiEAFAEBGpCEaD43jtEAgKZAFwcgj3oXB9F23rTuF2hEA9AUaEEDsq1YN1EVnVt1\nIAz5AMYGAjToYK0KuxCjgVGBAA30RWPfhSoRms8ANAsCNNCjBjG6qe7m7RsvtTxPiNTAeMBN\nQqBfGoNyO5vPcGsRGAloQQNK2L3t96begiYzMFoQoAHZmgq4cLcQEF4Ja/YXXFhz4+CaGwcP\nFPz6SljT0TXqMHrp4khKSnrx4gVCyNXVdfHixba2tgihkpKSnTt3ikQiJyenhIQEfaxlCagP\n7g0C7a4/v/9FzpHC6n/MWCYIIZFccuzhlVWDZgU592lDbjwer7i4mM/nq1IkEsmqVavS09MZ\nDIa5ufnHH3/88ccfN96NIvTSgl66dOn+/fv37dvn4eGRmppKJO7atSs8PDwlJcXd3V21jDkA\n6qARbeTKRTVf5Bx5JarpbtWVb2bNN7PubtX1lag6OSdNV+3o8PDwV69eFRQUPH369Lfffqup\noXTzXC8taHt7e2JDqVQSG+Xl5VVVVYMGDUIIjR8/PiEhITY2Vh9FA4rTeHNPLBbLZDIrK6sW\n7g86q6ynuYXV/3S36qqeaMO1+Lv6WebTm/P6jG9n/vfv37927VpZWRmPx0MI2dvbr1q1qp15\n6pW+RnEkJSUVFRU5OjquX78eIVRVVWVnZ0ej0RBC9vb2tbW1crmcxWIhhDAM+/nnn4mjJBKJ\nVCptVUEymUyhUOi6+k1SKBRSqZQ4EXIolUoMw1p7WdoDwzCEEI7jZJaoVCrJPEccx4mPkrQS\nicZKfX09mV8ePZ2jquGlcyWCV0TPRgM8lmnJ2/L253/37t3evXsT0dkg6CtAJycnKxSKgwcP\nnj59Ojo6Gsfxpr6XGIYdPHiQ2Obz+RKJpFUFkfyNVyqV+vt2NlUihmFk/iNEhGa5XE5miTiO\nt/ajbyfyP0qEkEQiIfnrqo+rqr/r1tTFwRGOdNFiILPZoRN6HAfNYDDGjRu3evXq6Ohoe3v7\nyspKIkxXVlZaWFgQzWeEEJfLPXPmDLGdlJRkbW3dwvxxHK+urjY3N1dlRQKhUGhmZkbmb0wg\nELDZbFNTU9JKlMlkOI5zOBzSStTSxdGArkZA4zguEonIbElhGCYQCKysrOh08oZO1dXV6eNu\nPJOpr7jhZe1UJxPzzRoGAaFM8o5Nt/bn37dv3wcPHgiFQkNpROv+u1JXV0cM4VAqlb/99pur\nqytCiM/n29ra5uTkIISysrKGDBmi83JBpwd3BTu9Ce4D/Ow9KsRv1BMrxAJfB/eJ3Qe2P39f\nX9+hQ4d++OGHAoEAIVRZWUn0wVKW7gO0VCrdtm3bnDlz5s2b9/Lly0WLFhHpS5YsOXLkyNy5\nc0tKSmbNmqXzcoGRgDDdidmaWKwfFt3dyunRm+cvhFUvhNWP3jzvbtX186HRtlyLtuUZEBDg\n7Ozs7Ozcv39/hNCJEyccHR19fHy6des2atQo1dC6BrtRhO7/VLG3t9++fXvjdA8Pj127dum8\nOGAkIC4biQBHr9QJn15+lvdE8BLHcU9rp1Eu/lacNvZICIXCBimmpqY7duzYsWOH9t0oAubi\nAIYH5uLo3Kw4vMlewzq6FpQAj3oDAwDNZ2CcIEADgwQhGxgDCNCA6nQyuRIAhggCNKA0iMLA\nmDUToGtra48fP75p0ybi5YMHDyoqKvRfKwBaBMI36Ny0jeLIz88PCQnp0qVLXl7eihUrEEIp\nKSkCgWD//v1kVQ8YOxitAYyZthb0kiVLFixYcPfuXVXK1KlTr169qvdKAQAA0B6g79y5ExER\noZ7C5/Nfvnyp5yoBAABASHuAtrCwaBCOc3Jyunbt2tT+AAAAdEhbgI6Ojl62bFlhYSFCqKam\n5vDhw3FxcXPnziWrbgAAYNS03ST8/PPPEUL9+vVDCNna2jKZzAULFqxcuZKkqgEAgHHTFqCZ\nTOaGDRuSkpIeP34sFAq9vb1bPlkzAACAdtIWoM+ePctisUJDQ/38/IiU8+fPKxSKiRNh5BMA\nAOidtj7oZcuWNVj9gU6n//vf/9ZzlQAAACCkPUD/888/Xl5e6imenp5PnjzRc5UAaDt4thB0\nJtoCtL29fX5+vnpKfn6+jY2NnqsEQBtBdAadjLYAPXv27IULF54+fVogEAgEgtOnTy9atCgq\nKoq0ygHQBhCmQafRzDA7DMNmzJghk8kQQmw2Oy4ujuJrLAKjBXEZdD7aAjSLxdq6devatWuL\niopoNJqXl5ehrFUOjE2D6AxrYhm0kjvP7p4vrHr2BiFk52rTN6SXRz+Xjq5Ux2hmTUKhUFhQ\nUFBRUaFUKv/55x8iMSwsTP8VA6AV8srL1V/6/3epZmBwsn/66+iac+Z2PK4JC0e0sgevfj90\nM/yzd4dMD2hDbjwer7i4mFi6u6qqis/nYxjWtooplcpNmzaR/KSetgB97ty52bNnYxhma2ur\nng4BGlBKeOyPDVLyysuhEW2Iyp9UHl1zzsHNjmPKIlJMzDkmFpxja8+5B3Tr4mnfgXWTy+Ub\nNmygUICOj4//8ssvY2JiaDQaaRUCACEUnHpQ/eWVqDla9nRs4i2I0Qan8PdiU3OuKjoT2CZs\nUwuTwt+LdRigL1++vGrVKrFYbGlp+f3333t7eyOEpk2bVlRUhGHYsGHDvv32WwaDIZVKHR0d\nlyxZ8vDhw4KCAqlUOmbMGEtLy4yMDF3VRDttAbqiogKiM6CC4NSDTcVoxz/b+BcroCDB61qW\nCatxOpvDFJS/bVue/fv3ZzAYCCGlUkmkVFRULFu27NKlS7a2ttnZ2TExMTdv3kQI7d6929HR\nEcfx+fPnHz16NDIyEiFUW1s7YsSI9evXYxhmZWV16dKlNp5bm2gL0KNGjbp+/frw4cNJqw0A\nqFHzWZWoMUa/HvI/32FVZ/RrPhMhtEIP1QP6w+IwlQpl43SlEmdxmrlh1pTc3Fz1PmiE0PXr\n1ysqKlST3dfV1REb6enpx44dQwhVVFRYWFgQAZrD4YwePbptRbeftnP28/ObPHnyzJkzPT09\niX+CCIsWLdJ/xQBotQa3CpHWpjegIGfvLpIDNy0dcPU/3HEcF9dJnb11duMXx3E/P7/MzEz1\nxJycnH379l29etXc3Pyrr74qKSkh0rlcbgf2Imh7UOXcuXNubm7Z2dmHDh06oIasugHQPFX8\nbRydgcHxCe7Rc5hHzYu3uBInUnAlXvPybc+hHn1G99RVKUFBQbdu3crOzkYIKZXK27dvI4Te\nvHnj4uJibm6OYdipU6caH8VkMhkMBvFQCGm0taCJegNAcVei5gSnHhTKZDw2G8EYO0PG4jCn\nrZlw9svf/jpXwDJhIYTLJVjfib3/tWx0m7s4GnN0dDx58mR8fLxYLMYwLCwsLDAwcOzYsQcO\nHJg0aZKpqWmvXr00Hrhw4cKAgAA+n09aTzQNx3FySmqJpKSk5OTkFu6M43h1dbWlpSWLpeGu\ngp4IhUIzMzMy/+QRCARsNtvU1JS0EmUyGY7jHA6HtBLFYrFMJrOyslKlNO6GbranwnrLZo2h\nWeOBOI6LRCIyH7zCMEwgENjY2DSYIVKv6urqzM3NdZ5tq36nbYGjF49eV5ZWI4Ts3Wyd3nFE\nxjpSQds/ShKJZMuWLRkZGWVlZXK5XJUuFAr1XzFg1IhGsfpL7fsTO+eVlzeI0Y5/YptK/v8h\nQxh1ZxhoyKmno1PPpsZPGhFtAToxMfHy5cvr168PCwu7ePFiXl7ezp07YckrQI7W3tzz5/Mb\ndEM3HoEHI6OBYdEWoNPT048ePRoUFIQQGjVq1JgxY0JDQ2NjYz/++GM91UapVEql0lYdIpPJ\nFAqFnurTmEKhkEqlZHZxKJVKDMNae1naQ6FQ4DhOZt8XhmFt+OhVQo4dJTZ8HRzyyst9HRxU\nb/k6OKhGvxKIUnAcJz7Ktla51Yhq1NfXk/nl0dM5NrikQH+0Beg3b944OzsjhKysrCorKx0d\nHX18fFQzcugDjUZTH8+nHRFB6HR6yw9pP6KGZP7GaDQayedIRGcyS6TT6a366NWNPZKm+jjy\nX7/2c3RECF2cFbEt+QLS9Cfy9o2XEpJCcBxvc4ntQafTyeyD1tM5wsNrpNEWoHv06PH48WN3\nd3d/f/9du3YlJiZmZWWx2Wz91YZGo7X8jh8RoJlMJpk3Cevr61ksFvkBmsxzJAI0mSXK5fJW\nffTqGnwWxEvtnxGLxcJxXCaTkXmOqoqRGaD19M2BAE0abd+VpUuXlpWVIYQ2bdq0Z88eHo83\nZ86cbdu2kVU3AJqhfiOR6IAm/h+cerCpvmbogwYGRFsLOjo6mtgYOHDgixcvSkpKnJyc9DFq\nBwAAQGMtHfvNZrN79tTZkzwAtJ968/mPZ8/Ut4e5uASnHryybg5Sm8sf2s7A4GgI0Nu2bfPx\n8QkNDW2qNyMhIUHPtQKgeapxeMGpB3mNbo2o3oW4DAyXhgB97tw5DMNCQ0PPnTun8RgI0IA6\nglMPNp6FA+blAJ2DhgB99epVhJBMJgsJCVm8eDGsQwiorKlYrJrHDia0A4aryVEcbDZ7w4YN\nZE4BAYCeaJxgGgDq03aTMCgo6Lfffhs7dixptQGgVbRH3pbHZWJPaGgDqtEWoPv27Ttt2rQp\nU6b06tWLy+Wq0mHCfkARqk6MZt9qSUcHdIYAqtEWoC9cuODp6Zmfn5+fn6+eDgEaUEqDqe+Q\n1sDdGHSAAMqCCftBZ9C45ds47GpsIDfYDRrRgFLImxYAgA4HjWVgWLQFaIlE8tlnn/n6+lpb\nW/PUkFY5ANqmPZ0bEMQBdWgL0ImJiRkZGevXrxcIBKdPn163bp21tfXWrVtJqxwAOteS+Asx\nmgoEUqmAxAm7qYlaE/YDoBMt6Uee8NMJ8ueDBs1S4Hja/XunHz58I5UihGxMTCa9884snz5M\nEqdppQ5qTdgPAGkyp02H/joKWn/t9+9u3+lqzjNjs2kIlQjeLL948UnNm89GjmxDbjwer7i4\nmM/nl5eXjx07dt68eXFxcVwut8FCMw8fPgwPD8/Ly9PNOeiOtn+UiAn7EULEhEG26YgAACAA\nSURBVP0ikSg9PV2vE/YDAIxZzovnu2/d8rK1seJyWXQ6k0634nC9bG2/yb2V/fx5m7N9/vx5\ncHDwokWL4uLiEELXrl3TXZX1CybsBwBQxR/PnllxTVj/25vBotOtTUyuP2vj3+6lpaXBwcGJ\niYnz588nUoYPH05sHDp0yMvLKyAgIDU1lUiRSqWWlpZr166dMWPGgAEDVEONL1++PHjwYD8/\nv+HDhxcWFiKEkpOTExMTiXdzcnL69+/ftuppBxP2AwCoolosYTM13BhgMxhVYnHb8pw4ceLm\nzZujoqIapD9//vzTTz+9ffu2k5PTvHnzVOm1tbVjxoz57LPPMjMz16xZk5mZWVFRsWzZskuX\nLtna2mZnZ8fExNy8efPDDz8MDAz8/PPPWSzWnj17VNFft7S1oE+fPl1fX09sExP2Q3QGAOiP\nBYeDaVoyXK5QWqnNNtEq48eP379/v1AobJB+48aNUaNGOTk5IYTUA7SpqSkxMsLHx6ekpAQh\ndP369YqKioiIiNDQ0M8++6yurg4h5OjoOHTo0NOnTwsEgvPnz8+cObNt1dNOW4CeO3euo6Nj\ndHR0VlYWhmH6KB4AAFQGOjm9kUgU/xujFUqloF4y0Mm5bXlu27bN39//3XffFTdqg6tW1FVf\nWle1TafTibiH47ifn9/5/yK6OBBCCxYs+P7771NTUydPnmxmZta26mmnLUC/fv36yJEjOI6H\nh4fz+fyPPvro8uXLCoVCH/UAAICRbm6zfPo8EbwRy//TIhTLsSeCNzN9+gS7ubU52927d3t6\neoaFham6BBBCQ4cOvXHjBhG1f/nlFy2HBwUF3bp1Kzs7GyGkVCpVHdPDhw+vrKzcsmWLnvo3\nkPYAzWKxJkyYcPDgwYqKih9//FEoFE6aNIn4iwAAAHSOTqNtHD166YCB5mzWvdev771+bc5m\nLR0wcOOo0XQarc3Z0mi0vXv38vn8KVOmyGQyItHZ2XnNmjWhoaFz5sx5+/atlsMdHR1PnjwZ\nHx/v6+vr4+Nz8uRJ1VvR0dEuLi6+vr5trpt2LVo0lsPhuLq6urm52dnZvXjxQk9VAQAAUxbr\n0yFDF/Uf8M/btwghNysrE2ZL17ZuTNX1TKfT1YdqEBuzZ8+ePXu2+v5cLlcgEBDbzs7OxcXF\nxPbw4cNzcnIa53/jxo2FCxe2uXrNaubhnPv3769evfqdd94JDAzMzc1dtWrV69ev9VcbAABA\nCJmyWL3s7HrZ2bUnOuvVP//806tXr/r6+hkzZuivFG0n36tXr0ePHg0bNmzp0qVTp051cHDQ\nXz0AAMCAuLq6/v333/ouRVuAnj9//vTp07t27arvSgAAAGhMW4AmHosEAADQIYxxgigAADAI\nEKABAICiIEADAABFkTeEpaSkZOfOnSKRyMnJKSEhAab1AAAA7ZpvQb9+/fqDDz7o0aNH9+7d\no6Ki2vygyq5du8LDw1NSUtzd3dPS0tqWCQAAGI/mA3RMTIyzs/OZM2cyMzN79eo1ffr0NhRT\nXl5eVVU1aNAghND48eNv3LjRhkwAAMCoaO7iWLBgwYYNG6ysrBBCRUVF6enppqamCKFly5bt\n3r27DcVUVVXZ2dnRaDSEkL29fW1trVwuJ2aNEovFqvmz+/btW1VV1aqctT9Erw9S0heyxDCs\n8URc+kbMqUim1n707Uf+R1lTU0NyierTA+mKXC7XeZ5AI80B2s3NrV+/fmvXro2KioqMjBw4\ncODEiRNpNNr58+fDw8PbUAyO47Qm5jrhcDibNm0its+fP9/yvmkcx4VCoampKZlLf0qlUg6H\n09S56INIJGKxWGSuNIZhGI7j6hMw6ptMJsMwjGgEkAPH8fr6em5bpxhuA6VSKRKJeDwemV8e\niURiYmKi82xhsV3SaA7Qn3766cyZM+Pi4lJSUr799tsRI0Zcu3ZNqVRu2rQpJCSkDcXY29tX\nVlYSYbqystLCwkL1+2cwGGPGjCG2r169yuFwWpgnEaBZLBaZoUQul5McoCUSCYPBaPllaT8a\njYbjOJklKhQKhUJBZok4jmMYRmaJxMzCbDabTuLq1DKZTB/nSOYpGLkmR3F069YtIyMjKytr\n2rRp77777tq1a9uzBDKfz7e1tc3JyRk8eHBWVtaQIUPanBUAABiJJv8lFAgE9+7dGzZs2N27\nd83MzPr27Zuent6ekpYsWXLkyJG5c+eWlJTMmjWrPVkBAIAx0NyCPnjwYFxcnKOjY1VVVUpK\nyrp16yIjIxcvXpySkkKsTdCGkjw8PHbt2tW+2gIAgBHR3IJeu3ZtXl7ew4cP//zzz6SkJISQ\np6dnVlbWhx9+OH78eHJrCAAARkpzgFYqlcS9BRMTE/VFCKdMmZKXl0dS1QAAwLhp7uJYuXJl\n7969PTw8nj59unPnTvW39LR4LQAAqMgqa6RPy3AcN+3uyrKz7ujqdBjNATo2NnbSpElPnz71\n9PSEhVQAAKTBBLX/bPuh7JtUGotJQ0gpx7otjHJN+JBpZdGG3Hg8XnFxMZ/PV6VwuVzyH1Bq\nsyaH2Tk4ODx48OCHH36oqKggXg4cODA4OBjGqAMA9ATHsMefbqzK+t3M25POZiOEcJn8ZWqG\nrKKq57frabpYn/DatWttO1ChUJAf/TT3QZeWlvr7+48bN+7w4cN37969e/fu4cOHx40b17dv\n39LSUnJrCAAwFjWXblSeuWji4UL/76OzNDbLxN2l8szF6ot/6KQIYmKJNWvWrFmzhki5c+eO\nv78/Qujy5cuDBw/28/MbPnx4YWEhQkgqlVpaWq5evXratGnnzp3TSQVaRXOA/uijj7p06VJW\nVvb3339fu3bt2rVrf//9d1lZGZ/Pnz9/PslVBAAYido795mW5jT6/zypS6PTmFYWdXcKdFhQ\nVFRUWloajuMIoUOHDs2ePbuiomLZsmXnzp3Lz8/fvHlzTEzMf6pUWztixIiffvpp0qRJOqxA\nC2n+k+HatWvZ2dlOTk7qiU5OTps2bRo6dCgpFQMAGB2FUIw09WPQmEysTqjDgoi7azdu3Bg0\naNBPP/10+/bt69evV1RUREREEDuoJgvjcDijR4/WYdGtojlA29raFhYW9u3bt0F6YWGhra2t\n/msFADBGbAdbVC9rnK6sl7EddBx5Zs+efejQodra2j59+nTp0gXHcT8/v8zMzAa7cblcMufe\naUBzgI6Li5s3b15OTk5wcLCDgwOO45WVlVeuXPnhhx+Sk5NJriIAwEhYjxz0NHk3y9GWrjbH\nk7K+Hnvz1iZ4sG7LCg8P//zzzysrK2fPno0QCgoKio2Nzc7OHjx4sFKp/OuvvwIDA3VbYhto\nDtDLly93dHT8+uuvv/vuO+JBFQaD4e/vv3fv3qioKHJrCAAwFuZ9e7slLijd/D2niz3DzBQh\npBCJZeWVbis+Ng/waVueAQEBxPR7Xbp0yc3NVaXb2NgMHjz44sWLhw8fRgg5OjqePHkyPj5e\nLBZjGBYWFkbdAI0QioqKioqKqq+vr6qqotFotra2ZM7NCAAwTq7LPjD1cqs4dUFa9gohZPqO\nh0PYOPtJY9uWm1DYsOdafRD0qVOn1N8aPnx4Tk6OegqXyxUIBG0rWie0jSuUSCRcLlf9VqFM\nJnv27FnbJksCAIDm0Wj2k8baTxqLyzEcx+ls8mZ7pyDNw+yqqqpCQkJ4PJ61tXViYqJM9p9u\n+8LCQi8vLxKrBwAwUjQW08ijM2qqBf3vf/+7qKjo2LFjtbW1mzdvvn379pkzZ8hckQgAAIDm\nFnRmZua2bdumTZs2b968O3fu0Gi00NBQ8lcRBQAAY6Y5QNfV1anmSDI3Nz979iyPxxs7diz5\nS2gDAIDR0hyge/bseffuXdVLDodz6tQpOzu7mTNnklUxAAAwdpoD9IwZM4ixgSocDufkyZP9\n+vUjpVYAAACaCNDLly+/detWg0Q2m/3zzz9LJBL91woAAEDTq3prRKPRuFyunqoCAABAXesC\nNAAAANJAgAYAAIqCAA0AABQFARoAACiqmQBtZWU1fvx41cvx48dbWVnpuUoAAAAQ0j6bHUJo\n3bp16rPZzZ07NyQkRM9VAgAAgFCzATouLk795bRp0/RZGQAAAP9PWxeHRCIhVr1VkclkxcXF\neq4SAAAAhGA+aAAAoCyYDxoAAChKc4DOzMzcvXv3lClTEELTp0+fMmVKaGjoL7/8ou/aKBSK\nxmuIaSeRSOrr6/VUn8bkcrlIJCKtOISQQqGQyWRKpZK0EpVKJY7jcrmctBIxDFMqla396NsD\nx3EMw0guESEkEoloNBppherpHImFpPVHqcBLiisqXtchhBwczT08HegM8i4apWgO0I3ng37/\n/ffHjh27efNmvdaGwWDweLwW7ozjuFQqNTExYbHIWxdHKBSamZmR/Btjs9lk/vkik8lwHCdz\njWCxWCyTyVr+0bcfjuMikYjMEjEMq6+vNzMzI1aYJkddXZ0+zpHBYOg8T5XK13UZx24X3HvB\n4TARjuplmI+v0+Twfg6OFm3IjcfjFRcX8/l8ndeTHDAfNACAKqRS+bFDN58WVzo7W9s7mNs7\nmjs7Wz99Unn80C2plLw/6agD5oMGAFDF/bvPix9VWNmaItXfqDRkZWP6pKgi/6+y9udfXl7u\n6elJbBcUFAQGBiKEpFKppaXl2rVrZ8yYMWDAgNu3bxM7HDp0qEePHoGBgStXrvT3929/6W0A\n80EDAKjixfM3JqYsGvqfLkQaopmYsl+UvdFfubW1tWPGjDl+/Pi6devWrFmDECorK1u5cuX1\n69dv375dU1Ojv6K10xygX79+vWTJkurqavXE6urqpUuXwtKxAAA9UWBKzTd4aAjD9Hhn0tTU\nNCgoCCHk4+NTUlKCEPrzzz+Dg4MdHR0RQjExMforWjvNAXrz5s21tbW2trbqiba2trW1tZs2\nbSKlYgAAo2Njx5PLNARiWT1ma6eDu51MJlM1IEr1eAdCSDXQgE6nYxjW/oJ0RXOA/uWXX2bN\nmtU4fdasWSQMtgMAGKc+/s4OXcwlkv+5HyiRyPldLH39ndufv62trVgsfvPmDULowoULWvYc\nMmTIlStXKisrEUKpqantL7ptNA+zKy0tVQ2zU2dvb19aWqrfGgEAjJWdPW90iPe+b6+b8Tgc\nLgshvL4eEwql700ebt+mYXYIoYCAAGJoY5cuXXJzczdv3hwSEuLl5dWtWzctR3Xr1m39+vVB\nQUF8Pt/f39/S0rJtpbeT5gBtY2NTVFTU+MZlUVGRjY2N/msFADBS/Qa48btY3r39T+XrOoSQ\nvaN530BXp27Wbcut8XM6c+bMmTNnjnoKl8sVCATEtrOzs2q6oRkzZkRHRyOEPvvss4EDB7at\nAu2kOUCPGzdu27ZtkydPZjL/fwe5XL5161aDmG60LjjY/MqVjq4FAKAtnLpZtzki61BSUtK1\na9cUCoWHh8cPP/zQIXXQHKDXrl0bGBgYGBi4bNmyXr164Tj+8OHDbdu2vXjxIj09neQqtoEi\nL6+jqwAAMGxffvllR1ehiZuEHh4eN27csLW1nTNnTv/+/QcMGDBnzhx7e/sbN264ubmRW8NW\nE1hbI4TqgoM7uiIAANAuTU7Y36tXr99++62qqurJkycIoe7du9vZ2ZFYsTYiojOCRjQAwPA1\nGaDv3LmTlZVVX18/fPjwsWPHklknXYGeaACAQdPcxZGenj5gwIBvv/32yJEj48aNW79+PcnV\nahtV85mgyMuDjg4AgOHSHKCTk5M//vjjFy9ePHny5JtvvtmwYQOlnq5pOejoAAAYLs0B+tGj\nR3PnziUeip83b159ff2zZ8/IrVirNWg+q0AjGgBgoDQHaKlUqpohnsPh0Ol0sVhMYq1arano\njKARDQAwWE3eJExMTFQ93ahUKpOSkqysrIiXBw4cIKFmOgR3CwEAhkhzgJ40aRKO46rHH//1\nr38hhFQvKcjqjR7nigUAgA6hOUCfPn2a5Hq0wagTy38avaKjawEAAPpC3vqVumW9632EUNj5\n9WMzIEYDADonQw3QhIKafzq6CgAAoC9N3iSkMqL5/MeLAjMmFyEUfDzhyoxtqncX//ukanvX\n5snkVw8AAHTCsFvQCKH8yhL1l+rRufFLAAAwIIYXoInms1AuQQiJ5FKEUF7Fk+DjCaiJcAwx\nGgBgoAysi4MIxER0JhDbRIx+bFWJEHpfMKmjqgcAADpkYC3ovIonDVIUSiVCSCiXqN46wjGA\nJQUAAKBZhhSg1Ts3CApcif4boyVSeQWtklgP+HFJ5eOSyg6qJgAA6IYhBWjt5DQ5nfE/pwMx\nGgBg0AymDzr4eIK/Q/cGiXdfFxNT7kmkckfMFiGETBDRiFaBkXYAAANlMAFaNdKZuE/YgB0R\nnRFCCJmYsLw87EmqFgAA6I1eAnRSUtKLFy8QQq6urosXL7a1tUUIlZSU7Ny5UyQSOTk5JSQk\nmJubtyHnxtFZKJfwWCZVzGr03zCtPooDms8AAMOllz7opUuX7t+/f9++fR4eHqmpqUTirl27\nwsPDU1JS3N3d09LS2l8KMWyDxzJBCJlwWQihKmY1jLEDAHQaegnQ9vb/6WFQKpXERnl5eVVV\n1aBBgxBC48ePv3Hjhm5L9HfobofZ2mG2p6zOqBKh+QwAMGj66oNOSkoqKipydHQkFpytqqqy\ns7MjbujZ29vX1tbK5XIWi4UQksvl33//PXFUXV2dSCTSkq1CoSA27lU9JTZwHCfSPVxtEEJP\nSmsUCsXWzyYghLRn1TZyuZzkxWWUSqVcLtfHuTSFuMhkrkKJYZhSqSTzHBFCJF9VorEiFouJ\nXwE5MAzTxzmqGl5A3/QVoJOTkxUKxcGDB0+fPh0dHY3jeFPfS6VS+ffff6teao8LRDjWku7h\nar1x4Tj9BRccx0lePxfHcaVSSWahxM+vqUutpxI75MKSWSJxPTEMIzNA6+kcIUCTRjcBOjMz\nk5jjPzY2NiAggEhkMBjjxo1bvXp1dHS0vb19ZWUlEaYrKystLCyI5jNCiMPhfPvtt8R2UlKS\nap0tja5H7CDuE/Z19MyreEJkqD78Tn1aO30QCoVmZmZk/sYEAgGbzVYtEUkCmUyG4ziHwyGt\nRLFYLJPJtH/0uoXjuEgk4vF4pJWIYZhAILCwsKDTyXv4oK6urm1347VjMg1m9Jeh082FnjBh\nwoQJE4jturq62tpaJycnpVL522+/ubq6IoT4fL6trW1OTs7gwYOzsrKGDBnS5rKuzNimGsvh\nY+MK3xUAQGel++gmlUq3bdtWU1NDp9N79OixaNEiIn3JkiXbt2//4YcfnJ2dly1b1p4iiGYy\njuPV1dVTL21UTwQAgE5D9wHa3t5++/btjdM9PDx27dql8+IuTtmk6i0BAIDOpPPMxQFAp3En\neOad4JkdXQvQ8aADFwAKUY/LxHa/K0c7rjqgg0ELGgCq0Nhqhqa0MYMADQAAFAUBGgAAKAoC\nNACGB+4iGgkI0AAYEvXQDGG6sZEjR27b1nkeiYAADQBVaBywoZ4IdxGbFRoa6uPj09G10BkY\nZgcAhcCgunZasWJFR1dBl6AFDQAgW3p6ure3N5fLdXBwiIyMJBIDAwMTExPHjRtnb2/v4eFx\n5sx/5nbHMGzlypVOTk48Hm/EiBH5+fmq9KSkpG7dunG5XG9v77Nnz6L/7eJo6kCNpVMTtKAB\nAKSqrq6OiIhISUkZNWpUdXV1Tk6O6q3du3f/8ssvw4cPP3Xq1IwZM4qKilxcXNasWXPp0qVj\nx4517do1LS0tJCSkqKjIwsJi1apVhw8f3r17t6+vb1FRUeM5JjUeKJfLmyqdgiBAA2Aw+l05\n2gl6nMvLy5VK5fjx4+3s7JycnHx9fVVvhYWFDR8+HCH0/vvvDx06dO/evStXrty+ffvt27d7\n9+6NEFqzZs2pU6eysrImTpy4Y8eO48ePh4WFIYS6d+/eoBSxWKzxQB8fn6ZKpyDo4gDAsBlc\nt3WvXr1GjBjh5eUVGRmZlpYmlUpVb/Xp00e17evr++jRo8ePH0ulUh8fH9p/5eXlPX369PHj\nxzKZbOTIkU2V0tSBWkqnIAjQABiSfleOEv+ptju6Rq1Gp9N//fXXU6dOde3adc2aNYGBgRKJ\nhHhLLperdpPJZAghoVCIEKqursbVrFixotkVf5o6UEvpFAQBGgCDZIihWYVOp48cOXLLli35\n+fmPHj3Kzc0l0m/evKna5+bNm++88463tzeTyczKymqQQ48ePdhs9tWrV5sqoqkDtZROQdAH\nDQAgVW5u7m+//Ub0Al+4cIFGo3l4eBBvXbt2bffu3RMmTDh06NC9e/cyMjKsra0XLFgQHx9P\np9MHDhxYUVGRnp4+d+5cb2/vuLi4hQsXIoT8/PweP36sVCpDQ0NVpTR1oEgkaqp0CoIADQAg\nlYWFxeXLl7/88kuhUPjOO++kp6c7OzsTby1btuzXX39dvnw5n88/fvw4sWDeV199ZWdnl5iY\n+PLlS0dHx+DgYDs7O4RQcnIynU5fsGBBTU1N9+7dt2zZ0qAgjQcyGIymSqcgGpmLNzcrKSkp\nOTm5hTsTS15ZWlqSuaIKLBqrD8SisVZWVqSV2FGLxtrY2HSCRWNb9TttucDAwOjoaNUieQBB\nHzQAAFAWBGgAAKAo6IMGAFDC7du3O7oKlAMtaAAAoCgI0AAAQFEQoAEAgKIgQAMAAEVBgAYA\nAIqCAA0AABQFARoAACgKAjQAAFAUBGgAAKAoCNAAAEBR1HrUW6FQiESiVh0ikUiIlRfIgWGY\nSCQiczY7pVIpl8tbe1naWSKO4xiGkVYihmFKpZLMcyROkOSrihASi8Vkfnn0dI7EuQASUCtA\n0+l0Lpfbwp1xHJdIJBwOh8kk7yyUSiWXyyXzNyaTyZhMZssvS/vJ5XIcx9lsNmklSqVS4sKS\nViLx5SGzRIVCUV9fz+Fw9Dfd6N3REartvr+lEYXq4xzJ/P4bOWoFaBqNxmAwWrgzMZM1nU5v\n+SHtR9SQzC8osd4lmeeoUChwHCf5qpJ8jjiOk18iQojBYOgpQN8Jnqn+tcwbE9nvylE9nSME\naNJAHzQABu9O8MwWJgLDAgEaAAAoilpdHAAAgBAKDg5WbV+5cqXN+fB4vOLiYj6fjxCqqqri\n8/lk3v1uP2hBAwCoRT06N35pVCBAA2Dw+l052sJELSjSZ60xHOs2RpeXl3t6ehLbBQUFgYGB\nCCGpVGplZbVq1aqePXv6+Pjk5+dPmzbN09Pz3XffJQbyTps2zc/Pr3fv3vPnz1coFMQhlpaW\na9eunTFjxoABA/SxIgwEaAA6gwbhuLXRuRPr37+/m5ubm5tbQECA9j3fvn0bEBDw8OHDiIiI\nsWPHbtiwobi4mMlknj59GiG0e/fu/Pz8goICHMePHv3P5a2trR0zZszx48fXrVu3Zs0anVce\n+qAB6CTaE5SJ5vOd4JmdL7Ln5uaq90Fr2ZPL5U6ZMgUhNGDAAG9vby8vL4RQ//79S0pKEELp\n6enHjh1DCFVUVFhYWERGRiKETE1Ng4KCEEI+Pj7EbroFARoAYFyYTKbqYUj155A5HA6xwWAw\nVA/4MBgMDMNycnL27dt39epVc3Pzr776ShWLWSwWsUGn0/Vx+xG6OAAwduq9zxTpiW6gPQM5\nGrO1tRWLxW/evEEIXbhwoSWHvHnzxsXFxdzcHMOwU6dO6bAy2kGABsCoNY7IHRujdRuLNaLR\naJs3bw4JCYmIiHj79m1LDhk7diybzZ40adLs2bN79eql7xqqQBcHAIBadBijhUKhatvOzk7V\nCzFnzpw5c+ao78nlcgUCAbE9cuTIkSNHEtsrVqwgNo4fP94gc/VDnJ2di4uLdVVtFWhBA2C8\nmmosU7OjwwhBCxqAzqxBqO18gzQ6N2hBA9Bp/T1xXoOUlt8PhEY0FUALGgAjBa1p6oMWNACd\nE/QvdwIQoAEAgKIgQAPQOTXVgwE9GwYEAjQAAFAUBGgAOq1ev/zYIAWaz4YFRnEA0Jk1FZFV\nE9d1yhnsOg1oQQNgdGAgh6GAFjQARooI03V5hVet/Ua+ye/o6gANoAUNgHGB5rMBgQANgJGq\nyysU/PGfZfSuWvt1bGWARhCgATAiquZzXV5hU28B6oAADYCRUgjFqv8jTSEbdDgI0AAYi6aa\nz6oYDR0dVAMBGgCj0KAHQxWUAZVBgAbAuNTlFTaOzhRsRAcHBwcHB3d0LToYjIMGwCioPy4I\n9wMNhV4CdH19/Z49e3Jzc+l0+nvvvTd16lSEUElJyc6dO0UikZOTU0JCgrm5uT6KBgA0i/rP\ndqvazsHBwe1ZQ5bH4/Xu3fvmzZvEy9jY2D179ty/f9/Hx0cHtdQ/vXRx7N+/H8Owffv2HThw\nYNiwYUTirl27wsPDU1JS3N3d09LS9FEuAAA0IJfLHz16hBCqr6+/fPmyi4tLR9eoFXTfgpZK\npZcuXfrxxx9ZLBZCiM/nI4TKy8urqqoGDRqEEBo/fnxCQkJsbCyxv1KpfPXqFbGtUCgUCkUL\nC8JxnDi85Ye0H47jCoWCRqORWSJRKGklKpVKkksk/xw75KoihBQKBfG9JYeezlHfp9Cg67md\njeioqKjU1NQvvvji7NmzISEhWVlZRPq0adOKioowDBs2bNi3337LYDAQQpcuXVqxYoVEImGz\n2RkZGR4eHhp3O3To0Pr1683NzUNCQjIzM/Py8srLy4cNG1ZcXIwQKigoiI6Ovn37NkLo8uXL\nq1atEovFlpaW33//vbe3d6sqr/sA/erVKwsLi4yMjPv37/N4vJiYGA8Pj6qqKjs7OyKu2dvb\n19bWyuVyIoJLpdJJkyYRx/bt2/fNmzetKq6urk7np6BdfX09ySVKJBKJREJyoSKRiOQSW/vR\ntx/5H+Xbt29JLlEmk+k8TwzDdJ6n/kydOnXcuHHJycmpqamrV69WBejdu3c7OjriOD5//vyj\nR49GRkaWl5fPnj374sWLPj4+YrGY+Heo8W7Pnz9fsWLFnTt3+Hy+qqGpVuTFTgAADUxJREFU\nUUVFxbJlyy5dumRra5udnR0TE6PqbGkh3QdohUJRVVXl4eExd+7c7OzsDRs2pKSk4DjeVKuT\ny+WeOXOG2N69e7e1tXULC8JxXCAQmJubM5nk3eoUi8UmJiZktqBra2vZbDaXyyWtRLlcjuM4\nm80mrUSpVCqTySwsLEgrEcdxiURiampKWokKhaK2ttbS0pJOJ2/olEgkMjMz03m2ev3FaRy5\n0Z5GNI/HCwgIyMjIKC0t7d+/vyo9PT392LFjCKGKigoLC4vIyMg//vgjKCiI6J5WfTca73bj\nxo1Ro0YRfQMxMTE5OTlNFX39+vWKioqIiAjiZRtak7q50JmZmadPn0YIxcbGdu/eHSEUFBSE\nEBo8ePDWrVtra2vt7e0rKyuJMF1ZWWlhYUE0nxFCdDrdycmJ2GYwGMRfEC1B/PtGp9Nbfkj7\n0Wg0BoNBZoCm0WhEoaSVSPwZTvJVJfkcia8iySUihBgMBpkBWk/nqL/vv5Zxde2J0VFRUeHh\n4cuXL1el5OTk7Nu37+rVq+bm5l999VVJSQn677dC/UCNu6lT9fYwmUyiFwup/dWC47ifn19m\nZmbbqo10dZNwwoQJe/fu3bt3b0BAgKWlpY+Pz927dxFC9+7ds7CwMDc35/P5tra2xD81WVlZ\nQ4YM0Um5AADQrDFjxqxevTomJkaV8ubNGxcXF3NzcwzDTp06RSQOHz78+vXrxB1FqVQqFos1\n7jZ06NDLly+Xl5cjhA4cOEAk2traEvsjhC5cuEAkBgUF3bp1Kzs7GyGkVCqJXulW0cufKgsX\nLtyxY0dKSoqpqWliYiLxj9KSJUu2b9/+ww8/ODs7L1u2TB/lAgAMWntuBmpBp9Pj4+PVU8aO\nHXvgwIFJkyaZmpr26tWLSHR0dExNTY2IiCDukJ04cULjbs7Ozps2bRo+fLiDg0NISAiRSKPR\nNm/eHBIS4uXl1a1bN1WGJ0+ejI+PF4vFGIaFhYUFBga2quY0Mu8pNyspKSk5ObmFO+M4Xl1d\nbWlpqeotIYFQKDQzMyOzi0MgELDZbDJ7S2UyGY7jHA6HtBLFYrFMJrOysiKtRBzHRSIRj8cj\nrUQMwwQCgY2NDZldHHV1dfp44KBVv9PO7eHDh+Hh4Xl5eXrKHx71BgAAiqLWo95isXjDhg0t\n318qlbLZbDKbJKrRgaSRyWR0Op3MkSrEOGgyb6BhGKZUKskcN4JI/yiVSqVMJuNwOGT++aWn\nc5TL5TrP00D17NlTf81nRLUuDqKnpoU719XVRUZGbt68uWfPnnqtlTot4wX1JCEhwd/fPzIy\nksxCSXb06NHc3NyvvvqKzEJJ/iiLioqWL1+emppqaWlJWqF6Okcmk0lmn5sxo1YLurWfel1d\nHZfLJXP8LPmIFnTnPkcajUbyOGjycbncuro6Ho/XuU8T6Bb0QQMAAEUx1q1b19F1aCMajebo\n6Ojr62tiYtLRddEjKysrb29ve3v7jq6IHpmamnp5ebm7u3d0RfSIyWR269bNx8eHzM59YOio\n1QcNAABABbo4AACAoqh1k1CLPXv23Lx5s6qq6vDhw6rbLBoXAegcKwN8/PHHYrGYGEH4ySef\n9OnTB3WWU2ugU54UMqZPEOgRbiAKCgpqamqmTZv29u1bVWJcXNyff/6J4/j+/fu/++47LYkG\nJzY2lpheSl3nOLUGOuVJ4cb0CQL9MZgujt69ezeYibTBIgA3btxoKrFz6JSn1ilPqilGdbJA\nJwymi6MxjYsAaFkZwOCsXLkSIdS3b9/o6GgTE5POdGoqnfKkVIzhEwR6ZTAt6MZwTU9JaUw0\nRGvXrt2zZ8+OHTtqa2uJKQ07zamp65QnRTCSTxDoFXUDdGZm5kcfffTRRx/99ddfGndQLQKA\nEFItAqAxkdR6t1WD8+Xz+TQazdTUdMKECcQEtYZ7alp0ypMiGMknCPSKul0cEyZMmDBhgpYd\nVIsADB48WLUIgMZEg6B+vhKJRCQS2dnZyeXyy5cvE09wGO6padEpTwoZ0ycI9MpgHlT55ptv\nbt++XVNTY2Vl5ebm9tlnnyGESkpKtm/fLhKJiEUAiGloNCYalsrKys8++0woFNLpdB8fnw8/\n/FA1gtDQT62xTnlSRvUJAv0xmAANAADGhrp90AAAYOQgQAMAAEVBgAYAAIqCAA0MzMiRI7dt\n29bRtQCADBCgDUlYWBiNRiNG1/r7+6enp6veKi0tjYmJ6dq1K5fL7dmz5yeffPL69WuE0Nmz\nZ8eOHWttbU2j0Z4/f95sEXFxcTQ1sbGx2tP37t3r6urq7u5++PBhVSZbtmxZuHChLs9cTWho\nqI+Pj54yB4BSYBSHISEC9MaNG8Vi8b59+77//vubN2/269fv0aNHQ4cOdXNzW758eY8ePSor\nK48cOWJhYfH1118fO3bs8ePHPB7vk08+KSsrc3Z21l5EXFzc48ePv/jiC+KlnZ0dcYjG9PLy\n8t69ez948KC+vr5v374lJSVWVlZlZWUjR468c+eOlZWVzq+ATCYjeW1ZADpSh0zRBNpm0qRJ\nc+bMIbaVSmWXLl3Wr1+P4/jo0aN9fX1lMpn6zlVVVart3NxchFBZWVmzRSxdujQiIqKF6Tdu\n3AgMDCS233nnnfz8fBzHJ0+efPDgwaby/+qrr3r16qV6KZPJrK2tT58+jeP4/v37+/bta2pq\n6uzsHB8fL5FIiH369eu3atWqCRMm8Hi8DRs2jBgxYuvWrcRbWg5Zu3ZtWFiYhYWFl5dXZmYm\nkS6Xy1etWuXs7MzhcHr16vXzzz8TiYmJiV27djUzMxs+fHheXl6zVwkAckAXh6Gi0WgmJib1\n9fVVVVWXL19OSEho8Nywra2tlsN37NjRVAv37NmzJiYmrq6uCxcufPv2rZZ0T0/PkpKSsrKy\noqKiyspKd3f38+fPV1ZWRkVFNVXu9OnTHz58eP/+feLlxYsXlUplaGgoQgjDsI0bNxYUFKSl\npV26dGnLli2qo3bt2rVkyZK3b98uXrxYPTcth+zevTs+Pl4gEKxYsSIiIkIkEiGEVq1atX//\n/q+//vrBgwdffvklccXWrFlz6dKlY8eO5efnjx49OiQkpLa2VsulA4A8Hf0vBGgFVQtaKpV+\n8803CKELFy7k5OQghG7duqXlwMYt6BMnToSGhjbe86effjp27Fh2dvb+/fudnZ0nTJigPf3g\nwYM9e/b09vbOyMiQSCQ9e/a8f//+559/3r179wkTJlRUVDQuIigoKCkpidiOiopS/U2g7vTp\n03369CG2+/XrN3fuXNVb6i1oLYcsWrSI2FYqlTweLzs7u66ujs1mnzp1Sv0okUjE5XILCgpU\nKf7+/seOHWucPwDkgwBtSCZNmkSj0RgMBo1G43K5n3/+OY7jRIDOzc3VcmDLuzjUXb16FSFU\nVFTUwvS1a9cmJCRcuXKld+/eQqFw9erVH3zwQeNsv/nmmx49euA4LpVKLS0tVf0Pt27dGjVq\nlGrWbysrKyK9X79+6hFZPUBrOWTHjh2qQ1xdXc+cOUPMQvXmzRv1yuTl5TVutWzcuLFVFwoA\nPYEuDgPzr3/9Ky8v7/Hjx7W1tatXr0YIde/enUaj/f333zovKzAwECFUWlrakvQnT56kpaWt\nXbv2999/f/fdd83MzGbOnPn77783znbq1KlPnjzJy8s7f/48k8kcO3YsQkgmk02cOHHw4MGF\nhYUYhl28eFEul6sO4XK5jfPRfkiDxbOVSiWu6X64UChECFVXV6v/KlasWKH1wgBAEgjQBsbK\nysrHx6d79+6qHmc7O7vg4OAvv/wSwzD1Paurq9tZ1u3btxFCxExszaYvWrRo48aNPB5PqVQS\nsx7T6XSFQtE4WwcHh+Dg4BMnThw/fnzy5MlMJhMhVFJSUllZmZiYyOfzGQxGfn5+s9Vr7SE9\nevRgs9lE81/F29ubyWRmZWU1WxwA5IMA3Rns3r372bNnQ4YMOXHiRF5e3q+//hodHU1M+CcQ\nCPLy8oqKihBChYWFeXl5RJvxp59+Gj9+fOOs5s6de+rUqZycnIMHD0ZGRo4fP97T01NLOiEj\nIwPH8alTpyKEBg8efOHCBZlMdvr06aZm1JwxY8bRo0fPnj0bHh5OpDg5OZmamhKBsrCwcOfO\nnc2edWsP4fF4cXFxCxcuPH369NOnT3/99dfz589bW1svWLAgPj7+6NGjJSUlOTk5CQkJhYWF\nzZYOABk6pmcFtIn6MLsGnjx5EhUVxefz2Wx2jx494uPjX79+jeP40aNHG3ziV65cwXF8+/bt\nlpaWjfMJDw/v2rUrm812d3ePi4sTCATa03EcFwqFnp6e6l3SS5cu7dq16+DBg589e6axtjU1\nNSwWy9HREcMwVWJGRoaHh4eTk9PgwYO/+eYbMzMzIr1fv367du1S7abeB93CQ1xdXYl7gzKZ\nbMWKFV26dOFwON7e3ufOncNxHMOwzz//3NXVlcViOTs7z549m7h0AHQ4eFAFAAAoCro4AACA\noiBAAwAARUGABgAAioIADQAAFAUBGgAAKAoCNAAAUBQEaAAAoCgI0AAAQFEQoAEAgKL+D7JZ\nxDmLyyL/AAAAAElFTkSuQmCC",
      "text/plain": [
       "plot without title"
      ]
     },
     "metadata": {
      "image/png": {
       "height": 180,
       "width": 240
      }
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "#plot PCA\n",
    "options(repr.plot.width = 8, repr.plot.height = 6)\n",
    "\n",
    "pcaData <- plotPCA(vsd, intgroup=c(\"species\", \"tissue\"), returnData=TRUE) %>% rownames_to_column(var = \"sample\") %>% \n",
    "    filter(sample != \"Human_Heart_R1\") # this sample is actually lung and is mentioned as getting mixed up on GEO. \n",
    "\n",
    "pcaData$tissue <- factor(pcaData$tissue, levels = c(\"GM12878\", \"LCL8664\", \"LCL\",\n",
    "             \"Heart\", \"Kidney\", \"Liver\", \"Lung\"))\n",
    "\n",
    "percentVar <- round(100 * attr(pcaData, \"percentVar\"))\n",
    "\n",
    "p <- ggplot(pcaData, aes(PC1, PC2, color=tissue, shape=species)) +\n",
    "  geom_point(size=2, alpha = 0.75) +\n",
    "  xlab(paste0(\"PC1: \",percentVar[1],\"% variance\")) +\n",
    "  ylab(paste0(\"PC2: \",percentVar[2],\"% variance\")) +\n",
    "  ggsci::scale_color_aaas() +\n",
    "  #scale_shape_manual(values = c(1,3)) +\n",
    "  theme_bw(base_size = 8)\n",
    "\n",
    "options(repr.plot.width = 4, repr.plot.height = 3)\n",
    "p\n",
    "\n",
    "ggsave(p, filename = \"../../results/results_human-evolution/RNA-expression-PCA_w-Gilad-data.pdf\", device = cairo_pdf, width = 4, height = 3)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "56d0ca99",
   "metadata": {},
   "source": [
    "## PCA with Human Blood Cells\n",
    "We will repeat this using RNA-seq data from a variety of human blood cells to see how our cells compare to other immune cells"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ad0128b1",
   "metadata": {},
   "source": [
    "Use data from this accesssion: GSE118165\n",
    "\n",
    "__RUNID__ __Cell-Type__  __DONOR__  \n",
    "SRR11007061\tCD8pos_T\t1001  \n",
    "SRR11007071\tEffector_CD4pos_T\t1001  \n",
    "SRR11007082\tRegulatory_T\t1001  \n",
    "SRR11007090\tMature_NK\t1001  \n",
    "SRR11007092\tMonocytes\t1001  \n",
    "SRR11007094\tMyeloid_DCs\t1001  \n",
    "SRR11007096\tCD8pos_T\t1002  \n",
    "SRR11007113\tRegulatory_T\t1002  \n",
    "SRR11007121\tMature_NK\t1002  \n",
    "SRR11007124\tMonocytes\t1002  \n",
    "SRR11007126\tMyeloid_DCs\t1002  \n",
    "SRR11007127\tCD8pos_T\t1003  \n",
    "SRR11007137\tEffector_CD4pos_T\t1003  \n",
    "SRR11007147\tRegulatory_T\t1003  \n",
    "SRR11007156\tMature_NK\t1003  \n",
    "SRR11007158\tMonocytes\t1003    \n",
    "SRR11007160\tCD8pos_T\t1004  \n",
    "SRR11007170\tEffector_CD4pos_T\t1004  \n",
    "SRR11007182\tRegulatory_T\t1004  \n",
    "SRR11007186\tMature_NK\t1004  \n",
    "SRR11007188\tMonocytes\t1004  \n",
    "SRR11007190\tMyeloid_DCs\t1004  \n",
    "SRR7647654\tBulk_B\t1001  \n",
    "SRR7647656\tMem_B\t1001  \n",
    "SRR7647658\tNaive_B\t1001  \n",
    "SRR7647696\tBulk_B\t1002  \n",
    "SRR7647698\tMem_B\t1002  \n",
    "SRR7647700\tNaive_B\t1002  \n",
    "SRR7647731\tBulk_B\t1003  \n",
    "SRR7647767\tBulk_B\t1004  \n",
    "SRR7647768\tMem_B\t1004  \n",
    "SRR7647769\tNaive_B\t1004  \n",
    "SRR7647807\tMem_B\t1010  \n",
    "SRR7647808\tNaive_B\t1010  "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "746d1218",
   "metadata": {},
   "source": [
    "### Download Data:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4262c426",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-03-21T22:01:54.466772Z",
     "start_time": "2022-03-21T15:32:32.999568Z"
    }
   },
   "outputs": [],
   "source": [
    "# bash kernel \n",
    "\n",
    "cd /data/hodges_lab/public_data/immune-cell_RNA-seq/Calderon_2019\n",
    "\n",
    "fasterq-dump -3 --skip-technical --mem 64G --threads 8 --progress \\\n",
    "    SRR11007061 SRR11007071 SRR11007082 SRR11007090 SRR11007092 SRR11007094 SRR11007096 \\\n",
    "    SRR11007113 SRR11007121 SRR11007124 SRR11007126 SRR11007127 SRR11007137 SRR11007147 SRR11007156 \\\n",
    "    SRR11007158 SRR11007160 SRR11007170 SRR11007182 SRR11007186 SRR11007188 SRR11007190 SRR7647654 \\\n",
    "    SRR7647656 SRR7647656 SRR7647658 SRR7647696 SRR7647698 SRR7647700 SRR7647731 SRR7647767 SRR7647768 \\\n",
    "    SRR7647769 SRR7647807 SRR7647808"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "3dd1c410",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-03-22T13:54:36.891891Z",
     "start_time": "2022-03-22T13:54:36.666982Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "#!/bin/bash\n",
      "#SBATCH --mail-user=tyler.j.hansen@vanderbilt.edu\n",
      "#SBATCH --mail-type=ALL\n",
      "#SBATCH --nodes=1\n",
      "#SBATCH --ntasks=1\n",
      "#SBATCH --cpus-per-task=12\n",
      "#SBATCH --time=01-00:00:00\n",
      "#SBATCH --mem=120G\n",
      "\n",
      "#download data\n",
      "cd /data/hodges_lab/public_data/immune-cell_RNA-seq/Calderon_2019\n",
      "\n",
      "fasterq-dump -3 --skip-technical --mem 120G --threads 12 \\\n",
      "    SRR11007126 SRR11007127 SRR11007137 SRR11007147 SRR11007156 \\\n",
      "    SRR11007158 SRR11007160 SRR11007170 SRR11007182 SRR11007186 SRR11007188 SRR11007190 SRR7647654 \\\n",
      "    SRR7647656 SRR7647656 SRR7647658 SRR7647696 SRR7647698 SRR7647700 SRR7647731 SRR7647767 SRR7647768 \\\n",
      "    SRR7647769 SRR7647807 SRR7647808\n"
     ]
    }
   ],
   "source": [
    "#stopped at SRR11007126. Run the rest as a slrm script. \n",
    "cat download_calderon-2019.slrm"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "66bcedbd",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-03-22T21:05:01.826862Z",
     "start_time": "2022-03-22T21:05:01.721520Z"
    }
   },
   "source": [
    "SRR7647656 did not work for some reason. This is Mem_B 1001, so i will just omit that one from the following steps."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2cf1433b",
   "metadata": {},
   "source": [
    "### Trim"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "94b61464",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-03-23T02:51:31.717906Z",
     "start_time": "2022-03-23T02:51:31.530925Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CD8pos_T_1001\tSRR11007061\n",
      "Effector_CD4pos_T_1001\tSRR11007071\n",
      "Regulatory_T_1001\tSRR11007082\n",
      "Mature_NK_1001\tSRR11007090\n",
      "Monocytes_1001\tSRR11007092\n",
      "Myeloid_DCs_1001\tSRR11007094\n",
      "CD8pos_T_1002\tSRR11007096\n",
      "Regulatory_T_1002\tSRR11007113\n",
      "Mature_NK_1002\tSRR11007121\n",
      "Monocytes_1002\tSRR11007124\n",
      "Myeloid_DCs_1002\tSRR11007126\n",
      "CD8pos_T_1003\tSRR11007127\n",
      "Effector_CD4pos_T_1003\tSRR11007137\n",
      "Regulatory_T_1003\tSRR11007147\n",
      "Mature_NK_1003\tSRR11007156\n",
      "Monocytes_1003\tSRR11007158\n",
      "CD8pos_T_1004\tSRR11007160\n",
      "Effector_CD4pos_T_1004\tSRR11007170\n",
      "Regulatory_T_1004\tSRR11007182\n",
      "Mature_NK_1004\tSRR11007186\n",
      "Monocytes_1004\tSRR11007188\n",
      "Myeloid_DCs_1004\tSRR11007190\n",
      "Bulk_B_1001\tSRR7647654\n",
      "Naive_B_1001\tSRR7647658\n",
      "Bulk_B_1002\tSRR7647696\n",
      "Mem_B_1002\tSRR7647698\n",
      "Naive_B_1002\tSRR7647700\n",
      "Bulk_B_1003\tSRR7647731\n",
      "Bulk_B_1004\tSRR7647767\n",
      "Mem_B_1004\tSRR7647768\n",
      "Naive_B_1004\tSRR7647769\n",
      "Mem_B_1010\tSRR7647807\n",
      "Naive_B_1010\tSRR7647808\n",
      "\n"
     ]
    }
   ],
   "source": [
    "cd /data/hodges_lab/public_data/immune-cell_RNA-seq/Calderon_2019\n",
    "cat expt-sra_key.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "b5d9714d",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-03-23T02:52:24.612853Z",
     "start_time": "2022-03-23T02:52:24.231801Z"
    },
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "#!/bin/bash\n",
      "#SBATCH --mail-user=tyler.j.hansen@vanderbilt.edu\n",
      "#SBATCH --mail-type=ALL\n",
      "#SBATCH --nodes=1\n",
      "#SBATCH --ntasks=1\n",
      "#SBATCH --cpus-per-task=12\n",
      "#SBATCH --time=01-00:00:00\n",
      "#SBATCH --mem=120G\n",
      "\n",
      "cd /data/hodges_lab/public_data/immune-cell_RNA-seq/Calderon_2019\n",
      "\n",
      "#trim\n",
      "# use read combined with while loop to read two variables at once.\n",
      "DIR='/data/hodges_lab/public_data/immune-cell_RNA-seq/Calderon_2019'\n",
      "\n",
      "while read -r a b ; do    \n",
      "    trim_galore --paired -j 12 --fastqc --fastqc_args \"--outdir ${DIR}/processing_results/fastQC\" --dont_gzip --basename \"$a\" \\\n",
      "    --output_dir ${DIR}/fastq/trimmed_reads ${DIR}/fastq/\"$b\"_1.fastq ${DIR}/fastq/\"$b\"_2.fastq \n",
      "done < expt-sra_key.txt\n",
      "Submitted batch job 39138666\n"
     ]
    }
   ],
   "source": [
    "#submit job.\n",
    "cat trim_calderon-2019.slrm\n",
    "\n",
    "sbatch trim_calderon-2019.slrm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "0658d8a3",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-03-23T13:44:28.659456Z",
     "start_time": "2022-03-23T13:44:28.493649Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Naive_B_1001\tSRR7647658\n",
      "Bulk_B_1002\tSRR7647696\n",
      "Mem_B_1002\tSRR7647698\n",
      "Naive_B_1002\tSRR7647700\n",
      "Bulk_B_1003\tSRR7647731\n",
      "Bulk_B_1004\tSRR7647767\n",
      "Mem_B_1004\tSRR7647768\n",
      "Naive_B_1004\tSRR7647769\n",
      "Mem_B_1010\tSRR7647807\n",
      "Naive_B_1010\tSRR7647808\n"
     ]
    }
   ],
   "source": [
    "#I forgot to move some fastqs over so, repeat but only on those.\n",
    "cat /data/hodges_lab/public_data/immune-cell_RNA-seq/Calderon_2019/expt-sra_key_2.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "801e548f",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-03-23T13:47:03.054430Z",
     "start_time": "2022-03-23T13:47:02.769763Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "#!/bin/bash\n",
      "#SBATCH --mail-user=tyler.j.hansen@vanderbilt.edu\n",
      "#SBATCH --mail-type=ALL\n",
      "#SBATCH --nodes=1\n",
      "#SBATCH --ntasks=1\n",
      "#SBATCH --cpus-per-task=12\n",
      "#SBATCH --time=08:00:00\n",
      "#SBATCH --mem=120G\n",
      "\n",
      "cd /data/hodges_lab/public_data/immune-cell_RNA-seq/Calderon_2019\n",
      "\n",
      "#trim\n",
      "# use read combined with while loop to read two variables at once.\n",
      "DIR='/data/hodges_lab/public_data/immune-cell_RNA-seq/Calderon_2019'\n",
      "\n",
      "while read -r a b ; do    \n",
      "    trim_galore --paired -j 12 --fastqc --fastqc_args \"--outdir ${DIR}/processing_results/fastQC\" --dont_gzip --basename \"$a\" \\\n",
      "    --output_dir ${DIR}/fastq/trimmed_reads ${DIR}/fastq/\"$b\"_1.fastq ${DIR}/fastq/\"$b\"_2.fastq \n",
      "done < expt-sra_key_2.txt\n",
      "Submitted batch job 39145478\n"
     ]
    }
   ],
   "source": [
    "cat trim_calderon-2019_2.slrm\n",
    "sbatch trim_calderon-2019_2.slrm"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0249510e",
   "metadata": {},
   "source": [
    "### Map"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "db90b26f",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-03-25T14:46:42.502123Z",
     "start_time": "2022-03-25T14:46:42.281850Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "#!/bin/bash\n",
      "#SBATCH --mail-user=tyler.j.hansen@vanderbilt.edu\n",
      "#SBATCH --mail-type=ALL\n",
      "#SBATCH --nodes=1\n",
      "#SBATCH --ntasks=1\n",
      "#SBATCH --cpus-per-task=12\n",
      "#SBATCH --time=01-00:00:00\n",
      "#SBATCH --mem=120G\n",
      "\n",
      "cd /data/hodges_lab/public_data/immune-cell_RNA-seq/Calderon_2019\n",
      "\n",
      "#trim\n",
      "# use read combined with while loop to read two variables at once.\n",
      "DIR='/data/hodges_lab/public_data/immune-cell_RNA-seq/Calderon_2019'\n",
      "\n",
      "while read -r a b ; do    \n",
      "    STAR --runMode alignReads --runThreadN 12 --genomeDir /data/hodges_lab/hg38_genome/hg38_GTF-ensembl.104_index_99 \\\n",
      "    --readFilesIn ${DIR}/fastq/trimmed_reads/\"$a\"_val_1.fq ${DIR}/fastq/trimmed_reads/\"$a\"_val_2.fq \\\n",
      "    --outFileNamePrefix ${DIR}/star-output/\"$a\"_ensembl.104_ --outSAMtype BAM SortedByCoordinate\n",
      "done < expt-sra_key.txt\n"
     ]
    }
   ],
   "source": [
    "cat map_calderon-2019.slrm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "2667e8c6",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-03-28T16:28:56.299313Z",
     "start_time": "2022-03-28T15:48:30.558806Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\tSTAR --runMode alignReads --runThreadN 8 --genomeDir /data/hodges_lab/hg38_genome/hg38_GTF-ensembl.104_index_99 --readFilesIn /data/hodges_lab/public_data/immune-cell_RNA-seq/Calderon_2019/fastq/trimmed_reads/Naive_B_1010_val_1.fq /data/hodges_lab/public_data/immune-cell_RNA-seq/Calderon_2019/fastq/trimmed_reads/Naive_B_1010_val_2.fq --outFileNamePrefix /data/hodges_lab/public_data/immune-cell_RNA-seq/Calderon_2019/star-output/Naive_B_1010_ensembl.104_ --outSAMtype BAM SortedByCoordinate\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Mar 28 10:48:31 ..... started STAR run\n",
      "Mar 28 10:48:46 ..... loading genome\n",
      "Mar 28 10:55:23 ..... started mapping\n",
      "Mar 28 11:20:28 ..... finished mapping\n",
      "Mar 28 11:20:32 ..... started sorting BAM\n",
      "Mar 28 11:28:55 ..... finished successfully\n"
     ]
    }
   ],
   "source": [
    "#It did all but the last one. So run that here. \n",
    "cd /data/hodges_lab/public_data/immune-cell_RNA-seq/Calderon_2019\n",
    "\n",
    "#trim\n",
    "# use read combined with while loop to read two variables at once.\n",
    "DIR='/data/hodges_lab/public_data/immune-cell_RNA-seq/Calderon_2019'\n",
    "\n",
    "STAR --runMode alignReads --runThreadN 8 --genomeDir /data/hodges_lab/hg38_genome/hg38_GTF-ensembl.104_index_99 \\\n",
    "    --readFilesIn ${DIR}/fastq/trimmed_reads/Naive_B_1010_val_1.fq ${DIR}/fastq/trimmed_reads/Naive_B_1010_val_2.fq \\\n",
    "    --outFileNamePrefix ${DIR}/star-output/Naive_B_1010_ensembl.104_ --outSAMtype BAM SortedByCoordinate"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "568f40a4",
   "metadata": {},
   "source": [
    "### Count"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "c2825873",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-03-29T10:27:18.020739Z",
     "start_time": "2022-03-29T10:27:17.666295Z"
    },
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "#!/bin/bash\n",
      "#SBATCH --mail-user=tyler.j.hansen@vanderbilt.edu\n",
      "#SBATCH --mail-type=ALL\n",
      "#SBATCH --nodes=1\n",
      "#SBATCH --ntasks=1\n",
      "#SBATCH --cpus-per-task=12\n",
      "#SBATCH --time=01-00:00:00\n",
      "#SBATCH --mem=120G\n",
      "\n",
      "HG38_FA='/data/hodges_lab/hg38_genome/hg38.fa'\n",
      "HG38_GTF='/data/hodges_lab/ATAC-STARR_B-cells/data/RNA-seq/Ensembl.v64.fullTransExon.hg19TorheMac2.hg38.gtf'\n",
      "INT_DIR='/data/hodges_lab/ATAC-STARR_B-cells/data/RNA-seq'\n",
      "PUB_DIR='/data/hodges_lab/public_data/immune-cell_RNA-seq/Calderon_2019'\n",
      "\n",
      "#Seperate into two files, one for paired end reads (ours) the other for single end reads (public data). Join afterwards. \n",
      "# For the PE file, will need to assess if fragments or individual reads are counted. \n",
      "# For both files, do unstranded mapping since I'm not sure if the public data is forward or reverse stranded. \n",
      "\n",
      "#Human\n",
      "featureCounts -s 0 -p -B -O -T 12 -F GTF -G $HG38_FA -t exon -g gene_id \\\n",
      "        -a $HG38_GTF -o ${INT_DIR}/count_matricies/Human_Immune-Cell_RNA-seq_counts_XSAnno_PE.tsv \\\n",
      "        ${PUB_DIR}/star-output/Bulk_B_1001_ensembl.104_Aligned.sortedByCoord.out.bam \\\n",
      "        ${PUB_DIR}/star-output/Bulk_B_1002_ensembl.104_Aligned.sortedByCoord.out.bam \\\n",
      "        ${PUB_DIR}/star-output/Bulk_B_1003_ensembl.104_Aligned.sortedByCoord.out.bam \\\n",
      "        ${PUB_DIR}/star-output/Bulk_B_1004_ensembl.104_Aligned.sortedByCoord.out.bam \\\n",
      "        ${PUB_DIR}/star-output/CD8pos_T_1001_ensembl.104_Aligned.sortedByCoord.out.bam \\\n",
      "        ${PUB_DIR}/star-output/CD8pos_T_1002_ensembl.104_Aligned.sortedByCoord.out.bam \\\n",
      "        ${PUB_DIR}/star-output/CD8pos_T_1003_ensembl.104_Aligned.sortedByCoord.out.bam \\\n",
      "        ${PUB_DIR}/star-output/CD8pos_T_1004_ensembl.104_Aligned.sortedByCoord.out.bam \\\n",
      "        ${PUB_DIR}/star-output/Effector_CD4pos_T_1001_ensembl.104_Aligned.sortedByCoord.out.bam \\\n",
      "        ${PUB_DIR}/star-output/Effector_CD4pos_T_1003_ensembl.104_Aligned.sortedByCoord.out.bam \\\n",
      "        ${PUB_DIR}/star-output/Effector_CD4pos_T_1004_ensembl.104_Aligned.sortedByCoord.out.bam \\\n",
      "        ${PUB_DIR}/star-output/Mature_NK_1001_ensembl.104_Aligned.sortedByCoord.out.bam \\\n",
      "        ${PUB_DIR}/star-output/Mature_NK_1002_ensembl.104_Aligned.sortedByCoord.out.bam \\\n",
      "        ${PUB_DIR}/star-output/Mature_NK_1003_ensembl.104_Aligned.sortedByCoord.out.bam \\\n",
      "        ${PUB_DIR}/star-output/Mature_NK_1004_ensembl.104_Aligned.sortedByCoord.out.bam \\\n",
      "        ${PUB_DIR}/star-output/Mem_B_1002_ensembl.104_Aligned.sortedByCoord.out.bam \\\n",
      "        ${PUB_DIR}/star-output/Mem_B_1002_ensembl.104_Aligned.sortedByCoord.out.bam \\\n",
      "        ${PUB_DIR}/star-output/Mem_B_1010_ensembl.104_Aligned.sortedByCoord.out.bam \\\n",
      "        ${PUB_DIR}/star-output/Monocytes_1001_ensembl.104_Aligned.sortedByCoord.out.bam \\\n",
      "        ${PUB_DIR}/star-output/Monocytes_1002_ensembl.104_Aligned.sortedByCoord.out.bam \\\n",
      "        ${PUB_DIR}/star-output/Monocytes_1003_ensembl.104_Aligned.sortedByCoord.out.bam \\\n",
      "        ${PUB_DIR}/star-output/Monocytes_1004_ensembl.104_Aligned.sortedByCoord.out.bam \\\n",
      "        ${PUB_DIR}/star-output/Myeloid_DCs_1001_ensembl.104_Aligned.sortedByCoord.out.bam \\\n",
      "        ${PUB_DIR}/star-output/Myeloid_DCs_1002_ensembl.104_Aligned.sortedByCoord.out.bam \\\n",
      "        ${PUB_DIR}/star-output/Myeloid_DCs_1004_ensembl.104_Aligned.sortedByCoord.out.bam \\\n",
      "        ${PUB_DIR}/star-output/Naive_B_1001_ensembl.104_Aligned.sortedByCoord.out.bam \\\n",
      "        ${PUB_DIR}/star-output/Naive_B_1002_ensembl.104_Aligned.sortedByCoord.out.bam \\\n",
      "        ${PUB_DIR}/star-output/Naive_B_1004_ensembl.104_Aligned.sortedByCoord.out.bam \\\n",
      "        ${PUB_DIR}/star-output/Naive_B_1010_ensembl.104_Aligned.sortedByCoord.out.bam \\\n",
      "        ${PUB_DIR}/star-output/Regulatory_T_1001_ensembl.104_Aligned.sortedByCoord.out.bam \\\n",
      "        ${PUB_DIR}/star-output/Regulatory_T_1002_ensembl.104_Aligned.sortedByCoord.out.bam \\\n",
      "        ${PUB_DIR}/star-output/Regulatory_T_1003_ensembl.104_Aligned.sortedByCoord.out.bam \\\n",
      "        ${PUB_DIR}/star-output/Regulatory_T_1004_ensembl.104_Aligned.sortedByCoord.out.bam\n"
     ]
    }
   ],
   "source": [
    "cat count_Calderon-2019.slrm"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6ad97c8f",
   "metadata": {},
   "source": [
    "### PCA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "d637ae21",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-07-27T14:53:57.321002Z",
     "start_time": "2022-07-27T14:53:54.344Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1mRows: \u001b[22m\u001b[34m27898\u001b[39m \u001b[1mColumns: \u001b[22m\u001b[34m8\u001b[39m\n",
      "\n",
      "\u001b[36m\u001b[39m \u001b[1mColumn specification\u001b[22m \u001b[36m\u001b[39m\n",
      "\u001b[1mDelimiter:\u001b[22m \"\\t\"\n",
      "\u001b[31mchr\u001b[39m (5): gene_id, Chr, Start, End, Strand\n",
      "\u001b[32mdbl\u001b[39m (3): Length, GM12878_R1, GM12878_R2\n",
      "\n",
      "\n",
      "\u001b[36m\u001b[39m Use `spec()` to retrieve the full column specification for this data.\n",
      "\u001b[36m\u001b[39m Specify the column types or set `show_col_types = FALSE` to quiet this message.\n",
      "\n",
      "\u001b[1mRows: \u001b[22m\u001b[34m27898\u001b[39m \u001b[1mColumns: \u001b[22m\u001b[34m39\u001b[39m\n",
      "\n",
      "\u001b[36m\u001b[39m \u001b[1mColumn specification\u001b[22m \u001b[36m\u001b[39m\n",
      "\u001b[1mDelimiter:\u001b[22m \"\\t\"\n",
      "\u001b[31mchr\u001b[39m  (5): gene_id, Chr, Start, End, Strand\n",
      "\u001b[32mdbl\u001b[39m (34): Length, Bulk_B_1001, Bulk_B_1002, Bulk_B_1003, Bulk_B_1004, CD8pos...\n",
      "\n",
      "\n",
      "\u001b[36m\u001b[39m Use `spec()` to retrieve the full column specification for this data.\n",
      "\u001b[36m\u001b[39m Specify the column types or set `show_col_types = FALSE` to quiet this message.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# r kernel\n",
    "suppressPackageStartupMessages(library(tidyverse))\n",
    "\n",
    "dir <- \"/data/hodges_lab/ATAC-STARR_B-cells/data/RNA-seq/count_matricies/\"\n",
    "\n",
    "#read in cts\n",
    "AS <- read_tsv(paste0(dir,\"Human_cross-tissue_RNA-seq_counts_XSAnno_PE.tsv\"), \n",
    "                     col_names = c(\"gene_id\", \"Chr\", \"Start\", \"End\", \"Strand\", \"Length\", \"GM12878_R1\", \"GM12878_R2\"), \n",
    "                     skip = 2) %>% dplyr::select(-Chr, -Start, -End, -Strand, -Length)\n",
    "                     \n",
    "C <- read_tsv(paste0(dir,\"Human_Immune-Cell_RNA-seq_counts_XSAnno_PE.tsv\"), \n",
    "                     col_names = c(\"gene_id\", \"Chr\", \"Start\", \"End\", \"Strand\", \"Length\", \n",
    "                     \"Bulk_B_1001\", \"Bulk_B_1002\", \"Bulk_B_1003\", \"Bulk_B_1004\",\n",
    "                     \"CD8pos_T_1001\", \"CD8pos_T_1002\", \"CD8pos_T_1003\", \"CD8pos_T_1004\",\n",
    "                     \"Effector_CD4pos_T_1001\", \"Effector_CD4pos_T_1003\", \"Effector_CD4pos_T_1004\",\n",
    "                     \"Mature_NK_1001\", \"Mature_NK_1002\", \"Mature_NK_1003\", \"Mature_NK_1004\",\n",
    "                     \"Mem_B_1002_1\", \"Mem_B_1002_2\",\"Mem_B_1010\",\n",
    "                     \"Monocytes_1001\", \"Monocytes_1002\", \"Monocytes_1003\", \"Monocytes_1004\",\n",
    "                     \"Myeloid_DCs_1001\", \"Myeloid_DCs_1002\",\"Myeloid_DCs_1004\",\n",
    "                     \"Naive_B_1001\", \"Naive_B_1002\", \"Naive_B_1004\", \"Naive_B_1010\",\n",
    "                     \"Regulatory_T_1001\", \"Regulatory_T_1002\", \"Regulatory_T_1003\", \"Regulatory_T_1004\"), \n",
    "                     skip = 2) %>% dplyr::select(-Chr, -Start, -End, -Strand, -Length, -Mem_B_1002_2, Mem_B_1002 = Mem_B_1002_1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "d50edc1a",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-07-27T14:53:57.560123Z",
     "start_time": "2022-07-27T14:53:54.347Z"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1mRows: \u001b[22m\u001b[34m27762\u001b[39m \u001b[1mColumns: \u001b[22m\u001b[34m8\u001b[39m\n",
      "\n",
      "\u001b[36m\u001b[39m \u001b[1mColumn specification\u001b[22m \u001b[36m\u001b[39m\n",
      "\u001b[1mDelimiter:\u001b[22m \"\\t\"\n",
      "\u001b[31mchr\u001b[39m (5): gene_id, Chr, Start, End, Strand\n",
      "\u001b[32mdbl\u001b[39m (3): Length, LCL8664_R1, LCL8664_R2\n",
      "\n",
      "\n",
      "\u001b[36m\u001b[39m Use `spec()` to retrieve the full column specification for this data.\n",
      "\u001b[36m\u001b[39m Specify the column types or set `show_col_types = FALSE` to quiet this message.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "M <- read_tsv(paste0(dir,\"Macaque_cross-tissue_RNA-seq_counts_XSAnno_PE.tsv\"), \n",
    "                     col_names = c(\"gene_id\", \"Chr\", \"Start\", \"End\", \"Strand\", \"Length\", \"LCL8664_R1\", \"LCL8664_R2\"), \n",
    "                     skip = 2) %>% dplyr::select(-Chr, -Start, -End, -Strand, -Length)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "438691a5",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-07-27T14:53:57.640478Z",
     "start_time": "2022-07-27T14:53:54.349Z"
    }
   },
   "outputs": [],
   "source": [
    "#join counts and assign gene_id to row_names\n",
    "cts <- inner_join(AS, M, by = \"gene_id\") %>% \n",
    "    inner_join(C, by = \"gene_id\") %>% \n",
    "    column_to_rownames(var = \"gene_id\") %>%\n",
    "    as.matrix()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "abc46c04",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-07-27T14:53:57.671584Z",
     "start_time": "2022-07-27T14:53:54.352Z"
    }
   },
   "outputs": [],
   "source": [
    "# Prepare dataframe of sample info\n",
    "RNames <- c( \"GM12878_R1\", \"GM12878_R2\", \"LCL8664_R1\", \"LCL8664_R2\",\n",
    "             \"Bulk_B_1001\", \"Bulk_B_1002\", \"Bulk_B_1003\", \"Bulk_B_1004\",\n",
    "             \"CD8pos_T_1001\", \"CD8pos_T_1002\", \"CD8pos_T_1003\", \"CD8pos_T_1004\",\n",
    "             \"Effector_CD4pos_T_1001\", \"Effector_CD4pos_T_1003\", \"Effector_CD4pos_T_1004\",\n",
    "             \"Mature_NK_1001\", \"Mature_NK_1002\", \"Mature_NK_1003\", \"Mature_NK_1004\",\n",
    "             \"Mem_B_1002\", \"Mem_B_1010\",\n",
    "             \"Monocytes_1001\", \"Monocytes_1002\", \"Monocytes_1003\", \"Monocytes_1004\",\n",
    "             \"Myeloid_DCs_1001\", \"Myeloid_DCs_1002\",\"Myeloid_DCs_1004\",\n",
    "             \"Naive_B_1001\", \"Naive_B_1002\", \"Naive_B_1004\", \"Naive_B_1010\",\n",
    "             \"Regulatory_T_1001\", \"Regulatory_T_1002\", \"Regulatory_T_1003\", \"Regulatory_T_1004\")\n",
    "\n",
    "cell_type <- c(\"GM12878\", \"GM12878\", \"LCL8664\", \"LCL8664\",\n",
    "             \"Bulk_B\", \"Bulk_B\", \"Bulk_B\", \"Bulk_B\",\n",
    "             \"CD8pos_T\", \"CD8pos_T\", \"CD8pos_T\", \"CD8pos_T\",\n",
    "             \"Effector_CD4pos_T\", \"Effector_CD4pos_T\", \"Effector_CD4pos_T\",\n",
    "             \"Mature_NK\", \"Mature_NK\", \"Mature_NK\", \"Mature_NK\",\n",
    "             \"Mem_B\", \"Mem_B\",\n",
    "             \"Monocytes\", \"Monocytes\", \"Monocytes\", \"Monocytes\",\n",
    "             \"Myeloid_DCs\", \"Myeloid_DCs\",\"Myeloid_DCs\",\n",
    "             \"Naive_B\", \"Naive_B\", \"Naive_B\", \"Naive_B\",\n",
    "             \"Regulatory_T\", \"Regulatory_T\", \"Regulatory_T\", \"Regulatory_T\")\n",
    "             \n",
    "replicates <- c(\"Rep1\", \"Rep2\", \"Rep1\", \"Rep2\",\n",
    "             \"Rep1\", \"Rep2\", \"Rep3\", \"Rep4\", \n",
    "             \"Rep1\", \"Rep2\", \"Rep3\", \"Rep4\",\n",
    "             \"Rep1\", \"Rep2\", \"Rep3\",\n",
    "             \"Rep1\", \"Rep2\", \"Rep3\", \"Rep4\", \n",
    "             \"Rep1\", \"Rep2\",\n",
    "             \"Rep1\", \"Rep2\", \"Rep3\", \"Rep4\", \n",
    "             \"Rep1\", \"Rep2\", \"Rep3\",\n",
    "             \"Rep1\", \"Rep2\", \"Rep3\", \"Rep4\",\n",
    "             \"Rep1\", \"Rep2\", \"Rep3\", \"Rep4\")\n",
    "\n",
    "coldata <- data.frame(row.names = RNames, cell_type, replicates)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "d95bc69e",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-07-27T14:53:57.696557Z",
     "start_time": "2022-07-27T14:53:54.356Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] TRUE\n"
     ]
    }
   ],
   "source": [
    "#check that the coldata matches data. It should print TRUE. \n",
    "print(all(rownames(coldata) == colnames(cts))) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "e88fd328",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-07-27T14:54:07.552364Z",
     "start_time": "2022-07-27T14:53:54.360Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: S4Vectors\n",
      "\n",
      "Loading required package: stats4\n",
      "\n",
      "Loading required package: BiocGenerics\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "\n",
      "The following objects are masked from package:dplyr:\n",
      "\n",
      "    combine, intersect, setdiff, union\n",
      "\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, append, as.data.frame, basename, cbind, colnames,\n",
      "    dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,\n",
      "    grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,\n",
      "    union, unique, unsplit, which.max, which.min\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "\n",
      "The following objects are masked from package:dplyr:\n",
      "\n",
      "    first, rename\n",
      "\n",
      "\n",
      "The following object is masked from package:tidyr:\n",
      "\n",
      "    expand\n",
      "\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Loading required package: IRanges\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "\n",
      "The following objects are masked from package:dplyr:\n",
      "\n",
      "    collapse, desc, slice\n",
      "\n",
      "\n",
      "The following object is masked from package:purrr:\n",
      "\n",
      "    reduce\n",
      "\n",
      "\n",
      "Loading required package: GenomicRanges\n",
      "\n",
      "Loading required package: GenomeInfoDb\n",
      "\n",
      "Loading required package: SummarizedExperiment\n",
      "\n",
      "Loading required package: MatrixGenerics\n",
      "\n",
      "Loading required package: matrixStats\n",
      "\n",
      "\n",
      "Attaching package: matrixStats\n",
      "\n",
      "\n",
      "The following object is masked from package:dplyr:\n",
      "\n",
      "    count\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: MatrixGenerics\n",
      "\n",
      "\n",
      "The following objects are masked from package:matrixStats:\n",
      "\n",
      "    colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,\n",
      "    colCounts, colCummaxs, colCummins, colCumprods, colCumsums,\n",
      "    colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,\n",
      "    colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,\n",
      "    colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,\n",
      "    colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,\n",
      "    colWeightedMeans, colWeightedMedians, colWeightedSds,\n",
      "    colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,\n",
      "    rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,\n",
      "    rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,\n",
      "    rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,\n",
      "    rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,\n",
      "    rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,\n",
      "    rowWeightedMads, rowWeightedMeans, rowWeightedMedians,\n",
      "    rowWeightedSds, rowWeightedVars\n",
      "\n",
      "\n",
      "Loading required package: Biobase\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: Biobase\n",
      "\n",
      "\n",
      "The following object is masked from package:MatrixGenerics:\n",
      "\n",
      "    rowMedians\n",
      "\n",
      "\n",
      "The following objects are masked from package:matrixStats:\n",
      "\n",
      "    anyMissing, rowMedians\n",
      "\n",
      "\n",
      "converting counts to integer mode\n",
      "\n",
      "Warning message in DESeqDataSet(se, design = design, ignoreRank):\n",
      "some variables in design formula are characters, converting to factors\n"
     ]
    }
   ],
   "source": [
    "#normalize counts with DESeq2:\n",
    "library(\"DESeq2\")\n",
    "dds <- DESeqDataSetFromMatrix(countData = cts,\n",
    "                              colData = coldata,\n",
    "                              design = ~ cell_type)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "c85302ff",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-07-27T14:54:48.602159Z",
     "start_time": "2022-07-27T14:54:44.780Z"
    }
   },
   "outputs": [],
   "source": [
    "#VST normalize and use variance accross all samples\n",
    "vsd <- vst(dds, blind=TRUE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "0ff0e16b",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-07-27T14:55:52.628704Z",
     "start_time": "2022-07-27T14:55:51.694Z"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message:\n",
      "This manual palette can handle a maximum of 10 values. You have supplied 11.\n",
      "Warning message:\n",
      "This manual palette can handle a maximum of 10 values. You have supplied 11.\n"
     ]
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAApQAAAHgCAIAAACU9mohAAAACXBIWXMAABJ0AAASdAHeZh94\nAAAgAElEQVR4nOzdd1xTV/sA8Ju9CGHvTVjKlClbqoCrjjqg7opiq/2pVTtsX0dtba2tr7a2\nFSeuiorbitQtIkNBhrKnMmWFEbJzf3+kb5oyIiIZyPP9w09y7npyiPfJPffcczAoiiIAAAAA\nGD6wqg4AAAAAAK8HkjcAAAAwzEDyBgAAAIYZSN4AAADAMAPJGwAAABhmIHkDAAAAwwwkbwAA\nAGCYgeQNAAAADDN4VQcwSF9++eXTp097FKIoiqIoBoPBYDAqiapHMAiCqEkkUC19EovFUC29\nqVW1oCiKxar4GsPNze3rr79WbQwA9DBck3dzc/PevXvNzc1lCwUCQXt7O4VCodFoqgpMis/n\n8/l8DQ0NVQeCCIVCFoulJtUiEAi4XC6dTld1IH9XC5lMVoe/kfpUi0gkamtrU5NqEQqFHA5H\ntdVSXV29Y8cOFQYAQJ+g2RwAAAAYZiB5AwAAAMMMJG8AAABgmIHkDQAAAAwzkLwBAACAYQaS\nNwAAADDMQPIGAAAAhhlI3gAAAMAwA8kbAAAAGGYgeQMAAADDzHAdHhUAWc3Nza2trQQCQV9f\nXx3G9QQAAIWC5A2Gt5ycnPv37+fk5FAoFJFYZGxkbGVlNWHCBBMTE1WHBgAAigLJGwxXKIpe\nvnz52rVr7h7uc6PmkkgkBEE6OzoLCgoOHz787rvvurq6qjpGAABQCEjeYLhKSUlJTk6eMmUK\nXfOfWafomnRfP9/KysrLly/r6OiYmZmpMEIAAFAQ6LAGhqWurq6UlJSAwADZzC1lbW1NIBL+\n+usv5QcGAABKAMkbDEtFRUXNLc2mpqb9reDs7JySksJisZQZFQAAKAckbzAsNTY2GhgYyFmB\nRqPRNGiNjY1KCwkAAJQGkjcYlgQCAR7/ih4beBxeKBQqJx4AAFCm4dphTSQStbe393iiF0VR\nBEG4XC6fz1dRXP8KBkVRgUCg6kD+xuPx1Kda2tra3nA/OByuuamZy+X2t4JQKGSxWBgMpr9j\nSb4tPB5PHf5GQ1UtQ0VNqgVBELFYrNpq6ejoEIlEKgwAgD4N1+SNw+EYDIa2trZsoUAgaG9v\nJ5PJNBpNVYFJ8fl8Pp+vDgOGSNIYiURSh2oRCARcLpdO76OX2WsZM2bMxUsXxSIxlUbtc4Xy\n8nJnZ2dHR0cstu/mJWm1qMPfaKiq5c2JRKK2tjY1qRahUMjhcFRbLR0dHTgcToUBANAnaDYH\nw5Kpqem40HFpaWmSC+geuru7sx5nBQYG9pe5AQBgWINTGxiupk2bpqOjc+fOHQ6HI1ve1NR0\nPel6cHCwr6+vqmIDAACFGq7N5gDQ6fSlS5deuHDhXOI5U1NTLW0tkUjU9LJJQ0Nj8uTJ48eP\nx2Awqo4RAAAUApI3GMY0NTUXLVoUGhpaUlLCYrFwOJyfr9/o0aN7dIYAAIC3DCRvMOxZWlpa\nWlqq6ugikQg6NAEAlOztT958Pr+oqOj58+ccDodGo1lZWdnb27/yEWEwEnA4nOzs7Lq6us7O\nTiKRaGJi4ubmZmFh8coNURR9+vRpdnZ2TU2NQCCgUCjW1tZjx441NzdXQtgAAPCW57CCgoJr\n1649SEnR1dUlEAh8Pr+5uTnsnXemTZtmbW2t6uiAKhUVFZ0+fbqrq8uWaaupqckX8HNycy5e\nujhl8pQpU6YQCIT+NuRyuWfPnr2R/NiO6WxkOIZEJHd3d+VmV1w8993ipdMnTJgA99oBAIr2\nNifv7OzsLzdudHBwCAwMlJ5PxWJxaUnJyZMn58yZY29vr9oIgaoUFRXt+u8uD3ePwKBAyVyi\nCILY2dm5u7nfvn2by+VGRUX1mYPFYvHp06cz08onRswhkyiSQjqdYWhoamvtdOrEdTweHxYW\nprxPAgAYkd7aR8VaWlquXbvm4upqbm4uexbGYrE2NjbNTU3nz59ns9kqjBCoCpfLvXz5sre3\nty3TtkeGpmvSJ4RPuJ9yPycnp89tnzx5cvdWblBApDRzS2lr6wWMnZCamvry5UtFhQ4AAAiC\nvMXJOzU1tbi42LCfuSvMzc2fZGdnZGQoOSqgDnJzc+vr6/trd6FSqa6urg8fPuxzaVZWloO9\nK4lEli1ks9llZWVZWVllZVU5WbWHDh1qb28f+rgBAOB/3trkXVpaamJsLGcFYxOT0tJSpcUD\n1EdlZaWFhYWcO9OWFpaPHj/q6urqUS4QCOrr642N/umVhqJoeXn59Wt3K8obeRwiIqZTyUY3\n/kr75ZdfMjMzFfUBAAAj3tuZvIVCYVdXl/yhvGk0Gkz2PDJ1dXXRNOR9NyhUChaD7ezs7FHO\n4XAKn9aRZBrMy8rKnuaX29g4WpjZ6GjrMTS1DfSMrS3ttemjf959HJp2AAAK8nYmbxwOh8Fg\nxGKxnHXEIhE8MDYykclkHo8nZwWBQCBGxWQyuUc5lUp1cjbhcLslb9va2vJzS60s7CjkfyZH\n4fE4JBLFxNjS3/edGzduqM9EYQCAt8nbmbwxGIyBgYH8C2sWi2VoaKi0kID6MDU1ra+rl7NC\nfV2982hnLS2tHuV4PN7MzKyurkrytqamRkdHv8f976aWGgN9EwRBjI0tWl5ioPEcAKAIb2fy\nRhDExcXl+fPn/V18C4XCmpoaFxcXJUcF1IGbmxuFQmlsbOxzqVgszs3L9fDw6POmuI+PT3Fp\nfjeHjSAIi8XSpP8rwbe21eMI3bY2TpK3ZqbWlZWVQx0+AAC8vcnbx8dnXFhYYWFh7ykjxWJx\nfn7+tOnTIXmPTDo6OoGBgffv3e/dNiMSiR48eGBtZR0YGNjnts7Ozr5j7U+e2vvn9YS8pw9L\nK7IaXlZJvmNtrMb8glQXZ28NDU3JyjQqvfeNcwAAeHNv7U1fAoEQHR0tFAoz0tPt7OwYDAYG\ng0FRtKWlpbSsbNy4cXPnzoXJnkes0NBQNpt99uxZp1FOdkw7DbqGQCBoaGh4mv+UyWS+//77\n0pFbZPH5/EuXLhUk3rfnIZ2lBWZCFFNTXil4kmeuo61tSKYIg4IDHR3cpOvz+NzeN84BAODN\nvbXJG0EQPT295cuXM5nMvLy8O3fuSIZHDQwKWr58eUhISJ9nZzBCYDCYyMhIIyOj7OzsgoKC\n8opyK0srHR2dmTNn+vn5EYnE3psIhcITJ05UX8yM8p2oSaa1tbWlpqZ2tHPNNEyq617W4Lpn\nhi0xNbGS3aSxscZ7rFXvXQHw5m7evDlhwoQjR44sXry491vw1nubkzeCIDQaberUqREREfXz\n53d3d2toaBgbG0MncyBhZWXl6OhIJpPZbDaRSKRQeg6aJuvBgwcliQ9mjhlPwhMQBNHW1vb0\n9MxMz2MyRztjcY/r8gsKn5gYW0rvlHd2tlc9L41dOVMZnwSAASspKfnjjz9mzpzp6uqq6ljA\n4I2IdmMikWhpaenk5GRubg6ZG/SAx+MZDIb8zC0SiTIyMvxsXCWZW8LIyMjCyuD5i3KRWORq\n6Nj2MOdlU51kUXd3V0rq9TlR4SqcqxSAPpWUlGzdujUvL0/VgYA3ApkMgFerq6ur+Ss3PHCa\nbCEGg3F2dkaQp2XlBXq6RrpkRk1tJYlEqa2tLC7Jf3dG0JQpU1QVMADg7TYirrwBeEOdnZ0a\nJAoW0/P/C4FAcHd39/IeTaLwxfyOu/euPq9LNTbHfv5l7KxZs6CZB8gnFAp3797t6elJo9Ho\ndLqrq+vmzZtll+7atcvd3Z1CodDp9NDQ0L/++usNj7hly5apU6ciCLJgwQIMBoPBYEJDQ+/c\nuYPBYLZt29Zj5QULFuDx+JqaGgRBEhMTMRhMQkLCl19+aWVlRSKR7Ozsdu/e3fsTDXnMoE9w\ncgHg1YhEIl8k6HMRBoMxNTU1NTXt0kBDZ3rNnDkTcjYYCKFQOGXKlOTk5JCQkE2bNmlqahYV\nFZ09e3br1q0IgohEonfffTc5OXn27NkxMTFcLvfEiRORkZEnT56Mjo4e9EEXL15MIpE2bty4\ncePGCRMmIAiipaXl7u7u4OBw6NChL7/8UvoMDovFOnfu3MSJE83MzKSbr1+/3tPTMzExUUND\nIz4+fu3atY2Njd99951kqYJiBn2Cs8yIg6Lo06dPc3NzGxoaeDweg8GwtbX18/PT1tZWdWjq\ny8jISNPflsXu1KLS+1wBRdD6juZgOzvI3GCA9u7dm5yc/PHHH+/Zs0faz1E6rlRcXFxSUtLh\nw4eXLFkiKVm9erWfn9/atWtnz5496K+ZlZWVZHwLJyen0NBQafny5cvXrVt38+bN8PBwScmJ\nEyc4HM7y5ctlNycSiefOnZMc/fvvv6+urv7hhx+WLl3KZDIVFzPoEzSbjyzd3d3x8fFffPFF\n6oMHrS0tPC63qrLy5IkTv/76a1ZWlqqjU18aGhp2dnZZ1QX9rVD+skYr2N7R0VGZUYFh7cSJ\nExQKZfv27bJj+UkvfI8dO2ZgYBAdHc39H5FIFB0d3djYmJubO+TBLF68mEwmHzhwQFpy4MAB\nMzOzSZMm9VhNNgcvW7ZMLBZfvHhRJTGPcPBTSOHq6urKysra2toIBIKBgYGjo6OGhoZKIhEK\nhSdPnrx961ZgQACB8HevaQaDYWxs/PLly6+++mrHjh3w9Eh/wsPDD1RX55YVu5k79Fj0srP1\nfknWgg/Wyu+yDoCskpISJpPZ39mgsLCwo6Ojz2/Uy5cvhzwYHR2dWbNmnTlzpqmpSV9fPzMz\nMy8vb9OmTTgcTnY1W1tb2bc2NjYIgpSXl6sk5hEOkrcCsdnsS5cunTh+XFtHh0ajicViFos1\nZsyYoKCgoKAgOfNJK0haWtq1a9f8fH17t18ZGBiMcnK6du0ak8mkUql9bj7C6enpzZ0798yZ\nM3WPHria2xtp6mIw2HZOZ3FDVX5N6cyvlnt5eak6RjCcoCgq5yQgFovt7OyOHTvWe5GCGnhW\nrFhx4sSJY8eOrVu37sCBA1gsdunSpT3W6TEdn+StbJu/kmMeySB5K0pnZ2d8fHxebm5gYKDs\nGJktLS0/7tzZ2to6ffp0ZcaDouijR49sbWz6u/NkbGyclp6el5fn5+enzMCGERsbmxUrVtx1\nvJtbWHgl+R6CQYzHOVuE2a4L+cDa2lrV0YFhxsHBoaCgoKurq8+Lb3t7+6dPnzo7Ow95Q11/\nvxgCAgKcnZ0PHjy4fPnyhISEiIgICwuLHus8ffq091vJ9bdCYwa9wT1vRbl8+fKT7GwPD48e\no1vr6up6e3sfPHAgJydHmfGwWKzk69f19fX7WwGDwRjo61dXVyszqmFHS0tr+vTpn3766bbE\n37Yk/PLpp58uXrwYMjcYhPnz53M4nP/85z+yhdKJlBYuXMjn89evX99jaqW6uro3PC6dTkcQ\npLW1tfei2NjYoqKiVatWdXV19eiqJnHkyJGGhgbJa4FA8NNPP2EwmGnTpik6ZtAbXHkrRE1N\nTUJCwlg/vz5/5FKpVFsm8969e25ubkprPO/u7sbicPI7fJJIJDabrZx4hjUcDqenp6fqKMDw\ntnLlyqtXr+7evTsnJ2fixImampqlpaXJycmSy9mVK1fevHkzLi7uyZMn06ZN09fXf/HiRVpa\nWm5u7hveP3ZzcyOTyb/88guRSNTS0jIwMAgLC5MsWrBgwWeffXbs2DFjY+M+hxiytbX19fVd\nsWKFhobGH3/8kZ6evmHDBjs7O+knUlDMoDdI3gpRXFysq6srZ+4TE2PjpGvXoqOjDQwMlBMS\nhUIRi0RCoVBO/ubx+XDDGwDlIBAISUlJu3fvPn78+ObNmwkEgrW19ezZsyVL8Xj8pUuXDhw4\nEB8f/9133wmFQiMjI3d39127dr3hcRkMxh9//LF169Y1a9bweLyQkBBp8mYwGHPnzj1y5MgH\nH3zQ54li48aN5eXl+/btq6mpMTc3/+mnn9auXStdqriYQW+QvBWCxWLRaDQ5K+DxeDKZ3NbW\nprTkra2tHR4R0VBfb2Rk1N86TU1NMBY3AEpDIBA2bNiwYcOGPpdisdjY2NjY2Ng+l44fP162\ndbrHW/lmzJgxY8aMPhcRiUQsFhsTE9NfSHICfmXMYAhB8lYIHA6HisWdnZ2tra1dXV0oilIo\nFH19fV1dXdmemT0ew1AoDAbj6en5888/6+vr93nchoYGT09PeFQMgBGrra3txIkTERERVlZW\nqo4FvAIkb4XQ0dEpLCysfv7c0MBAk8HAYbFdXV15+fk0KtXV1ZVCoXA4HE8vL6Vddkv4+/sX\nFRWl3L/v6urao02spaWloKBg2zffyG8wAACoORRFezzQJUtyYd27PCcnJz8///Dhw93d3Rs3\nblRkgGBoQPIeeiKR6OnTp9o6OrY2NrLPWggEgsKioozMTD9f3+rqav+AAE1NTWUGRiQS582b\nh8Phrly+bGFhoa2tjScQutnshoYGpp3dfzZtGjNmjDLjAQAMuWfPnkkGQO3TqVOnoqKiepef\nOHHip59+MjMz+/XXXwMDAxUZIBgakLyH3oMHD4RCYUxMzKGDBw0NDaXd1ggEgouzc15eXurD\nh/7+/pJZAZSMTqcvXbp0zJgxubm59fX1AoHA0MjIPyDA399fzlNkAIDhgslkPnr0qL+lPYZI\nk/rxxx9//PFHObudNWvWwO+pAyWA5D3ERCJRWlratGnTrKysWCzW+XPn7O3tDQ0NJU1VPB4P\nh8Px+Xxvb2/ZuXqUCYvFenp6enp6IggiFov7bEMDAAxTZDIZBvsbCSB5D7GamhoURe3t7UUi\n0Xvvvefo6JiSknL79m0qhSISiThc7pw5c9hstpynyJQJMjcAAAxHSk3eN2/evHfvXlVVFY/H\nMzExmTx5srTp+M8//4yLi5Ndedu2bW5ubsoMb0h0dHQwGAwcDicSiTAYjJ+fn4+PT3R0tGRi\nEkNDQ21t7evXr7e3t6s6UvBqAoGAxWKJxWIGg9FjpDwAAFAhpSbv27dvjx49etq0aVQq9eHD\nh7/88otQKJw4caJkKZ1O37Ztm3RlExMTZcY2VAgEAp/Ply3BYrFmZmayjeR8Pp9IJCo9NPAa\nmpub7969W1xc/PJGCgaD0Rk31srKKiQkBJ6DBwCoA6Um7+3bt0tfjxo1qrKyMjU1VZq8cTic\ndID74cvIyKi9vb29vb2/2SFRFK2urg4KClJyYGDgioqKzp8/z8gpDTWzNAgJxyCYtk72sxsP\nf45PmPH1RuiLCwBQOVXe8+bz+bIPOnd2di5cuFAoFJqZmU2bNi0gIECFsQ2alpaWpaXlgwcP\n+utMXlBQwGazHRx6zgkN1ERdXd25c+dcn7eNHvXPXRtdmkawraOtnuHFzd9r/nc7DGUDAFAt\nlSXvmzdvlpWVSSeuMTc3//DDDy0tLfl8/r1793bs2BETE/Puu+9K1y8pKUlMTJS+5XA43d3d\nXV1dsvsUi8UIgggEgh7lShYQEHD8+HEGgzF69OgekTQ2Np4/fz44OFgkEiktSMkDHiqvFgmx\nWKzMzy4/EqSvarl06ZJ+XrmdrWOP2x8IguhTaP6WtleuXDEzMxvaGx/qUy3wbemhu7tb8lUB\nQK2oJnmnpKTs27dv7dq10uloXF1dpVczLi4ubDb73Llzssm7trb2/Pnz0re2trY8Ho/L5fbe\nuVAoFAqFigz/FbS1tSMiIq5du1ZeXu7l5WVoaIggSEdHR35+fmZmpru7++jRo/uMXKFUXi2y\nRCKRqkP4m0gkkg2mra3t6clzC7wC+qsrOz2D9FsPC0JCHB0dFRHMkO9zcHpUi2qpNhIejwfJ\nG6ghFSTvpKSkQ4cOrV+/3s/Pr791nJycUlNTZafA8vLyOn78uHSFn376SVNTU0tLS3YroVDY\n1dVFIpH6u9+sNL6+vubm5rdu3UpMTMRgMFgsls/nm5iYzJ8/39LSsrKysq2tDYPB6OrqWllZ\nyZ+m882JRKLOzk51qBYEQYRCIV895i6TVAuRSJQNpq6uTo+uqafJkLOhpa4+m83u8d17Q+pT\nLWKxuKOjo0e1qIpIJOLxeKqNpKOjQ5lzEAAwQMpO3gkJCefPn//Pf/4j/zGwwsJCLS0t2axG\np9OdnJykb3E4HK7X7NSSFj8sFqvodDgQRkZGs2fPJhAIzc3NIpFIW1ubRqOlpqZevHiRQCAY\nGBigKJqamoogSGhoqLe3t6In9laTakFRFIPBqEMkEj2qRSQSkfAE+Y+/kwgE+TOrDoL6VIvk\nMldNvi0Igqi8WnA4nKL/bwIwCEr9X3HgwIFr164tX76cTqdXVFQgCEIgEMzNzREE+fXXX52c\nnIyNjfl8/v3791NTU5csWaLM2BSERCKZmpoiCCIWi8+ePVtbWztjxgwHBwfJ6UAsFj99+vT6\n9esNDQ2y9wiAqtDp9E4eB0VQDNLv+bqdy3FQ7qD0AADQg1KT9927d0Ui0e+//y4tMTIy2r9/\nP4IgRCLx9OnTLS0tRCLR1NR0w4YNb9nDVPfv36+rq1u2bBmdTpcWYrFYV1dXU1PTQ4cOmZiY\nwKCGKmdhYaER4vviZYuFtl6fK7D5vDpWa38DRAMAgHIoNXmfPHmyv0XLli1btmyZMoNRJg6H\nk5qaOnfuXNnMLaWrqxsREfHXX395eHjA3TXVIhKJ3t7eKT8fmOOhTej1t0AR9G5podviqCEf\nQYjD4eTl5bW2tvJ4PA0NDWtra3t7exi8FgDQH7W4rfXWKysr09DQYDKZ/a3g7OyclJRUXV39\nFgxTM9yNGzeusrLy4rW7kaPc6KR/hkTli4R3Swu5PqMXDPUNjuzs7KSkJO79R9Z6hiQ8/jmP\ne7+1yfTd8OnTp6tq9hoAgJqD5K0MLS0txsbGclbA4XCGhobNzc2QvFWOQCAsWrTogqbmsYMn\nmHqGhpoMDIJp7e4qfdlgPWvy0tmzh7af+YMHDy5t+T6U6WTjOZZE/Hu6GqFYlJFZcLS9XfJ4\nwhAeDgwXHR0dT548qa6u5nA4ZDLZwsLCw8NjaL97YFiD5K0MA5kHF4PBwHS5aoJMJkdHR/v7\n+z979uzly5disVhHW3uhg4OTk9PQdjyuqak5v2n7DDdvYzpD9slyPBYXYGNPKCtPTEz8+OOP\nFT0SPoqidXV1ra2tOBxOR0dHTfqZj2SPHj3666+/yqo4RsaWJDKNz+Pef3gnNTV13Lhxw3To\nSTDk4KaaMujq6jY0NMhZQSwWNzY26un13UkKqISlpeWkSZMWL178wQcfzJgxY9SoUUP+yFBK\nSsooI1NThnafS70sbLpup2VnZw/tQWWhKPr48eNdu3Zt3rz58OHDcfvjvtj4xeHDh8vLyxV3\nUCBfWlra9z8eQQh2gSFTmfau5hZ2tnYu/kGT8dRRu/eeunv37pvsvLu7+/vvv/f09KTT6XQ6\n3dnZecWKFfn5+ZKla9aswWAwPj4+spuUlZVhsVgMBpOeni4pycjImDlzprW1NQaDiYmJkV35\n8OHD48ePNzAw0NDQ8PDwOHTokHSRWCzevn27vb09hUIxMTGZN2/e8+fPJYu8vLww/4bD4To7\nO+VvNcJB8lYGJpPZ2dlZWVnZ3wrPnj1DEAQaSEcUkUhUXl7uZGTa3wpYDMbBwLi0tHRIDsfl\nchsaGurr66Wj+6EoevHixYOHDmpqakZFR02bPm3GjBlz585FEfS/u//78OHDITkueC3Nzc03\nb9709h1vZNLzbKBvYOrjF75n76m6urrB7bytrW3s2LHff//9e++9d/78+Xv37n399dd0On3h\nwoXSdQwNDfPy8p48eSIt2b9/v4WFhex+2Gw2k8n89ttve/fjOXbs2NixY48cOXL9+vXQ0NCY\nmJh9+/ZJFu3cuXPr1q1ffPFFfn7+yZMnc3Jypk2bJln0xx9/PJHBZDInTpwo6d4rZ6sRDtrH\nlIFKpfr5+V2+fDkmJoZGo/VYymKxJF90aK4cUdhsdtPNB1oBYXLW0aHSilmsNzxQbW3tnTt3\nKioqWHfTEQTRCvWztraW9Mu7cePG5MmTZR+CIJFILi4uJiYmhw4f0tfXlw5gDJQjIyOjthHr\n7df3TzodXUNTc9u0tLT33ntvEDvfsGFDaWlpXl6eNOmOGTNm5syZsgPQamlpjR8/Pi4uTpJ0\n+Xx+fHz8//3f//3nP/+RrhMWFhYWFoYgyA8//NDjELINA4GBgTk5OWfPnl2xYgWCIA8ePAgM\nDJQM4MFkMleuXLly5Uoej0cikezt7aVbZWVllZWV7dmzR/JWzlaDqIG3CVx5K0loaKient7B\ngwdlGyRRFC0sLDx48KCtrW2Ppirw1sPj8dqhfiK542YLxeI3/EmXlZV15MgRXtL9yE7MyuAJ\nq4LDIzsxgusPDhw4cOrUKf8A/z4fXzQ0NHRzc0tOToZ+GEpWUVFhbGotZwUTUxvJCFevi8/n\nJyQkLF68uPflco8nVGNjY//44w/JfDDnz58nk8nSiZtfF5fLlU4dGRoampWVlZaWhiBIfX39\nmTNnIiMje+fgX3/91draOjIy8rW2GoHgUk9JcDhcdHT0vXv3EhISNDQ0JF/ouro6gUAQHBw8\nduxYGIJxpKFQKJqamvXlL+30jfpbp669zcho7KAPUVZW9seG/0wc7W5jN0paqEejh9o5PS4u\nPP7gNuLv39+29vb2R+OPWlpaamtr6+jo2NrawhlTCTo7O6k0ebfPaDR6WyNbJBK97pgQNTU1\nbDbb2dlZWrJ06dKjR49KXufm5o4ePVryOigoyNzc/I8//li+fHlcXFxMTMzgxp84fPhwVlbW\nzz//LHm7bt06Pp8fHByMIIhQKIyIiJCdKFKira0tISFh8+bN0kEOBrLVyARX3sqDxWLHjRv3\nySefvPPOOyYmJqamppMmTVq7dq2/vz9k7hEIg8G4uLhkv6hCkb6vbtl8XlFjvYuLy+D2LxaL\nk5KSfKyYNroGvZdq44muBiYZGRmouI+j19XVZWVlNTa2nj6VfO7MvZ3fH/zpp5/u3r0L82sp\nGolEEgh6zkUri8/nEQiEQWTT3o0oX3/9dU5Ozr59+0QiUY+lsbGx+/fvLykpebutHhAAACAA\nSURBVPDgwdKlS1/3WAiCnD59euXKlUePHvX29paUJCYm7ty5c+/evdnZ2deuXaurq5szZ06P\n4x45ckQsFn/wwQfSkoFsNTLBlbeyUSgU2R+/YCQLCAjIy8tLySkOsO55a5kvEl57luPxQbSc\nsX3kq6mpqbp4Pdy/73vqKIraMnQfPsxuCPCXHYQARdGysrKy0lpTE0tjI4sx7gHWVg4oijY0\nvDgef7Wuri4qKgqGflMcU1PTzCd1hkbm/a3Q3FTnxOy3k6Mc5ubmVCr16dOnsscyNTXt80GY\nhQsXfv755x999NGkSZNMTU2bmppe61j79u375JNPTp06NX36dGnhJ598smjRotjYWARBXFxc\ntLW1x44dm5aW5v+/5h8URX///fdZs2bp6+sPfKsRC/4TAqAyVCo1Ojq6ydX2cn52Y2e75BJc\nJBaXNTcmZKXpTAmbOXPmoHfe0NBgSGf0HuRVgkwmC3g8PZpGa0urbHldXV1p8QtrS3u6BoPN\n7qTR6AiCYDAYY2OL8WHTb/315A0fVQLyubu7V1cVcbndfS7l83kV5c88PDwGsWcikRgVFRUf\nH19WVvbKlbW0tObMmXPr1q3ly5e/7oG+/vrrDRs2XL58WTZzIwjS3d0t+7NP8lq2r9xff/1V\nVlb24YcfvtZWIxZceQOgSiYmJrGxsclWyTfz8ngPbpPw+G4+zyAiJHTWx4GBgW/SW43P5/eX\nuREE0dXV7e7uRih4gVAgLURRtLq62tDQjEgkNTTUatIN9XT/uR9PJlO9PYPT0tL8/PzUYbbv\nt5Kjo+N70wKTb93wC4ggEsmyiwQC/qOMm5MjPF1dXQe38507d2ZkZHh7e3/66afe3t56enov\nXrw4cOAABoPpPRDQvn37fvzxR11d3d774XA4xcXFkhetra05OTkYDEYyy/OaNWt+/fXXX375\nRU9PLycnB0EQEokkmc15xowZcXFxrq6u/v7+NTU169evt7a29vT0lO72t99+c3V17TEKzSu3\nGrEgeQOgYpqamtOnTx83bhw7hs3hcOh0urGx8Zs3TWtqarZz+r6AQxCEQqE4ODg8epRqTaFI\nC1ks1suGDlsbMy6PW1CY5+UxvkcYhoamWU9SSktLJWdqoAjvvfeeUCj88/p5poObkbEFhULj\ncjmNDc9Li3MnjHOeM2fOoLvI6OjoZGRk7Nmz5+zZs998841QKDQxMQkJCcnIyJB9WEuCTCaT\nyeQ+91NcXCy9+i8pKblw4QIOh5MMEXjixAmhUCh79Wxrayu51t+zZ4++vv62bdtqa2u1tbUD\nAwO3b98u/RX4/PnzP//8c+/evT2OJX+rkWy4DskZGxv71VdfSeYClxIIBO3t7RQKpfez1MrH\n5/P5fL6GhoaqA0GEQiGLxVKTahEIBFwut8/Hk5RMUi1kMlkd/kaKqJaurq5du3aFdyAGGn3P\nPv6yo/2bkke2trZh74QxGAwEQerq6nKyS2hUzfyn2ZbmowPGhvfOEw/Tb06c4iV5zFfRhEKh\n5NeMEo7Vn+rq6h07dvz222/KPKhYLM7Ozs7MzHz58uWTvFoPV1NdXV1vb29vb2+YeBBIwJU3\nAG8nDQ0NNze3u4dPvefug8P0vI4XoeL7FcULFy7E4XCXLl7SN9DX0dFpbm7OzS1k0I2d7P08\n3P3gIQhVwWKxXl5eXl5e3d3dbDabSqWqwy9voFYgeQPw1oqIiHjx4sXl2+nhji404j9PaXfz\neTeKn5LD/KZOnUqhUIKDg0tKSlpbW5uamp7lVoe/856mphamV76XYLGa+7wPChSBSqVCEzHo\nEyRvAN5aZDJ5yZIl5xmMo/GnbfUMDOgMBEGaujrKmhpdF82ZOXMmhUJBEERfX1/ycI5IJGpq\nampuadDU7HvqycbGWitbBoyZCoDKQfIG4G1Go9EWLFgQGBhYUFDQ0tKCoqi5nl64k1OfM8fj\ncLjQ0NDdPx010DfR0ek5xx2X2/0o6978RZPhWhAAlYPkDcDbz9ra2tpa3ojZUp6enpOmliRf\nu+LtFWxuZiO57S0ZpCXryYN3wseEhIQoOFgAwKtB8gYA/AODwURGRpqbm2dkZFy6kqqjo49B\nMG2sZjtHvYVLpgYFBcHwagCoA0jeAIB/wWAwXl5eAQEBFRUVzc3NYrFYV1cXJiYBQK1A8gYA\n9IFAIDg4ODg4OKg6EABAH6AFDAAAABhmIHkDAAAAwwwkbwAAAGCYgeQNAAAADDOQvAEAAIBh\nBpI3AAAAMMxA8gYAAACGGUjeAACgvsRisapDAOoIBmkBACiJWCwWCAQwUttAlJeXZ2ZmVldX\nc7lcEolkYWHh5eUFY+YAKbjyBgAollAoTE9Pj4uL27Zt23fffffdd9+dOXOmtrZW1XGpKZFI\ndOnSpePLl9P+/HNaQ8PS7u4ZjY2M5OTTK1eePXtWIBAMes9r1qwxMzPrc1F3d/e3337r7u5O\no9G0tbW9vb23bNnCYrHkbCUWi7dv325vb0+hUExMTObNm/f8+XPp0vb29jVr1pibm5NIJCsr\nq2+//XYgiyQePnxIIBDweLi2lAdqBwCgQO3t7adOnXr27Nmo0aNc3VzxeHxXZ1dZedn15OsL\n5i8YN26cqgNUO9euXSvbt2+Zl5cOhSIp0aNSLbW0vE1NE+LjLyLI7Nmzh/aILBYrJCTk+fPn\nX3zxhY+Pj5aWVkFBwcGDBzU0NNavX9/fVjt37ty6deu+ffuCgoJevHixatWqadOmPXnyBEEQ\nLpc7btw4gUDw/fffM5nM1tbWzs5OyVZyFkk0NzdHR0dHRERcv359aD/mWwaSNwBAUfh8/smT\nJxsaG6ZNnya9kKLT6cYmxkwmMyEhgUQi+fv7qzZItVJdXf34v/+VzdxSmiTSXBeX/XFxxa6u\nQ9t+vmHDhtLS0ry8PCaTKSlxd3d///33m5qa5Gz14MGDwMDAJUuWIAjCZDJXrly5cuVKHo9H\nIpF27979/PnzkpISHR2dHlvJWYQgiFgsnjdv3pIlSzQ0NCB5ywfN5kAhurq6GhoaWltbobvN\nSJaamlpaWhoSEtK7CdTAwCAoOOjOnTvt7e0qiU09PX782NXISJdK7XMpg0weY2Ly+PHjITyi\nQCA4derUokWLpJlbSl9fX86GoaGhWVlZaWlpCILU19efOXMmMjJS0qEhMTExLCzsyy+/NDY2\ntrOzW758eUtLi2QrOYsQBNm2bRufz9+0adMQfsC3FVx5g6GEomheXt7Dhw8vXriAYDAIir4z\nfrybm1twcDCl15UEeLuhKJqdne3q5orD4fpcwczMLD8vPzc3Nzg4WMmxqa3nz5+P6+uSVIqp\no3NB5tbym3vx4gWbzXZxcXndDdetW8fn8yV/O6FQGBERkZiYKFlUXl6en58/Y8aMy5cvNzc3\nr169etKkSWlpaVgsVs6imzdv7tu3Lzs7G+aMHwhI3mDIiESic+fOHTt2zMbaOjg4mEwmi0Si\n1paWQwcPFhYWzps3T/4PefCWYbFY+U/z33//fTnrmJqavnjxQmkhqT8Oh0MjEuWsoEEk8ng8\nkUjU30+i14Wi6OA2TExM3Llz5969e/39/Wtqaj777LM5c+ZcvXoVg8GIRCItLa1jx44RiUQE\nQchkclhYWGpqalBQUH+L7Ozs5s+ff/ToUWNj4yH5XG89SN5gyCQlJZ06dWqsnx+ZTJaU4HA4\nfX19PT29/Ly8kwgSGxsLPUhHDi6Xi8ViiXJTEZlC5nA4SgtJ/VGp1C4+X84KnXw+WUdnqDI3\ngiDm5uY0Gi0/P/91N/zkk08WLVoUGxuLIIiLi4u2tvbYsWPT0tL8/f1NTEz09PSkf3pnZ2cE\nQaqqqoKCgvpbxGazGxsbJ0+eLClHUVQsFuPx+C+//HLr1q1D8knfMsP1TIqiKJ/P5/F4soUi\nkUjyb49ylRAKhWoSieSus6KDaWxs/OXnn319fQkEguQPIcve3j4lJcXe3j40NFRNqkWtvi0i\nkUgsFqtDJEP4bcHj8SKhqKurS84dk87OTgN9g/6OJRaLVV4tfD5/0Nemg2BpaVmanu6op9ff\nCqXNzVZ+fkN4RCKRGBUVFR8fv27duh63vZuamuS0lnV3d8u2b0teS/5bBQcHX7lyRSAQEAgE\nBEEKCgoQBLG2tpazyN3dXfYHRHx8/O7du3NycgwMDIbww75NhnHyFgqFPR55lJx3JANBqCiu\nf0hOx+oQieTUo+hgsrKydHR1KRRK78wtYWFu/uTJk4CAABRF1aFa1OrbIslS6hDJEH5bKBSK\ns7NzdXW1nZ1df8eqrqoe4zGmv2NJLr9UWy1CoVCZydvHxyful1+8zcyMNDR6L23p7s6ur1/o\n4zPo/QsEgpycHNkSZ2fnH374IT093dvb+4svvvD19WUwGJJHxSZNmiR5VKzPrWbMmBEXF+fq\n6ippNl+/fr21tbWnpyeCIOvWrTtx4kRMTMz69etbWlo++ugjX19fyWMF/S3CYrGSq3AJIyMj\n5H/X5aBPwzV5Y7FYKpWq8e/vt0Ag4PP5BAKBRqOpKjApPp/P5/M1+vofqGRCoZDH4ym6Wlgs\nloG+vpw2UkNDwzt37sTGxpJIJDWpFsm3RR2CEQgEXC5XHSKRXHMPVbUEBwcfO36MacskkfsY\nVa20pNTU1NTPz08kEhUWFtbV1QkEAjqdbmtra2tri8FghEIhh8NRbbVQqVRl9p8yNTUN+Oyz\nM7t2zXFx6ZG/m7u7z+Tnj1m92sbGZtD7f/nypYeHh2xJfX29kZFRRkbGrl27Tp48uWXLFhKJ\nZGdnFx0dvWzZMjlb7dmzR19ff9u2bbW1tdra2oGBgdu3b6dSqQiCODg43Lx589NPP/Xx8dHR\n0YmIiPjhhx8k1ShnEXgtGGX+qBxCsbGxX331lbm5uWyhQCBob2+nUCiQvGUJhUIWi6XoaomL\niyspLjYxMelvBRRFb9y8eeTIETqdTqfTFRfJAEmqhUwmq8PfSJK81aFaRCJRW1vbUFWLSCQ6\nceJE/tP8sHFhVNq/Hn+qqKh4mPpw9erVHR0d9+7de17Zpa9vTMATu7ra6xtqAkNGz5w5U1tb\nm8PhqLZaqqurd+zY8dtvvyntiGKx+MaNG6k7djgbGNjp6WmSSJ08Xnlra15Dg/cnn0ycOHEI\nb3iD4Wu4XnkDdaOpqdktt+dRd3e3v7+/OuQnoDQ4HC4qKop4nnjhwgWmHdPIyIhIIHZ0dlRV\nVmloaHz88ce1tbV/nEjy9Q6NmGAp3Uog4OfkpR86dGjhwoUj8AuDxWIjIiKcnZ0zMzPvVVez\n2WwqlWoxbtxSb28LCwtVRwfUBSRvMDRsbW3Pnjlja2ODwWD6XKGhocHTy4tMJnO5XCXHBlSI\nRCJFR0d7enrm5ubW1dXx+XxNTc133nnHx8envr7+xNGr74RN12L868lmAoHo7Rmc/ST13Llz\nCxcuVFXkqmVqajpjxgxVRwHUFyRvMDTc3Nz8AwKqq6qsrKx6L2Wz2dXV1f+3erXS4wJqwd7e\n3t7evkfh6dOnHRzcemRuKTdX3yt//uHjU+Tr66v4AAEYZqCbABgaJBJpxowZhkZGVVVVPTpS\nsFisx1lZsStWODo6qio8oG7YbHZdXZ2VZd8d0REEweHwZmbW5eXlyowKgOECkjcYMra2tu+/\n/76VtXXqw4elpaUvamqqqqqysrKIJNLq1aunTJmi6gCBGmlvby8uaKRR5d3S1qRrwcjnAPQJ\nms3BULKxsfn4449DQkIqKio6OjrIZLKJiYmbm5uurq6qQwPqBY/Hi1ExiqL9dZJAEEQkEsGQ\nfAD0Cf5jgCFGJBK9vLy8vLxUHQhQazo6Oq7uFi0tjfr6/Y5l3dzcMNpttDKjAmC4gGZzAIAK\n4PF4JyenopLc/lbo6Girb3g+atQoZUYFwHAByRsAoBqhoaEGxtii4j7yN5fHeZh+a3ZUuGSY\nTABAD5C8AQCqoa2tPWfOHATfkJZ+s729VVIoEgmrn5feuHk+MMRp4sSJqo0QALX1Gve8hUJh\nbm7uy5cvx44dq6WlpbiYAAAjhI2NzfLly2/cuFFS8rDsXgsBT+Bwu8d42SxbMcvX11csFguF\nQlXHCIA6GmjyPnXq1Nq1axsbGxEESUtL8/Pzq6urc3d337Vr1/z58xUZIQDgbaarqxsVFdXd\n3d3Y2Mjj8RgMhpGRkaQLumTmNwBAbwNqNk9OTp43b56ZmdnOnTulhSYmJq6urufOnVNYbACA\nkYJKpVpbWzs6OhobG8t5eAwAIDGg5L19+3Z3d/f09PRVq1bJlo8dOzY3t9/OogAAAABQhAEl\n76ysrPnz5/ceLcHCwqK+vl4BUQEAAACgXwNK3iKRiEQi9S5/+fIlgUAY6pAAAAAAIM+Akre9\nvf2DBw96FKIoevnyZWdnZwVEBQAAIx2Koq2trbW1tS0tLT0m+wFgQL3NFy1atGHDhvDw8Ojo\naElJV1fX+vXrMzMz4+LiFBkeAACMOEKhMD09PTMz8/KzFBweJxKKpjgFeHl5+fv7E4lE5cez\nePHi5ubmq1evIggya9YsPB6fkJCg/DCArAFdef/f//1fRETEBx98YGlpiSDIwoULdXV14+Li\npk6dGhMTo+AIAQAjQnNz86NHj27fvp2Wlvb8+fMR+5xYd3d3fHz8t+fiSvW6AxdMCVr0btDC\nqZXGgp1Xjxw6dKizs3PQe16zZg3mf2g0mrOz86+//jqEkavDEUeOAV154/H4K1euxMXFHTt2\njMvl1tfXOzs7L1y4cNWqVVgsjNEGAHgjLBbrzz//TD14WZ+urUGi8oT8xo4Wx5mBkZGRBgYG\nqo5OqVAUPXPmzNWqDK/p43CEv8/POAJe39pU19zo5u1MfELC0qVLB33iNTQ0vHv3LoIgbDY7\nMTFx1apVhoaGs2bNGqr41eGII8RAB2nB4XAfffTRRx99pNBoAAAjTXNzc3x8vDijdp7fJA0S\nVVIoEAmfZBadbDkZGRnp4+Oj2giV6dmzZycfJfnOmiDN3FJYPG7UOO/Eczc9sj0GPWsfHo93\ndHSUvPb09Ny/f//jx48lqTQ0NNTd3X337t2SpSdOnIiJieFyuXL2VllZOXHixNGjR588eZJM\nJr/uEcGbgOtmAIDKiESi06dPE7KawkePlWZuBEEIOLyPtbNDLfnq1autra0qjFDJcnNzTRys\n8aS+b2zjCHizUbZDMrqGQCA4c+ZMW1tbQEDA4Pbw+PHjsWPHRkZGnj17tr/MPbRHBLIGlLw3\nbdrk7Ozco7ujWCweNWrU1q1bFRMYAODtl5+fX3k5M9DOA4P0MaqavaElOb/t/v37yg9MVerr\n6xlGunJWYBjp1tfXD7rzeW1tLZlMJpPJJBJp3rx527dvnzp16iD2c/Xq1bCwsE8//XT37t3y\n2/CH6oighwE1m1+4cCEiIqLHmIVYLHbChAnnz5/fvHmzYmIDALzlCgsL7Y0s8Vhcfys4GVkX\nFRWhKDpCxkwVCAR4IkXOCngiQSQSicViHK7fSpNDegeaw+Gkp6evX7+ewWB8+OGHr7WTlJSU\nxMTEw4cPL1y4UDlHBL0N6Mq7srLSzs6ud7mjo2NVVdUQRwQAGDFaW1t1afKmKNTV0Kq+ns1m\ns5UWkmppampyOrrkrMDp6KLRaIPL3Mj/7kA7Ojp6eHh8+OGHCxYs2LRpk2RRjwtokUjU304c\nHBycnJyOHDnS1SUv1FceEbyJASVvsVjc0dHRu7yjo0MgEAx1SAAA8DcURQ1DR6k6CuWxtbV9\nWVErZ4WXFbVMJnMIj8jlciWN8AYGBs3NzdLywsLC/jYxMDC4e/duR0fHhAkTWCzWoI8I3sSA\nkrejo2NSUlKPQhRFk5KS7O3tFRAVAGBE0NPTa+5qk7NCSxeLSqXSaDSlhaRafn5+HnjT5qq6\nPpe21TU5i/T9/f0HvX+hUFhUVFRUVJSTk/P7778fPXp05syZklsS4eHhf/75Z0lJCYIgKSkp\nBw8elLMfXV3d27dvY7HYsLAw2ZT/WkcEb2JAyXv+/Pl3795du3attJGkq6tr9erV9+7dW7Bg\ngSLDAwC8zUaNGlXcUC0QCftboaC+wtHRceSc67W0tCZNmlR473FD6XPk35enTZW1z25ljB8/\n/k2efW9sbHRycnJycvLz89u1a9fatWt///13yaKFCxcuXbo0JCTEyMjov//972effSZ/VwwG\n46+//tLW1g4NDZUzQ5WcI4I3gRlI84VAIJg4ceKtW7coFIqdnR2KomVlZRwOJzw8/OrVqyqZ\nmyQ2Nvarr74yNzfvEWd7ezuFQlGH3+l8Pp/P52toaKg6EEQoFLJYLDWpFoFAwOVy6XS6qgP5\nu1rIZLI6/I3Up1pEIlFbW5vSqkUsFu/fv599pyzM0QfbK0Pn15SWmfHXrFmjpSXvvrhCVVdX\n79ix47ffflPmQfPz869du5bSXKhrYUSiUvgcbsuLRn8tu/DwcE9PT2VGAtTWgHqbEwiEpKSk\nvXv3njx5sri4GIPBjB49ev78+StXruw9TygAAAwQFoudO3duPCf+WkpKINNDi/r3zxeugJdV\nXdhsT3p3yrsqzNyq4uLiwmQyQ/Pyqqqquru7KXoUizEWbm5u6vD7G6iJgaZeAoGwdu3atWvX\nKjQaAMBIo62tvWzZsiSzpLsFBfzMFxpkKlfAb+5iuUe/syg8XEdHR9UBqgaFQvH19fX19VV1\nIK9WVlbW5+NI27Zt++qrr5QfzwgB180AABXT0NCYPXt2e3t7VVVVZ2cnmUw2MzMzMjISCoUc\nDkfV0YFXYDKZ0Htc+V4veYvF4s7Ozh5/pxHYqAUAGHIMBsPNzU3VUQAwPAwoeYvF4ri4uJ9/\n/rmiooLP5/dYCr+5AAAAAGUaUPL+5ptvNm/ebGdnN3PmTAaDoeiYAAAAACDHgJL3gQMHli5d\nun//fpi9GwwXAoGgtLS0rq6Ox+Npamra2NiYmpqqOigAABgaA0rejY2Ny5Ytg8wNhov8/Pzk\n5GQMBmNpaUkikaqqqu7du2dmZjZt2jRtbW1VRwcAAG9qQMnbwsKivb1d0aE8fvz4+PHjNTU1\nDAZj/Pjx0dHRI2dYJTCEHj58eOfOncjISA8PD+kvzq6uruvXrx84cOCDDz7Q09NTbYQAAPCG\nBnQx/cEHH/z8888K7ZhWXFz8zTffjBo1ateuXfPnzz9//vzJkycVdzjwtnrx4sXt27fff/99\nT09P2bYiDQ2N9957z9HR8ezZs2KxWIURAgDAmxvQlbe9vf2hQ4f8/PzmzZtnbm7e44J4+vTp\nbx7H+fPnTU1NY2NjEQSxtLSsr6+/dOnS7NmzSSTSm+8cjBz37t3z9PS0trbuvQiDwURGRv78\n888FBQXOzs7Kjw0MLR6P19zcLBQKtbS0oCMtGGkGlLxnz56NIEhFRUVmZmbvpUNyRV5YWBgS\nEiJ9O2bMmNOnT1dUVDg5Ob35zsEIwefzKysrx48f398KRCLR2dm5uLgYkvew1tTUdOvWrdLS\nUtadNCwWyxcKHedOCwkJGTWq38lDURStr6+vrq5ms9lUKtXc3NzMzAxuzIHha0DJ++zZswoN\nAkVRFosl25NI8rq1tVVa8uDBA9kp3A0MDFgsFpVK7bEfBEE4HA6Xy1VowAOEoiiPx1N1FH/j\ncrnqUy29RwsYEi0tLV1dXVgstqWlpb91SCRSUVGRdAUul6smfyPFVcsg8Hg8ta2WsrKyK1eu\n6D+rCjKx0PcOQhCEzecVP8jZn3DR78Ml48eP752Sm5qarl+/Xnj+gQFdm0Ig84T8xo4W++n+\n4eHhxsbG8gNgsVhCYb+TngGgKgNK3rNmzVJ0HK+Ex+Nl51zCYrEYDKZHB3gURcVice9ylZD8\nklCHSBAEEYlEGAxGHa4zFFoteDxe8hnl7B9FUTwej8ViURRFURS+Lb2JRCJEPYLpXS1NTU1X\nrlxxq2GNdvin7USDRPY0t7bWNbgYd4zBYPj5+cnupKam5syZM1pF3e97TyQTiJJCvlCQnVGU\n0Jowc+bMPm+ySElONUP5qQAYCmoxtjkGg9HS0mpra5OWSF7Lzkng5+d36dIl6dvY2FgGg9Hj\nsR/JlKBkMlkd5t5RtylBSSSSOlSLQue+1NTU1NTU7O7utrKy6m+djo4OS0tLbW1tabWow99I\n3aYEVZNqkYxtLlstV69etShv8HTo466HCZk8ycUj/cmToKAgafBcLjc5OdnqBc5r9L8yOhkh\nhzp559WU3rhxY9WqVXL+a3R0dOBwuCH6QAAMmYEmbxRFb968mZGR0dra2qOz7u7du988Dicn\np+zs7KVLl0reZmdnk8lkGxubN98zGDlwOJyTk1NGRkZ/ybuzs7OgoCAqKkq5cYGh0dXVVVFR\nMdWi3wtlKx391IephRMKvb29JSWPHj1i3S0Z79V3NwhXM7vKtDsPxzycMGGCQiIGQGEGlLw7\nOzsnTpyYmpra59IhSd4zZ8787LPP4uLiIiMjKyoqLly4MH36dOhqDl5XSEjIvn37MjMzfXx8\neiwSCATnzp2zsrKCH4XDVGNjIzc1W8t/nJx1TBk69fX10rdFRUVOxtYYpN92bydjm8LCQvVM\n3p2dnc+fP+/u7qZQKObm5tCjHsgaUPLevHlzWlra9u3bp0+fPmrUqKtXr9Lp9G+//batrW2o\n+rI5ODh8+eWXJ06cSE5OZjAYM2bMeP/994dkz2BE0dHRmTVr1tmzZxsbG4ODgyXnOxRFKysr\nk5OTsVjs3LlzVR0jGCQ+n098VQs2AYcTCATSt62treYa8nKeroZWIatOLBarwz1+qY6OjuvX\nrxcWFtY8eUkhUbl8rqm7np2d3cSJE1U1RODixYubm5uvXr2qkqOD3gaUvC9cuDBnzpwvvvhC\n0l1ZV1fXz88vICDAz89v7969O3fuHJJQvL29pY1dAAyavb390qVLr1+/vnv3bh0dHRKJxGKx\nBAKBt7d3aGgokUhUdYBgkBgMRhePKxSL8Nh+UziLwzaWuULFYDDyH2VVmHOuggAAIABJREFU\nw0kRm5qajh49Wv+k1d3e0z3w78/S1d2Zc/3JixcvFixYYGJiMrg9r1mzZs+ePZJft9JCKyur\nqKio77//Xv62gYGBXV1dgzvuK0OSvKZSqdbW1h9++OHKlSuH/EBvnwEl79ra2qCgIOR/3T4l\nP2xxOFxUVNQvv/wyVMkbgKFiZGS0ePHitrY26cQkFhYWkLaHOyMjI+NJYWXljY6GfWevbgG/\nurVlKpMpLdHV1W0qrjfR0u9vn02dbbrWuupz2S0UCk+dOsUq4Aa5j5Pt5a5BpQe4BueUZCUk\nJKxcuXLQtxTJZHJiYmJKSorklD5wMTExgzviKxkaGt69exdBEDabnZiYuGrVKkNDQ3V4xEnN\nDegrS6PRJAmbSCSSyeS6ujpJuaamZkNDgwKjA+ANaGtrjx49esyYMUwmEzL3WwCLxQYEBKRW\nlHTz+34G/X5Z4ah5M2W7K44aNaqgvkKM9j0gLoqgz+rKRo8erYhoBycrKyv/drGno0+fz6e5\n2Y0pTX2enp4+6P1bWlrOmjVr7dq1fTY5JCcnh4aG6unp0el0Hx+fpKQk6aLFixdPmTIFQZD4\n+HgajdbZ2SldlJCQQCQSm5ubJW+PHj3q6upKJpOtrKxWr14tu2af8Hi8o6Ojo6Ojp6fnd999\np6Oj8/jx40F/wJFjQMnbxsamuLhY8trNzS0hIQFFUaFQePr0aTMzM0WGBwAA//D19R21cHZi\nTmZT179SAk8o/Kson+01asaMGbLlXl5exhFuD8tyUaSPXJVZ+VQr1KHHc+Gq9ezZM1sz+/5a\nAjAYjJ25w7Nnz97kED/88MPTp0+PHz/ee1F7e/uKFStSUlIeP348efLkd999t/exZs+ejcVi\nz5w5Iy05evTolClTJPP97N27d/369Z999llBQcHJkycfPny4ZMmSAQYmEAjOnDnT1tYWEBAw\n2A83ggyo2Tw8PPzw4cN79uwhEAgxMTHLli1jMplisbiqquqbb75RdIgAACAh6XKYrKWV8ONe\nc21dY00tHBbb1s0ub35pN/fdpe+916M/F4FAiIqKOsY/lpzxcKytG4Py9/PfXbzu9Ip8gYfe\n/LlzyWSyKj5K35qbm3U05d3S1mHovmgplowvNLhDWFtbr1mzZuPGjbNmzeoxSOWcOXOkrzdv\n3nz79u3Tp09//fXXsuvQaLRZs2bFx8dLnuytr6+/cePG+fPnEQQRiURbtmzZsWPHvHnzEASx\nsbHZv3//mDFjGhsbDQ0N+4untrZW8ifg8/k4HG779u1Tp04d3EcbUQaUvD///PN58+ZJHu+O\niYlpb28/fPgwFovdsmXL559/ruAIAQDgH3g8fvLkyT4+Ps+ePWtoaBCIRGZaWmH29kwms898\nZmhouGzZsiSzpFP7L2hTNalEMkfAa2W3+38wddKkSbq6usr/CHKIxWKs3KyMQTDSwQEHfZSN\nGzfGx8f/8MMPW7ZskS2vr6///vvvU1JSXr58KRQK29vbLS0te2++ePHi0NDQ8vJyW1vb48eP\n6+rqTpo0CUGQysrKlpaWmJiYHjfIy8rK5CRv6T1vDoeTnp6+fv16BoPx4YcfDvrTjRADSt4M\nBkP2EcN169atW7dOYSEBAMAr6Ovrh4aGDnBlLS2t6OjoiRMnPn/+vKuri0qlWlhYyA7gqD50\ndHRY9W362v2muvYulpal1hv2sNPU1Ny2bduaNWt6ZNkpU6Zoamru2rXL0tKSQqEsXry4z/H2\ng4ODbWxs4uPjt23bduzYsfnz5+PxeOR/A+smJSVFRkYOPBjJPW/Jaw8Pj9zc3E2bNkHyfiV1\n6WMJAAAKpaWl5erq6u/v7+7urp6ZG0EQBweHitoyOQ+wldeWSlPdm1i6dKmdnd3GjRulJa2t\nrdnZ2Zs2bQoNDbW2tjYwMCgpKelzWwwGs2jRomPHjmVmZj579mzx4sWScltbW21t7YsXL75h\nbFwuVw0f4VM3/V55x8fHIwiyYMECHA4ned0f6V8OAADAm/D19c3MzCysejrK2qX30tIXRcbu\nOv7+/m9+ICwWu2vXrvHjx0sfxNDS0tLX109KSgoNDRWJRBs3bnzx4kXvkQolFi1atGXLlhUr\nVnh6erq4/B0qHo/funXrmjVr9PT05s6dSyKRCgsLz507d+zYMTmRCIXCoqIiBEG4XG5aWtrR\no0fnzJkDk8G8Ur/JW9JFMCoqCofDye8uCMkbAACGBJlMnj179smTJ7OKMl2Y7kT835lVKBI+\nq8jDmvPnzJwzVHPGhIWFTZ069fLly5K3WCz23Llzq1evNjY2ptFos2fPnj59en/bWlpahoWF\n3bp165dffpEt//jjj/X19Xft2vXTTz8RCAQbGxs5O5FobGx0cnJCEIREIpmbm69du/arr756\n4w/39sP01zpx8+ZNBEHCwsKwWKzkdX/Gj+970H+Fio2N/eqrr8zNzWULJbOKUSgUdZg+S91m\nFVOTalGf6bMk1UImk9Xhb6Q+1SKZVUxNqqX3rGLKV11dvWPHjt9++02ZB62vr79y5cr9C+kG\n2oYUEpXD625qezl2qufUqVN7nPTAiNXvlbdsSlZJegYAgJHJ2Nh42bJlkZGR1dXVbDabSqWa\nm5tbWVlBYzKQenWHte7u7s8//zwzM1MJ0QAAAEAQBIPBWFhYBAUFRUZGBgcHW1tbD9/MXVZW\nhukLDBPyJl79qBiFQtm1axc8NQ8AAGAQmEwm9B4fcq++8pb8AJSdIhcAAAAAKjSg57wXLFiw\ne/duoVCo6GgAAAAA8EoDGmHNyckpPj5+9OjRS5Yssba27jEb3SufBAAAAADAEBpQ8p47d67k\nxRdffNF7KdzMAAAAAJRpQMn77Nmzio4DAAAAAAM0oOQ9a9YsRccBAAAAgAGCiUkAAACAYWZA\nV94IgqAoevPmzYyMjNbWVsnE3lK7d+9WQGAAAAAA6NuAkndnZ+fEiRNTU1P7XArJGwAAAFCm\nATWbb968OS0tbfv27QUFBQiCXL169d69e+Hh4d7e3lVVVYoNEAAAAAD/NqDkfeHChTlz5nzx\nxRfW1tYIgujq6gYHB1+7dg1F0b179yo4QgAAAAD8y4CSd21tbVBQEIIgWCwWQRCBQIAgCA6H\ni4qKgqfIAAAAACUbUPKm0WiShE0kEslkcl1dnaRcU1OzoaFBgdEBAAAAoJcBdVizsbEpLi6W\nvHZzc0tISJgzZ45IJDp9+rSZmZkiwwMAgJFIJBJVVFRUVVVxOBwymWxhYcFkMvH4gT4fBN56\nA/oqhIeHHz58eM+ePQQCISYmZtmyZUwmUywWV1VVwYSsAAAwtKqrqy9fvnz52TNtE2MihSLg\nctsunJ/k4Dh16lRbW1tVRwfUwoCazT///PNbt25JHu+OiYn58ccfyWSyhobGli1bPv/8cwVH\nCAAAI0hpaenx48ezRMKxs2eNCglh+vg4BQePnTMnD487derUs2fPBr3nNWvWYDAYHx8f2cKy\nsjIsFovBYNLT09849gEFIEGj0ZydnX/99VdFH/RtNaArbwaDwWAwpG/XrVu3bt06hYUEAAAj\nFJvNPnfuXLWerpWTk2w5Bou1cHGpLS+/ePGimZmZ7An5tRgaGubl5T158sTDw0NSsn//fgsL\ni+rq6jcNfcAB3L17F0EQNpudmJi4atUqQ0NDGIF7EAZ05f3ixQtFx/H/7N1nXFPX3wDwm0Em\nO4QdNiJ7CLhAFBXBiQpaR3GBoFahKiL9u3BVcLTWCaKCiqNuq1IrVlBEQZbIEIggsmQHwghJ\nbvK8uG0eyogYIIlyvi/8JOec3PvLCebk3HsGAAAAkJqa+rKxUeu/LbeAmqFhRkd7X+tl9Yei\noqKXl1dkZCTylM1mx8TE+Pr6disWGxtrZWVFIBD09PQCAwOZTCaS7uXl9d133x08eFBLS0tW\nVnbx4sXt7e23bt0yMzMjk8lTp06trKwUHgAWix05cuTIkSNHjRr1888/Kysrp6eni/x2hrN+\nNd56enpTp06Ni4vr6OgY6oAAAACGraKiInVjIyEFNIyNi4qKBnIKf3//y5cvt7a2QhB069Yt\nAoHg4eHRtcDx48c3b94cEhKSn58fFxeXkpKyYsUKQe6jR4/evn37xx9/XLt2LT4+fv78+b/9\n9tuZM2eePHlSVVW1YcOGfobB4XB+//33pqam8ePHD+TtDFv9umz+/fff37hxIyEhQV5efsGC\nBcuXLwfVDQAAMOgYDAbJwEBIAZKCYvOHMhiGMRiMaKdwdnam0WiXL19evXp1ZGSkr69v10PB\nMLxr167w8PAlS5ZAEGRgYBAVFWVnZ1dTU6OmpgZBkJKS0vnz55GXLFmy5NSpU1VVVUjWxo0b\nf/zxR+Fnr6ysJBAIEASx2WwMBrN///5Zs2aJ9kaGuX71vGNiYj59+nT+/HlbW9uzZ886OTmN\nGDFi37594HI6AADAIMJgMDwYFlKAz4PRaDSyXpbI/P39o6KiioqKkpOTV61a1TWrtLS0oaHB\n19dXMLLMzs4OgiA6nY4UsLGxETT2WlpaWlpaSMsNQZCmpiaTyWxvbxdyajU1tezs7Ozs7IyM\njN9++23Pnj2nTp0ayHsZtvr7FyArK7t8+fLExMT379/v2rULhuFt27Yhl9OHND4AAIDhg0ql\nMuvrhBRoqaujUqkoFGogZ/Hx8cnPz1+7du306dO1tLS6ZsEwDEFQfHw8/78EV1txOJygMAqF\n6vYUgqBu2052I7jnbWtru2bNmu+//37Hjh0DeS/D1hf/fNPX19+5cyedTr9y5YqsrGxCQsJQ\nhAUAADAMWVhYVBUW8rjcXnP5PF5FQYGFhcUAz6KoqLhgwYInT56sXr26W5ahoaGSktKdO3cG\neIr+Y7FYfD5fbKf7Znzxej0sFuvOnTuxsbGPHz+GYZhGow1FWAAAAMOQra3t7LS0pJSUkU7O\nKPR/u9d8fnFqqruB4ejRowd+otOnTx86dIhCoXRLx2KxYWFhQUFBKioqCxcuxOPxBQUFN2/e\nvHDhwsBPCkEQl8t99+4dBEEsFuvly5exsbELFiwY4IWE4ekLGu+UlJTY2Nhr1641NzcTicSF\nCxcuX7588uTJQxccAADAsILBYBYtWsSOjU189KfBqFHyVCqEQkEQxKxvKM3MGK9M+e6772Rk\nZAZ+IgKBgAwc62n9+vVUKvXIkSOHDx+WkZExMDDw9PQc+BkRNTU1pqamEATh8Xgajfbjjz9u\n27ZtsA4+rKD6c71i3759sbGxxcXFEASNGzdu+fLlCxculJeXH/rw+uTr67t58+Zud2t4PB6H\nw8FgMNKwAjCPx+PxeNIQCZ/PRwZ2SkMwPB4PhuFB+fYZeCRS9dciJdUiVX8tfD4fhmHJRlJe\nXn706FHBxGjxaG9vT0hIePPmzfOqShyJxO7ocFLXsLCwmDp1qpycnDgjAaRWv/5XbNu2jUaj\n/fTTT8uXLzc2Nh7qmPoDjUYTCAQikdg1kcvlcjgcLBbbLV0iOBwOh8ORhkhgGGaz2VJSLVwu\nl81mS0MkMAwjjbc0BCM91cLj8ZDGWxqCgWG4s7NTspEQCIQBDu0WAYlEmj179rRp0xaVl7e1\ntZFIJG1tbWn4RADp0a/G+/Hjx66uruL/CxYChUL17BwgVxHQaLQ0dBqQvpQ0RIKQkmrh8/ko\nFEoaIkGAaukGGWwsJdUCQZDEqwWDwUjqjiwejzcyErZgi3Si0+m99vH27NkDrpAPon79r5gy\nZcpQxwEAAAB8A4yMjMDocTGQos40AAAAAAD9ARpvAAAAAPjKSMVtLWAo8Pn86urqioqKtrY2\nPp8/YsQIMpks6aAAAACAQQAa729TeXn5gwcPfs9MkFNRxOJlWptaOppb17h9N336dJF3AgYA\nAACkBGi8v0GFhYXXr18vlG0Zt9gDi5Ph8XgsFgtmcW5mPK+srFy+fLmKioqkYwQAAABE98X3\nvLlcbmJi4oULF548ecJms4ciJmAgGAzGzZs3P2rCho4WWNz/L/pBkCNZTBmdAVdcu3YNFrpt\nEQAAACDlvqznXVZW5uHhUVBQgDw1MjJ68ODBiBEjhiAwQETPnz/PYJdbjhzbLZ3D6qyvqkSh\n0WcfXSeRSDNmzFBVVZVIhAAAAMAAfVnPe8OGDerq6pmZmQwG482bNwYGBmvXrh2iyAAR8Pn8\ngoICTVP9bqnlOcVp1xOKL7xpfcagNFHvn338o8+Wa9euCd95FwAAAJBOwnre8fHxHh4eXVPS\n0tL++OMPW1tbCIKsrKwOHTok2OQVkAYsFqu9vV1WWeP/k/j8d88zqx6UGBlaKllRIQhqZzAp\n7TLmVuYpt1Nqa2tXrVpFIpEkFjEAAADw5YT1vD09PZcuXVpfXy9I0dTUfPr0qeDpkydPNDU1\nhzA6YMCqi8qqH5aajrRXUPjP3n+yRNmJdlPep1Tcu3dPUrEBAAAAohHWeGdkZNDpdFNT07i4\nOCTlp59+2rJli62t7bx580aNGvXjjz+GhoaKJU6gXwgEAolEam1s/uc5n1+WXUjTNsLj/39L\nA3YHC+lqo9FoR7Oxf8T8WV1dLZFoAQAAANEIa7wtLCxSUlL+97//+fv7T58+/ePHj/Pnz09O\nTrazs2ttbbW0tPz777+XLVsmtliBz0KhUGZmZlUFpcjT1sZm5utGCuX/r6LzefzWhmYqlYo8\nJRNl1ZQ1ioqKJBArAAAAIKrPjDZHo9FBQUGenp4BAQHm5ub79+9ft24duM8tzZycnHJycj68\nK9Ucqd/BbMPhCBgMhsvlIrmNFZ9MVHTU1NQE5eXJCk1NTRIKFgAAABBFv6aK6enp/fnnnxcu\nXNi4cePly5fPnj1rZmY21JEBolFUVJw/f/7169cLW3JlKQqC7X24nRxGdZ22jJK5uXnXLQ55\nfB4Gg5FQsAAA9K6lpSUrK6usrKyjo4NAIOjo6Nja2ioqKko6LkBafH6qWE1NTXl5OQRBPj4+\nBQUFenp6tra2YWFhYIUWqWViYrJ8+XInomFeQmpD1aeq4tKaoo+VBSVmFN1Ro0Z1G1vewKgD\nE74BQKq8fv362LFjr46cJiZlaGe/Jz/LzPgt+vjx4y9evJB0aF9g+fLlM2fOlHQU3yxhjXdl\nZeX48ePV1dV1dHTMzMxyc3OpVOqVK1du3bp19uxZW1vbV69eiS1Q4Itoa2uvXr366s+nZ3m7\nquDw401s50+abmFhgcfjuxara6rRsKGYmppKKk4AALp5+fLl76FhjtVt3rajR9H0TdU07Wj6\n86wdnOpYf+wKT0xMFPnIQUFBKBQKhUKh0Wh1dfUZM2ZkZmYOXuAiWrp0qaenpxhO1Nraiuqb\nGAIYXMIa78DAwNbW1qtXr965c0dLS2vp0qVI+owZM/Ly8iZOnOjs7BwYGCiWOIEvhkKhNDU1\nV69ezWA18rEwkUjsVqCd1f4q98WYMWPk5eUlEiEAAN3U19cnJCTMMLcxVOl+PUxHSWWWpd29\n3RFVVVUiH19NTa2goCAvL+/KlSsdHR3Tpk37eoe8cDicLypPIpGy/nXmzBkIguLi4gQpQxPj\nEBLWeD958uTcuXMLFy6cM2fOzZs3375929jYiGTJycmdOHEiMTHx8ePHYokTEJGent6GsICX\nuclv32ez2CwkEebBH6pLHqc+nLrYxdXVVbIRAgAgkJqaqphD11HqfesgDXlFE1WNly9finx8\nLBY7cuRIU1PTSZMm7dq1q76+Pj09XZAbGxtrZWVFIBD09PQCAwOZTCaSzuFwNm7cqKKiIisr\n6+3tfeLECSz2n/FSEydODAoKEhzh0qVLBAKh53kfPXo0ceJEFRUVOTk5R0fH+Ph4JN3X1zcu\nLu7u3btI9/f06dMQBPF4vH379unr6+NwOAMDg4iICMHYHS8vL29v75CQEC0tLSKRGB0dTSKR\nGAxGtwAaGhp6xoBGo23+ZWRkBEHQiBEjBCkiV6mkCBuwhsfjBZXS3NzM5/NxOFzXAuPHj/8a\nf7AMN/b29lsPbkxOTr774IYsSQ6DRre2t1q5jvRfsMLR0fFrvF4EAN+qkpISI6q6kALGquoZ\nJSUDP1FdXd358+chCBJ8qx8/fjwsLOzXX38dO3ZsdXV1UFDQihUrbty4AUHQnj17zp8/HxUV\nZW9v/+DBgx07dnzp6ZqbmwMCAqytrdFo9NWrV2fPnp2dnW1ubh4dHc1isVpbW+/cuSMoHBER\nsX///mPHjjk7Oz99+jQwMBCHwwl+Ity9e3fTpk2FhYUQBGGx2J9++unChQsbNmxAcqOiory8\nvCgUSs8YvjHCGu/58+cvWbJk5cqVeDz+0qVLbm5usrKy3cp0u4cKSCcdHZ1Vq1YtXry4traW\nw+EoKytraWmh0V+8pxwAAEOKyWTKE7rf4epKgUBqa2uDYVi0SSKVlZVIpxnZWtDFxcXZ2Rl5\numvXrvDw8CVLlkAQZGBgEBUVZWdnV1NTQ6VSf/311+3bt3t7e0MQ9MMPP7x+/Vqwclc/LViw\nQPB4586df//997Vr13bv3t2zJJ/PP3DgwKZNm1auXAlBkLGxcUlJyd69ewWNN41G27dvn+Dr\na+XKlZGRkUjj/e7du+fPnz9//vyL6+UrJOzrOyIiwtvbOzY29uTJk46OjhcuXBBbWMBQUFFR\nMTMzs7a2ptFooOUGACmEx+M7ucJu5bK4HBkZGZGnd6qpqWVnZ6enp584ccLAwODChQvIV0Fp\naWlDQ4Ovr69gAJednR0EQXQ6vby8nMlkjhs3TnCQro/7qbq6OjAw0M7OTltbW11dPS0t7cOH\nD72WrKioaG5unjBhgiDFxcWloaFBsBCkhYVF16+vgICAd+/eJScnQxAUGRlpYWHh5OT0peF9\njYT1vMlk8rFjx44dOya2aAAAAIYzLS2t8my6njK1rwLlTQ1aVqNEPj4Wi7WwsIAgyMbGprm5\nec6cOa9fv8ZisUhHPD4+3t3dvdtLysrKIAjqen9NcAcagqBu3QDkOD3NnDlTXl7+yJEjurq6\nRCJx+fLlfU02Rg7e83aeIKXb2Fs9PT0PD4/IyEgHB4eLFy/u2rWrj7f+rQHdLwAAAGlhY2OT\nW13exu7sNZfF5bypLEP2dRy4wMDAioqKyMhICIIMDQ2VlJS63ngWoNFocnJyXaeYdx0xp6qq\n2nXzqoKCgp5HaGxszMzM3LFjx8SJE/X19VVVVbsuyYzD4QRLQCKnU1BQSEpKEqQkJSVRKBR1\n9T6HAqxdu/bGjRuRkZEdHR3ff//9Z9/4t+HzjXd7e3vXmq2srPz999/j4uKKi4uHMjAAAIBh\nZ+TIkbYrF/+Rm9nB6d4x7eRyH+RmmSyeZ2VlNSjnIpFIGzZs2LNnT2trKxaLDQsLO3PmzLZt\n296+fVtUVHT37l0fHx/o30Wy9+/ff+PGjdLS0uPHj9+/f1/QD3Zzc3vw4AHSGD9//jw6Orrn\niRQVFalUanx8PJ/P53K5W7duRRb+QhgaGubk5BQUFNTX17NYLBQKFRoaevjw4fPnzxcXF0dF\nRR09enTbtm1C3oi7u7umpmZwcPCiRYsUFBQGpXKkn7DGG4bhVatWycvLy8nJHTx4EIKg6Oho\nQ0PDhQsXLl261MTEJCAgoOv1EwAAAGCA5s+fr+01Iy49JbuyrIXVwePzWztZb6vKL6e/UJ41\necGCBYM4Q+SHH35oa2s7fPgwBEHr16+Pi4v766+/HB0d7e3td+7caWBggBTbvn37ihUr/P39\nLS0tk5KStm7dKpgP5uPjs2rVKhcXF3V19V9++SUkJKTnWdBo9M2bNxMSEjQ0NExMTNBodNdV\nWfz9/c3NzceMGUOlUmNiYiAICg4O/umnn8LCwszMzA4cOBAWFiZ8QRE0Gu3n58dmswMCAgap\nYr4CKCGtb2RkZEBAgIGBgaqq6qtXry5durRs2TIbG5u5c+dyOJy4uDg6nX7q1CmJ1Je/v/+2\nbdtoNFrXRA6H09zcTCQSyWSy+EPqhs1ms9nsnuPzxY/L5TIYDCmpFg6Hw2Kx5OTkJB3IP9VC\nIBCk4TOSnmqBYbipqUlKqoXL5XZ0dEi2WsrKysLDw0+ePCnOk/J4vMzMzLS0tNra2qbEV0oT\nx1AoFAcHBwcHB2nYieCHH354+fJlRkaGpAP5f+vXr3/58mXXOevfPGED1s6dO2dra/vy5Us8\nHr9ly5aAgICxY8cmJiYifz3BwcFWVlbnzp0bVj92gG54PF5ubm5eXl59fT0Mw8rKyiYmJnZ2\ndjIyMpIODQC+Vmg02t7e3t7evr29vW1dG4lEkuwv7/fv3z9//tzFxQWNRj98+PDMmTPSM5C5\nqakpIyPj3LlzZ8+elXQsYiXssjmdTl+4cCEyk9vHx6e1tXXlypWC331kMtnHxyc/P18cYQJS\nqbm5OTo6evPpU/HVVYUK8u9VKInNjLDLccePHx/ICo4AACBIJBKVSpX4NTNk7TNra2szM7PI\nyMjTp0+vXr1asiEJjB07ds6cOT4+PgsXLhQk0un0Xhcw37t3rwRDHVzCet6dnZ2CHaiQB92u\npMnJyYG9xYat9vb2mJiY5Kam0fPmYf7tZ1NoNJqFZVZWJis2dtWqVUO9X1lra2tubm5VVVVr\na6uCgoKRkdHIkSOl4boiAHxLDAwMpHYbqnfv3vVMNDIy+ubHYwlrvHV0dFJTU9evXw9BUFpa\nGgRBiYmJyCI7iKdPn2pqag51iIB0evz48dPqKutp7ij0f4bPoNAog1Gjcl6k3Lt3b9WqVUO3\n/OrLly8fPnwIPXokz+NBMFwKw484HMyUKYGBgcjCxQAAAN+qzyyPun//fjU1NS0trQMHDujo\n6BQXFx84cMDb2xuG4TNnzty/fx9ZwQ4Ybjo6Ot68eWPo4NCt5RYwtB915ffr06ZN6zaocLAk\nJCQkbN8+AoLMWlvVZGVJeDwfghgsVvydO+vp9LCwMEdHx6E4LwAAgDQQ1nhv3Ljx9u3bR44c\ngSAIh8OdOXOGQqFMmjQpNDQUKaCiorJ9+3ZxhAlImbKyshc1NWOvpxPLAAAgAElEQVRdJvRV\nAIvHK6irlZaWDkXjXVJS8njHDsvKyrFqal2vzJNlZFbJySkUF+/evfvcuXNDfdEeAABAUoQ1\n3kpKSunp6ffv329ubp40aRIy5y8lJeXEiROfPn0yMzMLCgrS0dERV6iAFGlra8OTScLLEMjk\n1tbWoTj7s2fPqByOHYWi2mMgDwqC5mlr52dknDlz5n//+99QnB0AAEDihDXeEAQRCAQvL6+u\nKQ4ODsg8emA4IxAI3M7eV3AU4LA6u61CPCg4HE52dvbYujotXd1eC2AgaDyFcikzs7W1VRom\nKwMAAAw6sLY5IAptbe2OFiaLyeyrAA+GGZ8+aWtrD/qpmUwmJzFRi0RC9z0UToNI5BcUVFRU\nDPrZAQAApAFovAFRKCgorJo8uTQru68CFXl5M83MBMsrDiIZGRkeDAvf0pTD4xFxuPb29kE/\nOwAAgDQYUON94MABwQq3wHDj7u5ug8WWvXkD9ZhPWVta+uHNmxkzZgzFlGtZWVnIyeljR4eQ\nMuWdnVg0GlwzBwDgW/WZe97Ccbnczs/d+OwqISEhKSnpw4cPnZ2dmpqaM2bMmDp1KpL14MED\nZGc6gT179lhbWw8kPGBIqaioLF68+Nq1ay/i/9Q2M5VTUUGj0a2NjdXFdGssNmDDUE22RqFQ\nY8aMSb1xw43Pl+ntyjmDy33T0qLi7T1Es9QAAAAkTljj3XUn0F7xeLwvOtnff/9tbm4+Z84c\nEomUkpJy7NgxLpfr4eGB5MrJye3Zs0dQGCz/Iv10dXXXrVtnnZycl5fXUvoBgiAtItHVxsbZ\n2ZlCoQzdeb28vO7du3c1K2upjk631ruDx7tVV8dQV58+YcJQDJcDAACQBsIa70HfW2L//v2C\nx2ZmZqWlpS9evBA03hgMZihukQJDikwmT5s2bdq0aZ2dnTAMC9bTHeqTbt++PTQ0tDk/f46m\npjYej4IgNp9f2NaWyGCUKyuPWLhQcFEHAADg2yOs8UahUCoqKkIa1MrKyoEM6GWz2V2X0WAy\nmT4+PlwuV1tbe86cOePHj+9auLa2NicnR/CUw+Gw2exuF+1hGEb+/aKL+UOEy+VKSSTIBRIx\nBIPBYD57ChiGByUSU1PT/fv3Hzly5EVamiGTKYfFtvN4THV1LIVitnSpt7f3Z4ORqr+WwaqW\ngRPbX0t/SEO1sNnsb36VbOBrJKzx1tfX19fXT0hI6KvA3r17RV5hLSEhgU6nC7amodFoa9as\n0dXVZbPZSUlJ4eHhvr6+s2fPFpTPy8vbunWr4KmhoWFbWxuzt6lKyEbaokU16DgcjqRD+IdU\nVUuvH9yXotFoBw4ceP36dXZ2NrIFtY2OjqmpqYWFBRqN7ucpOByO9HxGg1ItgwJUi0BbWxvy\nO09SeDye8LkVwPAkrPG2s7N78uSJyIfOysoKCwtDHs+YMcPPz0+Q9fz589OnT//444/GxsZI\nipWVlZWVFfLY0tKyra3t5s2bXRtvQ0NDZIsUxN9//00kErvtlAfDMIvFkpGRweFwIoc9WJBO\ngzREwuPxOjo6xFktHA6ns7OTTCb33JUEhmEul4vsMysyPp9fV1fHYDAwGMy4ceOmTZsmwkGQ\nasFisQMMZlAMSrUMCj6f397eLiXVwuPxOByOZCMhEokSaTvfv3+flpZWVlbGYrHweLyOjo69\nvb2JiYn4IwGkk7DG297e/vbt26Wlpfr6+r0W0NfXnzx5cl8vNzU1PX78OPK466Sd+Pj4s2fP\nbt68ecyYMUJe++LFCy6Xi8X+E6GOjs6yZcsEBVJSUggEQrcRSRwOh8ViYbFYaRiphPR0pSES\nLpeLtFJDHUxnZ+erV6/evHnT2NgIQZCMjIyhoaGTk1PXpVo4HA6fzxc5Ej6fn5WV9ezZs6Tk\nAjyByOfxzUyUdXR03NzctLS0vuhQYquW/hhgtQwiGIaRxlsaguFyuTweT7KREAgEMTfeMAzf\nv3//9yt/GRqYamqaE1VIrM6O4oLy+AfH58yd4OnpOZDRSO3t7b/88sv169eLi4txOJyRkdGM\nGTOCgoIUFRWDgoKOHj0KQRAajZaTkzM0NHR1df3hhx90/13KkMfjHThwICYmpry8XElJadKk\nST///LMYVshubW2Vk5PrK3fY3tQQ1niHhISEhIQIKbBkyZIlS5b0lUsgEHousHX16tVbt25t\n375d+DSwgoICRUVFQcsNSL/6+vrLly8XJL030TUzVrLBojHMDmbeY/ofMX+uDl3h4uIy8FPw\n+fzbt29fuf73SDN795nfY7EyEAS1tzPf5OdWVp6fNWsWmFsIfAMePnz4x50X06Z6yckpICny\nkJIqVdPYyDzhUTwEQV33Zf4iDAbDxcXl48ePoaGhjo6OioqK+fn50dHRsrKymzdvhiBITU0t\nMTGRz+czmcw3b96cOHHi1KlT9+7dc3V1hSDo4MGDYWFhp0+fdnZ2Li8v/+GHH+bMmZOVlTVI\n77tPJBJJcJb09HQ/P7+4uDgzM7OhPq+UE2vreObMmYcPH65evVpOTq6kpASCIBkZGWQy7okT\nJ0xNTTU0NNhs9rNnz168eLFixQpxxgYMRHt7+6VLl2rftLiNniHophDwRKqiqp6G4dmIi0Qi\nceB7dCYmJl67+WzCRE8uzC8qKmYwGGw2G4vFysvLN7bi7927R6FQhmJBVgAQm7Kysitx8V1b\nbgESSdZ5vPvd2zesrKxEu34eHBxcXFyck5MjWIPBxsZm8eLFdXV1yFMsFjty5EjksaOj47Jl\nyyZNmuTj4/P+/Xs8Hp+cnOzk5IR8MxsZGa1bt27dunWdnZ14PB7ZAkNHR+fWrVuNjY3Ozs6R\nkZHIf0Yej/fzzz9HR0dXVlZqa2sHBAQEBwcjN9SSk5O3bt2ak5PD4/H09PR27949b968nmGj\n0WgbGxvkMYPBgCBoxIgRgpRhS6yNd2JiIgzDp06dEqSoq6tHRUVBEITD4a5du9bQ0IDD4bS0\ntIKDg52dncUZGzAQSUlJ71PKJzu497zJraJIHW0+LiEhwczMbCBLnjGZzBcvXtjZu1R/qk3P\nKJBXpMjKKhLIOBjmfqprYTTVk3Atag8fCoZAAsDXKD09XU93hLycYq+5ZLKckYFZenq6CI03\nh8O5cuXKsmXLeq6eRKVSe30JDofbunXr7Nmzk5OTJ0+ePHHixD179rx8+XLs2LHV1dW///67\nu7u7YETC3bt3Q0JC6HR6Z2fnkiVLPD09X79+jUKhIiIi9u/ff+zYMWdn56dPnwYGBuJwuKCg\nIA6HM3PmTD8/vwsXLqDR6Pz8fGkYZvEVEWvjHRcX11eWn59f1xFtwFcEhuHs7GwLQ+ueLTdC\nW1Un71VObm6ukFEOn1VQUFBU0krTY2VlF+vomRAI/z+hnESSU1JS/VCSe/7CA29vbyUlJZHP\nAgCS9fHjR0313rfLQ2ho0D5+fCvCkcvLy9va2iwtLb/oVUj59+/fT548edOmTWw2e8KECRAE\ncbncadOm3bhxo0tgGmFhYRgMBovFnjhxgkajJSQkTJky5cCBA5s2bVq5ciUEQcbGxiUlJXv3\n7g0KCmIwGM3NzR4eHshsZD09PRHe1HAGZiAAA1VXV5ebVKiqpC6kjDpFc4B7fNXW1hJJ8q/T\n8zVpBl1bbgQWK6NnYMFoYSUnJw/kLAAgWR0dHXiCsAF6BAIJWRDpS48s2sAu5FXI7/IbN24c\nPHjw+PHjmZmZDx8+rKqqWrBggeCwVlZWgr0MtLS0VFVVc3NzKyoqmpubkfYe4eLi0tDQUF1d\nTaVSly9f7u7u7ubmtm/fvrdvRflFMpyBxhsYKBaLhcXICB+Ri5fBdwjdSuSzuFwug9EsK6dA\nJJB7LYDFypDICtnZfW50BgDSj0QisVjCdsPr6GgjEAgibPlDo9HIZPKXtpHI0liGhoYQBG3c\nuHHZsmX+/v6WlpYeHh5RUVEPHz58+fIlUrLbwgDIqhJd2/6ukJTz589nZGS4ubklJyfb2tpG\nRER86ZsazkDjDQwUmUxmc9lcWNhK+O2stgHu8aWgoNDQUEuW7T6KR4DHg1EofktLi2SX1ACA\ngdDV1a2sKhNSoKqqTLQrzDgc7rvvvouJiaHT6d2yBAPWumGz2eHh4dra2k5OThAEtbe3d/2N\njjwW/HfLyMgQbMKbl5fX1NRkbm5Oo9EUFBSSkpIEr0pKSqJQKOrq/1yos7S03Lx5c3x8/JYt\nW7rtTQUIBxpvYKBUVFQcpllX1HzsqwCPx6usq+hrtYB+GjFiBKOxhgf3uexXQ/1HXW1q0XsG\ni8UayIkAQIIcHR0/fqQ3NdX3mtvCZLwvKRB54kZERIShoaGDg0NERERSUlJ2dvbly5ddXV1j\nY2ORAlwu9927d+/evXv9+nV0dLSjo2NOTk5sbCyyvtPcuXMjIyNjY2OLi4ufPn26Zs0afX39\nUaNGIa9ls9nLli3Lz89PS0vz8fGxtbWdOnUqCoUKDQ09fPjw+fPni4uLo6Kijh49um3bNgiC\nCgsLQ0NDX758WVlZmZWVlZiYaGFhIdr7Gp7ARGpgoFAo1OjRo88nXdakauFkehkvWliWb+Zi\nZG5uPpCzaGlpWZqp0UuzlJTdIaj7VTg2u6OiLMfR3lZJrkUaVhcBANFoaWktWznnatyfzuPd\nlZRUuma1tDQ9f/Fo4eJpIm/gpKysnJqaeuTIkbi4uF27duHxeGNj40WLFgkGC9fU1JiamqLR\naFlZWQMDgylTpty9e1ewSMvRo0epVOqePXsqKyuVlJScnJz2798v2Ito+vTppqamrq6uzc3N\nkyZNOn36NHJtPDg4mMvlhoWFVVZW0mi0sLCwwMBACILIZHJBQcHFixfr6uooFIqbm9uhQ4dE\ne1/DE6o/oxj4fD6yNwCBQBCeKDb+/v7btm3rtmEzh8Npbm7uuWyqRCArrA3wWvGg4HK5DAZj\nSKsFhuELFy5kPMwdb+1C6jKajM/nF5cXdqoyFi1ahNw2Q1bBE7JekhC3bt069MtZOUUTPcNR\nMrj/b6GZLfUlxa/MzYzkFZRMDQn9XCEAqRYCgSANn9FAqmVwwTCMLBQvDdWCrIIn2WopKysL\nDw8/efKk2M7I4/EeP34ce+6ujo6RlqYuiUju6Giv+vTxQ1nR4qXTPTw8RLjhPdS8vLywWOzV\nq1clHcgw0q+ed2pq6tixY6H/jlfsNREYnjAYzOLFiwmEWw8v3NXTNKAqqspgZVramss+leo6\naCyctxBpuQfI2dk5MzPz/Ye2N5kPFRTViEQFHo/bymxQkIUdHe309E2T/r69bNGagZ8IACQI\njUZPmzbNwsICWdv8U0MbiUQa5agTsG6eGNYiBb4W/Wq8CQRCzy/fXhOBYQuPxy9atMjBwSEn\nJ6e6urqT06aqoDDe2GvUqFGDdR2bSqVOnjy5MOrWlClTWB1t7e2taDRRXkFXU1OfD/HTXv7l\nPdcZLJoIfBu0tLTmzp0r6SjEjU6nC3ar6mrPnj3InXJAoF+Nt42NTc8Bir0mAsOckZFRz/Wb\nBtHEiRP5fP6xk9c0tQ00tfSJRFk2m/Wenlv24d0ib1dPT8++FooBAGDodF2tZSCMjIzApdx+\nAgPWgK8JCoVydXU1NTVNTU0tLS1tZbQSicQxo2hr/ab3+oMdAADgmwQab+Dro6Gh4enpKeko\nAAAAJAbM8wYAAACAr8xnGu/q6upNmza5u7uvWLEiJSWlW25UVBSYVg8AAAAAYibssnlDQ4OD\ng0NlZSXyNCYmZsuWLQcOHBCMCaqtrc3LyxvyGAEAAAAA6EJYz/vgwYOVlZWhoaGlpaWpqanT\npk2LiIgICAgAowEBAAAAQIKE9bz/+OOPCRMm7N+/H4IgPT29+Pj4kJCQgwcPItu1iitCAAAA\nAAD+Q1jj/eHDhzlz5gieolCoiIgIHA63b98+DAbz22+/DX14AAAAAAB0J6zxJpPJyOrlXe3d\nu5fFYh0+fFhGRkZJSWkoYwMAAAAAoBfCGu8RI0akpaX1TD906BCLxTpy5Ei3fUEAAAAAABAD\nYQPWpk+fnpKS8uHDh55Zx44d8/PzKy8vH6q4AAAAAADog7Ce96JFi2pra+l0up6eXrcsFAp1\n+vRpKpX6/v37IYwOACAIgqCOjo62tjYp2ewVAABA4oQ13vr6+r/++mtfuWg0et++fUMQEgD8\ng8fjZWVlpaWl1dTUJJa/mUizplAo9vb2Dg4OWCxY2RcAgOELfAMCUqqzs/P333+/lPZQx3oE\nZZyGC0mvs6Mzt6Lm4bUT83JdFi9eDHrhwLeNz+c3NTV1dHQQCARlZWWwYx7QlbB73iwWa/Lk\nyevXr+81NzAwcPLkySwWa2gCA4a727dv/5771H6uq5aZIUGWhEKj8WSihome/VzXhx/Tr127\nxuPxJB0jAAwJLpebnJz8yy+//G9ewAGfzdvmrzl8+HBiYiKbzR66k4aEhKipqaFQqF9//ZXP\n5//www9UKhWFQkVHRw/dSQGRCWu8z549m5iYuGbNml5z165dm5iYeO7cuaEJDBjWioqKopNu\nWU4dI0PAd8vCyGDNJztey3qcnZ0tkdgAYEi1t7fHxMT8se8crZi3wsnT13neKue5BqXoxwcv\nnT17lslkinzkoKAgVA/p6ekQBD179uzgwYM3b95sampas2bNw4cPo6OjHz582NTU5OPjI8K5\nli5dOhRb/7W3t+/bt8/GxoZMJispKTk4OOzatYvBYEBd3h0Gg1FUVBw1alRwcHBZWVnPg6Sk\npMjIyIjt1ltra2vPahcQ+bDCor9+/bqLi4uZmVmvuSYmJq6urtevX1+7dq3IpweAXmVlZakb\n6+JIhF5zsTgZLXPDrKwsOzs7MQfWKxiGGxsb2Wy2goKCrKyspMMBvmJ8Pv/333+vvp/lZT8F\nh5FBEmUwWEMqTZei+Tjh5VXs1VWrVqHRIm4Iqaqq+ujRo64pJiYmEAQVFRUpKSk5OTkhicXF\nxVpaWg4ODgN4KyLicDgyMjK9ZjEYDBcXl48fP4aGhjo6OioqKubn50dHR8vKym7evBmCIDU1\ntcTERD6fz2Qy37x5c+LEiVOnTt27d8/V1VVwkPr6+kWLFk2bNu3PP/8UzzsikUhZWVnI4/T0\ndD8/v7i4uL4a1v4T1njn5OT4+/sLKTBmzJiTJ08OMAIA6KmqqkpJU0VIASUNalVOFZ/Pl+yN\nQCaTmZSUlJub+2dxKgaLgbnwQns3JycnCwsLcIcSEEFeXt6buIRFoz0ELbcAFo2ZYjbm6o0/\nM21t7e3tRTu+jIyMjY1Nt8Tvvvvu2rVrEAR1+6NFniI33WNjYw8fPlxUVKSurj5nzpy9e/fK\nyckhxS5evHjkyJGCggIymWxvb3/p0qXQ0NC4uDjBEU6dOhUQEMDj8X7++efo6OjKykptbe2A\ngIDg4GCkgJeXFwqFMjAwuHTpUk1NTWtrK4HQyw/34ODg4uLinJwcIyMjJMXGxmbx4sV1dXX/\nVBEWO3LkSOSxo6PjsmXLJk2a5OPj8/79ezweD0EQj8dbsmTJihUrZGVluzbeXl5eEATp6Ojc\nunWrsbHR2dk5MjJSW1sbeUlfYScnJ2/dujUnJ4fH4+np6e3evXvevHk9w0aj0YI6Ry4SjBgx\nouen8KWE/XxjMpny8vJCCsjLy7e0tAwwAgDoqbOzE4vr/dc3AovHwTDM5XLFFlJPVVVVkZGR\nZ1PvVRtiXVbMcV42e+x37umoqo2n9t6+fRvckgdE8ObNG1NNAzwW12uuDAZrqW385s2bwT3p\n1atXT506RaFQ+P86ePCgoaEh8phAIBw/fnzz5s0hISH5+flxcXEpKSkrVqxAXvvbb7+tWrVq\nwYIFWVlZiYmJM2fO5HK50dHRS5YsmTNnDnKEgIAACIIiIiL279+/ffv2vLy8rVu37ty58+jR\no4IY7t69i0ajCwsLGQwG0tB2w+Fwrly5smzZMkHLLUClUnt9XzgcbuvWrZWVlcnJyUjKnj17\n2Gz2jh07eha+e/cuiUSi0+nV1dUyMjKenp7IFlx9hc3hcGbOnDl27Njs7Ozc3NyIiAgFBYUv\nrfmBENbzVlBQ+PTpk5AC1dXVioqKgx0SAEBycnJlrU1C/iuwmG3KRGJfl9fEgMlkXr58+Q22\n1tp9PPRvfwVPJtKsjKn6mlEPb8rKyrq5uUkqPOArVV1dPVKh96YIoaGgklZdLfI1p8rKyq4v\nxGAwn/0FDMPwrl27wsPDlyxZAkGQgYFBVFSUnZ1dTU2NiorKrl27NmzYEBoaihS2tLTs9SB8\nPv/AgQObNm1auXIlBEHGxsYlJSV79+4NCgpCCtBotH379gm5HVBeXt7W1tbX8fuClH///v3k\nyZMTEhJOnz6dmZnZ61k0NDTCwsIwGAyy8xaNRktISJgyZUpfYTMYjObmZg8PDwMDAwiCeq6G\nMtSE9bytra3j4+P76kDweLyHDx8OvO8PAD0ZGBjUllQKKVBbWon8n5GUxMTEF81Fhg7mUI/v\nUIIc2dzVMTU1tb6+XiKxAV8vDocjgxX2kxSHlYFhWOTrOqqqqlldZGRkfPYlpaWlDQ0Nvr6+\ngjFWyFgTOp1eUlLS1NTUnx+pFRUVzc3NEyZMEKS4uLg0NDRUV1cjTy0sLITfyBdtK2rkVSgU\n6tOnT0uXLo2JidHQ0Oi1pJWVFQaDQR5raWmpqqrm5uYKCZtKpS5fvtzd3d3NzW3fvn1v374V\nIbyBEFZZCxcuLC4ujoiI6DU3IiKisLBwwYIFQxMYMKyNHj3aBqPZWF7Ta25LbWNN8cdx48aJ\nOSoBGIZzc3N1rE16ttwIOapSFqdC/P+fga+dvLx8S0erkALNHa1kMlnQzHwp5J63gLW19Wdf\nAsMwBEHx8fH8/xo/frygafzsQfoqKUghEonCj0Cj0chk8pf+n8rJyYEgyNDQMDs7u6amZsaM\nGVgsFovFhoSEwDCMxWJ37tyJlORwOF1fiMzKEx72+fPnMzIy3NzckpOTbW1t+2orh4iwxnvZ\nsmXW1tahoaHff/99amoqcnWFy+WmpqYuXbo0NDTUxsZm2bJl4goVGEaUlZXd3NzyE1/Xfeje\n/26qqnv7+OW2Reu0tLQkEhsEQY2NjX+9f62oThFSRlFDpaqqSmwhAd8GQ0NDeu1HIQXoteU9\nb/oOKUNDQyUlpTt37vSV1W34OgKHw3W9IE+j0RQUFJKSkgQpSUlJFApFXV29n2HgcLjvvvsu\nJiaGTqd3yxIMWOuGzWaHh4dra2s7OTk5OTm9ffs2+18//vgjBoPJzs5et24dUjgjI6O9vR15\nnJeX19TUZG5u/tmwLS0tN2/eHB8fv2XLlsjIyH6+l0Eh7J43Ho+/f//+rFmzLl26dOnSJRQK\nRSKR2tvbkR8jNjY2f/zxBw7X+8AKABig0aNHH8BsefToUVruMxUdDTyZyO5gNZTX2OG0d3z3\ng2BOi0Sw2WwUGo0SepUPIyMzpEtqAN+kMWPGpKenl36q1Ffp5bdpRVNNu4XCQK45cTicbgsk\n6OvrCx9phcViw8LCgoKCVFRUFi5ciMfjCwoKbt68eeHCBQwGs2PHji1btlAoFGSEV2Ji4ty5\nczU0NAwNDRMSEgoKCqhUqqysLIFACA0N3b17t76+vpOT09OnT48ePfrzzz9/UfARERGvXr1y\ncHAIDQ0dPXq0goICMlVs+vTpyFQxLpf77t07CIKQqWLHjx+n0+n37t3D4XA4HM7CwkJwKKT1\n7ZrCZrOXLVsWFhbW2tq6Zs0aW1vbqVOnolCovsIuLCyMiYmZPXu2jo5ObW1tYmJi16OJwWdm\nqWtra6empl68ePH69etv375taWnR0NCwtLT09vb28fGR4HAhYDiwt7c3MTHJysr68OFDW2sb\nkUikTXSxs7OT+Eby8vLyPBhmt7P6mokOQVBHS5uCnlhHnwLfAEVFxenTp58PPuhsbDdCXRcF\n/f8F2/d15YmF6Yv2rFdVVRX5+LW1tba2tl1Trl+/jkyUEmL9+vVUKvXIkSOHDx+WkZExMDAQ\nLMASFBQkJyf366+/7ty5U05OztHR0dvbG4Igf3//5OTkMWPGtLS0IFPFgoODuVxuWFhYZWUl\njUYLCwsLDAz8ouCVlZVTU1OPHDkSFxe3a9cuPB5vbGy8aNEiPz8/pEBNTY2pqSkajZaVlTUw\nMJgyZcrdu3d1dXX7c/Dp06ebmpq6uro2NzdPmjTp9OnTyLXxvsImk8kFBQUXL16sq6ujUChu\nbm6HDh36orczQCjRRgFInL+//7Zt27ptKM7hcJqbm6Vk7yk2m81ms6VhyQ4ul8tgMKSkWjgc\nDovFEswQlSCkWggEgmif0alTp9LRVTRL415zeTCcdiMhfOWW/txThKSpWmAYbmpqErlaBheX\ny+3o6JBstZSVlYWHh4t5QYu3b98+fPiw5Tldj6JJxhPb2ayPDdXEcXpubm6jRo0SZyTDhJeX\nFxaLvXr1qqQD+QKf6Xnz+fykpKTi4mIKhTJlyhTh074BYPgYP378rdP7qHqaBLlefhKVZRd6\nGDiam5uLPzDgG2BpaWlkZJQzMefDhw/t7e2yRKK7jpu1tbU0/P4GpISwxrutrW369OnPnj1D\nnlKp1D///FNKFqQEAMmytLQMmLzgxI3LclQlFrON3dGJkcHKU5XUjHTaGcwRTLn5y+aDfUsB\nkRGJxNGjR48ePVrSgYgbnU43Nu7lgtaePXu2bdsm/ni+iDiDF/blcuDAgWfPnllbW0+bNq2o\nqOjOnTsrVqwY9MV9RIZMV+iW0u2BBCExSE8kkDQFIz2RQAOYP6qkpIR7w+HzW3CyRGUFVT6f\nx8ivf3HlvtVoQ+8dO7S0tPp/5G+mWgaXNFSLNNTDsGJkZCT+Or9x48agHEecwQtrvG/dumVq\navr69WtkYFpwcPChQ4fodLqYJyr0CobhlpYWZJ1YAaTWWCyWNIzyRX5bdJs7KEGdnZ3SUy3d\nPjhJRQJBUGdnp2if0cuXL+OOXp/tMo8oQ6qrq2ttbYVhGG8tQG8AACAASURBVK+jRzabUFSd\n//jxYwqF0v/dI6SnWhAiV8ug4/F4kq0WJpOJTHQGAKkirPEuKSkJDAwUDClfsGDBoUOHSkpK\npKHxxmAwCgoK3UYdIwPWCASCNNwZkrYBa3g8XhqqRXpGZgmqRYTPqLGxMSMjw2XUZDVldQiC\nlJWVu+bq6OjEP/rD0dGx/5sySU+1IAPWRKuWQScNA9ZaWlpEXhEFAIaOsJ4Bi8VSUfn/nZ2Q\nxd87OjqGPCgAkG6ZmZlNBe1Iy90TTgZvqmfx+vVrMUcFAMDw8cWbwoI7QADw8eNHjd7W0BDQ\npGrX1tayWCyxhQQAwLDymdGw169fRxasgSCotbUVgqATJ07cv3+/a5no6OghCg4ApFNHRwcB\n1+fyLBAEEXGEwicl7e3tvW5LDAAAMECfabzT0tLS0tK6piQkJHQrAxpvYLghkUh1ncJ2j+jo\n7DAZZ0AikcQWEgAAw4qwxhvctAOAXunq6r7565GJrmlfBSrrKtTN1UG3GwCAISKs8ba3txdb\nHADwFbGzs0sxT6mur9JQ0eyZy2KzCj7krl/hL/7AAAAYJr54wBoAAIqKiq6uri/fPq9t6r7j\neEdn+7Osv10XONnY2EgkNgAAhgOwfCMAiGLcuHG8UN7Tp0/pOYU0NV0Sgczhsj81VNMrijyW\nTp4/f37/V2gBAAD4UqDxBgAROTk5jRgx4tWrVyUlJZXMKqIC0diSttBuzsiRIyUdGgAA3zjQ\neAOA6FRVVWfPni3pKAAAGHbAlT0AAAAA+MqAnjcAAIA0YjKZHz9+bG9vJxKJNBpNQUFB0hEB\nUgQ03gAAANKlpaXlzz//LCgoaGhoIJKILBZLWUnZ2NjYw8Oj225MwLAFLpsDAABIkbq6uqio\nqLdv31rbWM+bP8/Dw2Pu3Ll2dnZFxUVRUVFVVVUiHzkoKAiFQjk6OnZNpNPpaDQahUK9evXq\ns0dYunSpp6enyAF8KSRgb2/vrol6enpbt24VFNDW1hZk1dbWOjg4jBw5srS0VGxBSgpovAEA\nAKQFl8u9cuUKm8N2neyqqKgoSJeTl5s4cSIGg7l69WpnZ6fIx1dTU8vJycnKyhKkREVF6ejo\nDCjofhB5e3gCgXDjxo3nz59/tmRRUdGYMWMIBEJKSoq+vr5op/uKgMYbAABAWmRkZNDp9DFj\nxqBQqJ65o+xHVVRW9KeL3BdFRUUvL6/IyEjkKZvNjomJ8fX17Vrm0aNHEydOVFFRkZOTc3R0\njI+PR9J9fX3j4uLu3r2LQqFQKNTp06chCJo4cWJQUJDgtZcuXRKsCuzl5eXt7R0SEqKlpUUk\nEpFN9mJjY62srAgEgp6eXmBgIJPJFB6wrq6ul5fXjz/+KHxDy5SUlHHjxjk4OCQkJCgrK39Z\npXydQOMNAOLQ3t6elpZ28+bNixcv3r59OzMzk81mSzooQOrk5eWNMBnR1wo/KBRq5MiReXl5\nAzmFv7//5cuXkV0ib926RSAQPDw8uhZobm4OCAh4/vx5enr6jBkzZs+ejZwxOjp6yZIlc+bM\n4fP5fD4/ICDgs+e6e/cuGo0uLCxkMBh4PP748eObN28OCQnJz8+Pi4tLSUlZsWLFZw8SERGR\nm5t78eLFvgrcvHlz8uTJy5cvv3r1Kh6P/3wVfBPAgDUAGHJZWVmPHj0qe12tTtHEy+A72BUP\n6xJGjNebNWsWWNEF6Kq+vp6mQxNSQEVFJS83j8/n99o17w9nZ2cajXb58uXVq1dHRkb6+vpi\nMJiuBRYsWCB4vHPnzr///vvatWu7d+8W4Vw0Gm3fvn3IbxEYhnft2hUeHr5kyRIIggwMDKKi\nouzs7GpqatTU1IQcRF9fPygo6KeffvLy8uq5U19NTc2CBQs2bdoUEREhQoRfL9DzBoCh9erV\nqyM/HSfVUaaNmWkzYpSpvoWdicOM8Z6oCuKeoPDc3FxJBwhIER6PJ3xhXRQKhXR8B3IWf3//\nqKiooqKi5OTkVatWdcutrq4ODAy0s7PT1tZWV1dPS0v78OGDaCeysLAQvJ3S0tKGhgZfX1/U\nv+zs7CAIotPpnz3OTz/9xOVye22elZSUXF1dz507l56eLlqQXynQeAPAEKqtrX38+LGTjYuu\nhn7XrhIKhTKmjXQ0G3vv3r3P3vYDhg9lZeWmxiYhBZqamhQVFQe4cr6Pj09+fv7atWunT5+u\npaXVLXfmzJk5OTlHjhx5/vx5dna2i4uLkFs83SKBYbjrUyKR2C0rPj6e/1/jx4//bMDy8vJ7\n9uw5ePBgRUVFtywcDnf//v0JEya4uromJSV99lDfDNB4A8AQevHiRRsdVqf0snMoBEF6mgY1\nOYzU1FQxRwVILRMTk+LiYiEd66KiooHfalFUVFywYMGTJ09Wr17dLauxsTEzM3PHjh0TJ07U\n19dXVVUtKioS5OJwOC6X27W8qqpqfX294GlBQUFfJzU0NFRSUrpz545oMa9atcrY2Pinn37q\nmYXH469fvz5v3jwPD48HDx6IdvyvDmi8AWAI0el0XQ1hs1Z0NfT7c9kQGCZGjx6trq7+9u3b\nXnPfFbxTVFAcN27cwE90+vTpurq66dOnd0tXVFSkUqlI/5jL5W7durW8vFyQa2homJOTU1BQ\nUF9fj4wed3Nze/DgAdLAP3/+PDo6uq8zYrHYsLCwM2fObNu27e3bt0VFRXfv3vXx8elnwGg0\n+siRI5cuXfr06VPPXAwGc/78eT8/v7lz5169erWfx/yqgcYbAIYKl8tta2uTI8kJKSNPkmcw\nGGILCZByBALB29ub2cJMfZXa9WI1l8PNyMior6+fN2+erKzsoJxIRUWl56g3NBp98+bNhIQE\nDQ0NExMTNBrddVUWf39/c3PzMWPGUKnUmJgYCIJ8fHxWrVrl4uKirq7+yy+/hISECDnp+vXr\n4+Li/vrrL0dHR3t7+507dxoYGPQ/ZldX11mzZvU1zR2FQh09ejQ0NHTJkiVnzpzp/2G/UqgB\nDnyQFH9//23bttFo/xmWyeFwmpubiUQimUyWVGACbDabzWYPyn+zAeJyuQwGQ0qqhcPhsFgs\nOTlh7Zl4INVCIBCG7jPi8/l79+6lcUyU5PuceFpVX9mh2rB+/XopqRYYhpuamoa0WvqPy+V2\ndHRItlrKysrCw8NPnjwpzpNWV1f/8ccfaWlpaupqZBK5vb29pqbG1tZ21qxZ3b70gGELTBUD\ngKGCQqFUVVXr8mqFNN51TTVGVtp95QLDk4aGhp+fn7u7e1lZWVtbG4lEotFoenp6Ik8PA749\n4LI5AAwha2vrwrICmAf3mtvJ6XxfUWxtbS3mqADph0KhdHR0nJ2d3d3dJ0yYoK+v/6223HQ6\nHdWbvXv3Sjo0qQZ63gAwhOzt7TOnZb569WKshVO3STVcmPviTZLrAiczM7NuI3gBYPgwMjL6\nSu/eShboeQPAEMJisYsWLaI5UBPS4qvqKng8HgRBMMz9WPPh0asH5q5GXl5e32qPCgCAoQN6\n3gAwtJSUlPz8/J6ZPMvKynr+JBEnI9PJYdtNsVgy02vcuHEyMjKSDhAAgK8PaLwBMens7Ozo\n6MDhcJIORAIIBIKbm9uUKVPqlte1tbXJyspSqVTQ4QYAQGSg8QaGFpfLff36dWZmZl1dXWLZ\nhwk0HXV19YkTJ5qamg631guNRgvfgAEAAKCfwD1vYAgxmcxz587tvnz5jQyWZT/KadGiNhvr\nlPb2DUd/vXXrVrdlkAEAAIB+Aj1vYKhwudwrV648+vjRfs4cjMw/f2lkGRkdazntkSbRfz3G\n4/EzZ86UbJAAAABfI9DzBoZKenr6rbc5ZhMnClpuAYKcnIWr65HbtyorKyUSGwAAwFcNNN7A\nUMnKyqKZW/RsuRFkJUWqnn5mZqaYowIAAPgGgMYbGBIcDqempkZZq/etMBFKmhqg5w0AACAC\n0HgDQ4LFYiWWfcAKnRgmg8cjuwoCAAAAXwQ03sCQIJFIkw0MWa1tQsqwWtukYR8tAACAr45Y\nR5s/ePAgMjKya8qePXsEuzKkp6dfvHixoqJCQUFhypQpixYtGm7zgL8lGAxGV1e3uOyDnAql\nrzJ1ZR/0p7qJMyoAAIBvg7inisnJye3Zs0fwVFPzn3uihYWFe/fu9fDw2Lhx4/v370+ePMnj\n8ZYuXSrm8IBBNHbs2NiICHUjI5KCQs/cug8fHGTl7O3txR8YAADA107cjTcGgzEwMOiZfuvW\nLS0tLX9/fwiCdHV1q6ur79696+3tjcfjxRwhMFiMjY0DZ806+fgvi0mTZCn/6X/XlpYWJr/w\nXbdOXl5eUuEBgDSDYbikpOTDhw8dHR0EAkFHR8fIyAiLBStzAP8Q958Ck8n08fHhcrna2tpz\n5swZP348kl5QUODi4iIoZmdnd+3atZKSElNTUzFHCAyimTNn4vH48N+vUWg6ylqaWBy+o5VZ\nU1o6Xpnit26dra2tpAMEAGlUVlZ27969R48eUZSVcXg8m81ubGiYPGXKrFmzDA0NJR0dIBXE\n2njTaLQ1a9bo6uqy2eykpKTw8HBfX9/Zs2fz+XwGg6GkpCQoiTxubGwUpOTl5V28eFHwtK2t\nra2tjclkdj0+st8im81GHkgWj8fj8XjdIpQIZK9cSVXLuHHjogwNc3JyKisrWYxmMplMc5lo\nb28vKysr2cpBaoPD4UjDZyRtfy1SUi18Ph+GYclG0tbWJuZ1fIuLi69du1ZZWTnB2RmDwSCJ\nPB6PXlx85cqVuXPnmpubi3bkoKCgo0ePOjg4pKWlCRLpdPqIESP4fP7Lly/HjBkzCG9gMCxd\nurS1tfXOnTuSDkR6DWHjnZWVFRYWhjyeMWOGn5+flZWVlZUVkmJpadnW1nbz5s3Zs2f352i1\ntbUJCQmCp4aGhmw2u7Ozs2dJGIalZ9FsqYpEUsEoKipOmDChW2Kvn534gb+WXoFqEWCz2cgP\nGvFAvhXramuN/tvDRqPR+vr6VVVVd+7c0dbWVuhtHEl/qKmp5eTkZGVlCa57RUVF6ejolJWV\nDTR0QLyGsPE2NTU9fvw48lhWVrbXAi9evOByuVgsVlFRsampSZCFPFZWVhakjBkz5u7du4Kn\nO3fulJeX79pZhyCIy+UymUwCgUAkEgf3vYiAw+FwOBwSiSTpQCAYhltaWqSkWrhcbmdnJ5lM\nlnQg/1QLHo+Xhs9IeqqFx+M1NzdLSbXAMMxisSRbLS0tLYLurxikpqZmZWY6Ojr2mqupqZmR\nmfnixYvp06eLdnxFRcUpU6ZERkaePn0agiA2mx0TE7Nhw4bt27cLyvB4vJ9//jk6OrqyslJb\nWzsgICA4OBiZ++Pl5YXBYExNTc+dO9fS0uLk5BQVFYWMOxbyKgiCLl68eOTIkYKCAjKZbG9v\nf+nSpUePHq1evbqqqkpRUREpc+nSJV9f38rKypCQkLi4OAiCkJefOnUqICAAgqDY2NjDhw8X\nFRWpq6vPmTNn7969yFzT5OTkrVu35uTk8Hg8PT293bt3z5s3T7T6+YoMYeNNIBC0tbWFFCgo\nKFBUVESGYJiammZmZq5atQrJyszMJBAIXYe2EYlELS0twVPMv7oeELkQikKhxPmfrS8wDEtJ\nJEi/QUqC4fF4UhIJqBYhpCQYPp8v8UgwGIw456wWFRUJ5uD0SktLq6ioSOTGG4Igf3//GTNm\nHDp0SFZW9tatWwQCwcPDo2vjHRERsX///mPHjjk7Oz99+jQwMBCHwwUFBSG5d+7csbS0LCoq\nYrFYM2bMWL9+/c2bN4W/6rffftu8eXNYWJinpyeXy01MTORyud7e3hs3brxw4cKGDRuQI0dF\nRXl5eVEolOjoaBaL1e2y+fHjx8PCwn799dexY8dWV1cHBQWtWLHixo0bHA5n5syZfn5+Fy5c\nQKPR+fn5w2SYs1jveZ84ccLU1FRDQ4PNZj979uzFixcrVqxAsubNmxcSEhIZGenu7l5SUnL7\n9m1PT89h8hkAAAAgGAxGr9cpBWTJ5ObmZhiGRf5N4+zsTKPRLl++vHr16sjISF9f366H4vP5\nBw4c2LRp08qVKyEIMjY2Likp2bt3r6DxHjFixLZt2yAIIhAI69atQ/rEQl4Fw/CuXbs2bNgQ\nGhqKHMHS0hJ5sHLlysjISKTxfvfu3fPnz58/f95rzMhBwsPDlyxZAkGQgYFBVFSUnZ1dTU0N\nGo1ubm728PBAOnt6enqiVctXR6wrrOFwuGvXrm3fvj08PPzjx4/BwcFz585FskxMTP73v//l\n5+f/+OOPFy5cmDt3LvIhAQAADB8YDEb4PX4ej4dGo9HoAX11+/v7R0VFFRUVJScnC653Iioq\nKpqbm7sOUnFxcWloaKiurkaejhw5UpClqqrKZDLb29uFvKqkpKSpqcnNrZflmAICAt69e5ec\nnAxBUGRkpIWFhZOTU68Bl5aWNjQ0+Pr6ov5lZ2cHQRCdTqdSqcuXL3d3d3dzc9u3b9/bt29F\nr5evilh73n5+fn5+fn3lOjg4ODg4iDMeAAAAqUKlUss+fKBQ+lyXsLm5eaSp6QCv5Pv4+Gzd\nunXt2rXTp0/X0tKqq6sTZAluJ3V7iSClZ4+fx+MJeVVfWRAE6enpeXh4REZGOjg4XLx4cdeu\nXX0FjPygiY+Pd3d375l7/vz5jRs3Pnr06MmTJzt37ty/f/+WLVv6OtQ3A6xtDgAAIC0sLCzK\ny8v76nzzeLyPHz9aWFgM8CyKiooLFix48uTJ6tWru2XRaDQFBYWkpCRBSlJSEoVCUVdXF3JA\nIa8yNDRUUlJ69OhRry9cu3btjRs3IiMjOzo6vv/+e0E6DofjcrmCp8hBhMwcs7S03Lx5c3x8\n/JYtW7otwv2tAuv1AAAASAtbW9tp7u6pqamWFhbdeqt8Pv9dYeHESZNGjx498BOdPn360KFD\nPbv4KBQqNDR09+7d+vr6Tk5OT58+PXr06M8//yz8aEJehcFgduzYsWXLFgqF4unpyefzExMT\n586dq6GhAUGQu7u7pqZmcHDw999/33X+m6GhYUJCQkFBAZVKlZWVJRAIYWFhQUFBKioqCxcu\nxOPxBQUFN2/evHDhQmFhYUxMzOzZs3V0dGpraxMTEwf+4+arABpvAAAAaYHBYBYtWsRms1+9\nfGlsbKygoIA04S0tLcXFxaPs7b/77jsZGZmBn4hAIBAIhF6zgoODuVxuWFhYZWUljUYLCwsL\nDAz87AGFvCooKEhOTu7XX3/duXOnnJyco6Ojt7c3koVGo/38/EJDQ5GBbwL+/v7Jycljxoxp\naWlBpoqtX7+eSqUeOXLk8OHDMjIyBgYGnp6eEASRyeSCgoKLFy/W1dVRKBQ3N7dDhw4NqGq+\nEihxrj8wiPz9/bdt20aj0bomcjic5uZmIpEoDfNl2Ww2m80WPnBUPLhcLoPBkJJq4XA4LBZL\nGnYCRaqFQCBIw2ckPdUCw3BTU5OUVAuXy+3o6JBstZSVlYWHh588eVKcJ21vb09ISHjz5k1a\naioej+9ksx0cHCwsLKZO/b/27juuqXv/H/gnAZIAIYIYFRCZoQYBERVHUSxyrbQ47i2IONAq\nGKsXtS2gtHqLvU4cdeCAOlBBpVUc7b3Wq1gHOFlaVFBkiKOK7Jn9++P8vmkaEBUtyZHX86+T\nz+eTz3nnzXnwzhk552+6sJG8XeHh4ZcvX87MzNR2IDSDPW8AAN1iZGQ0bty4Dz/8sKysrKGh\nwcjIqFevXrpwk6W3q6qqKisra/fu3bt27dJ2LPSD4g0AoIvYbLajo6O2o/gLDR06tKysLCQk\nJCgoSNux0A+KNwAAaEF+fr62Q6Ax/FQMAACAZlC8AQAAaAbFGwAAgGZQvAEAAGgGxRsAAIBm\nULwBAABoBsUbAACAZlC8AQAAaAbFGwAAgGZQvAEAAGgGxRsAAIBmULwBAABoBsUbAACAZvBU\nMQAAnVNbW5uTk1NaWtrU1MThcHr37t2/f39TU1NtxwW6AnveAAC65fr161u2bLl//76Dg8OQ\nIUMEAsGDBw/i4uIyMjK0HdoLzZgxw9/f/02GHTp0SF8f+5OvCsUbAECHXL58+cyZM5988snM\nmTPff//9fv36DRs2bPr06cHBwRkZGefOnWv3zAsXLmQwGJ6enuqNhYWFTCaTwWBcuXLlTUN/\nBV5eXr6+vu17LxU/g8HQ09MzNTUdMGBAZGRkaWmp+pjGxsYVK1a4u7sbGxubmZkNGjQoJiam\nurr6bcSuW/A1BwBAVzx//vzMmTOTJ092cHDQ6HJwcAgODk5MTHRycrK0tGzf/D169Lh582ZO\nTk7//v2ploSEhN69e2uUwL9OaGjom7y9R48e586dUyqVdXV1N27c2Lp16/bt20+cOOHj40MI\nqa6u9vb2fvDgQXR0tKenp6mp6e3bt3fu3MnlciMiIt7SJ9AV2PMGANAVV69edXJyalm5KdbW\n1q6urpcvX273/KampgEBAfHx8dRLiUSSmJioXlCTkpKMjIzUd1WTkpI4HE5FRQX1cu/evW5u\nbhwOx9bWdsGCBXV1dS3XolAoVqxYYWdnx2Kx7O3tY2NjlUol1aV+2Fwuly9atIjP53O53MDA\nwOfPn780fn19/T59+giFQk9Pz7CwsGvXrvXr1y8kJEQsFhNCIiMj7927d/369aioqJEjR7q7\nu0+ePPns2bPTp08nhKSnp3t5efF4PC6X6+Likpqa2o4E6g4UbwAAXVFUVOTs7NzGgL59+xYV\nFb3JKkQi0YEDB+rr6wkhqampHA7Hz89P1RsYGMjlcvft26dqSUhICAgIMDc3J4TExcVFREQs\nWrTo9u3bycnJly5d+vTTT1uuIjY2duXKlUuXLr1169bixYu/+eabTZs2tRy2cuXK7du3b968\n+ebNm56enkuXLn3dz8JisRYvXvzo0aP09HSpVHrw4MHp06c7OjpqDOPz+VKp1N/ff+jQobm5\nuXl5ebGxsV26dHnd1ekUHDYHANAVdXV1bV9SbmZm1tDQIJfL9fT02reK4cOHW1tbHzhwYPbs\n2fHx8aGhoepTsdnsmTNnxsfHz58/nxCSn59/8eLFixcvEkLkcnlMTMyaNWumTJlCCLG3t09I\nSPDw8Hj69GmPHj1UMyiVytWrV3/55ZczZ84khAgEgqKiouXLly9cuFA9DIVCsW7dukWLFgUH\nBxNCIiMjr1+/3o69YVdXV0LI/fv37ezsGhoaqJctVVdX19TU+Pn52dvbE0JsbW1fd0W6Bnve\nAAC6gs1mNzc3tzGgqanJwMCg3ZWbIhKJEhIS7t69m56ePmvWLI3eOXPm5Ofnp6enE0Li4+Nd\nXFy8vLwIIcXFxRUVFaGhoYz/4+HhQQgpLCxUf/vDhw9rampGjBihavH29q6oqHjy5InGsNra\n2uHDh6sPa8dnoQ7IMxgM1ZH5VvH5/BkzZowZM2b06NErVqz47bff2rEunYLiDQCgK6ysrIqL\ni9sYUFxcbGVl9YZrCQkJuX379ty5cz/66KOWs9na2vr5+cXHx4vF4v3794tEIqpdLpcTQk6e\nPKn8s/fff1/97apqqjGtRgs1jM1mq1rUl1/dzZs3CSEODg7W1tbGxsZtVOU9e/ZkZWWNHj06\nPT29f//+sbGx7Vid7kDxBgDQFe7u7llZWa1eBUYIaWpqunr1qupC8XYzNTWdOHFiWlra7Nmz\nWx0wd+7cw4cPx8fHNzU1TZs2jWp0cHAwMzM7duxY25NbW1t36dLl/Pnzqpbz58+bm5v37NlT\nYxiPx8vNzVW15OTkvO4HkUgka9as6dWrl5eXF4vFmjRpUmJiosaRAEJIeXk5teDq6hoREXHy\n5MmoqCjVVXs0hXPeAAC6ok+fPjY2NocOHZoyZYqRkZF6V3Nzc0pKCp/Pd3Nze/MV7dixY926\nddRlaC2NGTPG0tIyMjJy2rRpqgu79PX1ly1btnDhwm7dugUFBbHZ7Dt37hw5ckT96jZCCIPB\niI6O/vbbb+3s7Ly8vH799ddNmzatWrVKYxVMJvOLL75YvXr1qFGjHB0dL1y4kJSU9NKwZTJZ\nfn4+IYT6qVhcXFxhYeGJEydYLBYhJDY29sqVK4MGDYqOjh48eHCXLl2on4p99NFHY8eOTUxM\nHDduXO/evZ89e3bu3DkXF5d25E13oHgDAOiQTz75JCUlZfv27V5eXu+99x6Px6uvr6fOT5uY\nmAQHB7c8It0OHA6Hw+G8qJfJZIaFhUVHR8+ZM0e9PTw8nM/nb9iwYf369QYGBvb29hMmTGj5\n9sjISJlMtmzZskePHllbWy9btmzBggUth3399df19fVDhgzhcDjOzs6LFy9+6QXnT58+FQqF\nTCaTy+Xa29v7+voeP37cxsaG6u3atevVq1c3bNiQnJwcExPDZrMFAkFwcHBYWFh1dfWdO3f2\n799fXl5ubm4+evTodevWvTxNOuwlJ/l1lkgkWrJkibW1tXqjVCqtqakxNDQ0NjbWVmAqEolE\nIpFwuVxtB0JkMll1dbWOpEUqlTY3N5uYmGg7kP+fFg6Howt/I91Ji1wur6qq0pG0yGSypqYm\n7aaltLR0zZo127Zt68iVKhSK7Ozsa9euPXv2jGoxNzcfNGjQoEGD3vBStVcXHh5++fLlzMzM\njlkdvC7seQMA6BYmkzlw4MCBAwc2NjY2NDQYGRl15DfvqqqqrKys3bt379q1q8NWCq8LF6wB\nAOgoIyMjPp/fwcfMhg4dOn78+JCQkKCgoI5cL6WwsJDRmuXLl3d8MLoMe94AAPAH6oowbXF0\ndKTpydwOhj1vAAAAmkHxBgAAoBm6HjZXKBRisVjjPoLUDYBkMlnb9xfsGDKZTC6X60IkCoWC\nEKIjwcjlch2JBGlplU6lRaFQaD0SsVhM5QRAp9C1eBNCqDvzabRoLGidLkRCxdAyXdoNRtuB\nIC1tQVo0YgDQNXQt3kwmk8PhGBoaqjdSP5bV19fXaNcK6nfeuhAJ9WNZHUmLVCpVKpW6EAnS\n0iq5XN7Y2KgjaZHJZAqFQruRcDgcJhOnF0HnYKMElMDtKgAAEkxJREFUAACgGRRvAAAAmkHx\nBgAAoBkUbwAAAJpB8QYAAKAZFG8AAACaQfEGAACgGRRvAAAAmkHxBgAAoBkUbwAAAJpB8QYA\nAKAZFG8AAACaoeuDSQDaTSqVNjQ04GkTAEBfKN7QiRQUFFy6dKmsrOxsac4IK1cOh+Ph4eHj\n48NisbQdGgDAa0Dxhk5BoVD8/PPPm08m9errwPewHO49VtwsflD66MKppMLCwilTppiZmWk7\nRgCAV4Ujh9ApnDlzZltaisc4b1sPoXHXLnoG+mxjw+72vdz9R1yoKUhOThaLxdqOEQDgVaF4\nw7uvvLz86tWrfX08jbqYaHQx9ZhC7wG/lGZeunRJK7EBALQDije8+3Jzc3Plj034rR8YZzCZ\nvd2ccnJylEplBwcGANA+KN7w7nv06JGZVfc2BphZ8U/culhXV9dhIQEAvAkUb3j3icVifZZB\nGwP0WSwGg9Hc3NxhIQEAvAkUb3j3mZiYiOsb2xjQXN840rqfiYnmGXEAAN2E4g3vPjs7u/KS\nx+TFp7TLSx5ZWFgYGhp2ZFQAAO2G4g3vPnd39+HdhI8LSlrtba5vLLt5b9iwYR0bFABA+6F4\nw7vP0NBw3LhxhVd+e1JQorH/3Vhdd/OXDJFPYN++fbUVHgDA68Id1qBTEAqFWz//9/HjxzPy\nz/PtLDlcI6lY8qz0cfXDZ1GfhI0ZM4bBYGg7RgCAV4XiDZ2FUCi0sbHxys4uKiqqra5lsVg8\n5z4Dpg0QCATaDg0A4PWgeEMnYmRk5OXl5eXlRQiRyWTV1dUcDkfbQQEAvDac8wYAAKAZFG8A\nAACaQfEGAACgGRRvAAAAmkHxBgAAoBkUbwAAAJpB8QYAAKAZFG8AAACaQfEGAACgGRRvAAAA\nmqHx7VGPHj3apUsX9Ra5XN7c3GxgYMBisbQVlXowcrlcFyJRKBRNTU26kxaZTMZms7UdCNLS\nOp1Ki0KhkEql2k1LVVWVFtcO8CJ0Ld5Tpkx5/vy5RmNlZWVWVpa9vb1QKNRKVOqUSqVCodDT\n09N2IKS6ujorK8vGxsbFxUXbsehQWmpqarKysnr37u3q6qrtWHQoLbW1tVlZWdbW1m5ubtqO\nRSfSYmJiMnDgQC0GANAquhbvESNGtGzMzc3dsWOHm5vbP/7xj44PSWfl5eVt27bN2dkZaVFX\nUFCwdetWgUCAtKi7f/9+XFycnZ0d0gKgy3DOGwAAgGZQvAEAAGgGxRsAAIBmGEqlUtsxAAAA\nwGvAnjcAAADNoHgDAADQDIo3AAAAzdD1d94qSqXy8OHDaWlpz58/NzY2dnNzCwkJ4fP5VG9m\nZub+/fsfPnzYpUsXX1/f4OBgBoOh3YA7WH5+fnR0tFKpPHbsmKoRaUEGNGA7AaAX2u95p6am\nHjx4MCAgYMuWLREREUVFRStWrKC6CgoKli9f7uzsvGHDhqlTp6ampiYnJ2s32g5WW1u7du3a\n/v37qzciLciABmwnALRD+z3v27dvOzs7+/r6EkIsLCw+/vjjHTt2SKVSAwOD1NRUKysrkUhE\nCLGxsXny5Mnx48cDAwN14Q7SHUCpVK5fv97X15fD4WRnZ6vaO3laCDLwZ9hOAOiI9nverq6u\nhYWF+fn5hJCqqqr09HQPDw8DAwNCyJ07dzw8PFQjPTw8mpubi4qKtBZrxzp06JBMJps0aZJG\neydPC0EG/gzbCQAd0X7Pe8KECTKZLDo6mhAil8s9PDwWL15MCFEqldXV1WZmZqqR1HJlZaW2\nQu1IN27c+OWXXzZu3KhxkrKTp4UgA3+G7QSApmhWvHNycpYtW0Ytf/zxx2FhYRkZGampqSKR\nSCgUPn/+PDExMTY2dunSpdqNs4NppCUgIGD9+vULFy5U/+cLoKGqqgrbCQBN0ax4C4XCuLg4\napnL5RJCdu3a5ePjM2bMGEKIjY0Nl8uNjIwsKCjo06ePqamp+rN4qeWuXbtqI/C/lkZaioqK\nqqurv/32W6pFqVQqlcoJEyZMnDhx8uTJnSctrWIwGJ08AyrFxcXYTgBoimbFm8Ph9OrVS71F\nLBYzmX+cuaeO/snlckKIUCjMzs6eNWsW1ZWdnc3hcOzt7Tsw3g6ikRZnZ+ctW7aoXqalpZ04\ncWLTpk2mpqakM6XlRZABCrYTAPrSi4mJ0XYMb+TJkydpaWnm5uZsNrukpOT7779nsVhTp07V\n19fv3r17ampqTU0Nn8/PycnZt2/f+PHj1a/BeVfp6+ubqikpKblx48acOXM4HA4hpNOmRQUZ\noGA7AaAv2j+YRCwWp6SkpKenV1ZWGhsbOzs7h4SEWFhYUL3Xr19PSkoqKyuj7jIxefLkTniX\niaNHj+7du1f95htICzLQErYTABqhffEGAADobGj/O28AAIDOBsUbAACAZlC8AQAAaAbFGwAA\ngGZQvAEAAGgGxRsAAIBmULwB/lpnzpxhMBiJiYnaDgQA3h0o3p1IZmYmQ42RkZGLi0tMTExj\nY6P6MIlE8v333//tb3/j8/ksFqtbt26+vr47duxobm5WjYmNjZ04caKjoyOTyWQwGDKZrB3x\nlJSUrFq1avjw4T169OByuS4uLosXL66oqNAYVllZOX/+fFtbWxaLZWlpGRoa+vvvv6sPaGho\n+Oc//2lhYdG1a9eAgIDHjx+r9+bl5bHZ7JMnT7YjQgAA3YSbtHQimZmZgwYN8vT0DAwMJISU\nl5cfP368oKBg5MiRZ8+epW6e9fDhw7Fjx+bm5goEgjFjxlhYWNTW1l65cuX8+fMffPBBWloa\nNZW+vr6JiYmHh8dvv/1WXl4ulUr19V/7PvkLFy7csmXLsGHDBg0axGaz09PT09PTe/Xqde3a\nNdU98iorK4cMGVJYWBgcHOzp6VlUVJSQkNCzZ8+rV692796dGhMeHp6SknLgwAFTU9NPP/2U\nx+NlZGRQXQqFYtiwYY6OjklJSW+ewPZRKBQSicTAwEBPT09bMQDAu0YJncb169cJIbNmzVK1\niMViNzc3Qsivv/6qVColEom7uzshZOXKlXK5XP29t27dmjdvnurl/fv3qYXBgwcTQqRSaTvi\nOX36dHFxsXpLVFQUIeSLL75QtVBPZ1+zZo2qhfoCERYWpmqxtLSMjo6mlg8dOkQI+f3336mX\n3333Xbdu3crLy9sR3ptraGjQynoB4J2Hw+adGovFop6meu/ePULIvn37cnNzp0yZEh0drf6s\nNkKIs7Oz6qmjhJBXebpUbm7ulStXlC8+tOPr62tra6ve8tlnnxFCbt26pWo5e/YsISQsLEzV\n4uPjIxAIDh48qDqML5PJ2Gw2tUw9VEMikRBCSktLlyxZsnHjxm7durUd6unTpxkMxooVKzTa\nZ8yYoaenV1ZWRgipqalZsmTJ4MGDu3Xrxmaz7e3tIyIi6uvrVYMPHz7MYDB++OGHZcuWCQQC\nFotFPW1T45z3K85z5MiRNWvWODk5sdns3r17r1ixQiOTMpls48aNAwYMMDY2NjExcXNz++ab\nb9R7N2zY4O7ubmhoaGJiMnLkyP/9739tJwEAaIRmjwSFt+7u3buEEHNzc0LIjz/+SAgJDw9/\nKzMHBATcv3+/qamJKqiv4tmzZ4QQ1TFzQohYLGYwGIaGhurDjI2N6+vr8/LyBg4cSAjx9fVN\nTk4ODQ3l8Xjbtm0TCoXW1taEEJFINGLEiClTprx0vaNGjbK2tt67d+/XX3+tamxoaDhy5IiP\njw81W1lZWUJCQkBAQHBwMIvFunDhwoYNG65du3b+/Hn1x3UsWrTIyspq5cqVPXv2NDAwaLmu\nV5wnKirKyclp8+bNpqamO3fuXLJkibm5+Zw5c6hemUzm7+9/6tQpb2/vf/3rXzweLz8//8cf\nf1y2bBkhRC6Xjxs37tSpU4GBgaGhoc3NzUlJSWPGjElOTg4ODn6VPwQA6Dpt7/pDx6EOmwcF\nBRUXFxcXF1+/fj0iIoIQwufza2pqlEqlpaUlg8F4rWPgbRw2d3BwIIQ0NTW94lQKheLDDz8k\nhGRkZKgaqWLz3//+V9Xy8OFD6tvAkSNHqJanT5+OHj2awWAwmUw3N7ebN28qlcq9e/dyudzS\n0tJXXPtXX32lseo9e/YQQpKSkqiXzc3NEolE/S3Unvrp06epl9RXHycnJ41snD59mhCyZ8+e\n15pn4MCBCoWCapHL5QKBQCgUqt7y3XffEULCw8NVY6hh1MLWrVsJIbt371Z1SSQSDw+PHj16\ntO8EBwDoGhTvToQq3hpcXV0zMzOpAVwu19jY+LXmfJNz3hoiIyMJIVFRUeqNV65c0dfXt7Cw\nOHjwYGFh4ZkzZ/r3789isQgh+/fvVx9ZV1dXUVFBLT979szc3Hzz5s1KpfLw4cMDBgywsLAY\nP358SUnJi9ZOHYGYPXu2qsXb25vH4zU2NrYcLJFImpqabt++TQj597//TTVSRXfVqlUagzWK\n9yvOs2nTJvWR1G66qjwPGDDA0NCwrq6u1c8yePDg7t27N/3Z2rVrCSGqvzUA0BrOeXc6vr6+\nR48ePXbs2C+//HL37t2bN28OGDCA6qJqVft+9/WGli5dunbt2tDQ0NWrV6u3Dx48+PDhw0wm\nMzg42NHR0dfXt2fPntTRYx6Ppz6Sy+V27dqVWp4/f75AIJg3b15GRkZAQEBAQMDx48dra2v9\n/Pxe9OkEAsGwYcNSUlKoU+klJSUXLlyYOHGi+hH7xMTEYcOGGRsbs1gsQ0NDZ2dnQkhlZaX6\nPHZ2di/9sK8yD3WsXoXH40kkkrq6Ourl3bt3HR0duVxuq/PfuXPn2bNnhn9GfTeiTkwAAN3h\nnHenY2NjM2HChFa7XF1dHz9+nJ2d7enp2ZEhLVq0KDY2ViQSbd++Xf28L2X8+PH+/v55eXkV\nFRW2trb29vZU/EKhsNXZ/vOf/6SmpmZnZzOZzN27d/fr14+6ZH39+vUeHh7p6ekjR45s9Y0z\nZsyYPXv2sWPHJk2atHfvXqVSOWPGDFXvhg0bvvzyy7Fjx+7cudPS0pLNZldUVPj7+ysUCvVJ\nVJfOvcgrztMyD4QQ5f9ds6ZUKlsdQFEoFAKBYN++fS27+vTp03Z4AEALKN7wh8DAwFOnTsXF\nxbX6f/8vsmDBgs2bN8+bN2/Lli0vKkh6enr9+vWjlisrK0+fPi0QCAQCQcuRdXV1n3322Vdf\nfdW3b19CSFlZmerCeGrhwYMHL4okKChowYIFe/fuDQoK2rdvn0AgeP/991W9u3btsrOzO378\nuCrIixcvtuPzvpV53nvvvdu3b9fX17e68+3k5JSXl+fi4vKiXXMAoDscNoc/TJs2zd3dff/+\n/evWrVP++YdJBQUFCxcufK3ZXvpTMaVSKRKJNm/e/Pnnn8fFxb2ocovFYtWyVCoNCwtrbGxc\nsmRJq4MXL15sYmISHR1NvbS0tCwtLaWWqbJtZWX1onh4PN7f//7306dPp6SkFBUVTZ8+Xb2X\nyWQqlUq5XE69lMvlK1eufNFUbXgr80ydOrWpqWnp0qXqjapUh4SESCSSiIgIjeRr3HsOAOgL\ne97wBxaL9dNPP/n7+0dGRu7evdvPz69nz561tbVXr15NS0tTP9qcnJxcXFxMCHn06BEhZOXK\nlUwmU19fnzpATXnpT8WWLl2akJBgbW3dtWvX5cuXq9q7d+8+e/Zs1UsrKytfX18HB4eGhoaf\nfvqpqKho7ty5ISEhLSfMyMhISEi4cOECdUUbIWT69Ok+Pj5btmz54IMPoqKiHBwchg8f3kYG\nZsyYceDAAZFIxGQyNVYREBAQExPj5+c3ceLEurq6Q4cOtfG9pA1vZZ558+b9/PPPGzduzM3N\n9fPz4/F49+7dO3XqVF5eHtV75syZ+Pj4nJyc8ePH8/n8srKyy5cv37hxA+e8Ad4R2rlODrSh\n5R3WWtXc3BwfH+/j42Nubq6vr29mZjZy5Mi4uDj1665HjRrVcltis9nq87z0p2JBQUGtbpN9\n+/ZVHyYSiZycnAwNDXk8nre39w8//NDqbGKxWCgUhoeHa7Tv2bPHycmJy+X6+PgUFBS0/dnl\ncjl1pZivr69Gl1QqXb58uYODA4vFsra2/vzzz6mvLwsWLKAGUFeJHz16VOONGlebt28ekUhE\nCKmqqlK1SCSS2NhYV1dXDodD3aQlJiZG/YPs2LFjyJAhXC6Xw+HY2tpOmDBB4/p8AKAv3Nsc\nAACAZnDOGwAAgGZQvAEAAGgGxRsAAIBmULwBAABoBsUbAACAZlC8AQAAaAbFGwAAgGZQvAEA\nAGgGxRsAAIBmULwBAABoBsUbAACAZv4f3mNlPPFRCKEAAAAASUVORK5CYII=",
      "text/plain": [
       "plot without title"
      ]
     },
     "metadata": {
      "image/png": {
       "height": 240,
       "width": 330
      }
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "#plot PCA\n",
    "options(repr.plot.width = 8, repr.plot.height = 6)\n",
    "\n",
    "pcaData <- plotPCA(vsd, intgroup=c(\"cell_type\"), returnData=TRUE)\n",
    "\n",
    "pcaData$cell_type <- factor(pcaData$cell_type, levels = c(\"GM12878\", \"LCL8664\", \"Bulk_B\", \"Naive_B\", \"Mem_B\", \n",
    "                                        \"Regulatory_T\", \"CD8pos_T\",\"Effector_CD4pos_T\", \n",
    "                                        \"Mature_NK\", \"Monocytes\", \"Myeloid_DCs\"))\n",
    "\n",
    "percentVar <- round(100 * attr(pcaData, \"percentVar\"))\n",
    "\n",
    "p <- ggplot(pcaData, aes(PC1, PC2, fill=cell_type)) +\n",
    "  geom_point(size=3, alpha = 0.5, shape = 21, color = \"black\") +\n",
    "  xlab(paste0(\"PC1: \",percentVar[1],\"% variance\")) +\n",
    "  ylab(paste0(\"PC2: \",percentVar[2],\"% variance\")) +\n",
    "  ggsci::scale_fill_aaas() +\n",
    "  theme_bw() +\n",
    "  theme(legend.position = \"right\")\n",
    "options(repr.plot.width = 5.5, repr.plot.height = 4)\n",
    "p\n",
    "\n",
    "ggsave(p, filename = \"../../results/results_human-evolution/RNA-expression-PCA_w-Immune-Cell-data.pdf\", device = cairo_pdf, width = 5.5, height = 4)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.1.2"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": true,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {},
   "toc_section_display": true,
   "toc_window_display": true
  },
  "varInspector": {
   "cols": {
    "lenName": 16,
    "lenType": 16,
    "lenVar": 40
   },
   "kernels_config": {
    "python": {
     "delete_cmd_postfix": "",
     "delete_cmd_prefix": "del ",
     "library": "var_list.py",
     "varRefreshCmd": "print(var_dic_list())"
    },
    "r": {
     "delete_cmd_postfix": ") ",
     "delete_cmd_prefix": "rm(",
     "library": "var_list.r",
     "varRefreshCmd": "cat(var_dic_list()) "
    }
   },
   "types_to_exclude": [
    "module",
    "function",
    "builtin_function_or_method",
    "instance",
    "_Feature"
   ],
   "window_display": false
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
