{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-05-13T17:40:35.882461Z",
     "start_time": "2022-05-13T17:40:35.872612Z"
    }
   },
   "outputs": [],
   "source": [
    "import argparse\n",
    "import glob\n",
    "import itertools \n",
    "import numpy as np\n",
    "import os, sys\n",
    "import subprocess\n",
    "import time\n",
    "\n",
    "\n",
    "sys.path.append(\"/dors/capra_lab/users/fongsl/tools/py_/\")\n",
    "sys.path.append(\"/dors/capra_lab/users/fongsl/tools/genome/\")\n",
    "import config_readwrite as crw\n",
    "import chr_functions\n",
    "import split_filename"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-05-13T17:40:36.120997Z",
     "start_time": "2022-05-13T17:40:36.090993Z"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "usage: ipykernel_launcher.py [-h] [-b BEDFILE]\n",
      "                             [-br {hg38,rheMac8,hg38-rheMac8}]\n",
      "                             [-msa {20way,30way,100way}]\n",
      "                             [-mod {full,rheMac8_noOWM,hg38_noAPES}]\n",
      "                             [-o OUTDIR] [-l LINES]\n",
      "ipykernel_launcher.py: error: unrecognized arguments: -f /data1/home/fongsl/.local/share/jupyter/runtime/kernel-174a0763-c380-4c63-8929-20d596ff413b.json\n"
     ]
    },
    {
     "ename": "SystemExit",
     "evalue": "2",
     "output_type": "error",
     "traceback": [
      "An exception has occurred, use %tb to see the full traceback.\n",
      "\u001b[0;31mSystemExit\u001b[0m\u001b[0;31m:\u001b[0m 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/fongsl/.conda/envs/sfenv/lib/python3.6/site-packages/IPython/core/interactiveshell.py:3351: UserWarning: To exit: use 'exit', 'quit', or Ctrl-D.\n",
      "  warn(\"To exit: use 'exit', 'quit', or Ctrl-D.\", stacklevel=1)\n"
     ]
    }
   ],
   "source": [
    "arg_parser = argparse.ArgumentParser(description=\" describe argparse\")\n",
    "\n",
    "\n",
    "arg_parser.add_argument(\"-b\", \"--bedfile\", help='bed file to run phyloP on')\n",
    "\n",
    "arg_parser.add_argument(\"-br\", \"--branches\", \n",
    "                        choices = ['hg38', 'rheMac8', 'hg38-rheMac8'], \n",
    "                        help='hg38, rheMac8')\n",
    "\n",
    "arg_parser.add_argument(\n",
    "    \"-msa\", \"--multiz\", \n",
    "    choices =[\"20way\", \"30way\", \"100way\"],\n",
    "    default=\"30way\",\n",
    "    help='20-, 30-, 100-way multiz in hg38')\n",
    "\n",
    "arg_parser.add_argument(\n",
    "    \"-mod\", \"--model\", \n",
    "    choices= ['full', 'rheMac8_noOWM', 'hg38_noAPES'],\n",
    "    help='full\", hg38-rheMac8', default=\"full\")\n",
    "\n",
    "arg_parser.add_argument(\"-o\", \"--outdir\", help='path to output directory')\n",
    "arg_parser.add_argument(\"-l\", \"--lines\", help='path to output directory')\n",
    "\n",
    "\n",
    "# PARSE THE ARGUMENTS\n",
    "args = arg_parser.parse_args()\n",
    "\n",
    "BEDF = args.bedfile  # the chromosome\n",
    "BRANCHES = args.branches  # the branches to test.\n",
    "MSAWAY = args.multiz  # multiple sequence alignment.\n",
    "MODELS = args.model\n",
    "DATA_PATH = args.outdir\n",
    "LINES = args.lines  # nlines to split mapping file into. \n",
    "PATH, FILENAME, SAMPLE_ID = split_filename.split_filename(BEDF)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-05-13T19:43:25.194670Z",
     "start_time": "2022-05-13T19:43:25.186963Z"
    }
   },
   "outputs": [],
   "source": [
    "# the bed file\n",
    "BEDF = \"/data/hodges_lab/ATAC-STARR_B-cells/results/results_human-evolution/regions/all_uniq_diffAct_regions2.bed\"\n",
    "# the shuffle file\n",
    "BEDF = \"/data/hodges_lab/ATAC-STARR_B-cells/data/hansen-fong/bkgd_sharedAcc_regions/shuffles/shuf-all_uniq_diffAct_regions2-all.bed\"\n",
    "\n",
    "LINES = 100 # split file into n lines to run in array. \n",
    "BRANCHES =['hg38']# 'rheMac8', 'hg38-rheMac8']\n",
    "\n",
    "MSAWAY =\"30way\"\n",
    "MODELS =['full' ]#'rheMac8_noOWM', 'hg38_noAPES']\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-05-13T19:43:26.666335Z",
     "start_time": "2022-05-13T19:43:26.655665Z"
    },
    "code_folding": []
   },
   "outputs": [],
   "source": [
    "# %% functions\n",
    "\n",
    "def split_by_chr(f):\n",
    "    \n",
    "    \"\"\"\n",
    "    \n",
    "    return path containing bedfile into chromosomes .bed files\n",
    "    \n",
    "    1. get file name\n",
    "    2. make chr_path name\n",
    "    3. make chr_path dir\n",
    "    4. split the bed file by chromosome, \n",
    "    5. go to the directory\n",
    "    6. return chr_path, list of split files\n",
    "    \n",
    "    \"\"\"\n",
    "    \n",
    "    #1\n",
    "    path, file_name, sample = split_filename.split_filename(f)\n",
    "    \n",
    "    #2\n",
    "    chr_path = os.path.join(path, f\"chr-{sample}\") # make a path\n",
    "    \n",
    "    #3\n",
    "    if os.path.exists(chr_path) is False:\n",
    "        os.mkdir(chr_path)  # make the chr dir\n",
    "    \n",
    "    # 4\n",
    "    chr_functions.split_into_chr_bed(f, chr_path)  # split by chromosome\n",
    "    \n",
    "    #5\n",
    "    os.chdir(chr_path)\n",
    "    \n",
    "    return chr_path, glob.glob(\"chr*.bed\"), sample"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-05-13T19:43:27.764490Z",
     "start_time": "2022-05-13T19:43:27.749865Z"
    },
    "code_folding": []
   },
   "outputs": [],
   "source": [
    "# in case you need to split file on size before getting started\n",
    "def split_by_line(f, data_path, chr_num, nlines):\n",
    "    \n",
    "    \"\"\"\n",
    "    return list of files split up into n equal lines\n",
    "    \n",
    "    1. make a chr-specific directory\n",
    "    2. check that you haven't already split\n",
    "    3. if not, split the files by n lines\n",
    "    4. move files to chr-specific directory\n",
    "    5. return a list of the splits. \n",
    "    \n",
    "    \"\"\"\n",
    "    \n",
    "    #1\n",
    "    chr_path = os.path.join(data_path, chr_num) # make dir for chromosome splits\n",
    "    \n",
    "    try:\n",
    "        os.mkdir(chr_path)\n",
    "    except FileExistsError:\n",
    "        pass\n",
    "    \n",
    "    # change dir to the output chr path (not the original CHR_PATH, \n",
    "    # where file is split on CHR, but not line number)\n",
    "    \n",
    "    #2\n",
    "    split_fs = glob.glob(os.path.join(chr_path, f\"{chr_num}-*\"))\n",
    "\n",
    "    #3 split the file in command line into sizes of nlines\n",
    "    os.chdir(data_path)\n",
    "    \n",
    "    cmd = f\"split -l {nlines} {f} {chr_num}-\"\n",
    "    \n",
    "    if len(split_fs) ==0:\n",
    "        print(\"splitting\")\n",
    "        subprocess.call(cmd, shell = True)\n",
    "        \n",
    "        #4\n",
    "        mv_cmd = f\"mv {chr_num}-* {chr_path}\"\n",
    "        subprocess.call(mv_cmd, shell = True)\n",
    "        \n",
    "        split_fs = glob.glob(f\"{chr_path}/{chr_num}-*\")\n",
    "\n",
    "    else:\n",
    "        print(\"already split\")\n",
    "    \n",
    "    #5\n",
    "    return split_fs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-05-13T19:43:28.739184Z",
     "start_time": "2022-05-13T19:43:28.724931Z"
    },
    "code_folding": []
   },
   "outputs": [],
   "source": [
    "def make_run_list(branches, models, chrs):\n",
    "    \n",
    "    \"\"\"\n",
    "    return list of branch, model, and chr pairs to write in mapping file, run\n",
    "    \n",
    "    exclude specific branch, model tuples that don't make sense to test. \n",
    "    \n",
    "    inputs \n",
    "        branches (list)\n",
    "        models (list)\n",
    "        chr_list (list)\n",
    "        \n",
    "    output \n",
    "        runs (list of lists)\n",
    "    \"\"\"\n",
    "    runs = []\n",
    "    no_runs = [('hg38', 'rheMac8_noOWM') , \n",
    "               ('hg38-rheMac8', 'rheMac8_noOWM'), \n",
    "               ('hg38-rheMac8', 'hg38_noAPES'),\n",
    "              ('rheMac8', 'hg38_noAPES') \n",
    "              ] # don't run these tuples. Not interested yet in these results\n",
    "    for b in branches:\n",
    "        for m in models:\n",
    "            for c in chrs:           \n",
    "                combo = [b, m, c]\n",
    "                if combo not in runs and (b,m) not in no_runs:\n",
    "                    runs.append(combo)\n",
    "                    \n",
    "    return runs\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-05-13T19:43:29.396782Z",
     "start_time": "2022-05-13T19:43:29.381358Z"
    }
   },
   "outputs": [],
   "source": [
    "def write_mapping_file(mapping_file, run_list, n, msaway):\n",
    "    \n",
    "    \"\"\"\n",
    "    write to mapping file for slurm array. Line includes index, branch, model, and file to run\n",
    "    \n",
    "    0. make the outfile that the results will be written to\n",
    "    1. make the row as a tab-delimited string\n",
    "    2. open the mapping file \n",
    "    3. append the row to the file\n",
    "    4. increase the n counter by the number of rows,\n",
    "    5. return n to count for next set of rows. \n",
    "    \n",
    "    inputs: mapping file (.txt), run_list (list), n (int)\n",
    "    outputs: n (int)\n",
    "    \"\"\"\n",
    "    \n",
    "    for r in run_list:\n",
    "        \n",
    "        #0 make the outfile that results will be written to\n",
    "        branch, model, chr_file = r  # make variables from list\n",
    "        chr_, chr_n = (chr_file.split(\"/\")[-1]).split(\"-\")  # chr3-aj\n",
    "        \n",
    "        phylop_path = \"/\".join(chr_file.split(\"/\")[:-1])  # path to chr_file\n",
    "        \n",
    "        outpath = os.path.join(\n",
    "        phylop_path, f\"multiz{msaway}_br-{branch}_mod-{model}\") # path to write to\n",
    "        \n",
    "        outf = os.path.join(outpath, f\"{chr_}_{chr_n}_conacc.bed\")  # file to write during phylop run \n",
    "\n",
    "\n",
    "        #1 make the row as tabbed-string (index, branch, model, and file to run)\n",
    "        row =  str(n) + \"\\t\"+ \"\\t\".join(r) +\"\\t\"+ msaway + \"\\t\"+ outf + \"\\n\"\n",
    "\n",
    "        #2 open the mapping file\n",
    "        with open(mapping_file, \"a\") as writer:\n",
    "\n",
    "            #3 write the row \n",
    "            writer.write(row)\n",
    "            writer.close()\n",
    "        #4\n",
    "        n+=1 # for counting the next row. \n",
    "        \n",
    "    #5\n",
    "    return n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-05-13T19:43:31.526423Z",
     "start_time": "2022-05-13T19:43:31.516888Z"
    }
   },
   "outputs": [],
   "source": [
    "def run_conacc_slurm_array(mapping_file, num_files):\n",
    "    \n",
    "    \"\"\"\n",
    "    return slurm script command with array argument\n",
    "    \n",
    "    inputs \n",
    "        mapping file - branch, model, run file, and msaway info in tab-seperated format (txt)\n",
    "        num_files - length of mapping file/ number of files to run (int)\n",
    "\n",
    "    hardcoded\n",
    "        slurm script path\n",
    "        memory for array run\n",
    "        limit to n files run in array (%10)\n",
    "        \n",
    "    output\n",
    "        slurm command (str) with mapping file, array format, and memory specified\n",
    "    \"\"\"\n",
    "\n",
    "    script = os.path.join(\"/data/hodges_lab/ATAC-STARR_B-cells/bin_human-evolution/phylop\", \"conacc_array-mapping.slurm\")\n",
    "\n",
    "\n",
    "    mem = \"--mem=36GB\"\n",
    "    array = f\"--array [0-{num_files}]%30\"  # only run ten files at a time\n",
    "\n",
    "    # make the command\n",
    "\n",
    "    cmd = f\"sbatch {array} {mem} {script} {mapping_file}\"  #{chrnum} {branches} {msaway} {mod}\"\n",
    "    \n",
    "    print(cmd)\n",
    "\n",
    "    # return it\n",
    "    return cmd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-05-13T19:43:46.792029Z",
     "start_time": "2022-05-13T19:43:35.041482Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "already split\n",
      "91 phylop chr x branch x model comparisons to run\n",
      "already split\n",
      "94 phylop chr x branch x model comparisons to run\n",
      "already split\n",
      "90 phylop chr x branch x model comparisons to run\n",
      "already split\n",
      "72 phylop chr x branch x model comparisons to run\n",
      "already split\n",
      "97 phylop chr x branch x model comparisons to run\n",
      "already split\n",
      "85 phylop chr x branch x model comparisons to run\n",
      "already split\n",
      "84 phylop chr x branch x model comparisons to run\n",
      "already split\n",
      "96 phylop chr x branch x model comparisons to run\n",
      "already split\n",
      "92 phylop chr x branch x model comparisons to run\n",
      "already split\n",
      "89 phylop chr x branch x model comparisons to run\n",
      "already split\n",
      "74 phylop chr x branch x model comparisons to run\n",
      "already split\n",
      "100 phylop chr x branch x model comparisons to run\n",
      "already split\n",
      "94 phylop chr x branch x model comparisons to run\n",
      "already split\n",
      "95 phylop chr x branch x model comparisons to run\n",
      "already split\n",
      "96 phylop chr x branch x model comparisons to run\n",
      "already split\n",
      "91 phylop chr x branch x model comparisons to run\n",
      "already split\n",
      "94 phylop chr x branch x model comparisons to run\n",
      "already split\n",
      "59 phylop chr x branch x model comparisons to run\n",
      "already split\n",
      "94 phylop chr x branch x model comparisons to run\n",
      "already split\n",
      "85 phylop chr x branch x model comparisons to run\n",
      "already split\n",
      "85 phylop chr x branch x model comparisons to run\n",
      "already split\n",
      "92 phylop chr x branch x model comparisons to run\n",
      "sbatch --array [0-1949]%30 --mem=36GB /data/hodges_lab/ATAC-STARR_B-cells/bin_human-evolution/phylop/conacc_array-mapping.slurm /data/hodges_lab/ATAC-STARR_B-cells/data/hansen-fong/bkgd_sharedAcc_regions/shuffles/chr-shuf-all_uniq_diffAct_regions2-all/mapping_shuf-all_uniq_diffAct_regions2-all.txt\n"
     ]
    }
   ],
   "source": [
    "def main(argv):\n",
    "    \"\"\"\n",
    "    launch a slurm array with .bed file split on chromosome number and equal lines\n",
    "    \n",
    "    inputs - \n",
    "        bedfile (.bed) \n",
    "        data_path to save the split up files (dir)\n",
    "        n lines - number of lines to write per file\n",
    "        BRANCHES - list of branches to test\n",
    "        MODELS - list of neutral models to test acceleration against. \n",
    "        MSAWAY - multiz alignment to use. \n",
    "        \n",
    "    outputs - \n",
    "        mapping_file with all parameters for running phyloP (branch, model, multiz)\n",
    "        launched slurm command\n",
    "    \"\"\"\n",
    "\n",
    "    #1 split bedfile by chr\n",
    "    chr_path, chrs, sample_id = split_by_chr(BEDF)\n",
    "    \n",
    "    #2 go to the chr dir\n",
    "    os.chdir(chr_path)\n",
    "\n",
    "    #3 exclude these chromosomes by taking set difference.\n",
    "    excl_chr = set(['chrX.bed', 'chrY.bed', 'chrM.bed', \n",
    "                    'chr14_KI270726v1_random.bed', 'chr16_KI270728v1_random.bed'\n",
    "                    'chr14_KI270722v1_random.bed', 'chr16_KI270728v1_random.bed', \n",
    "                    'chr14_KI270722v1_random.bed'\n",
    "                   ])\n",
    "    \n",
    "    chrs_ = list(set(chrs).difference(excl_chr))\n",
    "\n",
    "\n",
    "    mapping_file = os.path.join(chr_path, f\"mapping_{sample_id}.txt\")\n",
    "\n",
    "    n=0 # count the number of lines in a file. \n",
    "    \n",
    "    # iterate through chromosomes\n",
    "    for chr_ in chrs_: \n",
    "\n",
    "        chr_num = chr_.split(\".bed\")[0]\n",
    "        \n",
    "        #print(chr_num, chr_)\n",
    "\n",
    "        # split chr_files into 1000 line files.\n",
    "        split_fs = split_by_line(chr_, chr_path, chr_num, LINES)\n",
    "\n",
    "        run_list = make_run_list(BRANCHES, MODELS, split_fs)\n",
    "        #print(run_list)\n",
    "        \n",
    "        # write to mapping file. \n",
    "        n = write_mapping_file(mapping_file, run_list, n, MSAWAY)\n",
    "    \n",
    "        print(len(run_list),\"phylop chr x branch x model comparisons to run\")\n",
    "    \n",
    "    # launch the slurm job\n",
    "    cmd = run_conacc_slurm_array(mapping_file, n)\n",
    "\n",
    "    #subprocess.call(cmd, shell = True)\n",
    "\n",
    "if __name__ == \"__main__\":\n",
    "    main(sys.argv[1:])"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python (sfenv)",
   "language": "python",
   "name": "sfenv"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.10"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {},
   "toc_section_display": true,
   "toc_window_display": false
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
